{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AnnotationReports {}\n",
      "AssembliesReport {'target_set_accession': 'GCF_002925485.2'}\n",
      "BuildInfo {}\n",
      "FeatureCounts {}\n",
      "FeatureStats {}\n"
     ]
    }
   ],
   "source": [
    "import xml.etree.ElementTree as ET\n",
    "tree = ET.parse('Homo_sapiens_AR109.20190905_annotation_report.xml')\n",
    "root = tree.getroot()\n",
    "print(root.tag, root.attrib)\n",
    "for child in root:\n",
    "    print(child.tag, child.attrib)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4\n"
     ]
    }
   ],
   "source": [
    "# children\n",
    "count = 0 \n",
    "for child in root:\n",
    "    count += 1\n",
    "print(count)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Child:\n",
      "AssembliesReport {'target_set_accession': 'GCF_002925485.2'}\n",
      "Grandchildren:\n",
      "FullAssembly {}\n",
      "1\n",
      "\n",
      "\n",
      "\n",
      "Child:\n",
      "BuildInfo {}\n",
      "Grandchildren:\n",
      "Taxid {}\n",
      "ScientificName {}\n",
      "AnnotationReleaseId {}\n",
      "FreezeDate {}\n",
      "FreezeDateId {}\n",
      "IdLoadDate {}\n",
      "SoftwareMajorVersion {}\n",
      "SoftwareMinorVersion {}\n",
      "11\n",
      "\n",
      "\n",
      "\n",
      "Child:\n",
      "FeatureCounts {}\n",
      "Grandchildren:\n",
      "51\n",
      "\n",
      "\n",
      "\n",
      "Child:\n",
      "FeatureStats {}\n",
      "Grandchildren:\n",
      "91\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# grandchildren\n",
    "count = 0\n",
    "for child in root:\n",
    "    print('Child:')\n",
    "    print(child.tag, child.attrib)\n",
    "    print('Grandchildren:')\n",
    "    for grandchild in child:\n",
    "        count += 1\n",
    "        if count < 10:\n",
    "            print(grandchild.tag, grandchild.attrib)\n",
    "    print(count)\n",
    "    print('\\n\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Child:\n",
      "AssembliesReport {'target_set_accession': 'GCF_002925485.2'}\n",
      "Grandchildren:\n",
      "FullAssembly {}\n",
      "Great Grandchildren:  Name {}\n",
      "Great Grandchildren:  Accession {}\n",
      "Great Grandchildren:  IsReference {}\n",
      "Great Grandchildren:  SubmitterOrganization {}\n",
      "Great Grandchildren:  SubmitterDate {}\n",
      "Great Grandchildren:  AssemblyContents {}\n",
      "Great Great Grandchildren:  Chromosomes {}\n",
      "Great Great Grandchildren:  AlternateLoci {}\n",
      "203\n",
      "\n",
      "\n",
      "\n",
      "Child:\n",
      "BuildInfo {}\n",
      "Grandchildren:\n",
      "213\n",
      "\n",
      "\n",
      "\n",
      "Child:\n",
      "FeatureCounts {}\n",
      "Grandchildren:\n",
      "2621\n",
      "\n",
      "\n",
      "\n",
      "Child:\n",
      "FeatureStats {}\n",
      "Grandchildren:\n",
      "6391\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# great grandchildren\n",
    "count = 0\n",
    "for child in root:\n",
    "    print('Child:')\n",
    "    print(child.tag, child.attrib)\n",
    "    print('Grandchildren:')\n",
    "    for grandchild in child:\n",
    "        count += 1\n",
    "        if count < 10:\n",
    "            print(grandchild.tag, grandchild.attrib)\n",
    "        for greatgrandchild in grandchild:\n",
    "            count += 1\n",
    "            if count < 10:\n",
    "                print('Great Grandchildren: ', greatgrandchild.tag, greatgrandchild.attrib)\n",
    "            for greatgreatgrandchild in greatgrandchild:\n",
    "                count += 1\n",
    "                if count < 10:\n",
    "                    print('Great Great Grandchildren: ', greatgreatgrandchild.tag, greatgreatgrandchild.attrib)\n",
    "                for ggggchild in greatgreatgrandchild:\n",
    "                    count += 1\n",
    "                    if count < 10:\n",
    "                        print('Great Great Great Grandchildren: ', ggggchild.tag, ggggchild.attrib, '\\n')\n",
    "    print(count)\n",
    "    print('\\n\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6396\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['AnnotationReports',\n",
       " 'AssembliesReport',\n",
       " 'FullAssembly',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsReference',\n",
       " 'SubmitterOrganization',\n",
       " 'SubmitterDate',\n",
       " 'AssemblyContents',\n",
       " 'Chromosomes',\n",
       " 'AlternateLoci',\n",
       " 'AssemblyPatches',\n",
       " 'UnlocalizedScaffolds',\n",
       " 'UnplacedScaffolds',\n",
       " 'Units',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'BuildInfo',\n",
       " 'Taxid',\n",
       " 'ScientificName',\n",
       " 'AnnotationReleaseId',\n",
       " 'FreezeDate',\n",
       " 'FreezeDateId',\n",
       " 'IdLoadDate',\n",
       " 'SoftwareMajorVersion',\n",
       " 'SoftwareMinorVersion',\n",
       " 'MaskingSoftware',\n",
       " 'FtpSite',\n",
       " 'FeatureCounts',\n",
       " 'OverallCounts',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'AssemblyCounts',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'AssemblyCounts',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'AssemblyCounts',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'AssemblyCounts',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'AssemblyCounts',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'AssemblyCounts',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'AssemblyCounts',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'AssemblyCounts',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'AssemblyCounts',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'AssemblyCounts',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'AssemblyCounts',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'AssemblyCounts',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " ...]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tags = []\n",
    "for element in root.iter():\n",
    "    tags.append(element.tag)\n",
    "print(len(tags))\n",
    "tags"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[[('Name', {}, 'GRCh38.p13')],\n",
       " [('Accession', {}, 'GCF_000001405.39')],\n",
       " [('IsReference', {}, 'Yes')],\n",
       " [('SubmitterOrganization', {}, 'Genome Reference Consortium')],\n",
       " [('SubmitterDate', {}, '02-28-2019')],\n",
       " [('AssemblyContents', {}, '\\n      '),\n",
       "  ('Chromosomes', {}, '25'),\n",
       "  ('AlternateLoci', {}, None),\n",
       "  ('AssemblyPatches', {}, None),\n",
       "  ('UnlocalizedScaffolds', {}, None),\n",
       "  ('UnplacedScaffolds', {}, None)],\n",
       " [('Units', {}, '\\n      '),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'Primary Assembly'),\n",
       "  ('Accession', {}, 'GCF_000001305.15'),\n",
       "  ('IsPrimary', {}, 'Yes'),\n",
       "  ('Type', {}, 'HaploidUnit'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'PATCHES'),\n",
       "  ('Accession', {}, 'GCF_000005045.25'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AssemblyPatches'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_1'),\n",
       "  ('Accession', {}, 'GCF_000001315.2'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_2'),\n",
       "  ('Accession', {}, 'GCF_000001325.3'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_3'),\n",
       "  ('Accession', {}, 'GCF_000001335.2'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_4'),\n",
       "  ('Accession', {}, 'GCF_000001345.2'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_5'),\n",
       "  ('Accession', {}, 'GCF_000001355.2'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_6'),\n",
       "  ('Accession', {}, 'GCF_000001365.3'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_7'),\n",
       "  ('Accession', {}, 'GCF_000001375.2'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_8'),\n",
       "  ('Accession', {}, 'GCF_000510225.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_9'),\n",
       "  ('Accession', {}, 'GCF_000510235.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_10'),\n",
       "  ('Accession', {}, 'GCF_000509915.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_11'),\n",
       "  ('Accession', {}, 'GCF_000509925.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_12'),\n",
       "  ('Accession', {}, 'GCF_000509935.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_13'),\n",
       "  ('Accession', {}, 'GCF_000509945.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_14'),\n",
       "  ('Accession', {}, 'GCF_000509955.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_15'),\n",
       "  ('Accession', {}, 'GCF_000509965.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_16'),\n",
       "  ('Accession', {}, 'GCF_000509975.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_17'),\n",
       "  ('Accession', {}, 'GCF_000509985.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_18'),\n",
       "  ('Accession', {}, 'GCF_000509995.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_19'),\n",
       "  ('Accession', {}, 'GCF_000510005.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_20'),\n",
       "  ('Accession', {}, 'GCF_000510015.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_21'),\n",
       "  ('Accession', {}, 'GCF_000510025.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_22'),\n",
       "  ('Accession', {}, 'GCF_000510035.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_23'),\n",
       "  ('Accession', {}, 'GCF_000510045.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_24'),\n",
       "  ('Accession', {}, 'GCF_000510055.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_25'),\n",
       "  ('Accession', {}, 'GCF_000510065.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_26'),\n",
       "  ('Accession', {}, 'GCF_000510075.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_27'),\n",
       "  ('Accession', {}, 'GCF_000510085.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_28'),\n",
       "  ('Accession', {}, 'GCF_000510095.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_29'),\n",
       "  ('Accession', {}, 'GCF_000510105.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_30'),\n",
       "  ('Accession', {}, 'GCF_000510125.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_31'),\n",
       "  ('Accession', {}, 'GCF_000510135.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_32'),\n",
       "  ('Accession', {}, 'GCF_000510145.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_33'),\n",
       "  ('Accession', {}, 'GCF_000510155.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_34'),\n",
       "  ('Accession', {}, 'GCF_000510165.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_35'),\n",
       "  ('Accession', {}, 'GCF_000510175.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'non-nuclear'),\n",
       "  ('Accession', {}, 'GCF_000006015.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'HaploidUnit')]]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_assembly_collection = []\n",
    "for i in root[0][0]:\n",
    "    tags = []\n",
    "    text = []\n",
    "    attributes = []\n",
    "    for element in i.iter():\n",
    "        tags.append(element.tag)\n",
    "        text.append(element.text)\n",
    "        attributes.append(element.attrib)\n",
    "    thing = list(zip(tags,attributes,text))\n",
    "    full_assembly_collection.append(thing)\n",
    "full_assembly_collection"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[[('Chromosomes', {}, '25')],\n",
       " [('AlternateLoci', {}, None)],\n",
       " [('AssemblyPatches', {}, None)],\n",
       " [('UnlocalizedScaffolds', {}, None)],\n",
       " [('UnplacedScaffolds', {}, None)]]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "assembly_contents_collection = []\n",
    "for i in root[0][0][5]:\n",
    "    tags = []\n",
    "    text = []\n",
    "    attributes = []\n",
    "    for element in i.iter():\n",
    "        tags.append(element.tag)\n",
    "        text.append(element.text)\n",
    "        attributes.append(element.attrib)\n",
    "    thing = list(zip(tags,attributes,text))\n",
    "    assembly_contents_collection.append(thing)\n",
    "assembly_contents_collection"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "38\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[[('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'Primary Assembly'),\n",
       "  ('Accession', {}, 'GCF_000001305.15'),\n",
       "  ('IsPrimary', {}, 'Yes'),\n",
       "  ('Type', {}, 'HaploidUnit')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'PATCHES'),\n",
       "  ('Accession', {}, 'GCF_000005045.25'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AssemblyPatches')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_1'),\n",
       "  ('Accession', {}, 'GCF_000001315.2'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_2'),\n",
       "  ('Accession', {}, 'GCF_000001325.3'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_3'),\n",
       "  ('Accession', {}, 'GCF_000001335.2'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_4'),\n",
       "  ('Accession', {}, 'GCF_000001345.2'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_5'),\n",
       "  ('Accession', {}, 'GCF_000001355.2'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_6'),\n",
       "  ('Accession', {}, 'GCF_000001365.3'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_7'),\n",
       "  ('Accession', {}, 'GCF_000001375.2'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_8'),\n",
       "  ('Accession', {}, 'GCF_000510225.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_9'),\n",
       "  ('Accession', {}, 'GCF_000510235.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_10'),\n",
       "  ('Accession', {}, 'GCF_000509915.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_11'),\n",
       "  ('Accession', {}, 'GCF_000509925.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_12'),\n",
       "  ('Accession', {}, 'GCF_000509935.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_13'),\n",
       "  ('Accession', {}, 'GCF_000509945.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_14'),\n",
       "  ('Accession', {}, 'GCF_000509955.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_15'),\n",
       "  ('Accession', {}, 'GCF_000509965.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_16'),\n",
       "  ('Accession', {}, 'GCF_000509975.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_17'),\n",
       "  ('Accession', {}, 'GCF_000509985.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_18'),\n",
       "  ('Accession', {}, 'GCF_000509995.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_19'),\n",
       "  ('Accession', {}, 'GCF_000510005.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_20'),\n",
       "  ('Accession', {}, 'GCF_000510015.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_21'),\n",
       "  ('Accession', {}, 'GCF_000510025.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_22'),\n",
       "  ('Accession', {}, 'GCF_000510035.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_23'),\n",
       "  ('Accession', {}, 'GCF_000510045.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_24'),\n",
       "  ('Accession', {}, 'GCF_000510055.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_25'),\n",
       "  ('Accession', {}, 'GCF_000510065.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_26'),\n",
       "  ('Accession', {}, 'GCF_000510075.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_27'),\n",
       "  ('Accession', {}, 'GCF_000510085.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_28'),\n",
       "  ('Accession', {}, 'GCF_000510095.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_29'),\n",
       "  ('Accession', {}, 'GCF_000510105.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_30'),\n",
       "  ('Accession', {}, 'GCF_000510125.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_31'),\n",
       "  ('Accession', {}, 'GCF_000510135.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_32'),\n",
       "  ('Accession', {}, 'GCF_000510145.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_33'),\n",
       "  ('Accession', {}, 'GCF_000510155.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_34'),\n",
       "  ('Accession', {}, 'GCF_000510165.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_35'),\n",
       "  ('Accession', {}, 'GCF_000510175.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'non-nuclear'),\n",
       "  ('Accession', {}, 'GCF_000006015.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'HaploidUnit')]]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "assembly_unit_collection = []\n",
    "for i in root[0][0][6]:\n",
    "    tags = []\n",
    "    text = []\n",
    "    attributes = []\n",
    "    for element in i.iter():\n",
    "        tags.append(element.tag)\n",
    "        text.append(element.text)\n",
    "        attributes.append(element.attrib)\n",
    "    thing = list(zip(tags, attributes, text))\n",
    "    assembly_unit_collection.append(thing)\n",
    "print(len(assembly_unit_collection))\n",
    "assembly_unit_collection"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[[('Taxid', {}, '9606')],\n",
       " [('ScientificName', {}, 'Homo sapiens')],\n",
       " [('AnnotationReleaseId', {}, '109')],\n",
       " [('FreezeDate', {}, 'Sep 5 2019')],\n",
       " [('FreezeDateId', {}, '20190905')],\n",
       " [('IdLoadDate', {}, 'Sep 9 2019')],\n",
       " [('SoftwareMajorVersion', {}, '8')],\n",
       " [('SoftwareMinorVersion', {}, '2')],\n",
       " [('MaskingSoftware', {}, 'repeatmasker')],\n",
       " [('FtpSite',\n",
       "   {},\n",
       "   'ftp://ftp.ncbi.nlm.nih.gov/genomes/all/annotation_releases/9606/109.20190905')]]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "build_info_collection = []\n",
    "for i in root[1]:\n",
    "    tags = []\n",
    "    text = []\n",
    "    attributes = []\n",
    "    for element in i.iter():\n",
    "        tags.append(element.tag)\n",
    "        text.append(element.text)\n",
    "        attributes.append(element.attrib)\n",
    "    thing = list(zip(tags,attributes,text))\n",
    "    build_info_collection.append(thing)\n",
    "print(len(build_info_collection))\n",
    "build_info_collection"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "40\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[[('OverallCounts', {}, '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '54639'),\n",
       "  ('Subcount', {'subtype': 'other'}, '47'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '15204'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1135'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '19965'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '364'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '397'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '18012'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '19844'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '112430'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '172'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '26'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '1542'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '58361'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '54069'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '1541'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '112283'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '88'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '1'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '47041'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '15'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '7'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '196'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '29863'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '15319'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '196'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '45168'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1442'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '-2'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '74'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '108'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1334'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '74'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1431'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '113017'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '2600'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '514'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '1109'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '58374'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '54069'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '534'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '112283'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '118'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '69'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '503'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '57'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '35')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000001405.39', 'Type': 'full assembly'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '54639'),\n",
       "  ('Subcount', {'subtype': 'other'}, '47'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '15204'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1135'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '19965'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '364'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '397'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '18012'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '19844'),\n",
       "  ('Subcount', {'subtype': 'multiply placed'}, '3433'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '112430'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '172'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '26'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '1542'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '58361'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '54069'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '1541'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '112283'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '88'),\n",
       "  ('Subcount', {'subtype': 'multiply placed'}, '2901'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '1'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '47041'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '15'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '7'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '196'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '29863'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '15319'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '196'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '45168'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'multiply placed'}, '695'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1442'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '-2'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '74'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '108'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1334'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '74'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1431'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'multiply placed'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '113017'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '2600'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '514'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '1109'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '58374'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '54069'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '534'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '112283'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '118'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '69'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '503'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '57'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '35')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000001305.15', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '54312'),\n",
       "  ('Subcount', {'subtype': 'other'}, '45'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '15137'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1127'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '19886'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '366'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '388'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '17860'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '19755'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '112169'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '172'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '31'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '1565'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '58188'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '53981'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '1564'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '112038'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '78'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '1'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '45167'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '14'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '4'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '195'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '28479'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '15310'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '195'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '43775'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1426'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '1'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '78'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '97'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1329'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '78'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1417'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '112558'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '2602'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '348'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '938'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '58188'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '53981'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '548'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '112038'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '104'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '66'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '493'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '61'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '45')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000001405.39', 'Type': 'grouped alt loci'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '2398'),\n",
       "  ('Subcount', {'subtype': 'other'}, '8'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '631'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '90'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '606'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '297'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '161'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '676'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '832'),\n",
       "  ('Subcount', {'subtype': 'multiply placed'}, '658'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '1917'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '267'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '818'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '136'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1781'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '818'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1906'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '7'),\n",
       "  ('Subcount', {'subtype': 'multiply placed'}, '638'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '1820'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '1'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '56'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '166'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '1013'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '506'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '170'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1519'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'multiply placed'}, '183'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '109'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '9'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '65'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '8'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '101'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '65'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '107'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'multiply placed'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '2077'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '28'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '364'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '654'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '136'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1778'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '491'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1906'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '10'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '61'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '544'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '26'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '28')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000001315.2', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '1942'),\n",
       "  ('Subcount', {'subtype': 'other'}, '8'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '515'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '71'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '486'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '305'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '161'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '529'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '658'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '1505'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '281'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '743'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '82'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1423'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '743'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1498'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '4'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '1038'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '1'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '71'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '162'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '582'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '417'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '166'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '999'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '85'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '6'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '53'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '6'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '79'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '53'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '84'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '1665'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '25'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '384'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '651'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '82'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1420'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '488'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1498'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '6'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '45'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '443'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '27'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '39')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000001325.3', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '784'),\n",
       "  ('Subcount', {'subtype': 'other'}, '2'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '182'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '31'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '210'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '24'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '241'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '328'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '700'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '44'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '428'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '19'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '681'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '428'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '697'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '2'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '470'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '8'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '103'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '215'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '208'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '104'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '423'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '38'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '3'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '27'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '36'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '27'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '37'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '700'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '9'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '38'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '207'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '19'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '681'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '207'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '697'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '3'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '24'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '183'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '17'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '17')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000001335.2', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '398'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '86'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '12'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '127'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '15'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '112'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '188'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '410'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '22'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '227'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '5'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '405'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '227'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '409'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '209'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '6'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '70'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '48'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '138'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '71'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '186'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '17'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '1'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '14'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '17'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '14'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '17'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '410'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '2'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '21'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '135'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '5'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '405'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '135'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '409'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '1'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '8'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '127'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '5'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '4')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000001345.2', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '340'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '83'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '10'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '111'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '13'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '92'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '155'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '366'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '29'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '213'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '7'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '359'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '213'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '366'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '176'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '1'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '58'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '35'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '113'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '60'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '148'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '15'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '11'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '15'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '11'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '15'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '366'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '3'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '24'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '128'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '7'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '359'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '128'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '366'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '13'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '115'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '6'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '8')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000001355.2', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '346'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '69'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '10'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '113'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '18'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '92'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '175'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '402'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '32'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '234'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '8'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '394'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '234'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '402'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '164'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '7'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '62'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '29'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '105'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '63'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '134'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '15'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '1'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '11'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '15'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '11'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '15'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '402'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '3'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '30'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '148'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '8'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '394'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '148'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '402'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '16'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '132'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '14'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '14')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000001365.3', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '367'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '75'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '9'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '123'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '7'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '104'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '179'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '408'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '12'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '276'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '6'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '402'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '276'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '408'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '198'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '2'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '71'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '36'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '130'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '72'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '166'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '13'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '10'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '13'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '10'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '13'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '408'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '3'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '12'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '134'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '6'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '402'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '134'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '408'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '8'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '126'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '4'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '6')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000001375.2', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '385'),\n",
       "  ('Subcount', {'subtype': 'other'}, '1'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '82'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '13'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '124'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '16'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '112'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '177'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '420'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '37'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '265'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '5'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '415'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '265'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '420'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '206'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '5'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '73'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '46'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '125'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '74'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '171'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '17'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '13'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '17'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '13'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '17'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '419'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '2'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '28'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '171'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '5'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '414'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '171'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '420'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '13'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '160'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '7'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '7')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510225.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '45'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '5'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '24'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '3'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '8'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '31'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '92'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '3'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '52'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '92'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '52'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '92'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '21'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '2'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '7'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '14'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '7'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '21'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '92'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '4'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '2'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '40'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '92'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '40'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '92'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '39'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '1')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510235.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '65'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '9'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '2'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '34'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '1'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '11'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '43'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '126'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '2'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '73'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '4'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '122'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '73'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '126'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '32'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '14'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '18'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '32'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '6'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '6'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '6'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '126'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '4'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '2'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '49'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '4'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '122'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '49'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '126'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '48'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '1')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000509915.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '15'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '2'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '5'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '11'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '19'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '10'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '19'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '10'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '19'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '19'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '9'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '19'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '9'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '19'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '9'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000509925.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '11'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '2'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '4'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '7'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '10'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '9'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '10'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '9'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '10'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '10'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '10'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '10'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000509935.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '8'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '0'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '4'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '0'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '7'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '14'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '9'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '14'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '9'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '14'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '0'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '0'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '14'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '9'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '14'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '9'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '14'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '9'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000509945.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '11'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '2'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '3'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '7'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '9'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '9'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '9'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '9'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '9'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '9'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000509955.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '14'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '2'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '5'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '1'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '10'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '17'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '2'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '17'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '17'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '17'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '2'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '11'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '17'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '11'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '17'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '11'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000509965.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '15'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '3'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '4'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '2'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '9'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '13'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '8'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '13'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '8'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '13'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '3'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '3'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '13'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '7'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '13'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '7'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '13'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '7'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000509975.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '8'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '0'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '3'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '0'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '7'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '13'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '13'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '13'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '0'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '0'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '13'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '13'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '13'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000509985.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '11'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '2'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '3'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '7'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '9'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '9'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '9'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '9'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '9'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '9'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000509995.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '10'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '0'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '4'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '0'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '9'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '17'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '17'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '17'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '0'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '0'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '17'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '17'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '17'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510005.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '11'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '2'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '3'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '7'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '9'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '7'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '9'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '7'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '9'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '9'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '9'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '9'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510015.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '14'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '2'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '5'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '1'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '10'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '17'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '2'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '17'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '17'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '17'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '2'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '17'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '17'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510025.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '11'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '2'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '3'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '7'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '9'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '9'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '9'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '9'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '9'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '9'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510035.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '10'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '0'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '4'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '0'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '9'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '17'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '7'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '17'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '7'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '17'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '0'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '0'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '17'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '7'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '17'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '7'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '17'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '7'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510045.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '13'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '3'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '5'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '1'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '9'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '22'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '5'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '22'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '22'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '0'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '0'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '22'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '5'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '9'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '22'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '9'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '22'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '7'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '2'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '2')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510055.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '9'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '2'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '2'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '1'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '5'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '6'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '1'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '6'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '6'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '6'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '1'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '6'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '6'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510065.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '11'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '2'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '4'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '7'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '10'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '10'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '10'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '10'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '10'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '10'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510075.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '13'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '2'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '5'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '2'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '9'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '22'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '5'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '14'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '22'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '14'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '22'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '1'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '22'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '5'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '13'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '22'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '13'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '22'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '11'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '2'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '2')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510085.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '19'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '3'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '7'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '1'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '14'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '33'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '5'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '15'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '33'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '15'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '33'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '33'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '5'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '14'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '33'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '14'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '33'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '2'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '2')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510095.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '13'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '3'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '6'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '1'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '9'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '23'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '5'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '13'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '23'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '13'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '23'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '0'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '0'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '23'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '5'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '11'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '23'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '11'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '23'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '9'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '2'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '2')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510105.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '10'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '0'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '6'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '2'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '0'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '9'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '27'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '5'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '27'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '27'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '0'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '0'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '1'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '27'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '5'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '27'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '27'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '4'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '2'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '2')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510125.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '13'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '3'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '5'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '2'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '9'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '22'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '6'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '22'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '22'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '0'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '0'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '22'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '6'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '11'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '22'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '11'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '22'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '9'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '2'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '2')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510135.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '12'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '4'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '9'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '12'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '12'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '12'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '12'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '10'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '12'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '10'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '12'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '10'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510145.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '12'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '3'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '3'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '7'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '9'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '8'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '9'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '8'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '9'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '9'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '9'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '9'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510155.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '15'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '3'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '5'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '10'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '17'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '17'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '17'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '17'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '17'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '17'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510165.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '11'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '2'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '4'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '7'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '10'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '10'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '10'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '10'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '10'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '10'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '8'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '10'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '8'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '10'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '8'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510175.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '10'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '0'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '4'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '1'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '0'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '9'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '17'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '2'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '10'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '17'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '10'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '17'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '0'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '0'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '17'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '2'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '17'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '17'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000005045.25', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '1529'),\n",
       "  ('Subcount', {'subtype': 'other'}, '12'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '431'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '80'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '380'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '101'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '22'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '386'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '598'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '1382'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '176'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '492'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '37'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1345'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '492'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1377'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '3'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '758'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '39'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '104'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '371'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '359'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '105'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '730'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '90'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '6'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '48'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '3'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '87'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '48'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '90'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '1390'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '23'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '161'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '274'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '37'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1331'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '252'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1377'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '4'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '57'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '200'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '36'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '40')]]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "feature_counts_collection = []\n",
    "for i in root[2]:\n",
    "    tags = []\n",
    "    text = []\n",
    "    attributes = []\n",
    "    for element in i.iter():\n",
    "        tags.append(element.tag)\n",
    "        text.append(element.text)\n",
    "        attributes.append(element.attrib)\n",
    "    thing = list(zip(tags, attributes ,text))\n",
    "    feature_counts_collection.append(thing)\n",
    "print(len(feature_counts_collection))\n",
    "feature_counts_collection"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "40\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[[('OverallStats', {}, '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '113'),\n",
       "  ('Mean', {}, '4.181'),\n",
       "  ('Median', {}, '2'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '363'),\n",
       "  ('Mean', {}, '11.6507'),\n",
       "  ('Median', {}, '9'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '159471'),\n",
       "  ('Min', {}, '16'),\n",
       "  ('Max', {}, '109224'),\n",
       "  ('Mean', {}, '3565.83'),\n",
       "  ('Median', {}, '2847'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '112443'),\n",
       "  ('Min', {}, '75'),\n",
       "  ('Max', {}, '107976'),\n",
       "  ('Mean', {}, '1986.65'),\n",
       "  ('Median', {}, '1461'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '37904'),\n",
       "  ('Min', {}, '33'),\n",
       "  ('Max', {}, '2473592'),\n",
       "  ('Mean', {}, '45183.9'),\n",
       "  ('Median', {}, '14182'),\n",
       "  ('LengthStats', {'category': 'RNase_MRP_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '1'),\n",
       "  ('Min', {}, '277'),\n",
       "  ('Max', {}, '277'),\n",
       "  ('Mean', {}, '277'),\n",
       "  ('Median', {}, '277'),\n",
       "  ('LengthStats', {'category': 'RNase_P_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '1'),\n",
       "  ('Min', {}, '341'),\n",
       "  ('Max', {}, '341'),\n",
       "  ('Mean', {}, '341'),\n",
       "  ('Median', {}, '341'),\n",
       "  ('LengthStats', {'category': 'Single-exon transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '1800'),\n",
       "  ('Min', {}, '180'),\n",
       "  ('Max', {}, '91671'),\n",
       "  ('Mean', {}, '2035.75'),\n",
       "  ('Median', {}, '1432'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '1403'),\n",
       "  ('Min', {}, '180'),\n",
       "  ('Max', {}, '12344'),\n",
       "  ('Mean', {}, '1839.05'),\n",
       "  ('Median', {}, '1198'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in non-coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '397'),\n",
       "  ('Min', {}, '200'),\n",
       "  ('Max', {}, '91671'),\n",
       "  ('Mean', {}, '2730.89'),\n",
       "  ('Median', {}, '2064'),\n",
       "  ('LengthStats', {'category': 'Y_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '113'),\n",
       "  ('Mean', {}, '98.75'),\n",
       "  ('Median', {}, '102'),\n",
       "  ('LengthStats', {'category': 'antisense_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '22'),\n",
       "  ('Min', {}, '255'),\n",
       "  ('Max', {}, '37027'),\n",
       "  ('Mean', {}, '4169.59'),\n",
       "  ('Median', {}, '1662'),\n",
       "  ('LengthStats', {'category': 'guide_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '29'),\n",
       "  ('Min', {}, '125'),\n",
       "  ('Max', {}, '421'),\n",
       "  ('Mean', {}, '206.793'),\n",
       "  ('Median', {}, '148'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '27344'),\n",
       "  ('Min', {}, '99'),\n",
       "  ('Max', {}, '91671'),\n",
       "  ('Mean', {}, '2029.63'),\n",
       "  ('Median', {}, '1376'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '112430'),\n",
       "  ('Min', {}, '153'),\n",
       "  ('Max', {}, '109224'),\n",
       "  ('Mean', {}, '4072.68'),\n",
       "  ('Median', {}, '3313'),\n",
       "  ('LengthStats', {'category': 'miRNA'}, '\\n      '),\n",
       "  ('Count', {}, '3173'),\n",
       "  ('Min', {}, '16'),\n",
       "  ('Max', {}, '28'),\n",
       "  ('Mean', {}, '21.6306'),\n",
       "  ('Median', {}, '22'),\n",
       "  ('LengthStats', {'category': 'misc_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '15218'),\n",
       "  ('Min', {}, '158'),\n",
       "  ('Max', {}, '25390'),\n",
       "  ('Mean', {}, '3605.73'),\n",
       "  ('Median', {}, '2981'),\n",
       "  ('LengthStats', {'category': 'premsg'}, '\\n      '),\n",
       "  ('Count', {}, '1913'),\n",
       "  ('Min', {}, '41'),\n",
       "  ('Max', {}, '180'),\n",
       "  ('Mean', {}, '81.9226'),\n",
       "  ('Median', {}, '80'),\n",
       "  ('LengthStats', {'category': 'rRNA'}, '\\n      '),\n",
       "  ('Count', {}, '39'),\n",
       "  ('Min', {}, '121'),\n",
       "  ('Max', {}, '13373'),\n",
       "  ('Mean', {}, '2736.56'),\n",
       "  ('Median', {}, '157'),\n",
       "  ('LengthStats', {'category': 'scRNA'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '58'),\n",
       "  ('Max', {}, '299'),\n",
       "  ('Mean', {}, '238.75'),\n",
       "  ('Median', {}, '299'),\n",
       "  ('LengthStats', {'category': 'snRNA'}, '\\n      '),\n",
       "  ('Count', {}, '63'),\n",
       "  ('Min', {}, '63'),\n",
       "  ('Max', {}, '332'),\n",
       "  ('Mean', {}, '134.921'),\n",
       "  ('Median', {}, '121'),\n",
       "  ('LengthStats', {'category': 'snoRNA'}, '\\n      '),\n",
       "  ('Count', {}, '541'),\n",
       "  ('Min', {}, '33'),\n",
       "  ('Max', {}, '250'),\n",
       "  ('Mean', {}, '105.649'),\n",
       "  ('Median', {}, '94'),\n",
       "  ('LengthStats', {'category': 'tRNA'}, '\\n      '),\n",
       "  ('Count', {}, '597'),\n",
       "  ('Min', {}, '59'),\n",
       "  ('Max', {}, '84'),\n",
       "  ('Mean', {}, '73.7672'),\n",
       "  ('Median', {}, '73'),\n",
       "  ('LengthStats', {'category': 'telomerase_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '1'),\n",
       "  ('Min', {}, '451'),\n",
       "  ('Max', {}, '451'),\n",
       "  ('Mean', {}, '451'),\n",
       "  ('Median', {}, '451'),\n",
       "  ('LengthStats', {'category': 'vault_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '88'),\n",
       "  ('Max', {}, '108'),\n",
       "  ('Mean', {}, '95.5'),\n",
       "  ('Median', {}, '98')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000001405.39', 'Type': 'full assembly'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '113'),\n",
       "  ('Mean', {}, '4.181'),\n",
       "  ('Median', {}, '2'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '363'),\n",
       "  ('Mean', {}, '11.6507'),\n",
       "  ('Median', {}, '9'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '411273'),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '91671'),\n",
       "  ('Mean', {}, '441.791'),\n",
       "  ('Median', {}, '149'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '352207'),\n",
       "  ('Min', {}, '26'),\n",
       "  ('Max', {}, '1160411'),\n",
       "  ('Mean', {}, '8185.69'),\n",
       "  ('Median', {}, '1873'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '159471'),\n",
       "  ('Min', {}, '16'),\n",
       "  ('Max', {}, '109224'),\n",
       "  ('Mean', {}, '3565.83'),\n",
       "  ('Median', {}, '2847'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '112443'),\n",
       "  ('Min', {}, '75'),\n",
       "  ('Max', {}, '107976'),\n",
       "  ('Mean', {}, '1986.65'),\n",
       "  ('Median', {}, '1461'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '327126'),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '33290'),\n",
       "  ('Mean', {}, '413.104'),\n",
       "  ('Median', {}, '143'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '156519'),\n",
       "  ('Min', {}, '2'),\n",
       "  ('Max', {}, '91671'),\n",
       "  ('Mean', {}, '417.327'),\n",
       "  ('Median', {}, '147'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '37904'),\n",
       "  ('Min', {}, '33'),\n",
       "  ('Max', {}, '2473592'),\n",
       "  ('Mean', {}, '45183.9'),\n",
       "  ('Median', {}, '14182'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '288015'),\n",
       "  ('Min', {}, '26'),\n",
       "  ('Max', {}, '1160411'),\n",
       "  ('Mean', {}, '7752.87'),\n",
       "  ('Median', {}, '1759'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '132792'),\n",
       "  ('Min', {}, '28'),\n",
       "  ('Max', {}, '773192'),\n",
       "  ('Mean', {}, '7841.45'),\n",
       "  ('Median', {}, '1965'),\n",
       "  ('LengthStats', {'category': 'RNase_MRP_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '1'),\n",
       "  ('Min', {}, '277'),\n",
       "  ('Max', {}, '277'),\n",
       "  ('Mean', {}, '277'),\n",
       "  ('Median', {}, '277'),\n",
       "  ('LengthStats', {'category': 'RNase_P_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '1'),\n",
       "  ('Min', {}, '341'),\n",
       "  ('Max', {}, '341'),\n",
       "  ('Mean', {}, '341'),\n",
       "  ('Median', {}, '341'),\n",
       "  ('LengthStats', {'category': 'Single-exon transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '1800'),\n",
       "  ('Min', {}, '180'),\n",
       "  ('Max', {}, '91671'),\n",
       "  ('Mean', {}, '2035.75'),\n",
       "  ('Median', {}, '1432'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '1403'),\n",
       "  ('Min', {}, '180'),\n",
       "  ('Max', {}, '12344'),\n",
       "  ('Mean', {}, '1839.05'),\n",
       "  ('Median', {}, '1198'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in non-coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '397'),\n",
       "  ('Min', {}, '200'),\n",
       "  ('Max', {}, '91671'),\n",
       "  ('Mean', {}, '2730.89'),\n",
       "  ('Median', {}, '2064'),\n",
       "  ('LengthStats', {'category': 'Y_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '113'),\n",
       "  ('Mean', {}, '98.75'),\n",
       "  ('Median', {}, '102'),\n",
       "  ('LengthStats', {'category': 'antisense_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '22'),\n",
       "  ('Min', {}, '255'),\n",
       "  ('Max', {}, '37027'),\n",
       "  ('Mean', {}, '4169.59'),\n",
       "  ('Median', {}, '1662'),\n",
       "  ('LengthStats', {'category': 'guide_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '29'),\n",
       "  ('Min', {}, '125'),\n",
       "  ('Max', {}, '421'),\n",
       "  ('Mean', {}, '206.793'),\n",
       "  ('Median', {}, '148'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '27344'),\n",
       "  ('Min', {}, '99'),\n",
       "  ('Max', {}, '91671'),\n",
       "  ('Mean', {}, '2029.63'),\n",
       "  ('Median', {}, '1376'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '112430'),\n",
       "  ('Min', {}, '153'),\n",
       "  ('Max', {}, '109224'),\n",
       "  ('Mean', {}, '4072.68'),\n",
       "  ('Median', {}, '3313'),\n",
       "  ('LengthStats', {'category': 'miRNA'}, '\\n      '),\n",
       "  ('Count', {}, '3173'),\n",
       "  ('Min', {}, '16'),\n",
       "  ('Max', {}, '28'),\n",
       "  ('Mean', {}, '21.6306'),\n",
       "  ('Median', {}, '22'),\n",
       "  ('LengthStats', {'category': 'misc_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '15218'),\n",
       "  ('Min', {}, '158'),\n",
       "  ('Max', {}, '25390'),\n",
       "  ('Mean', {}, '3605.73'),\n",
       "  ('Median', {}, '2981'),\n",
       "  ('LengthStats', {'category': 'premsg'}, '\\n      '),\n",
       "  ('Count', {}, '1913'),\n",
       "  ('Min', {}, '41'),\n",
       "  ('Max', {}, '180'),\n",
       "  ('Mean', {}, '81.9226'),\n",
       "  ('Median', {}, '80'),\n",
       "  ('LengthStats', {'category': 'rRNA'}, '\\n      '),\n",
       "  ('Count', {}, '39'),\n",
       "  ('Min', {}, '121'),\n",
       "  ('Max', {}, '13373'),\n",
       "  ('Mean', {}, '2736.56'),\n",
       "  ('Median', {}, '157'),\n",
       "  ('LengthStats', {'category': 'scRNA'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '58'),\n",
       "  ('Max', {}, '299'),\n",
       "  ('Mean', {}, '238.75'),\n",
       "  ('Median', {}, '299'),\n",
       "  ('LengthStats', {'category': 'snRNA'}, '\\n      '),\n",
       "  ('Count', {}, '63'),\n",
       "  ('Min', {}, '63'),\n",
       "  ('Max', {}, '332'),\n",
       "  ('Mean', {}, '134.921'),\n",
       "  ('Median', {}, '121'),\n",
       "  ('LengthStats', {'category': 'snoRNA'}, '\\n      '),\n",
       "  ('Count', {}, '541'),\n",
       "  ('Min', {}, '33'),\n",
       "  ('Max', {}, '250'),\n",
       "  ('Mean', {}, '105.649'),\n",
       "  ('Median', {}, '94'),\n",
       "  ('LengthStats', {'category': 'tRNA'}, '\\n      '),\n",
       "  ('Count', {}, '597'),\n",
       "  ('Min', {}, '59'),\n",
       "  ('Max', {}, '84'),\n",
       "  ('Mean', {}, '73.7672'),\n",
       "  ('Median', {}, '73'),\n",
       "  ('LengthStats', {'category': 'telomerase_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '1'),\n",
       "  ('Min', {}, '451'),\n",
       "  ('Max', {}, '451'),\n",
       "  ('Mean', {}, '451'),\n",
       "  ('Median', {}, '451'),\n",
       "  ('LengthStats', {'category': 'vault_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '88'),\n",
       "  ('Max', {}, '108'),\n",
       "  ('Mean', {}, '95.5'),\n",
       "  ('Median', {}, '98')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000001305.15', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '113'),\n",
       "  ('Mean', {}, '4.18929'),\n",
       "  ('Median', {}, '2'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '363'),\n",
       "  ('Mean', {}, '11.7428'),\n",
       "  ('Median', {}, '9'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '378131'),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '91671'),\n",
       "  ('Mean', {}, '451.678'),\n",
       "  ('Median', {}, '149'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '323298'),\n",
       "  ('Min', {}, '26'),\n",
       "  ('Max', {}, '1160411'),\n",
       "  ('Mean', {}, '8626.91'),\n",
       "  ('Median', {}, '2004'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '157336'),\n",
       "  ('Min', {}, '16'),\n",
       "  ('Max', {}, '109224'),\n",
       "  ('Mean', {}, '3594.62'),\n",
       "  ('Median', {}, '2873'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '112169'),\n",
       "  ('Min', {}, '75'),\n",
       "  ('Max', {}, '107976'),\n",
       "  ('Mean', {}, '1988.61'),\n",
       "  ('Median', {}, '1461'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '299699'),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '33290'),\n",
       "  ('Mean', {}, '424.016'),\n",
       "  ('Median', {}, '144'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '147954'),\n",
       "  ('Min', {}, '2'),\n",
       "  ('Max', {}, '91671'),\n",
       "  ('Mean', {}, '418.11'),\n",
       "  ('Median', {}, '146'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '37660'),\n",
       "  ('Min', {}, '33'),\n",
       "  ('Max', {}, '2473592'),\n",
       "  ('Mean', {}, '45795.7'),\n",
       "  ('Median', {}, '14350'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '263421'),\n",
       "  ('Min', {}, '26'),\n",
       "  ('Max', {}, '1160411'),\n",
       "  ('Mean', {}, '8220.54'),\n",
       "  ('Median', {}, '1904'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '125972'),\n",
       "  ('Min', {}, '28'),\n",
       "  ('Max', {}, '773192'),\n",
       "  ('Mean', {}, '8006.62'),\n",
       "  ('Median', {}, '2015'),\n",
       "  ('LengthStats', {'category': 'RNase_MRP_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '1'),\n",
       "  ('Min', {}, '277'),\n",
       "  ('Max', {}, '277'),\n",
       "  ('Mean', {}, '277'),\n",
       "  ('Median', {}, '277'),\n",
       "  ('LengthStats', {'category': 'RNase_P_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '1'),\n",
       "  ('Min', {}, '341'),\n",
       "  ('Max', {}, '341'),\n",
       "  ('Mean', {}, '341'),\n",
       "  ('Median', {}, '341'),\n",
       "  ('LengthStats', {'category': 'Single-exon transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '1769'),\n",
       "  ('Min', {}, '180'),\n",
       "  ('Max', {}, '91671'),\n",
       "  ('Mean', {}, '2041.42'),\n",
       "  ('Median', {}, '1434'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '1376'),\n",
       "  ('Min', {}, '180'),\n",
       "  ('Max', {}, '12344'),\n",
       "  ('Mean', {}, '1844.77'),\n",
       "  ('Median', {}, '1201'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in non-coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '393'),\n",
       "  ('Min', {}, '200'),\n",
       "  ('Max', {}, '91671'),\n",
       "  ('Mean', {}, '2729.97'),\n",
       "  ('Median', {}, '2064'),\n",
       "  ('LengthStats', {'category': 'Y_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '113'),\n",
       "  ('Mean', {}, '98.75'),\n",
       "  ('Median', {}, '102'),\n",
       "  ('LengthStats', {'category': 'antisense_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '22'),\n",
       "  ('Min', {}, '255'),\n",
       "  ('Max', {}, '37027'),\n",
       "  ('Mean', {}, '4169.59'),\n",
       "  ('Median', {}, '1662'),\n",
       "  ('LengthStats', {'category': 'guide_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '29'),\n",
       "  ('Min', {}, '125'),\n",
       "  ('Max', {}, '421'),\n",
       "  ('Mean', {}, '206.793'),\n",
       "  ('Median', {}, '148'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '26085'),\n",
       "  ('Min', {}, '99'),\n",
       "  ('Max', {}, '91671'),\n",
       "  ('Mean', {}, '2044.77'),\n",
       "  ('Median', {}, '1388'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '112169'),\n",
       "  ('Min', {}, '153'),\n",
       "  ('Max', {}, '109224'),\n",
       "  ('Mean', {}, '4077.3'),\n",
       "  ('Median', {}, '3319'),\n",
       "  ('LengthStats', {'category': 'miRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2875'),\n",
       "  ('Min', {}, '16'),\n",
       "  ('Max', {}, '28'),\n",
       "  ('Mean', {}, '21.6303'),\n",
       "  ('Median', {}, '22'),\n",
       "  ('LengthStats', {'category': 'misc_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '15084'),\n",
       "  ('Min', {}, '158'),\n",
       "  ('Max', {}, '25390'),\n",
       "  ('Mean', {}, '3614.11'),\n",
       "  ('Median', {}, '2987'),\n",
       "  ('LengthStats', {'category': 'premsg'}, '\\n      '),\n",
       "  ('Count', {}, '1913'),\n",
       "  ('Min', {}, '41'),\n",
       "  ('Max', {}, '180'),\n",
       "  ('Mean', {}, '81.9226'),\n",
       "  ('Median', {}, '80'),\n",
       "  ('LengthStats', {'category': 'rRNA'}, '\\n      '),\n",
       "  ('Count', {}, '37'),\n",
       "  ('Min', {}, '121'),\n",
       "  ('Max', {}, '13373'),\n",
       "  ('Mean', {}, '2816.57'),\n",
       "  ('Median', {}, '157'),\n",
       "  ('LengthStats', {'category': 'scRNA'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '58'),\n",
       "  ('Max', {}, '299'),\n",
       "  ('Mean', {}, '238.75'),\n",
       "  ('Median', {}, '299'),\n",
       "  ('LengthStats', {'category': 'snRNA'}, '\\n      '),\n",
       "  ('Count', {}, '63'),\n",
       "  ('Min', {}, '63'),\n",
       "  ('Max', {}, '332'),\n",
       "  ('Mean', {}, '134.921'),\n",
       "  ('Median', {}, '121'),\n",
       "  ('LengthStats', {'category': 'snoRNA'}, '\\n      '),\n",
       "  ('Count', {}, '541'),\n",
       "  ('Min', {}, '33'),\n",
       "  ('Max', {}, '250'),\n",
       "  ('Mean', {}, '105.649'),\n",
       "  ('Median', {}, '94'),\n",
       "  ('LengthStats', {'category': 'tRNA'}, '\\n      '),\n",
       "  ('Count', {}, '416'),\n",
       "  ('Min', {}, '71'),\n",
       "  ('Max', {}, '84'),\n",
       "  ('Mean', {}, '73.9423'),\n",
       "  ('Median', {}, '73'),\n",
       "  ('LengthStats', {'category': 'telomerase_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '1'),\n",
       "  ('Min', {}, '451'),\n",
       "  ('Max', {}, '451'),\n",
       "  ('Mean', {}, '451'),\n",
       "  ('Median', {}, '451'),\n",
       "  ('LengthStats', {'category': 'vault_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '88'),\n",
       "  ('Max', {}, '108'),\n",
       "  ('Mean', {}, '95.5'),\n",
       "  ('Median', {}, '98')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000001405.39', 'Type': 'grouped alt loci'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '36'),\n",
       "  ('Mean', {}, '2.14198'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '66'),\n",
       "  ('Mean', {}, '7.05033'),\n",
       "  ('Median', {}, '5'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '24956'),\n",
       "  ('Min', {}, '2'),\n",
       "  ('Max', {}, '18726'),\n",
       "  ('Mean', {}, '316.512'),\n",
       "  ('Median', {}, '143'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '21748'),\n",
       "  ('Min', {}, '30'),\n",
       "  ('Max', {}, '232263'),\n",
       "  ('Mean', {}, '2901.07'),\n",
       "  ('Median', {}, '742'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '3737'),\n",
       "  ('Min', {}, '17'),\n",
       "  ('Max', {}, '22824'),\n",
       "  ('Mean', {}, '2135.54'),\n",
       "  ('Median', {}, '1649'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '1917'),\n",
       "  ('Min', {}, '75'),\n",
       "  ('Max', {}, '22257'),\n",
       "  ('Mean', {}, '1561.41'),\n",
       "  ('Median', {}, '1182'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '20697'),\n",
       "  ('Min', {}, '6'),\n",
       "  ('Max', {}, '18726'),\n",
       "  ('Mean', {}, '278.942'),\n",
       "  ('Median', {}, '135'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '6523'),\n",
       "  ('Min', {}, '2'),\n",
       "  ('Max', {}, '9896'),\n",
       "  ('Mean', {}, '396.419'),\n",
       "  ('Median', {}, '162'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '1517'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '1296601'),\n",
       "  ('Mean', {}, '22661.2'),\n",
       "  ('Median', {}, '7270'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '18560'),\n",
       "  ('Min', {}, '33'),\n",
       "  ('Max', {}, '197352'),\n",
       "  ('Mean', {}, '2376.67'),\n",
       "  ('Median', {}, '647'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '5178'),\n",
       "  ('Min', {}, '30'),\n",
       "  ('Max', {}, '232263'),\n",
       "  ('Mean', {}, '4504.26'),\n",
       "  ('Median', {}, '983'),\n",
       "  ('LengthStats', {'category': 'Single-exon transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '159'),\n",
       "  ('Min', {}, '252'),\n",
       "  ('Max', {}, '8155'),\n",
       "  ('Mean', {}, '1895.09'),\n",
       "  ('Median', {}, '1100'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '127'),\n",
       "  ('Min', {}, '445'),\n",
       "  ('Max', {}, '7528'),\n",
       "  ('Mean', {}, '1752.8'),\n",
       "  ('Median', {}, '1041'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in non-coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '32'),\n",
       "  ('Min', {}, '252'),\n",
       "  ('Max', {}, '8155'),\n",
       "  ('Mean', {}, '2459.81'),\n",
       "  ('Median', {}, '2175'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '1204'),\n",
       "  ('Min', {}, '99'),\n",
       "  ('Max', {}, '11661'),\n",
       "  ('Mean', {}, '1654.36'),\n",
       "  ('Median', {}, '1162'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '1917'),\n",
       "  ('Min', {}, '264'),\n",
       "  ('Max', {}, '22824'),\n",
       "  ('Mean', {}, '2828.26'),\n",
       "  ('Median', {}, '2252'),\n",
       "  ('LengthStats', {'category': 'miRNA'}, '\\n      '),\n",
       "  ('Count', {}, '215'),\n",
       "  ('Min', {}, '17'),\n",
       "  ('Max', {}, '25'),\n",
       "  ('Mean', {}, '21.6326'),\n",
       "  ('Median', {}, '22'),\n",
       "  ('LengthStats', {'category': 'misc_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '232'),\n",
       "  ('Min', {}, '207'),\n",
       "  ('Max', {}, '6314'),\n",
       "  ('Mean', {}, '2368.12'),\n",
       "  ('Median', {}, '2216'),\n",
       "  ('LengthStats', {'category': 'premsg'}, '\\n      '),\n",
       "  ('Count', {}, '71'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '153'),\n",
       "  ('Mean', {}, '80.7324'),\n",
       "  ('Median', {}, '77'),\n",
       "  ('LengthStats', {'category': 'snRNA'}, '\\n      '),\n",
       "  ('Count', {}, '1'),\n",
       "  ('Min', {}, '165'),\n",
       "  ('Max', {}, '165'),\n",
       "  ('Mean', {}, '165'),\n",
       "  ('Median', {}, '165'),\n",
       "  ('LengthStats', {'category': 'snoRNA'}, '\\n      '),\n",
       "  ('Count', {}, '11'),\n",
       "  ('Min', {}, '63'),\n",
       "  ('Max', {}, '175'),\n",
       "  ('Mean', {}, '96.9091'),\n",
       "  ('Median', {}, '78'),\n",
       "  ('LengthStats', {'category': 'tRNA'}, '\\n      '),\n",
       "  ('Count', {}, '157'),\n",
       "  ('Min', {}, '72'),\n",
       "  ('Max', {}, '83'),\n",
       "  ('Mean', {}, '73.9299'),\n",
       "  ('Median', {}, '73')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000001315.2', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '21'),\n",
       "  ('Mean', {}, '2.13567'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '60'),\n",
       "  ('Mean', {}, '7.20447'),\n",
       "  ('Median', {}, '5'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '8112'),\n",
       "  ('Min', {}, '2'),\n",
       "  ('Max', {}, '18726'),\n",
       "  ('Mean', {}, '360.229'),\n",
       "  ('Median', {}, '146'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '6903'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '232263'),\n",
       "  ('Mean', {}, '4463.71'),\n",
       "  ('Median', {}, '1276'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '2543'),\n",
       "  ('Min', {}, '17'),\n",
       "  ('Max', {}, '22824'),\n",
       "  ('Mean', {}, '2349.12'),\n",
       "  ('Median', {}, '1890'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '1505'),\n",
       "  ('Min', {}, '75'),\n",
       "  ('Max', {}, '22257'),\n",
       "  ('Mean', {}, '1550.77'),\n",
       "  ('Median', {}, '1233'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '6114'),\n",
       "  ('Min', {}, '6'),\n",
       "  ('Max', {}, '18726'),\n",
       "  ('Mean', {}, '308.109'),\n",
       "  ('Median', {}, '137'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '2695'),\n",
       "  ('Min', {}, '2'),\n",
       "  ('Max', {}, '9896'),\n",
       "  ('Mean', {}, '441.574'),\n",
       "  ('Median', {}, '171'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '1195'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '1296601'),\n",
       "  ('Mean', {}, '23813.6'),\n",
       "  ('Median', {}, '7853'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '5448'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '176607'),\n",
       "  ('Mean', {}, '4070.42'),\n",
       "  ('Median', {}, '1211'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '2067'),\n",
       "  ('Min', {}, '54'),\n",
       "  ('Max', {}, '232263'),\n",
       "  ('Mean', {}, '5248.83'),\n",
       "  ('Median', {}, '1413'),\n",
       "  ('LengthStats', {'category': 'Single-exon transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '143'),\n",
       "  ('Min', {}, '252'),\n",
       "  ('Max', {}, '8155'),\n",
       "  ('Mean', {}, '1922.1'),\n",
       "  ('Median', {}, '1096'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '113'),\n",
       "  ('Min', {}, '445'),\n",
       "  ('Max', {}, '7528'),\n",
       "  ('Mean', {}, '1757.2'),\n",
       "  ('Median', {}, '1039'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in non-coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '30'),\n",
       "  ('Min', {}, '252'),\n",
       "  ('Max', {}, '8155'),\n",
       "  ('Mean', {}, '2543.2'),\n",
       "  ('Median', {}, '2435'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '763'),\n",
       "  ('Min', {}, '128'),\n",
       "  ('Max', {}, '11661'),\n",
       "  ('Mean', {}, '1726.92'),\n",
       "  ('Median', {}, '1230'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '1505'),\n",
       "  ('Min', {}, '264'),\n",
       "  ('Max', {}, '22824'),\n",
       "  ('Mean', {}, '2853.48'),\n",
       "  ('Median', {}, '2304'),\n",
       "  ('LengthStats', {'category': 'miRNA'}, '\\n      '),\n",
       "  ('Count', {}, '80'),\n",
       "  ('Min', {}, '17'),\n",
       "  ('Max', {}, '25'),\n",
       "  ('Mean', {}, '21.7375'),\n",
       "  ('Median', {}, '22'),\n",
       "  ('LengthStats', {'category': 'misc_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '167'),\n",
       "  ('Min', {}, '207'),\n",
       "  ('Max', {}, '6314'),\n",
       "  ('Mean', {}, '2141.28'),\n",
       "  ('Median', {}, '2006'),\n",
       "  ('LengthStats', {'category': 'premsg'}, '\\n      '),\n",
       "  ('Count', {}, '59'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '153'),\n",
       "  ('Mean', {}, '79.4407'),\n",
       "  ('Median', {}, '77'),\n",
       "  ('LengthStats', {'category': 'snRNA'}, '\\n      '),\n",
       "  ('Count', {}, '1'),\n",
       "  ('Min', {}, '165'),\n",
       "  ('Max', {}, '165'),\n",
       "  ('Mean', {}, '165'),\n",
       "  ('Median', {}, '165'),\n",
       "  ('LengthStats', {'category': 'snoRNA'}, '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '63'),\n",
       "  ('Max', {}, '175'),\n",
       "  ('Mean', {}, '95.1111'),\n",
       "  ('Median', {}, '77'),\n",
       "  ('LengthStats', {'category': 'tRNA'}, '\\n      '),\n",
       "  ('Count', {}, '18'),\n",
       "  ('Min', {}, '72'),\n",
       "  ('Max', {}, '83'),\n",
       "  ('Mean', {}, '73.7222'),\n",
       "  ('Median', {}, '73')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000001325.3', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '36'),\n",
       "  ('Mean', {}, '2.10694'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '66'),\n",
       "  ('Mean', {}, '7.41941'),\n",
       "  ('Median', {}, '5'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '3876'),\n",
       "  ('Min', {}, '6'),\n",
       "  ('Max', {}, '14433'),\n",
       "  ('Mean', {}, '355.506'),\n",
       "  ('Median', {}, '142'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '3336'),\n",
       "  ('Min', {}, '33'),\n",
       "  ('Max', {}, '197352'),\n",
       "  ('Mean', {}, '3610.37'),\n",
       "  ('Median', {}, '656'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '1170'),\n",
       "  ('Min', {}, '18'),\n",
       "  ('Max', {}, '16756'),\n",
       "  ('Mean', {}, '2275.08'),\n",
       "  ('Median', {}, '1870'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '700'),\n",
       "  ('Min', {}, '180'),\n",
       "  ('Max', {}, '16239'),\n",
       "  ('Mean', {}, '1541.12'),\n",
       "  ('Median', {}, '1044'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3119'),\n",
       "  ('Min', {}, '6'),\n",
       "  ('Max', {}, '14433'),\n",
       "  ('Mean', {}, '305.422'),\n",
       "  ('Median', {}, '134'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '1087'),\n",
       "  ('Min', {}, '12'),\n",
       "  ('Max', {}, '7568'),\n",
       "  ('Mean', {}, '456.215'),\n",
       "  ('Median', {}, '165'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '571'),\n",
       "  ('Min', {}, '61'),\n",
       "  ('Max', {}, '460024'),\n",
       "  ('Mean', {}, '19216.8'),\n",
       "  ('Median', {}, '5361'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '2766'),\n",
       "  ('Min', {}, '33'),\n",
       "  ('Max', {}, '197352'),\n",
       "  ('Mean', {}, '2895.93'),\n",
       "  ('Median', {}, '565'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '862'),\n",
       "  ('Min', {}, '71'),\n",
       "  ('Max', {}, '124245'),\n",
       "  ('Mean', {}, '5756.43'),\n",
       "  ('Median', {}, '1182'),\n",
       "  ('LengthStats', {'category': 'Single-exon transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '65'),\n",
       "  ('Min', {}, '445'),\n",
       "  ('Max', {}, '4834'),\n",
       "  ('Mean', {}, '1556.38'),\n",
       "  ('Median', {}, '1075'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '56'),\n",
       "  ('Min', {}, '445'),\n",
       "  ('Max', {}, '4834'),\n",
       "  ('Mean', {}, '1451.48'),\n",
       "  ('Median', {}, '1048'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in non-coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '770'),\n",
       "  ('Max', {}, '4805'),\n",
       "  ('Mean', {}, '2209.11'),\n",
       "  ('Median', {}, '2083'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '312'),\n",
       "  ('Min', {}, '99'),\n",
       "  ('Max', {}, '9818'),\n",
       "  ('Mean', {}, '1756.58'),\n",
       "  ('Median', {}, '1219'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '700'),\n",
       "  ('Min', {}, '331'),\n",
       "  ('Max', {}, '16756'),\n",
       "  ('Mean', {}, '2663.46'),\n",
       "  ('Median', {}, '2131'),\n",
       "  ('LengthStats', {'category': 'miRNA'}, '\\n      '),\n",
       "  ('Count', {}, '30'),\n",
       "  ('Min', {}, '18'),\n",
       "  ('Max', {}, '23'),\n",
       "  ('Mean', {}, '21.4667'),\n",
       "  ('Median', {}, '22'),\n",
       "  ('LengthStats', {'category': 'misc_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '86'),\n",
       "  ('Min', {}, '310'),\n",
       "  ('Max', {}, '6314'),\n",
       "  ('Mean', {}, '2855.48'),\n",
       "  ('Median', {}, '2746'),\n",
       "  ('LengthStats', {'category': 'premsg'}, '\\n      '),\n",
       "  ('Count', {}, '19'),\n",
       "  ('Min', {}, '61'),\n",
       "  ('Max', {}, '110'),\n",
       "  ('Mean', {}, '83.7895'),\n",
       "  ('Median', {}, '86'),\n",
       "  ('LengthStats', {'category': 'snoRNA'}, '\\n      '),\n",
       "  ('Count', {}, '6'),\n",
       "  ('Min', {}, '63'),\n",
       "  ('Max', {}, '132'),\n",
       "  ('Mean', {}, '81.6667'),\n",
       "  ('Median', {}, '77'),\n",
       "  ('LengthStats', {'category': 'tRNA'}, '\\n      '),\n",
       "  ('Count', {}, '36'),\n",
       "  ('Min', {}, '72'),\n",
       "  ('Max', {}, '83'),\n",
       "  ('Mean', {}, '73.9444'),\n",
       "  ('Median', {}, '73')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000001335.2', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '13'),\n",
       "  ('Mean', {}, '2.08392'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '66'),\n",
       "  ('Mean', {}, '8.05705'),\n",
       "  ('Median', {}, '5'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '2281'),\n",
       "  ('Min', {}, '6'),\n",
       "  ('Max', {}, '8265'),\n",
       "  ('Mean', {}, '301.163'),\n",
       "  ('Median', {}, '143'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '2003'),\n",
       "  ('Min', {}, '48'),\n",
       "  ('Max', {}, '110063'),\n",
       "  ('Mean', {}, '1855.19'),\n",
       "  ('Median', {}, '433'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '619'),\n",
       "  ('Min', {}, '20'),\n",
       "  ('Max', {}, '16756'),\n",
       "  ('Mean', {}, '2027.55'),\n",
       "  ('Median', {}, '1675'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '410'),\n",
       "  ('Min', {}, '180'),\n",
       "  ('Max', {}, '16239'),\n",
       "  ('Mean', {}, '1595.51'),\n",
       "  ('Median', {}, '1071'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '1961'),\n",
       "  ('Min', {}, '6'),\n",
       "  ('Max', {}, '8265'),\n",
       "  ('Mean', {}, '273.348'),\n",
       "  ('Median', {}, '138'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '519'),\n",
       "  ('Min', {}, '11'),\n",
       "  ('Max', {}, '7568'),\n",
       "  ('Mean', {}, '366.651'),\n",
       "  ('Median', {}, '154'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '300'),\n",
       "  ('Min', {}, '63'),\n",
       "  ('Max', {}, '152246'),\n",
       "  ('Mean', {}, '11334.9'),\n",
       "  ('Median', {}, '5586'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '1751'),\n",
       "  ('Min', {}, '48'),\n",
       "  ('Max', {}, '35672'),\n",
       "  ('Mean', {}, '1161.36'),\n",
       "  ('Median', {}, '366'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '424'),\n",
       "  ('Min', {}, '71'),\n",
       "  ('Max', {}, '110063'),\n",
       "  ('Mean', {}, '4483.21'),\n",
       "  ('Median', {}, '964'),\n",
       "  ('LengthStats', {'category': 'Single-exon transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '24'),\n",
       "  ('Min', {}, '963'),\n",
       "  ('Max', {}, '2624'),\n",
       "  ('Mean', {}, '1634.71'),\n",
       "  ('Median', {}, '1348'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '18'),\n",
       "  ('Min', {}, '963'),\n",
       "  ('Max', {}, '2517'),\n",
       "  ('Mean', {}, '1531.33'),\n",
       "  ('Median', {}, '1260'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in non-coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '6'),\n",
       "  ('Min', {}, '1180'),\n",
       "  ('Max', {}, '2624'),\n",
       "  ('Mean', {}, '1944.83'),\n",
       "  ('Median', {}, '2083'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '139'),\n",
       "  ('Min', {}, '149'),\n",
       "  ('Max', {}, '8477'),\n",
       "  ('Mean', {}, '1434.51'),\n",
       "  ('Median', {}, '893'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '410'),\n",
       "  ('Min', {}, '402'),\n",
       "  ('Max', {}, '16756'),\n",
       "  ('Mean', {}, '2428.95'),\n",
       "  ('Median', {}, '2059'),\n",
       "  ('LengthStats', {'category': 'miRNA'}, '\\n      '),\n",
       "  ('Count', {}, '20'),\n",
       "  ('Min', {}, '20'),\n",
       "  ('Max', {}, '23'),\n",
       "  ('Mean', {}, '21.65'),\n",
       "  ('Median', {}, '22'),\n",
       "  ('LengthStats', {'category': 'misc_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '30'),\n",
       "  ('Min', {}, '505'),\n",
       "  ('Max', {}, '4543'),\n",
       "  ('Mean', {}, '1928.1'),\n",
       "  ('Median', {}, '1801'),\n",
       "  ('LengthStats', {'category': 'premsg'}, '\\n      '),\n",
       "  ('Count', {}, '11'),\n",
       "  ('Min', {}, '63'),\n",
       "  ('Max', {}, '110'),\n",
       "  ('Mean', {}, '81.8182'),\n",
       "  ('Median', {}, '81'),\n",
       "  ('LengthStats', {'category': 'snoRNA'}, '\\n      '),\n",
       "  ('Count', {}, '6'),\n",
       "  ('Min', {}, '63'),\n",
       "  ('Max', {}, '132'),\n",
       "  ('Mean', {}, '81.6667'),\n",
       "  ('Median', {}, '77'),\n",
       "  ('LengthStats', {'category': 'tRNA'}, '\\n      '),\n",
       "  ('Count', {}, '14'),\n",
       "  ('Min', {}, '72'),\n",
       "  ('Max', {}, '82'),\n",
       "  ('Mean', {}, '73.0714'),\n",
       "  ('Median', {}, '72')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000001345.2', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '13'),\n",
       "  ('Mean', {}, '2.26432'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '66'),\n",
       "  ('Mean', {}, '7.55642'),\n",
       "  ('Median', {}, '5'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '1708'),\n",
       "  ('Min', {}, '6'),\n",
       "  ('Max', {}, '12739'),\n",
       "  ('Mean', {}, '306.721'),\n",
       "  ('Median', {}, '149'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '1492'),\n",
       "  ('Min', {}, '68'),\n",
       "  ('Max', {}, '110356'),\n",
       "  ('Mean', {}, '1996.42'),\n",
       "  ('Median', {}, '535'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '542'),\n",
       "  ('Min', {}, '19'),\n",
       "  ('Max', {}, '16756'),\n",
       "  ('Mean', {}, '1938.63'),\n",
       "  ('Median', {}, '1603'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '366'),\n",
       "  ('Min', {}, '180'),\n",
       "  ('Max', {}, '16239'),\n",
       "  ('Mean', {}, '1458.86'),\n",
       "  ('Median', {}, '1002'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '1463'),\n",
       "  ('Min', {}, '6'),\n",
       "  ('Max', {}, '12739'),\n",
       "  ('Mean', {}, '279.636'),\n",
       "  ('Median', {}, '142'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '432'),\n",
       "  ('Min', {}, '23'),\n",
       "  ('Max', {}, '7568'),\n",
       "  ('Mean', {}, '343.002'),\n",
       "  ('Median', {}, '153'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '247'),\n",
       "  ('Min', {}, '61'),\n",
       "  ('Max', {}, '152594'),\n",
       "  ('Mean', {}, '10004.8'),\n",
       "  ('Median', {}, '5355'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '1293'),\n",
       "  ('Min', {}, '68'),\n",
       "  ('Max', {}, '53017'),\n",
       "  ('Mean', {}, '1237.51'),\n",
       "  ('Median', {}, '453'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '361'),\n",
       "  ('Min', {}, '71'),\n",
       "  ('Max', {}, '110356'),\n",
       "  ('Mean', {}, '4303.32'),\n",
       "  ('Median', {}, '861'),\n",
       "  ('LengthStats', {'category': 'Single-exon transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '20'),\n",
       "  ('Min', {}, '770'),\n",
       "  ('Max', {}, '2624'),\n",
       "  ('Mean', {}, '1417.75'),\n",
       "  ('Median', {}, '1136'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '15'),\n",
       "  ('Min', {}, '963'),\n",
       "  ('Max', {}, '2481'),\n",
       "  ('Mean', {}, '1310.47'),\n",
       "  ('Median', {}, '1075'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in non-coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '5'),\n",
       "  ('Min', {}, '770'),\n",
       "  ('Max', {}, '2624'),\n",
       "  ('Mean', {}, '1739.6'),\n",
       "  ('Median', {}, '1889'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '108'),\n",
       "  ('Min', {}, '99'),\n",
       "  ('Max', {}, '8477'),\n",
       "  ('Mean', {}, '1469.68'),\n",
       "  ('Median', {}, '955'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '366'),\n",
       "  ('Min', {}, '402'),\n",
       "  ('Max', {}, '16756'),\n",
       "  ('Mean', {}, '2290.78'),\n",
       "  ('Median', {}, '1970'),\n",
       "  ('LengthStats', {'category': 'miRNA'}, '\\n      '),\n",
       "  ('Count', {}, '18'),\n",
       "  ('Min', {}, '19'),\n",
       "  ('Max', {}, '23'),\n",
       "  ('Mean', {}, '21.5'),\n",
       "  ('Median', {}, '22'),\n",
       "  ('LengthStats', {'category': 'misc_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '26'),\n",
       "  ('Min', {}, '505'),\n",
       "  ('Max', {}, '6314'),\n",
       "  ('Mean', {}, '1975.19'),\n",
       "  ('Median', {}, '1571'),\n",
       "  ('LengthStats', {'category': 'premsg'}, '\\n      '),\n",
       "  ('Count', {}, '10'),\n",
       "  ('Min', {}, '61'),\n",
       "  ('Max', {}, '110'),\n",
       "  ('Mean', {}, '82.7'),\n",
       "  ('Median', {}, '87'),\n",
       "  ('LengthStats', {'category': 'snoRNA'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '76'),\n",
       "  ('Max', {}, '132'),\n",
       "  ('Mean', {}, '90.75'),\n",
       "  ('Median', {}, '78'),\n",
       "  ('LengthStats', {'category': 'tRNA'}, '\\n      '),\n",
       "  ('Count', {}, '20'),\n",
       "  ('Min', {}, '72'),\n",
       "  ('Max', {}, '83'),\n",
       "  ('Mean', {}, '73.95'),\n",
       "  ('Median', {}, '73')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000001355.2', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '13'),\n",
       "  ('Mean', {}, '2.17004'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '66'),\n",
       "  ('Mean', {}, '8.30224'),\n",
       "  ('Median', {}, '6'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '2015'),\n",
       "  ('Min', {}, '11'),\n",
       "  ('Max', {}, '7195'),\n",
       "  ('Mean', {}, '277.085'),\n",
       "  ('Median', {}, '141'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '1759'),\n",
       "  ('Min', {}, '30'),\n",
       "  ('Max', {}, '39615'),\n",
       "  ('Mean', {}, '1315.9'),\n",
       "  ('Median', {}, '398'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '566'),\n",
       "  ('Min', {}, '20'),\n",
       "  ('Max', {}, '16756'),\n",
       "  ('Mean', {}, '1938.92'),\n",
       "  ('Median', {}, '1632'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '402'),\n",
       "  ('Min', {}, '180'),\n",
       "  ('Max', {}, '16239'),\n",
       "  ('Mean', {}, '1513.62'),\n",
       "  ('Median', {}, '1044'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '1800'),\n",
       "  ('Min', {}, '12'),\n",
       "  ('Max', {}, '7195'),\n",
       "  ('Mean', {}, '261.909'),\n",
       "  ('Median', {}, '138'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '432'),\n",
       "  ('Min', {}, '11'),\n",
       "  ('Max', {}, '3210'),\n",
       "  ('Mean', {}, '295.051'),\n",
       "  ('Median', {}, '143'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '267'),\n",
       "  ('Min', {}, '61'),\n",
       "  ('Max', {}, '152244'),\n",
       "  ('Mean', {}, '9200.34'),\n",
       "  ('Median', {}, '5056'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '1605'),\n",
       "  ('Min', {}, '48'),\n",
       "  ('Max', {}, '36388'),\n",
       "  ('Mean', {}, '1140.18'),\n",
       "  ('Median', {}, '352'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '347'),\n",
       "  ('Min', {}, '30'),\n",
       "  ('Max', {}, '39615'),\n",
       "  ('Mean', {}, '2001.61'),\n",
       "  ('Median', {}, '594'),\n",
       "  ('LengthStats', {'category': 'Single-exon transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '24'),\n",
       "  ('Min', {}, '407'),\n",
       "  ('Max', {}, '2547'),\n",
       "  ('Mean', {}, '1351'),\n",
       "  ('Median', {}, '1136'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '16'),\n",
       "  ('Min', {}, '963'),\n",
       "  ('Max', {}, '2481'),\n",
       "  ('Mean', {}, '1335'),\n",
       "  ('Median', {}, '1106'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in non-coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '8'),\n",
       "  ('Min', {}, '407'),\n",
       "  ('Max', {}, '2547'),\n",
       "  ('Mean', {}, '1383'),\n",
       "  ('Median', {}, '1332'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '89'),\n",
       "  ('Min', {}, '333'),\n",
       "  ('Max', {}, '4070'),\n",
       "  ('Mean', {}, '1238.73'),\n",
       "  ('Median', {}, '940'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '402'),\n",
       "  ('Min', {}, '402'),\n",
       "  ('Max', {}, '16756'),\n",
       "  ('Mean', {}, '2303.87'),\n",
       "  ('Median', {}, '2005'),\n",
       "  ('LengthStats', {'category': 'miRNA'}, '\\n      '),\n",
       "  ('Count', {}, '22'),\n",
       "  ('Min', {}, '20'),\n",
       "  ('Max', {}, '23'),\n",
       "  ('Mean', {}, '21.6818'),\n",
       "  ('Median', {}, '22'),\n",
       "  ('LengthStats', {'category': 'misc_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '29'),\n",
       "  ('Min', {}, '505'),\n",
       "  ('Max', {}, '6314'),\n",
       "  ('Mean', {}, '2024.31'),\n",
       "  ('Median', {}, '1603'),\n",
       "  ('LengthStats', {'category': 'premsg'}, '\\n      '),\n",
       "  ('Count', {}, '12'),\n",
       "  ('Min', {}, '61'),\n",
       "  ('Max', {}, '110'),\n",
       "  ('Mean', {}, '81.6667'),\n",
       "  ('Median', {}, '86'),\n",
       "  ('LengthStats', {'category': 'snoRNA'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '76'),\n",
       "  ('Max', {}, '132'),\n",
       "  ('Mean', {}, '90.75'),\n",
       "  ('Median', {}, '78'),\n",
       "  ('LengthStats', {'category': 'tRNA'}, '\\n      '),\n",
       "  ('Count', {}, '20'),\n",
       "  ('Min', {}, '72'),\n",
       "  ('Max', {}, '83'),\n",
       "  ('Mean', {}, '74.1'),\n",
       "  ('Median', {}, '73')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000001365.3', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '13'),\n",
       "  ('Mean', {}, '2.19923'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '66'),\n",
       "  ('Mean', {}, '8.15505'),\n",
       "  ('Median', {}, '6'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '2083'),\n",
       "  ('Min', {}, '6'),\n",
       "  ('Max', {}, '12739'),\n",
       "  ('Mean', {}, '293.904'),\n",
       "  ('Median', {}, '144'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '1832'),\n",
       "  ('Min', {}, '45'),\n",
       "  ('Max', {}, '110042'),\n",
       "  ('Mean', {}, '1644.74'),\n",
       "  ('Median', {}, '437'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '606'),\n",
       "  ('Min', {}, '20'),\n",
       "  ('Max', {}, '16756'),\n",
       "  ('Mean', {}, '1950.13'),\n",
       "  ('Median', {}, '1621'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '408'),\n",
       "  ('Min', {}, '180'),\n",
       "  ('Max', {}, '16239'),\n",
       "  ('Mean', {}, '1524.63'),\n",
       "  ('Median', {}, '1071'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '1817'),\n",
       "  ('Min', {}, '6'),\n",
       "  ('Max', {}, '12739'),\n",
       "  ('Mean', {}, '266.216'),\n",
       "  ('Median', {}, '138'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '476'),\n",
       "  ('Min', {}, '23'),\n",
       "  ('Max', {}, '7568'),\n",
       "  ('Mean', {}, '347.704'),\n",
       "  ('Median', {}, '154'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '283'),\n",
       "  ('Min', {}, '61'),\n",
       "  ('Max', {}, '152236'),\n",
       "  ('Mean', {}, '9397.71'),\n",
       "  ('Median', {}, '5464'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '1624'),\n",
       "  ('Min', {}, '68'),\n",
       "  ('Max', {}, '35669'),\n",
       "  ('Mean', {}, '1151.79'),\n",
       "  ('Median', {}, '382'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '394'),\n",
       "  ('Min', {}, '45'),\n",
       "  ('Max', {}, '110042'),\n",
       "  ('Mean', {}, '3451.84'),\n",
       "  ('Median', {}, '791'),\n",
       "  ('LengthStats', {'category': 'Single-exon transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '20'),\n",
       "  ('Min', {}, '770'),\n",
       "  ('Max', {}, '2624'),\n",
       "  ('Mean', {}, '1561.95'),\n",
       "  ('Median', {}, '1332'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '14'),\n",
       "  ('Min', {}, '963'),\n",
       "  ('Max', {}, '2517'),\n",
       "  ('Mean', {}, '1525.79'),\n",
       "  ('Median', {}, '1260'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in non-coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '6'),\n",
       "  ('Min', {}, '770'),\n",
       "  ('Max', {}, '2624'),\n",
       "  ('Mean', {}, '1646.33'),\n",
       "  ('Median', {}, '1889'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '118'),\n",
       "  ('Min', {}, '149'),\n",
       "  ('Max', {}, '8477'),\n",
       "  ('Mean', {}, '1466.86'),\n",
       "  ('Median', {}, '940'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '408'),\n",
       "  ('Min', {}, '402'),\n",
       "  ('Max', {}, '16756'),\n",
       "  ('Mean', {}, '2322.43'),\n",
       "  ('Median', {}, '2028'),\n",
       "  ('LengthStats', {'category': 'miRNA'}, '\\n      '),\n",
       "  ('Count', {}, '22'),\n",
       "  ('Min', {}, '20'),\n",
       "  ('Max', {}, '23'),\n",
       "  ('Mean', {}, '21.6818'),\n",
       "  ('Median', {}, '22'),\n",
       "  ('LengthStats', {'category': 'misc_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '30'),\n",
       "  ('Min', {}, '505'),\n",
       "  ('Max', {}, '6314'),\n",
       "  ('Mean', {}, '1951.17'),\n",
       "  ('Median', {}, '1603'),\n",
       "  ('LengthStats', {'category': 'premsg'}, '\\n      '),\n",
       "  ('Count', {}, '12'),\n",
       "  ('Min', {}, '61'),\n",
       "  ('Max', {}, '110'),\n",
       "  ('Mean', {}, '81.6667'),\n",
       "  ('Median', {}, '86'),\n",
       "  ('LengthStats', {'category': 'snoRNA'}, '\\n      '),\n",
       "  ('Count', {}, '6'),\n",
       "  ('Min', {}, '63'),\n",
       "  ('Max', {}, '132'),\n",
       "  ('Mean', {}, '81.6667'),\n",
       "  ('Median', {}, '77'),\n",
       "  ('LengthStats', {'category': 'tRNA'}, '\\n      '),\n",
       "  ('Count', {}, '22'),\n",
       "  ('Min', {}, '72'),\n",
       "  ('Max', {}, '83'),\n",
       "  ('Mean', {}, '74.3182'),\n",
       "  ('Median', {}, '73')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000001375.2', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '13'),\n",
       "  ('Mean', {}, '2.24715'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '65'),\n",
       "  ('Mean', {}, '8.01861'),\n",
       "  ('Median', {}, '6'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '2142'),\n",
       "  ('Min', {}, '6'),\n",
       "  ('Max', {}, '12739'),\n",
       "  ('Mean', {}, '286.005'),\n",
       "  ('Median', {}, '145'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '1890'),\n",
       "  ('Min', {}, '30'),\n",
       "  ('Max', {}, '110010'),\n",
       "  ('Mean', {}, '1795.23'),\n",
       "  ('Median', {}, '425'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '626'),\n",
       "  ('Min', {}, '19'),\n",
       "  ('Max', {}, '16756'),\n",
       "  ('Mean', {}, '1974.92'),\n",
       "  ('Median', {}, '1678'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '420'),\n",
       "  ('Min', {}, '180'),\n",
       "  ('Max', {}, '16239'),\n",
       "  ('Mean', {}, '1455.1'),\n",
       "  ('Median', {}, '1038'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '1852'),\n",
       "  ('Min', {}, '6'),\n",
       "  ('Max', {}, '12739'),\n",
       "  ('Mean', {}, '257.789'),\n",
       "  ('Median', {}, '138'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '564'),\n",
       "  ('Min', {}, '13'),\n",
       "  ('Max', {}, '7568'),\n",
       "  ('Mean', {}, '328.339'),\n",
       "  ('Median', {}, '154'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '290'),\n",
       "  ('Min', {}, '61'),\n",
       "  ('Max', {}, '152236'),\n",
       "  ('Mean', {}, '10236'),\n",
       "  ('Median', {}, '5628'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '1668'),\n",
       "  ('Min', {}, '68'),\n",
       "  ('Max', {}, '52905'),\n",
       "  ('Mean', {}, '1189.78'),\n",
       "  ('Median', {}, '365'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '466'),\n",
       "  ('Min', {}, '30'),\n",
       "  ('Max', {}, '110010'),\n",
       "  ('Mean', {}, '3478.97'),\n",
       "  ('Median', {}, '568'),\n",
       "  ('LengthStats', {'category': 'Single-exon transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '23'),\n",
       "  ('Min', {}, '590'),\n",
       "  ('Max', {}, '4525'),\n",
       "  ('Mean', {}, '1549.43'),\n",
       "  ('Median', {}, '1180'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '17'),\n",
       "  ('Min', {}, '590'),\n",
       "  ('Max', {}, '4525'),\n",
       "  ('Mean', {}, '1515.24'),\n",
       "  ('Median', {}, '1106'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in non-coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '6'),\n",
       "  ('Min', {}, '770'),\n",
       "  ('Max', {}, '2624'),\n",
       "  ('Mean', {}, '1646.33'),\n",
       "  ('Median', {}, '1889'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '121'),\n",
       "  ('Min', {}, '99'),\n",
       "  ('Max', {}, '8477'),\n",
       "  ('Mean', {}, '1400.26'),\n",
       "  ('Median', {}, '940'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '420'),\n",
       "  ('Min', {}, '402'),\n",
       "  ('Max', {}, '16756'),\n",
       "  ('Mean', {}, '2337.2'),\n",
       "  ('Median', {}, '2042'),\n",
       "  ('LengthStats', {'category': 'miRNA'}, '\\n      '),\n",
       "  ('Count', {}, '19'),\n",
       "  ('Min', {}, '19'),\n",
       "  ('Max', {}, '23'),\n",
       "  ('Mean', {}, '21.4737'),\n",
       "  ('Median', {}, '22'),\n",
       "  ('LengthStats', {'category': 'misc_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '36'),\n",
       "  ('Min', {}, '505'),\n",
       "  ('Max', {}, '6314'),\n",
       "  ('Mean', {}, '2293.11'),\n",
       "  ('Median', {}, '1960'),\n",
       "  ('LengthStats', {'category': 'premsg'}, '\\n      '),\n",
       "  ('Count', {}, '11'),\n",
       "  ('Min', {}, '61'),\n",
       "  ('Max', {}, '107'),\n",
       "  ('Mean', {}, '82.8182'),\n",
       "  ('Median', {}, '86'),\n",
       "  ('LengthStats', {'category': 'snoRNA'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '76'),\n",
       "  ('Max', {}, '132'),\n",
       "  ('Mean', {}, '95.3333'),\n",
       "  ('Median', {}, '78'),\n",
       "  ('LengthStats', {'category': 'tRNA'}, '\\n      '),\n",
       "  ('Count', {}, '27'),\n",
       "  ('Min', {}, '72'),\n",
       "  ('Max', {}, '83'),\n",
       "  ('Mean', {}, '74.037'),\n",
       "  ('Median', {}, '73')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510225.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '8'),\n",
       "  ('Mean', {}, '2.89744'),\n",
       "  ('Median', {}, '2'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '18'),\n",
       "  ('Mean', {}, '7.90265'),\n",
       "  ('Median', {}, '7'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '358'),\n",
       "  ('Min', {}, '12'),\n",
       "  ('Max', {}, '4033'),\n",
       "  ('Mean', {}, '248.601'),\n",
       "  ('Median', {}, '154'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '324'),\n",
       "  ('Min', {}, '69'),\n",
       "  ('Max', {}, '36335'),\n",
       "  ('Mean', {}, '1632.85'),\n",
       "  ('Median', {}, '673'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '113'),\n",
       "  ('Min', {}, '21'),\n",
       "  ('Max', {}, '5116'),\n",
       "  ('Mean', {}, '2015.02'),\n",
       "  ('Median', {}, '1870'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '92'),\n",
       "  ('Min', {}, '240'),\n",
       "  ('Max', {}, '3189'),\n",
       "  ('Mean', {}, '1241.15'),\n",
       "  ('Median', {}, '933'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '325'),\n",
       "  ('Min', {}, '12'),\n",
       "  ('Max', {}, '4033'),\n",
       "  ('Mean', {}, '241.431'),\n",
       "  ('Median', {}, '148'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '93'),\n",
       "  ('Min', {}, '28'),\n",
       "  ('Max', {}, '4033'),\n",
       "  ('Mean', {}, '275.516'),\n",
       "  ('Median', {}, '162'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '39'),\n",
       "  ('Min', {}, '72'),\n",
       "  ('Max', {}, '36881'),\n",
       "  ('Mean', {}, '11086.5'),\n",
       "  ('Median', {}, '7776'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '299'),\n",
       "  ('Min', {}, '69'),\n",
       "  ('Max', {}, '21374'),\n",
       "  ('Mean', {}, '1558.13'),\n",
       "  ('Median', {}, '646'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '77'),\n",
       "  ('Min', {}, '71'),\n",
       "  ('Max', {}, '36335'),\n",
       "  ('Mean', {}, '1841.68'),\n",
       "  ('Median', {}, '539'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '419'),\n",
       "  ('Max', {}, '2020'),\n",
       "  ('Mean', {}, '970.889'),\n",
       "  ('Median', {}, '1044'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '92'),\n",
       "  ('Min', {}, '526'),\n",
       "  ('Max', {}, '4875'),\n",
       "  ('Mean', {}, '2096.65'),\n",
       "  ('Median', {}, '1956'),\n",
       "  ('LengthStats', {'category': 'miRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '21'),\n",
       "  ('Max', {}, '22'),\n",
       "  ('Mean', {}, '21.5'),\n",
       "  ('Median', {}, '22'),\n",
       "  ('LengthStats', {'category': 'misc_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '10'),\n",
       "  ('Min', {}, '885'),\n",
       "  ('Max', {}, '5116'),\n",
       "  ('Mean', {}, '2602.4'),\n",
       "  ('Median', {}, '2364'),\n",
       "  ('LengthStats', {'category': 'premsg'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '72'),\n",
       "  ('Max', {}, '75'),\n",
       "  ('Mean', {}, '73.5'),\n",
       "  ('Median', {}, '75')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510235.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '8'),\n",
       "  ('Mean', {}, '2.92593'),\n",
       "  ('Median', {}, '2'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '18'),\n",
       "  ('Mean', {}, '8.02532'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '503'),\n",
       "  ('Min', {}, '12'),\n",
       "  ('Max', {}, '4033'),\n",
       "  ('Mean', {}, '248.541'),\n",
       "  ('Median', {}, '155'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '460'),\n",
       "  ('Min', {}, '69'),\n",
       "  ('Max', {}, '36335'),\n",
       "  ('Mean', {}, '1682.65'),\n",
       "  ('Median', {}, '652'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '158'),\n",
       "  ('Min', {}, '21'),\n",
       "  ('Max', {}, '5116'),\n",
       "  ('Mean', {}, '2097.47'),\n",
       "  ('Median', {}, '2056'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '126'),\n",
       "  ('Min', {}, '240'),\n",
       "  ('Max', {}, '3189'),\n",
       "  ('Mean', {}, '1266.02'),\n",
       "  ('Median', {}, '1035'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '451'),\n",
       "  ('Min', {}, '12'),\n",
       "  ('Max', {}, '4033'),\n",
       "  ('Mean', {}, '243.96'),\n",
       "  ('Median', {}, '142'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '142'),\n",
       "  ('Min', {}, '28'),\n",
       "  ('Max', {}, '4033'),\n",
       "  ('Mean', {}, '264.437'),\n",
       "  ('Median', {}, '171'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '54'),\n",
       "  ('Min', {}, '72'),\n",
       "  ('Max', {}, '36881'),\n",
       "  ('Mean', {}, '11802.4'),\n",
       "  ('Median', {}, '9771'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '419'),\n",
       "  ('Min', {}, '69'),\n",
       "  ('Max', {}, '21374'),\n",
       "  ('Mean', {}, '1586.19'),\n",
       "  ('Median', {}, '619'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '118'),\n",
       "  ('Min', {}, '71'),\n",
       "  ('Max', {}, '36335'),\n",
       "  ('Mean', {}, '1686.7'),\n",
       "  ('Median', {}, '448'),\n",
       "  ('LengthStats', {'category': 'Single-exon transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '1'),\n",
       "  ('Min', {}, '2050'),\n",
       "  ('Max', {}, '2050'),\n",
       "  ('Mean', {}, '2050'),\n",
       "  ('Median', {}, '2050'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '1'),\n",
       "  ('Min', {}, '2050'),\n",
       "  ('Max', {}, '2050'),\n",
       "  ('Mean', {}, '2050'),\n",
       "  ('Median', {}, '2050'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '16'),\n",
       "  ('Min', {}, '419'),\n",
       "  ('Max', {}, '2184'),\n",
       "  ('Mean', {}, '1075.81'),\n",
       "  ('Median', {}, '1009'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '126'),\n",
       "  ('Min', {}, '526'),\n",
       "  ('Max', {}, '5097'),\n",
       "  ('Mean', {}, '2208.09'),\n",
       "  ('Median', {}, '2149'),\n",
       "  ('LengthStats', {'category': 'miRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '21'),\n",
       "  ('Max', {}, '22'),\n",
       "  ('Mean', {}, '21.5'),\n",
       "  ('Median', {}, '22'),\n",
       "  ('LengthStats', {'category': 'misc_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '14'),\n",
       "  ('Min', {}, '885'),\n",
       "  ('Max', {}, '5116'),\n",
       "  ('Mean', {}, '2566.14'),\n",
       "  ('Median', {}, '2364'),\n",
       "  ('LengthStats', {'category': 'premsg'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '72'),\n",
       "  ('Max', {}, '75'),\n",
       "  ('Mean', {}, '73.5'),\n",
       "  ('Median', {}, '75')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000509915.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '5'),\n",
       "  ('Mean', {}, '1.75'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.33333'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '95'),\n",
       "  ('Min', {}, '34'),\n",
       "  ('Max', {}, '774'),\n",
       "  ('Mean', {}, '197.316'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '90'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '9888'),\n",
       "  ('Mean', {}, '1889.01'),\n",
       "  ('Median', {}, '1273'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '21'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '1877'),\n",
       "  ('Mean', {}, '1436.57'),\n",
       "  ('Median', {}, '1522'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '19'),\n",
       "  ('Min', {}, '321'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '991.263'),\n",
       "  ('Median', {}, '1029'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '91'),\n",
       "  ('Min', {}, '34'),\n",
       "  ('Max', {}, '774'),\n",
       "  ('Mean', {}, '195.242'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '12'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '16787'),\n",
       "  ('Mean', {}, '12227.3'),\n",
       "  ('Median', {}, '14312'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '87'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9888'),\n",
       "  ('Mean', {}, '1934.48'),\n",
       "  ('Median', {}, '1304'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '19'),\n",
       "  ('Min', {}, '979'),\n",
       "  ('Max', {}, '1877'),\n",
       "  ('Mean', {}, '1509.32'),\n",
       "  ('Median', {}, '1573')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000509925.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '2'),\n",
       "  ('Mean', {}, '1.5'),\n",
       "  ('Median', {}, '2'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.16667'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '62'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '207.242'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '57'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '9892'),\n",
       "  ('Mean', {}, '1948.54'),\n",
       "  ('Median', {}, '1113'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '12'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1523.5'),\n",
       "  ('Median', {}, '1592'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '10'),\n",
       "  ('Min', {}, '669'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '1056.6'),\n",
       "  ('Median', {}, '1047'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '58'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '204.672'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '8'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '16787'),\n",
       "  ('Mean', {}, '12586'),\n",
       "  ('Median', {}, '14521'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '54'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9892'),\n",
       "  ('Mean', {}, '2025.11'),\n",
       "  ('Median', {}, '1464'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '10'),\n",
       "  ('Min', {}, '1477'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1679.1'),\n",
       "  ('Median', {}, '1614')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000509935.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '5'),\n",
       "  ('Mean', {}, '2'),\n",
       "  ('Median', {}, '2'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '6'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.71429'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '59'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '688'),\n",
       "  ('Mean', {}, '210.644'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '58'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9892'),\n",
       "  ('Mean', {}, '2216.9'),\n",
       "  ('Median', {}, '1514'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '14'),\n",
       "  ('Min', {}, '979'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1573.36'),\n",
       "  ('Median', {}, '1582'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '14'),\n",
       "  ('Min', {}, '321'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '970.286'),\n",
       "  ('Median', {}, '1029'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '59'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '688'),\n",
       "  ('Mean', {}, '210.644'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '7'),\n",
       "  ('Min', {}, '10911'),\n",
       "  ('Max', {}, '16733'),\n",
       "  ('Mean', {}, '14087.1'),\n",
       "  ('Median', {}, '14345'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '58'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9892'),\n",
       "  ('Mean', {}, '2216.9'),\n",
       "  ('Median', {}, '1514'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '14'),\n",
       "  ('Min', {}, '979'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1573.36'),\n",
       "  ('Median', {}, '1582')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000509945.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '2'),\n",
       "  ('Mean', {}, '1.375'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.18182'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '62'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '207.613'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '56'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '9890'),\n",
       "  ('Mean', {}, '1850.05'),\n",
       "  ('Median', {}, '1113'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '11'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1524.82'),\n",
       "  ('Median', {}, '1605'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '915'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '1156'),\n",
       "  ('Median', {}, '1134'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '58'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '205.069'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '8'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '16751'),\n",
       "  ('Mean', {}, '12591.4'),\n",
       "  ('Median', {}, '14530'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '53'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9890'),\n",
       "  ('Mean', {}, '1922.49'),\n",
       "  ('Median', {}, '1273'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '1477'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1698'),\n",
       "  ('Median', {}, '1614')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000509955.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '5'),\n",
       "  ('Mean', {}, '1.72727'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '6.89474'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '84'),\n",
       "  ('Min', {}, '34'),\n",
       "  ('Max', {}, '774'),\n",
       "  ('Mean', {}, '204.393'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '77'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '7715'),\n",
       "  ('Mean', {}, '1864.09'),\n",
       "  ('Median', {}, '1273'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '19'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1467.21'),\n",
       "  ('Median', {}, '1573'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '17'),\n",
       "  ('Min', {}, '321'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '980.824'),\n",
       "  ('Median', {}, '1029'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '80'),\n",
       "  ('Min', {}, '34'),\n",
       "  ('Max', {}, '774'),\n",
       "  ('Mean', {}, '202.362'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '245'),\n",
       "  ('Median', {}, '269'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '11'),\n",
       "  ('Min', {}, '1644'),\n",
       "  ('Max', {}, '15237'),\n",
       "  ('Mean', {}, '11532.4'),\n",
       "  ('Median', {}, '14312'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '74'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '7715'),\n",
       "  ('Mean', {}, '1916.54'),\n",
       "  ('Median', {}, '1464'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '17'),\n",
       "  ('Min', {}, '979'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1552.12'),\n",
       "  ('Median', {}, '1582')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000509965.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '3'),\n",
       "  ('Mean', {}, '1.45455'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.0625'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '81'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '191.099'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '74'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '9888'),\n",
       "  ('Mean', {}, '1841.31'),\n",
       "  ('Median', {}, '1273'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '16'),\n",
       "  ('Min', {}, '536'),\n",
       "  ('Max', {}, '1877'),\n",
       "  ('Mean', {}, '1388.94'),\n",
       "  ('Median', {}, '1582'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '13'),\n",
       "  ('Min', {}, '828'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '1073.31'),\n",
       "  ('Median', {}, '1047'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '74'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '188.716'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '7'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '216.286'),\n",
       "  ('Median', {}, '197'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '11'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '16787'),\n",
       "  ('Mean', {}, '11811.2'),\n",
       "  ('Median', {}, '14015'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '69'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9888'),\n",
       "  ('Mean', {}, '1856.09'),\n",
       "  ('Median', {}, '1304'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '5'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '5470'),\n",
       "  ('Mean', {}, '1637.4'),\n",
       "  ('Median', {}, '849'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '13'),\n",
       "  ('Min', {}, '1101'),\n",
       "  ('Max', {}, '1877'),\n",
       "  ('Mean', {}, '1553.54'),\n",
       "  ('Median', {}, '1592'),\n",
       "  ('LengthStats', {'category': 'misc_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '1'),\n",
       "  ('Min', {}, '536'),\n",
       "  ('Max', {}, '536'),\n",
       "  ('Mean', {}, '536'),\n",
       "  ('Median', {}, '536')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000509975.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '5'),\n",
       "  ('Mean', {}, '1.85714'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '6'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.76923'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '59'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '688'),\n",
       "  ('Mean', {}, '211.017'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '57'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9891'),\n",
       "  ('Mean', {}, '2103.23'),\n",
       "  ('Median', {}, '1273'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '13'),\n",
       "  ('Min', {}, '979'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1578.23'),\n",
       "  ('Median', {}, '1582'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '13'),\n",
       "  ('Min', {}, '321'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '1008.46'),\n",
       "  ('Median', {}, '1029'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '59'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '688'),\n",
       "  ('Mean', {}, '211.017'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '7'),\n",
       "  ('Min', {}, '10951'),\n",
       "  ('Max', {}, '16752'),\n",
       "  ('Mean', {}, '13916.7'),\n",
       "  ('Median', {}, '14344'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '57'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9891'),\n",
       "  ('Mean', {}, '2103.23'),\n",
       "  ('Median', {}, '1273'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '13'),\n",
       "  ('Min', {}, '979'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1578.23'),\n",
       "  ('Median', {}, '1582')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000509985.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '2'),\n",
       "  ('Mean', {}, '1.375'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.18182'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '62'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '207.613'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '56'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '9890'),\n",
       "  ('Mean', {}, '1849.8'),\n",
       "  ('Median', {}, '1113'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '11'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1524.82'),\n",
       "  ('Median', {}, '1605'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '915'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '1156'),\n",
       "  ('Median', {}, '1134'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '58'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '205.069'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '8'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '16751'),\n",
       "  ('Mean', {}, '12589.6'),\n",
       "  ('Median', {}, '14519'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '53'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9890'),\n",
       "  ('Mean', {}, '1922.23'),\n",
       "  ('Median', {}, '1273'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '1477'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1698'),\n",
       "  ('Median', {}, '1614')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000509995.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '5'),\n",
       "  ('Mean', {}, '1.88889'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '6'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.76471'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '75'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '694'),\n",
       "  ('Mean', {}, '200.12'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '73'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9888'),\n",
       "  ('Mean', {}, '2023.41'),\n",
       "  ('Median', {}, '1476'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '17'),\n",
       "  ('Min', {}, '979'),\n",
       "  ('Max', {}, '1877'),\n",
       "  ('Mean', {}, '1524.71'),\n",
       "  ('Median', {}, '1573'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '17'),\n",
       "  ('Min', {}, '321'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '979.941'),\n",
       "  ('Median', {}, '1017'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '75'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '694'),\n",
       "  ('Mean', {}, '200.12'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '9465'),\n",
       "  ('Max', {}, '16787'),\n",
       "  ('Mean', {}, '13526'),\n",
       "  ('Median', {}, '14313'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '73'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9888'),\n",
       "  ('Mean', {}, '2023.41'),\n",
       "  ('Median', {}, '1476'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '17'),\n",
       "  ('Min', {}, '979'),\n",
       "  ('Max', {}, '1877'),\n",
       "  ('Mean', {}, '1524.71'),\n",
       "  ('Median', {}, '1573')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510005.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '2'),\n",
       "  ('Mean', {}, '1.375'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.18182'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '62'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '207.613'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '56'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '9889'),\n",
       "  ('Mean', {}, '1849.36'),\n",
       "  ('Median', {}, '1113'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '11'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1524.82'),\n",
       "  ('Median', {}, '1605'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '915'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '1156'),\n",
       "  ('Median', {}, '1134'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '58'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '205.069'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '8'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '16765'),\n",
       "  ('Mean', {}, '12586.6'),\n",
       "  ('Median', {}, '14520'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '53'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9889'),\n",
       "  ('Mean', {}, '1921.75'),\n",
       "  ('Median', {}, '1273'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '1477'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1698'),\n",
       "  ('Median', {}, '1614')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510015.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '5'),\n",
       "  ('Mean', {}, '1.72727'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '6.89474'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '84'),\n",
       "  ('Min', {}, '34'),\n",
       "  ('Max', {}, '774'),\n",
       "  ('Mean', {}, '210.167'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '77'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '7715'),\n",
       "  ('Mean', {}, '1875.66'),\n",
       "  ('Median', {}, '1273'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '19'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1492.89'),\n",
       "  ('Median', {}, '1581'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '17'),\n",
       "  ('Min', {}, '321'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '962.471'),\n",
       "  ('Median', {}, '1017'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '80'),\n",
       "  ('Min', {}, '34'),\n",
       "  ('Max', {}, '774'),\n",
       "  ('Mean', {}, '208.45'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '11'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '15868'),\n",
       "  ('Mean', {}, '11657.5'),\n",
       "  ('Median', {}, '14312'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '74'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '7715'),\n",
       "  ('Mean', {}, '1928.58'),\n",
       "  ('Median', {}, '1464'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '17'),\n",
       "  ('Min', {}, '979'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1580.82'),\n",
       "  ('Median', {}, '1583')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510025.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '2'),\n",
       "  ('Mean', {}, '1.375'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.18182'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '62'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '207.613'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '56'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '9890'),\n",
       "  ('Mean', {}, '1849.16'),\n",
       "  ('Median', {}, '1113'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '11'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1524.82'),\n",
       "  ('Median', {}, '1605'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '915'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '1156'),\n",
       "  ('Median', {}, '1134'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '58'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '205.069'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '8'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '16751'),\n",
       "  ('Mean', {}, '12585.1'),\n",
       "  ('Median', {}, '14514'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '53'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9890'),\n",
       "  ('Mean', {}, '1921.55'),\n",
       "  ('Median', {}, '1273'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '1477'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1698'),\n",
       "  ('Median', {}, '1614')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510035.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '5'),\n",
       "  ('Mean', {}, '1.88889'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '6'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.76471'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '75'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '688'),\n",
       "  ('Mean', {}, '193.573'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '73'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9888'),\n",
       "  ('Mean', {}, '2020.36'),\n",
       "  ('Median', {}, '1464'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '17'),\n",
       "  ('Min', {}, '979'),\n",
       "  ('Max', {}, '1877'),\n",
       "  ('Mean', {}, '1495.94'),\n",
       "  ('Median', {}, '1522'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '17'),\n",
       "  ('Min', {}, '321'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '979.941'),\n",
       "  ('Median', {}, '1017'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '75'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '688'),\n",
       "  ('Mean', {}, '193.573'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '9461'),\n",
       "  ('Max', {}, '16787'),\n",
       "  ('Mean', {}, '13446.7'),\n",
       "  ('Median', {}, '14313'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '73'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9888'),\n",
       "  ('Mean', {}, '2020.36'),\n",
       "  ('Median', {}, '1464'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '17'),\n",
       "  ('Min', {}, '979'),\n",
       "  ('Max', {}, '1877'),\n",
       "  ('Mean', {}, '1495.94'),\n",
       "  ('Median', {}, '1522')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510045.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '8'),\n",
       "  ('Mean', {}, '2.4'),\n",
       "  ('Median', {}, '2'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '5.54167'),\n",
       "  ('Median', {}, '5'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '76'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '1772'),\n",
       "  ('Mean', {}, '221.921'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '70'),\n",
       "  ('Min', {}, '77'),\n",
       "  ('Max', {}, '13683'),\n",
       "  ('Mean', {}, '2227.69'),\n",
       "  ('Median', {}, '1420'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '24'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '2476'),\n",
       "  ('Mean', {}, '1684.92'),\n",
       "  ('Median', {}, '1723'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '22'),\n",
       "  ('Min', {}, '537'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '901.773'),\n",
       "  ('Median', {}, '912'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '72'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '1772'),\n",
       "  ('Mean', {}, '220.667'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '10'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '17186'),\n",
       "  ('Mean', {}, '12181.9'),\n",
       "  ('Median', {}, '14529'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '67'),\n",
       "  ('Min', {}, '77'),\n",
       "  ('Max', {}, '13683'),\n",
       "  ('Mean', {}, '2301.9'),\n",
       "  ('Median', {}, '1428'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '22'),\n",
       "  ('Min', {}, '821'),\n",
       "  ('Max', {}, '2476'),\n",
       "  ('Mean', {}, '1770.32'),\n",
       "  ('Median', {}, '1826')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510055.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '2'),\n",
       "  ('Mean', {}, '1.33333'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '8'),\n",
       "  ('Mean', {}, '6.125'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '40'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '205.225'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '35'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '5467'),\n",
       "  ('Mean', {}, '1581'),\n",
       "  ('Median', {}, '934'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '8'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1433.12'),\n",
       "  ('Median', {}, '1592'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '6'),\n",
       "  ('Min', {}, '1026'),\n",
       "  ('Max', {}, '1335'),\n",
       "  ('Mean', {}, '1134'),\n",
       "  ('Median', {}, '1134'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '36'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '200.861'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '6'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '14531'),\n",
       "  ('Mean', {}, '9614.83'),\n",
       "  ('Median', {}, '12048'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '32'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '5467'),\n",
       "  ('Mean', {}, '1675.75'),\n",
       "  ('Median', {}, '999'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '6'),\n",
       "  ('Min', {}, '1477'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1662.33'),\n",
       "  ('Median', {}, '1614')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510065.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '2'),\n",
       "  ('Mean', {}, '1.5'),\n",
       "  ('Median', {}, '2'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.16667'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '62'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '207.242'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '57'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '9890'),\n",
       "  ('Mean', {}, '1948.89'),\n",
       "  ('Median', {}, '1113'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '12'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1523.5'),\n",
       "  ('Median', {}, '1592'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '10'),\n",
       "  ('Min', {}, '669'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '1056.6'),\n",
       "  ('Median', {}, '1047'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '58'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '204.672'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '8'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '16789'),\n",
       "  ('Mean', {}, '12588.8'),\n",
       "  ('Median', {}, '14530'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '54'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9890'),\n",
       "  ('Mean', {}, '2025.48'),\n",
       "  ('Median', {}, '1464'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '10'),\n",
       "  ('Min', {}, '1477'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1679.1'),\n",
       "  ('Median', {}, '1614')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510075.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '8'),\n",
       "  ('Mean', {}, '2.4'),\n",
       "  ('Median', {}, '2'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '5.54167'),\n",
       "  ('Median', {}, '5'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '76'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '1772'),\n",
       "  ('Mean', {}, '221.921'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '70'),\n",
       "  ('Min', {}, '77'),\n",
       "  ('Max', {}, '13683'),\n",
       "  ('Mean', {}, '2216.06'),\n",
       "  ('Median', {}, '1401'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '24'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '2476'),\n",
       "  ('Mean', {}, '1684.92'),\n",
       "  ('Median', {}, '1723'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '22'),\n",
       "  ('Min', {}, '537'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '901.773'),\n",
       "  ('Median', {}, '912'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '72'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '1772'),\n",
       "  ('Mean', {}, '220.667'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '10'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '17186'),\n",
       "  ('Mean', {}, '12100.6'),\n",
       "  ('Median', {}, '14528'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '67'),\n",
       "  ('Min', {}, '77'),\n",
       "  ('Max', {}, '13683'),\n",
       "  ('Mean', {}, '2289.75'),\n",
       "  ('Median', {}, '1420'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '22'),\n",
       "  ('Min', {}, '821'),\n",
       "  ('Max', {}, '2476'),\n",
       "  ('Mean', {}, '1770.32'),\n",
       "  ('Median', {}, '1826')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510085.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '8'),\n",
       "  ('Mean', {}, '2.33333'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '6.17143'),\n",
       "  ('Median', {}, '7'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '117'),\n",
       "  ('Min', {}, '34'),\n",
       "  ('Max', {}, '1772'),\n",
       "  ('Mean', {}, '202.171'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '111'),\n",
       "  ('Min', {}, '77'),\n",
       "  ('Max', {}, '13683'),\n",
       "  ('Mean', {}, '2122.87'),\n",
       "  ('Median', {}, '1420'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '35'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '2476'),\n",
       "  ('Mean', {}, '1564.23'),\n",
       "  ('Median', {}, '1573'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '33'),\n",
       "  ('Min', {}, '321'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '884.364'),\n",
       "  ('Median', {}, '912'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '113'),\n",
       "  ('Min', {}, '34'),\n",
       "  ('Max', {}, '1772'),\n",
       "  ('Mean', {}, '200.673'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '15'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '17186'),\n",
       "  ('Mean', {}, '12135.1'),\n",
       "  ('Median', {}, '14159'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '108'),\n",
       "  ('Min', {}, '77'),\n",
       "  ('Max', {}, '13683'),\n",
       "  ('Mean', {}, '2166'),\n",
       "  ('Median', {}, '1464'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '33'),\n",
       "  ('Min', {}, '821'),\n",
       "  ('Max', {}, '2476'),\n",
       "  ('Mean', {}, '1613.85'),\n",
       "  ('Median', {}, '1582')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510095.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '8'),\n",
       "  ('Mean', {}, '2.5'),\n",
       "  ('Median', {}, '2'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '5.6'),\n",
       "  ('Median', {}, '5'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '76'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '1772'),\n",
       "  ('Mean', {}, '221.632'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '71'),\n",
       "  ('Min', {}, '77'),\n",
       "  ('Max', {}, '13683'),\n",
       "  ('Mean', {}, '2301.82'),\n",
       "  ('Median', {}, '1420'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '25'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '2476'),\n",
       "  ('Mean', {}, '1677.92'),\n",
       "  ('Median', {}, '1614'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '23'),\n",
       "  ('Min', {}, '537'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '883.174'),\n",
       "  ('Median', {}, '864'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '72'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '1772'),\n",
       "  ('Mean', {}, '220.361'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '10'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '17186'),\n",
       "  ('Mean', {}, '12180.8'),\n",
       "  ('Median', {}, '14519'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '68'),\n",
       "  ('Min', {}, '77'),\n",
       "  ('Max', {}, '13683'),\n",
       "  ('Mean', {}, '2378.21'),\n",
       "  ('Median', {}, '1464'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '23'),\n",
       "  ('Min', {}, '821'),\n",
       "  ('Max', {}, '2476'),\n",
       "  ('Mean', {}, '1759'),\n",
       "  ('Median', {}, '1723')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510105.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '8'),\n",
       "  ('Mean', {}, '3'),\n",
       "  ('Median', {}, '2'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '6'),\n",
       "  ('Median', {}, '6'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '73'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '1772'),\n",
       "  ('Mean', {}, '224.945'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '72'),\n",
       "  ('Min', {}, '77'),\n",
       "  ('Max', {}, '13683'),\n",
       "  ('Mean', {}, '2515.99'),\n",
       "  ('Median', {}, '1514'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '27'),\n",
       "  ('Min', {}, '821'),\n",
       "  ('Max', {}, '2476'),\n",
       "  ('Mean', {}, '1692.26'),\n",
       "  ('Median', {}, '1583'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '27'),\n",
       "  ('Min', {}, '321'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '841'),\n",
       "  ('Median', {}, '828'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '73'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '1772'),\n",
       "  ('Mean', {}, '224.945'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '3950'),\n",
       "  ('Max', {}, '17186'),\n",
       "  ('Mean', {}, '13325.7'),\n",
       "  ('Median', {}, '14341'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '72'),\n",
       "  ('Min', {}, '77'),\n",
       "  ('Max', {}, '13683'),\n",
       "  ('Mean', {}, '2515.99'),\n",
       "  ('Median', {}, '1514'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '27'),\n",
       "  ('Min', {}, '821'),\n",
       "  ('Max', {}, '2476'),\n",
       "  ('Mean', {}, '1692.26'),\n",
       "  ('Median', {}, '1583')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510125.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '8'),\n",
       "  ('Mean', {}, '2.4'),\n",
       "  ('Median', {}, '2'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '5.45833'),\n",
       "  ('Median', {}, '5'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '74'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '1772'),\n",
       "  ('Mean', {}, '226.311'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '68'),\n",
       "  ('Min', {}, '77'),\n",
       "  ('Max', {}, '13683'),\n",
       "  ('Mean', {}, '2272.18'),\n",
       "  ('Median', {}, '1428'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '24'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '2476'),\n",
       "  ('Mean', {}, '1684.92'),\n",
       "  ('Median', {}, '1723'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '22'),\n",
       "  ('Min', {}, '537'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '901.773'),\n",
       "  ('Median', {}, '912'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '70'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '1772'),\n",
       "  ('Mean', {}, '225.271'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '10'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '17186'),\n",
       "  ('Mean', {}, '12027.2'),\n",
       "  ('Median', {}, '14529'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '65'),\n",
       "  ('Min', {}, '77'),\n",
       "  ('Max', {}, '13683'),\n",
       "  ('Mean', {}, '2350.72'),\n",
       "  ('Median', {}, '1464'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '22'),\n",
       "  ('Min', {}, '821'),\n",
       "  ('Max', {}, '2476'),\n",
       "  ('Mean', {}, '1770.32'),\n",
       "  ('Median', {}, '1826')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510135.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '2'),\n",
       "  ('Mean', {}, '1.4'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.28571'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '78'),\n",
       "  ('Min', {}, '34'),\n",
       "  ('Max', {}, '774'),\n",
       "  ('Mean', {}, '206.141'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '71'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '9889'),\n",
       "  ('Mean', {}, '1863.48'),\n",
       "  ('Median', {}, '1113'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '14'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1536.86'),\n",
       "  ('Median', {}, '1602'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '12'),\n",
       "  ('Min', {}, '669'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '1050.75'),\n",
       "  ('Median', {}, '1047'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '74'),\n",
       "  ('Min', {}, '34'),\n",
       "  ('Max', {}, '774'),\n",
       "  ('Mean', {}, '204.068'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '10'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '16769'),\n",
       "  ('Mean', {}, '12516.1'),\n",
       "  ('Median', {}, '14535'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '68'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9889'),\n",
       "  ('Mean', {}, '1920.53'),\n",
       "  ('Median', {}, '1464'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '12'),\n",
       "  ('Min', {}, '1477'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1668.75'),\n",
       "  ('Median', {}, '1614')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510145.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '2'),\n",
       "  ('Mean', {}, '1.375'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.18182'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '62'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '207.613'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '56'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '9892'),\n",
       "  ('Mean', {}, '1833.68'),\n",
       "  ('Median', {}, '1113'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '11'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1524.82'),\n",
       "  ('Median', {}, '1605'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '915'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '1156'),\n",
       "  ('Median', {}, '1134'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '58'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '205.069'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '8'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '16768'),\n",
       "  ('Mean', {}, '12476.5'),\n",
       "  ('Median', {}, '14514'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '53'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9892'),\n",
       "  ('Mean', {}, '1905.19'),\n",
       "  ('Median', {}, '1273'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '1477'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1698'),\n",
       "  ('Median', {}, '1614')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510155.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '5'),\n",
       "  ('Mean', {}, '1.72727'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.26316'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '87'),\n",
       "  ('Min', {}, '34'),\n",
       "  ('Max', {}, '774'),\n",
       "  ('Mean', {}, '209.793'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '82'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '9888'),\n",
       "  ('Mean', {}, '2009.87'),\n",
       "  ('Median', {}, '1464'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '19'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1492.89'),\n",
       "  ('Median', {}, '1581'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '17'),\n",
       "  ('Min', {}, '321'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '962.471'),\n",
       "  ('Median', {}, '1017'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '83'),\n",
       "  ('Min', {}, '34'),\n",
       "  ('Max', {}, '774'),\n",
       "  ('Mean', {}, '208.12'),\n",
       "  ('Median', {}, '104'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '11'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '16787'),\n",
       "  ('Mean', {}, '12692.8'),\n",
       "  ('Median', {}, '14342'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '79'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9888'),\n",
       "  ('Mean', {}, '2064.53'),\n",
       "  ('Median', {}, '1476'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '17'),\n",
       "  ('Min', {}, '979'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1580.82'),\n",
       "  ('Median', {}, '1583')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510165.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '2'),\n",
       "  ('Mean', {}, '1.5'),\n",
       "  ('Median', {}, '2'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.16667'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '62'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '207.242'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '57'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '9889'),\n",
       "  ('Mean', {}, '1945.98'),\n",
       "  ('Median', {}, '1113'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '12'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1523.5'),\n",
       "  ('Median', {}, '1592'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '10'),\n",
       "  ('Min', {}, '669'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '1056.6'),\n",
       "  ('Median', {}, '1047'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '58'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '204.672'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '8'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '16765'),\n",
       "  ('Mean', {}, '12568'),\n",
       "  ('Median', {}, '14527'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '54'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9889'),\n",
       "  ('Mean', {}, '2022.41'),\n",
       "  ('Median', {}, '1464'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '10'),\n",
       "  ('Min', {}, '1477'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1679.1'),\n",
       "  ('Median', {}, '1614')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510175.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '5'),\n",
       "  ('Mean', {}, '1.88889'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '5'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.41176'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '73'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '694'),\n",
       "  ('Mean', {}, '198.123'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '69'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '7716'),\n",
       "  ('Mean', {}, '1891.99'),\n",
       "  ('Median', {}, '1476'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '17'),\n",
       "  ('Min', {}, '979'),\n",
       "  ('Max', {}, '1877'),\n",
       "  ('Mean', {}, '1524.71'),\n",
       "  ('Median', {}, '1573'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '17'),\n",
       "  ('Min', {}, '321'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '979.941'),\n",
       "  ('Median', {}, '1017'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '73'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '694'),\n",
       "  ('Mean', {}, '198.123'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '8713'),\n",
       "  ('Max', {}, '15021'),\n",
       "  ('Mean', {}, '12624.4'),\n",
       "  ('Median', {}, '14015'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '69'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '7716'),\n",
       "  ('Mean', {}, '1891.99'),\n",
       "  ('Median', {}, '1476'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '17'),\n",
       "  ('Min', {}, '979'),\n",
       "  ('Max', {}, '1877'),\n",
       "  ('Mean', {}, '1524.71'),\n",
       "  ('Median', {}, '1573')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000005045.25', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '33'),\n",
       "  ('Mean', {}, '2.14199'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '93'),\n",
       "  ('Mean', {}, '7.87595'),\n",
       "  ('Median', {}, '5'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '8186'),\n",
       "  ('Min', {}, '7'),\n",
       "  ('Max', {}, '14433'),\n",
       "  ('Mean', {}, '366.997'),\n",
       "  ('Median', {}, '138'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '7161'),\n",
       "  ('Min', {}, '30'),\n",
       "  ('Max', {}, '334517'),\n",
       "  ('Mean', {}, '4315.23'),\n",
       "  ('Median', {}, '1113'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '2140'),\n",
       "  ('Min', {}, '17'),\n",
       "  ('Max', {}, '21046'),\n",
       "  ('Mean', {}, '2885.79'),\n",
       "  ('Median', {}, '2234'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '1382'),\n",
       "  ('Min', {}, '132'),\n",
       "  ('Max', {}, '16965'),\n",
       "  ('Mean', {}, '1728.69'),\n",
       "  ('Median', {}, '1284'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '6730'),\n",
       "  ('Min', {}, '7'),\n",
       "  ('Max', {}, '14433'),\n",
       "  ('Mean', {}, '339.759'),\n",
       "  ('Median', {}, '133'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '2042'),\n",
       "  ('Min', {}, '9'),\n",
       "  ('Max', {}, '10558'),\n",
       "  ('Mean', {}, '427.424'),\n",
       "  ('Median', {}, '153'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '996'),\n",
       "  ('Min', {}, '51'),\n",
       "  ('Max', {}, '650936'),\n",
       "  ('Mean', {}, '25456.7'),\n",
       "  ('Median', {}, '8349'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '6034'),\n",
       "  ('Min', {}, '30'),\n",
       "  ('Max', {}, '334517'),\n",
       "  ('Mean', {}, '3872.72'),\n",
       "  ('Median', {}, '1015'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '1642'),\n",
       "  ('Min', {}, '38'),\n",
       "  ('Max', {}, '112483'),\n",
       "  ('Mean', {}, '5694.09'),\n",
       "  ('Median', {}, '1571'),\n",
       "  ('LengthStats', {'category': 'Single-exon transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '118'),\n",
       "  ('Min', {}, '452'),\n",
       "  ('Max', {}, '18485'),\n",
       "  ('Mean', {}, '3601.76'),\n",
       "  ('Median', {}, '2485'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '104'),\n",
       "  ('Min', {}, '452'),\n",
       "  ('Max', {}, '18485'),\n",
       "  ('Mean', {}, '3517.5'),\n",
       "  ('Median', {}, '2462'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in non-coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '14'),\n",
       "  ('Min', {}, '829'),\n",
       "  ('Max', {}, '12682'),\n",
       "  ('Mean', {}, '4227.71'),\n",
       "  ('Median', {}, '2879'),\n",
       "  ('LengthStats', {'category': 'guide_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '266'),\n",
       "  ('Max', {}, '278'),\n",
       "  ('Mean', {}, '272'),\n",
       "  ('Median', {}, '278'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '475'),\n",
       "  ('Min', {}, '137'),\n",
       "  ('Max', {}, '13054'),\n",
       "  ('Mean', {}, '1834.45'),\n",
       "  ('Median', {}, '1308'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '1382'),\n",
       "  ('Min', {}, '234'),\n",
       "  ('Max', {}, '21046'),\n",
       "  ('Mean', {}, '3470.96'),\n",
       "  ('Median', {}, '2585'),\n",
       "  ('LengthStats', {'category': 'miRNA'}, '\\n      '),\n",
       "  ('Count', {}, '83'),\n",
       "  ('Min', {}, '17'),\n",
       "  ('Max', {}, '26'),\n",
       "  ('Mean', {}, '21.6386'),\n",
       "  ('Median', {}, '22'),\n",
       "  ('LengthStats', {'category': 'misc_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '161'),\n",
       "  ('Min', {}, '173'),\n",
       "  ('Max', {}, '12682'),\n",
       "  ('Mean', {}, '2863.29'),\n",
       "  ('Median', {}, '2760'),\n",
       "  ('LengthStats', {'category': 'premsg'}, '\\n      '),\n",
       "  ('Count', {}, '57'),\n",
       "  ('Min', {}, '51'),\n",
       "  ('Max', {}, '138'),\n",
       "  ('Mean', {}, '83.7368'),\n",
       "  ('Median', {}, '80'),\n",
       "  ('LengthStats', {'category': 'rRNA'}, '\\n      '),\n",
       "  ('Count', {}, '25'),\n",
       "  ('Min', {}, '121'),\n",
       "  ('Max', {}, '13351'),\n",
       "  ('Mean', {}, '1715.6'),\n",
       "  ('Median', {}, '121'),\n",
       "  ('LengthStats', {'category': 'snoRNA'}, '\\n      '),\n",
       "  ('Count', {}, '10'),\n",
       "  ('Min', {}, '59'),\n",
       "  ('Max', {}, '197'),\n",
       "  ('Mean', {}, '98.2'),\n",
       "  ('Median', {}, '75'),\n",
       "  ('LengthStats', {'category': 'tRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '82'),\n",
       "  ('Max', {}, '82'),\n",
       "  ('Mean', {}, '82'),\n",
       "  ('Median', {}, '82')]]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "feature_stats_collection = []\n",
    "for i in root[3]:\n",
    "    tags = []\n",
    "    text = []\n",
    "    attributes = []\n",
    "    for element in i.iter():\n",
    "        tags.append(element.tag)\n",
    "        text.append(element.text)\n",
    "        attributes.append(element.attrib)\n",
    "    thing = list(zip(tags,attributes,text))\n",
    "    feature_stats_collection.append(thing)\n",
    "print(len(feature_stats_collection))\n",
    "feature_stats_collection"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'GCF_000001405.39'"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_accession = full_assembly_collection[1][0][2]\n",
    "human_accession"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "From this 'human_accession', at the ncbi online database located at https://www.ncbi.nlm.nih.gov/assembly/GCF_000001405.39/, the full sequence report can be pulled down into a text file seen below. This information contains information about genetic information located on chromosomes, in other words, it's a type of dna data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "import csv\n",
    "txt_file_dna = r\"GCF_000001405.39_GRCh38.p13_assembly_report.txt\"\n",
    "csv_file_dna = r\"GCF_000001405.39_GRCh38.p13_assembly_report.csv\"\n",
    "in_txt_dna = csv.reader(open(txt_file_dna, \"r\", encoding='utf-8'), delimiter = '\\t')\n",
    "out_csv_dna = csv.writer(open(csv_file_dna, 'w'))\n",
    "out_csv_dna.writerows(in_txt_dna)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Assembly name:  GRCh38.p13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Description:    Genome Reference Consortium Hu...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Organism name:  Homo sapiens (human)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Taxid:          9606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>BioProject:     PRJNA31257</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>Submitter:      Genome Reference Consortium</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>Date:           2019-02-28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>Assembly type:  haploid-with-alt-loci</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>Release type:   patch</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>Assembly level: Chromosome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>Genome representation: full</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>RefSeq category: Reference Genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>GenBank assembly accession: GCA_000001405.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>RefSeq assembly accession: GCF_000001405.39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>RefSeq assembly and GenBank assemblies identic...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>Assembly-Units:</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           Assembly name:  GRCh38.p13\n",
       "1   Description:    Genome Reference Consortium Hu...\n",
       "3                Organism name:  Homo sapiens (human)\n",
       "5                                Taxid:          9606\n",
       "7                          BioProject:     PRJNA31257\n",
       "9         Submitter:      Genome Reference Consortium\n",
       "11                         Date:           2019-02-28\n",
       "13              Assembly type:  haploid-with-alt-loci\n",
       "15                              Release type:   patch\n",
       "17                         Assembly level: Chromosome\n",
       "19                        Genome representation: full\n",
       "21                  RefSeq category: Reference Genome\n",
       "23       GenBank assembly accession: GCA_000001405.28\n",
       "25        RefSeq assembly accession: GCF_000001405.39\n",
       "27  RefSeq assembly and GenBank assemblies identic...\n",
       "31                                    Assembly-Units:"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "to_skip_info = []\n",
    "for i in range(34,1406):\n",
    "    to_skip_info.append(i)\n",
    "information_about_dna_data = pd.read_csv('GCF_000001405.39_GRCh38.p13_assembly_report_fixed.csv', skiprows=(to_skip_info), header=0)\n",
    "pd.DataFrame(information_about_dna_data['Assembly name:  GRCh38.p13'].dropna(axis=0))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "dna_data = pd.read_csv('GCF_000001405.39_GRCh38.p13_assembly_report_fixed.csv', header=124)\n",
    "dna_data = dna_data.dropna(axis=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th># Sequence-Name</th>\n",
       "      <th>Sequence-Role</th>\n",
       "      <th>Assigned-Molecule</th>\n",
       "      <th>Assigned-Molecule-Location/Type</th>\n",
       "      <th>GenBank-Accn</th>\n",
       "      <th>Relationship</th>\n",
       "      <th>RefSeq-Accn</th>\n",
       "      <th>Assembly-Unit</th>\n",
       "      <th>Sequence-Length</th>\n",
       "      <th>UCSC-style-name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>881</td>\n",
       "      <td>HSCHR7_1_CTG6</td>\n",
       "      <td>alt-scaffold</td>\n",
       "      <td>7</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>GL383534.2</td>\n",
       "      <td>=</td>\n",
       "      <td>NW_003315922.2</td>\n",
       "      <td>ALT_REF_LOCI_1</td>\n",
       "      <td>119183.0</td>\n",
       "      <td>chr7_GL383534v2_alt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>559</td>\n",
       "      <td>HG1708_PATCH</td>\n",
       "      <td>fix-patch</td>\n",
       "      <td>11</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>KV766195.1</td>\n",
       "      <td>=</td>\n",
       "      <td>NW_017363816.1</td>\n",
       "      <td>PATCHES</td>\n",
       "      <td>140877.0</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>317</td>\n",
       "      <td>HSCHRUN_RANDOM_143</td>\n",
       "      <td>unplaced-scaffold</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>KI270529.1</td>\n",
       "      <td>=</td>\n",
       "      <td>NT_187439.1</td>\n",
       "      <td>Primary Assembly</td>\n",
       "      <td>1899.0</td>\n",
       "      <td>chrUn_KI270529v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>339</td>\n",
       "      <td>HSCHRUN_RANDOM_161</td>\n",
       "      <td>unplaced-scaffold</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>KI270584.1</td>\n",
       "      <td>=</td>\n",
       "      <td>NT_187453.1</td>\n",
       "      <td>Primary Assembly</td>\n",
       "      <td>4513.0</td>\n",
       "      <td>chrUn_KI270584v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>263</td>\n",
       "      <td>HSCHRUN_RANDOM_120</td>\n",
       "      <td>unplaced-scaffold</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>KI270422.1</td>\n",
       "      <td>=</td>\n",
       "      <td>NT_187416.1</td>\n",
       "      <td>Primary Assembly</td>\n",
       "      <td>1445.0</td>\n",
       "      <td>chrUn_KI270422v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>945</td>\n",
       "      <td>HG151_NOVEL_TEST</td>\n",
       "      <td>alt-scaffold</td>\n",
       "      <td>11</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>JH159137.1</td>\n",
       "      <td>=</td>\n",
       "      <td>NW_003871074.1</td>\n",
       "      <td>ALT_REF_LOCI_1</td>\n",
       "      <td>191409.0</td>\n",
       "      <td>chr11_JH159137v1_alt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>473</td>\n",
       "      <td>HSCHR4_8_CTG12</td>\n",
       "      <td>novel-patch</td>\n",
       "      <td>4</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>KQ090014.1</td>\n",
       "      <td>=</td>\n",
       "      <td>NW_013171800.1</td>\n",
       "      <td>PATCHES</td>\n",
       "      <td>163749.0</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>607</td>\n",
       "      <td>HG2291_PATCH</td>\n",
       "      <td>fix-patch</td>\n",
       "      <td>13</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>KN538372.1</td>\n",
       "      <td>=</td>\n",
       "      <td>NW_011332699.1</td>\n",
       "      <td>PATCHES</td>\n",
       "      <td>356766.0</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1007</td>\n",
       "      <td>HSCHR15_1_CTG3</td>\n",
       "      <td>alt-scaffold</td>\n",
       "      <td>15</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>KI270848.1</td>\n",
       "      <td>=</td>\n",
       "      <td>NT_187603.1</td>\n",
       "      <td>ALT_REF_LOCI_1</td>\n",
       "      <td>327382.0</td>\n",
       "      <td>chr15_KI270848v1_alt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>641</td>\n",
       "      <td>HSCHR15_6_CTG8</td>\n",
       "      <td>novel-patch</td>\n",
       "      <td>15</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>KQ031389.1</td>\n",
       "      <td>=</td>\n",
       "      <td>NW_012132920.1</td>\n",
       "      <td>PATCHES</td>\n",
       "      <td>2365364.0</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         # Sequence-Name      Sequence-Role Assigned-Molecule  \\\n",
       "881        HSCHR7_1_CTG6       alt-scaffold                 7   \n",
       "559         HG1708_PATCH          fix-patch                11   \n",
       "317   HSCHRUN_RANDOM_143  unplaced-scaffold                na   \n",
       "339   HSCHRUN_RANDOM_161  unplaced-scaffold                na   \n",
       "263   HSCHRUN_RANDOM_120  unplaced-scaffold                na   \n",
       "945     HG151_NOVEL_TEST       alt-scaffold                11   \n",
       "473       HSCHR4_8_CTG12        novel-patch                 4   \n",
       "607         HG2291_PATCH          fix-patch                13   \n",
       "1007      HSCHR15_1_CTG3       alt-scaffold                15   \n",
       "641       HSCHR15_6_CTG8        novel-patch                15   \n",
       "\n",
       "     Assigned-Molecule-Location/Type GenBank-Accn Relationship  \\\n",
       "881                       Chromosome   GL383534.2            =   \n",
       "559                       Chromosome   KV766195.1            =   \n",
       "317                               na   KI270529.1            =   \n",
       "339                               na   KI270584.1            =   \n",
       "263                               na   KI270422.1            =   \n",
       "945                       Chromosome   JH159137.1            =   \n",
       "473                       Chromosome   KQ090014.1            =   \n",
       "607                       Chromosome   KN538372.1            =   \n",
       "1007                      Chromosome   KI270848.1            =   \n",
       "641                       Chromosome   KQ031389.1            =   \n",
       "\n",
       "         RefSeq-Accn     Assembly-Unit  Sequence-Length       UCSC-style-name  \n",
       "881   NW_003315922.2    ALT_REF_LOCI_1         119183.0   chr7_GL383534v2_alt  \n",
       "559   NW_017363816.1           PATCHES         140877.0                    na  \n",
       "317      NT_187439.1  Primary Assembly           1899.0      chrUn_KI270529v1  \n",
       "339      NT_187453.1  Primary Assembly           4513.0      chrUn_KI270584v1  \n",
       "263      NT_187416.1  Primary Assembly           1445.0      chrUn_KI270422v1  \n",
       "945   NW_003871074.1    ALT_REF_LOCI_1         191409.0  chr11_JH159137v1_alt  \n",
       "473   NW_013171800.1           PATCHES         163749.0                    na  \n",
       "607   NW_011332699.1           PATCHES         356766.0                    na  \n",
       "1007     NT_187603.1    ALT_REF_LOCI_1         327382.0  chr15_KI270848v1_alt  \n",
       "641   NW_012132920.1           PATCHES        2365364.0                    na  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dna_data.sample(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "Int64Index: 640 entries, 1 to 1279\n",
      "Data columns (total 10 columns):\n",
      "# Sequence-Name                    640 non-null object\n",
      "Sequence-Role                      640 non-null object\n",
      "Assigned-Molecule                  640 non-null object\n",
      "Assigned-Molecule-Location/Type    640 non-null object\n",
      "GenBank-Accn                       640 non-null object\n",
      "Relationship                       640 non-null object\n",
      "RefSeq-Accn                        640 non-null object\n",
      "Assembly-Unit                      640 non-null object\n",
      "Sequence-Length                    640 non-null float64\n",
      "UCSC-style-name                    640 non-null object\n",
      "dtypes: float64(1), object(9)\n",
      "memory usage: 55.0+ KB\n"
     ]
    }
   ],
   "source": [
    "dna_data.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "640"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(dna_data)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "From the same https://www.ncbi.nlm.nih.gov/assembly/GCF_000001405.39/ page, under genome information, https://www.ncbi.nlm.nih.gov/genome/?term=txid9606[orgn], and in the Genome Assembly and Annotation report, https://www.ncbi.nlm.nih.gov/genome/genomes/51?, in the protein section at https://www.ncbi.nlm.nih.gov/genome/proteins/51?genome_assembly_id=582967, a comprehensive report of human proteins can be found. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "txt_file_protein = r\"ProteinTable51_582967.txt\"\n",
    "csv_file_protein = r\"ProteinTable51_582967.csv\"\n",
    "in_txt_protein = csv.reader(open(txt_file_protein, \"r\"), delimiter = '\\t')\n",
    "out_csv_protein = csv.writer(open(csv_file_protein, 'w'))\n",
    "out_csv_protein.writerows(in_txt_protein)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Replicon Name</th>\n",
       "      <th>Replicon Accession</th>\n",
       "      <th>Start</th>\n",
       "      <th>Stop</th>\n",
       "      <th>Strand</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>Locus</th>\n",
       "      <th>Protein product</th>\n",
       "      <th>Length</th>\n",
       "      <th>Protein name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>15509</td>\n",
       "      <td>1</td>\n",
       "      <td>NC_000001.11</td>\n",
       "      <td>196289844</td>\n",
       "      <td>196608309</td>\n",
       "      <td>-</td>\n",
       "      <td>343450</td>\n",
       "      <td>KCNT2</td>\n",
       "      <td>XP_016856668.1</td>\n",
       "      <td>923</td>\n",
       "      <td>potassium channel subfamily T member 2 isoform X3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14107</td>\n",
       "      <td>1</td>\n",
       "      <td>NC_000001.11</td>\n",
       "      <td>157796112</td>\n",
       "      <td>157820037</td>\n",
       "      <td>-</td>\n",
       "      <td>115350</td>\n",
       "      <td>FCRL1</td>\n",
       "      <td>XP_011507427.1</td>\n",
       "      <td>424</td>\n",
       "      <td>Fc receptor-like protein 1 isoform X1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>58847</td>\n",
       "      <td>9</td>\n",
       "      <td>NC_000009.12</td>\n",
       "      <td>41127130</td>\n",
       "      <td>41128383</td>\n",
       "      <td>-</td>\n",
       "      <td>653404</td>\n",
       "      <td>FOXD4L6</td>\n",
       "      <td>NP_001078945.1</td>\n",
       "      <td>417</td>\n",
       "      <td>forkhead box protein D4-like 6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>100154</td>\n",
       "      <td>18</td>\n",
       "      <td>NC_000018.10</td>\n",
       "      <td>2539067</td>\n",
       "      <td>2567216</td>\n",
       "      <td>-</td>\n",
       "      <td>64863</td>\n",
       "      <td>METTL4</td>\n",
       "      <td>XP_011524032.1</td>\n",
       "      <td>325</td>\n",
       "      <td>N(6)-adenine-specific DNA methyltransferase ME...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>73097</td>\n",
       "      <td>11</td>\n",
       "      <td>NC_000011.10</td>\n",
       "      <td>108287651</td>\n",
       "      <td>108365508</td>\n",
       "      <td>+</td>\n",
       "      <td>472</td>\n",
       "      <td>ATM</td>\n",
       "      <td>XP_006718908.1</td>\n",
       "      <td>1708</td>\n",
       "      <td>serine-protein kinase ATM isoform X8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>38450</td>\n",
       "      <td>5</td>\n",
       "      <td>NC_000005.10</td>\n",
       "      <td>37107602</td>\n",
       "      <td>37247698</td>\n",
       "      <td>-</td>\n",
       "      <td>65250</td>\n",
       "      <td>CPLANE1</td>\n",
       "      <td>XP_005248407.1</td>\n",
       "      <td>3208</td>\n",
       "      <td>ciliogenesis and planar polarity effector 1 is...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34748</td>\n",
       "      <td>4</td>\n",
       "      <td>NC_000004.12</td>\n",
       "      <td>53061810</td>\n",
       "      <td>53365941</td>\n",
       "      <td>-</td>\n",
       "      <td>152579</td>\n",
       "      <td>SCFD2</td>\n",
       "      <td>XP_011532677.1</td>\n",
       "      <td>533</td>\n",
       "      <td>sec1 family domain-containing protein 2 isofor...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>72346</td>\n",
       "      <td>11</td>\n",
       "      <td>NC_000011.10</td>\n",
       "      <td>77323274</td>\n",
       "      <td>77379385</td>\n",
       "      <td>-</td>\n",
       "      <td>5058</td>\n",
       "      <td>PAK1</td>\n",
       "      <td>NP_001363233.1</td>\n",
       "      <td>447</td>\n",
       "      <td>serine/threonine-protein kinase PAK 1 isoform 13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>44764</td>\n",
       "      <td>6</td>\n",
       "      <td>NC_000006.12</td>\n",
       "      <td>42228345</td>\n",
       "      <td>42269590</td>\n",
       "      <td>-</td>\n",
       "      <td>55809</td>\n",
       "      <td>TRERF1</td>\n",
       "      <td>NP_277037.1</td>\n",
       "      <td>1200</td>\n",
       "      <td>transcriptional-regulating factor 1 isoform 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>51667</td>\n",
       "      <td>7</td>\n",
       "      <td>NC_000007.14</td>\n",
       "      <td>103312569</td>\n",
       "      <td>103344622</td>\n",
       "      <td>-</td>\n",
       "      <td>27000</td>\n",
       "      <td>DNAJC2</td>\n",
       "      <td>NP_055192.1</td>\n",
       "      <td>621</td>\n",
       "      <td>dnaJ homolog subfamily C member 2 isoform 1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       #Replicon Name Replicon Accession      Start       Stop Strand  GeneID  \\\n",
       "15509               1       NC_000001.11  196289844  196608309      -  343450   \n",
       "14107               1       NC_000001.11  157796112  157820037      -  115350   \n",
       "58847               9       NC_000009.12   41127130   41128383      -  653404   \n",
       "100154             18       NC_000018.10    2539067    2567216      -   64863   \n",
       "73097              11       NC_000011.10  108287651  108365508      +     472   \n",
       "38450               5       NC_000005.10   37107602   37247698      -   65250   \n",
       "34748               4       NC_000004.12   53061810   53365941      -  152579   \n",
       "72346              11       NC_000011.10   77323274   77379385      -    5058   \n",
       "44764               6       NC_000006.12   42228345   42269590      -   55809   \n",
       "51667               7       NC_000007.14  103312569  103344622      -   27000   \n",
       "\n",
       "          Locus Protein product  Length  \\\n",
       "15509     KCNT2  XP_016856668.1     923   \n",
       "14107     FCRL1  XP_011507427.1     424   \n",
       "58847   FOXD4L6  NP_001078945.1     417   \n",
       "100154   METTL4  XP_011524032.1     325   \n",
       "73097       ATM  XP_006718908.1    1708   \n",
       "38450   CPLANE1  XP_005248407.1    3208   \n",
       "34748     SCFD2  XP_011532677.1     533   \n",
       "72346      PAK1  NP_001363233.1     447   \n",
       "44764    TRERF1     NP_277037.1    1200   \n",
       "51667    DNAJC2     NP_055192.1     621   \n",
       "\n",
       "                                             Protein name  \n",
       "15509   potassium channel subfamily T member 2 isoform X3  \n",
       "14107               Fc receptor-like protein 1 isoform X1  \n",
       "58847                      forkhead box protein D4-like 6  \n",
       "100154  N(6)-adenine-specific DNA methyltransferase ME...  \n",
       "73097                serine-protein kinase ATM isoform X8  \n",
       "38450   ciliogenesis and planar polarity effector 1 is...  \n",
       "34748   sec1 family domain-containing protein 2 isofor...  \n",
       "72346    serine/threonine-protein kinase PAK 1 isoform 13  \n",
       "44764       transcriptional-regulating factor 1 isoform 2  \n",
       "51667         dnaJ homolog subfamily C member 2 isoform 1  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "protein_data = pd.read_csv('ProteinTable51_582967.csv')\n",
    "protein_data.sample(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 119489 entries, 0 to 119488\n",
      "Data columns (total 10 columns):\n",
      "#Replicon Name        119489 non-null object\n",
      "Replicon Accession    119489 non-null object\n",
      "Start                 119489 non-null int64\n",
      "Stop                  119489 non-null int64\n",
      "Strand                119489 non-null object\n",
      "GeneID                119489 non-null int64\n",
      "Locus                 119489 non-null object\n",
      "Protein product       119489 non-null object\n",
      "Length                119489 non-null int64\n",
      "Protein name          119489 non-null object\n",
      "dtypes: int64(4), object(6)\n",
      "memory usage: 9.1+ MB\n"
     ]
    }
   ],
   "source": [
    "protein_data.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "119489"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(protein_data)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The protein and dna information is related through the RefSeq-Accn column on the dna_data table and the Replicon Accession column on the protein_data table. The two cells below show that many types of protein information is stored in chromosomes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th># Sequence-Name</th>\n",
       "      <th>Sequence-Role</th>\n",
       "      <th>Assigned-Molecule</th>\n",
       "      <th>Assigned-Molecule-Location/Type</th>\n",
       "      <th>GenBank-Accn</th>\n",
       "      <th>Relationship</th>\n",
       "      <th>RefSeq-Accn</th>\n",
       "      <th>Assembly-Unit</th>\n",
       "      <th>Sequence-Length</th>\n",
       "      <th>UCSC-style-name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>12</td>\n",
       "      <td>assembled-molecule</td>\n",
       "      <td>12</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>CM000674.2</td>\n",
       "      <td>=</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>Primary Assembly</td>\n",
       "      <td>133275309.0</td>\n",
       "      <td>chr12</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   # Sequence-Name       Sequence-Role Assigned-Molecule  \\\n",
       "23              12  assembled-molecule                12   \n",
       "\n",
       "   Assigned-Molecule-Location/Type GenBank-Accn Relationship   RefSeq-Accn  \\\n",
       "23                      Chromosome   CM000674.2            =  NC_000012.12   \n",
       "\n",
       "       Assembly-Unit  Sequence-Length UCSC-style-name  \n",
       "23  Primary Assembly      133275309.0           chr12  "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dna_data[dna_data['RefSeq-Accn'] == 'NC_000012.12']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Replicon Name</th>\n",
       "      <th>Replicon Accession</th>\n",
       "      <th>Start</th>\n",
       "      <th>Stop</th>\n",
       "      <th>Strand</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>Locus</th>\n",
       "      <th>Protein product</th>\n",
       "      <th>Length</th>\n",
       "      <th>Protein name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>74388</td>\n",
       "      <td>12</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>66883</td>\n",
       "      <td>175033</td>\n",
       "      <td>+</td>\n",
       "      <td>440073</td>\n",
       "      <td>IQSEC3</td>\n",
       "      <td>NP_001164209.1</td>\n",
       "      <td>1182</td>\n",
       "      <td>IQ motif and SEC7 domain-containing protein 3 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74389</td>\n",
       "      <td>12</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>66883</td>\n",
       "      <td>171308</td>\n",
       "      <td>+</td>\n",
       "      <td>440073</td>\n",
       "      <td>IQSEC3</td>\n",
       "      <td>XP_016874800.1</td>\n",
       "      <td>1041</td>\n",
       "      <td>IQ motif and SEC7 domain-containing protein 3 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74390</td>\n",
       "      <td>12</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>66883</td>\n",
       "      <td>171321</td>\n",
       "      <td>+</td>\n",
       "      <td>440073</td>\n",
       "      <td>IQSEC3</td>\n",
       "      <td>XP_011519262.1</td>\n",
       "      <td>1062</td>\n",
       "      <td>IQ motif and SEC7 domain-containing protein 3 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74391</td>\n",
       "      <td>12</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>66883</td>\n",
       "      <td>171308</td>\n",
       "      <td>+</td>\n",
       "      <td>440073</td>\n",
       "      <td>IQSEC3</td>\n",
       "      <td>XP_016874799.1</td>\n",
       "      <td>1041</td>\n",
       "      <td>IQ motif and SEC7 domain-containing protein 3 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74392</td>\n",
       "      <td>12</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>66883</td>\n",
       "      <td>174693</td>\n",
       "      <td>+</td>\n",
       "      <td>440073</td>\n",
       "      <td>IQSEC3</td>\n",
       "      <td>XP_011519260.1</td>\n",
       "      <td>1063</td>\n",
       "      <td>IQ motif and SEC7 domain-containing protein 3 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>80340</td>\n",
       "      <td>12</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>133218885</td>\n",
       "      <td>133234356</td>\n",
       "      <td>-</td>\n",
       "      <td>647589</td>\n",
       "      <td>ANHX</td>\n",
       "      <td>XP_006719806.1</td>\n",
       "      <td>492</td>\n",
       "      <td>anomalous homeobox protein isoform X1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>80341</td>\n",
       "      <td>12</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>133218885</td>\n",
       "      <td>133234356</td>\n",
       "      <td>-</td>\n",
       "      <td>647589</td>\n",
       "      <td>ANHX</td>\n",
       "      <td>XP_011533124.1</td>\n",
       "      <td>336</td>\n",
       "      <td>anomalous homeobox protein isoform X3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>80342</td>\n",
       "      <td>12</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>133221223</td>\n",
       "      <td>133234356</td>\n",
       "      <td>-</td>\n",
       "      <td>647589</td>\n",
       "      <td>ANHX</td>\n",
       "      <td>XP_011533127.1</td>\n",
       "      <td>322</td>\n",
       "      <td>anomalous homeobox protein isoform X5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>80343</td>\n",
       "      <td>12</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>133221223</td>\n",
       "      <td>133234356</td>\n",
       "      <td>-</td>\n",
       "      <td>647589</td>\n",
       "      <td>ANHX</td>\n",
       "      <td>XP_011533126.1</td>\n",
       "      <td>331</td>\n",
       "      <td>anomalous homeobox protein isoform X4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>80344</td>\n",
       "      <td>12</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>133221223</td>\n",
       "      <td>133234356</td>\n",
       "      <td>-</td>\n",
       "      <td>647589</td>\n",
       "      <td>ANHX</td>\n",
       "      <td>XP_011533125.1</td>\n",
       "      <td>331</td>\n",
       "      <td>anomalous homeobox protein isoform X4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5957 rows  10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #Replicon Name Replicon Accession      Start       Stop Strand  GeneID  \\\n",
       "74388             12       NC_000012.12      66883     175033      +  440073   \n",
       "74389             12       NC_000012.12      66883     171308      +  440073   \n",
       "74390             12       NC_000012.12      66883     171321      +  440073   \n",
       "74391             12       NC_000012.12      66883     171308      +  440073   \n",
       "74392             12       NC_000012.12      66883     174693      +  440073   \n",
       "...              ...                ...        ...        ...    ...     ...   \n",
       "80340             12       NC_000012.12  133218885  133234356      -  647589   \n",
       "80341             12       NC_000012.12  133218885  133234356      -  647589   \n",
       "80342             12       NC_000012.12  133221223  133234356      -  647589   \n",
       "80343             12       NC_000012.12  133221223  133234356      -  647589   \n",
       "80344             12       NC_000012.12  133221223  133234356      -  647589   \n",
       "\n",
       "        Locus Protein product  Length  \\\n",
       "74388  IQSEC3  NP_001164209.1    1182   \n",
       "74389  IQSEC3  XP_016874800.1    1041   \n",
       "74390  IQSEC3  XP_011519262.1    1062   \n",
       "74391  IQSEC3  XP_016874799.1    1041   \n",
       "74392  IQSEC3  XP_011519260.1    1063   \n",
       "...       ...             ...     ...   \n",
       "80340    ANHX  XP_006719806.1     492   \n",
       "80341    ANHX  XP_011533124.1     336   \n",
       "80342    ANHX  XP_011533127.1     322   \n",
       "80343    ANHX  XP_011533126.1     331   \n",
       "80344    ANHX  XP_011533125.1     331   \n",
       "\n",
       "                                            Protein name  \n",
       "74388  IQ motif and SEC7 domain-containing protein 3 ...  \n",
       "74389  IQ motif and SEC7 domain-containing protein 3 ...  \n",
       "74390  IQ motif and SEC7 domain-containing protein 3 ...  \n",
       "74391  IQ motif and SEC7 domain-containing protein 3 ...  \n",
       "74392  IQ motif and SEC7 domain-containing protein 3 ...  \n",
       "...                                                  ...  \n",
       "80340              anomalous homeobox protein isoform X1  \n",
       "80341              anomalous homeobox protein isoform X3  \n",
       "80342              anomalous homeobox protein isoform X5  \n",
       "80343              anomalous homeobox protein isoform X4  \n",
       "80344              anomalous homeobox protein isoform X4  \n",
       "\n",
       "[5957 rows x 10 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "protein_data[protein_data['Replicon Accession']== 'NC_000012.12']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "def get_keys(path):\n",
    "    with open(path) as f:\n",
    "        return json.load(f)\n",
    "keys = get_keys(\"/Users/eannefawcett/.secret/ncbi_api.json\")\n",
    "api_key = keys['api_key']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Entrez\n",
    "Entrez.api_key = api_key\n",
    "Entrez.email = 'elizabethfawcett47@gmail.com'\n",
    "Entrez.tool = 'locus_by_amino_acid'\n",
    "handle = Entrez.einfo()\n",
    "result = handle.read()\n",
    "handle.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<?xml version=\"1.0\" encoding=\"UTF-8\" ?>\n",
      "<!DOCTYPE eInfoResult PUBLIC \"-//NLM//DTD einfo 20190110//EN\" \"https://eutils.ncbi.nlm.nih.gov/eutils/dtd/20190110/einfo.dtd\">\n",
      "<eInfoResult>\n",
      "<DbList>\n",
      "\n",
      "\t<DbName>pubmed</DbName>\n",
      "\t<DbName>protein</DbName>\n",
      "\t<DbName>nuccore</DbName>\n",
      "\t<DbName>ipg</DbName>\n",
      "\t<DbName>nucleotide</DbName>\n",
      "\t<DbName>structure</DbName>\n",
      "\t<DbName>sparcle</DbName>\n",
      "\t<DbName>genome</DbName>\n",
      "\t<DbName>annotinfo</DbName>\n",
      "\t<DbName>assembly</DbName>\n",
      "\t<DbName>bioproject</DbName>\n",
      "\t<DbName>biosample</DbName>\n",
      "\t<DbName>blastdbinfo</DbName>\n",
      "\t<DbName>books</DbName>\n",
      "\t<DbName>cdd</DbName>\n",
      "\t<DbName>clinvar</DbName>\n",
      "\t<DbName>gap</DbName>\n",
      "\t<DbName>gapplus</DbName>\n",
      "\t<DbName>grasp</DbName>\n",
      "\t<DbName>dbvar</DbName>\n",
      "\t<DbName>gene</DbName>\n",
      "\t<DbName>gds</DbName>\n",
      "\t<DbName>geoprofiles</DbName>\n",
      "\t<DbName>homologene</DbName>\n",
      "\t<DbName>medgen</DbName>\n",
      "\t<DbName>mesh</DbName>\n",
      "\t<DbName>ncbisearch</DbName>\n",
      "\t<DbName>nlmcatalog</DbName>\n",
      "\t<DbName>omim</DbName>\n",
      "\t<DbName>orgtrack</DbName>\n",
      "\t<DbName>pmc</DbName>\n",
      "\t<DbName>popset</DbName>\n",
      "\t<DbName>probe</DbName>\n",
      "\t<DbName>proteinclusters</DbName>\n",
      "\t<DbName>pcassay</DbName>\n",
      "\t<DbName>biosystems</DbName>\n",
      "\t<DbName>pccompound</DbName>\n",
      "\t<DbName>pcsubstance</DbName>\n",
      "\t<DbName>seqannot</DbName>\n",
      "\t<DbName>snp</DbName>\n",
      "\t<DbName>sra</DbName>\n",
      "\t<DbName>taxonomy</DbName>\n",
      "\t<DbName>biocollections</DbName>\n",
      "\t<DbName>gtr</DbName>\n",
      "</DbList>\n",
      "\n",
      "</eInfoResult>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "handle = Entrez.einfo()\n",
    "record = Entrez.read(handle)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['DbList'])"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "record.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['pubmed', 'protein', 'nuccore', 'ipg', 'nucleotide', 'structure', 'sparcle', 'genome', 'annotinfo', 'assembly', 'bioproject', 'biosample', 'blastdbinfo', 'books', 'cdd', 'clinvar', 'gap', 'gapplus', 'grasp', 'dbvar', 'gene', 'gds', 'geoprofiles', 'homologene', 'medgen', 'mesh', 'ncbisearch', 'nlmcatalog', 'omim', 'orgtrack', 'pmc', 'popset', 'probe', 'proteinclusters', 'pcassay', 'biosystems', 'pccompound', 'pcsubstance', 'seqannot', 'snp', 'sra', 'taxonomy', 'biocollections', 'gtr']"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "record['DbList']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Protein sequence record'"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from Bio import Entrez\n",
    "Entrez.api_key = api_key\n",
    "Entrez.email = 'elizabethfawcett47@gmail.com'\n",
    "Entrez.tool = 'locus_by_amino_acid'\n",
    "handle = Entrez.einfo(db=\"protein\")\n",
    "record = Entrez.read(handle)\n",
    "record[\"DbInfo\"][\"Description\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'831488909'"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "record[\"DbInfo\"][\"Count\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'2020/03/16 21:46'"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "record[\"DbInfo\"][\"LastUpdate\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['DbName', 'MenuName', 'Description', 'DbBuild', 'Count', 'LastUpdate', 'FieldList', 'LinkList'])"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "record[\"DbInfo\"].keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ALL, All Fields, All terms from all searchable fields\n",
      "UID, UID, Unique number assigned to each sequence\n",
      "FILT, Filter, Limits the records\n",
      "WORD, Text Word, Free text associated with record\n",
      "TITL, Title, Words in definition line\n",
      "KYWD, Keyword, Nonstandardized terms provided by submitter\n",
      "AUTH, Author, Author(s) of publication\n",
      "JOUR, Journal, Journal abbreviation of publication\n",
      "VOL, Volume, Volume number of publication\n",
      "ISS, Issue, Issue number of publication\n",
      "PAGE, Page Number, Page number(s) of publication\n",
      "ORGN, Organism, Scientific and common names of organism, and all higher levels of taxonomy\n",
      "ACCN, Accession, Accession number of sequence\n",
      "PACC, Primary Accession, Does not include retired secondary accessions\n",
      "GENE, Gene Name, Name of gene associated with sequence\n",
      "PROT, Protein Name, Name of protein associated with sequence\n",
      "ECNO, EC/RN Number, EC number for enzyme or CAS registry number\n",
      "PDAT, Publication Date, Date sequence added to GenBank\n",
      "MDAT, Modification Date, Date of last update\n",
      "SUBS, Substance Name, CAS chemical name or MEDLINE Substance Name\n",
      "PROP, Properties, Classification by source qualifiers and molecule type\n",
      "SQID, SeqID String, String identifier for sequence\n",
      "GPRJ, BioProject, BioProject\n",
      "SLEN, Sequence Length, Length of sequence\n",
      "MLWT, Molecular Weight, Molecular Weight\n",
      "FKEY, Feature key, Feature annotated on sequence\n",
      "PORG, Primary Organism, Scientific and common names of primary organism, and all higher levels of taxonomy\n",
      "ASSM, Assembly, Assembly\n",
      "DIV, Division, Division\n",
      "STRN, Strain, Strain\n",
      "ISOL, Isolate, Isolate\n",
      "CULT, Cultivar, Cultivar\n",
      "BRD, Breed, Breed\n"
     ]
    }
   ],
   "source": [
    "for field in record[\"DbInfo\"][\"FieldList\"]:\n",
    "    print(\"%(Name)s, %(FullName)s, %(Description)s\" % field)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Entrez\n",
    "Entrez.api_key = api_key\n",
    "Entrez.email = 'elizabethfawcett47@gmail.com'\n",
    "Entrez.tool = 'locus_by_amino_acid'\n",
    "handle = Entrez.einfo(db=\"protein\", ACCN='NP_001164209.1')\n",
    "record = Entrez.read(handle)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'DbInfo': {'DbName': 'protein', 'MenuName': 'Protein', 'Description': 'Protein sequence record', 'DbBuild': 'Build200315-0206m.1', 'Count': '831488909', 'LastUpdate': '2020/03/16 21:46', 'FieldList': [{'Name': 'ALL', 'FullName': 'All Fields', 'Description': 'All terms from all searchable fields', 'TermCount': '4427725646', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'N', 'Hierarchy': 'N', 'IsHidden': 'N'}, {'Name': 'UID', 'FullName': 'UID', 'Description': 'Unique number assigned to each sequence', 'TermCount': '0', 'IsDate': 'N', 'IsNumerical': 'Y', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'Y'}, {'Name': 'FILT', 'FullName': 'Filter', 'Description': 'Limits the records', 'TermCount': '260', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}, {'Name': 'WORD', 'FullName': 'Text Word', 'Description': 'Free text associated with record', 'TermCount': '1159081284', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'N', 'Hierarchy': 'N', 'IsHidden': 'N'}, {'Name': 'TITL', 'FullName': 'Title', 'Description': 'Words in definition line', 'TermCount': '134102182', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'N', 'Hierarchy': 'N', 'IsHidden': 'N'}, {'Name': 'KYWD', 'FullName': 'Keyword', 'Description': 'Nonstandardized terms provided by submitter', 'TermCount': '32361', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}, {'Name': 'AUTH', 'FullName': 'Author', 'Description': 'Author(s) of publication', 'TermCount': '2816926', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}, {'Name': 'JOUR', 'FullName': 'Journal', 'Description': 'Journal abbreviation of publication', 'TermCount': '85573', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}, {'Name': 'VOL', 'FullName': 'Volume', 'Description': 'Volume number of publication', 'TermCount': '4633', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}, {'Name': 'ISS', 'FullName': 'Issue', 'Description': 'Issue number of publication', 'TermCount': '4639', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}, {'Name': 'PAGE', 'FullName': 'Page Number', 'Description': 'Page number(s) of publication', 'TermCount': '204739', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}, {'Name': 'ORGN', 'FullName': 'Organism', 'Description': 'Scientific and common names of organism, and all higher levels of taxonomy', 'TermCount': '3676060', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'Y', 'IsHidden': 'N'}, {'Name': 'ACCN', 'FullName': 'Accession', 'Description': 'Accession number of sequence', 'TermCount': '1664479509', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}, {'Name': 'PACC', 'FullName': 'Primary Accession', 'Description': 'Does not include retired secondary accessions', 'TermCount': '1662924655', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}, {'Name': 'GENE', 'FullName': 'Gene Name', 'Description': 'Name of gene associated with sequence', 'TermCount': '1273647323', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}, {'Name': 'PROT', 'FullName': 'Protein Name', 'Description': 'Name of protein associated with sequence', 'TermCount': '6847131', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}, {'Name': 'ECNO', 'FullName': 'EC/RN Number', 'Description': 'EC number for enzyme or CAS registry number', 'TermCount': '6214', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}, {'Name': 'PDAT', 'FullName': 'Publication Date', 'Description': 'Date sequence added to GenBank', 'TermCount': '11535', 'IsDate': 'Y', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}, {'Name': 'MDAT', 'FullName': 'Modification Date', 'Description': 'Date of last update', 'TermCount': '6916', 'IsDate': 'Y', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}, {'Name': 'SUBS', 'FullName': 'Substance Name', 'Description': 'CAS chemical name or MEDLINE Substance Name', 'TermCount': '913681', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}, {'Name': 'PROP', 'FullName': 'Properties', 'Description': 'Classification by source qualifiers and molecule type', 'TermCount': '161343', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}, {'Name': 'SQID', 'FullName': 'SeqID String', 'Description': 'String identifier for sequence', 'TermCount': '623643845', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}, {'Name': 'GPRJ', 'FullName': 'BioProject', 'Description': 'BioProject', 'TermCount': '140917', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'Y', 'IsHidden': 'N'}, {'Name': 'SLEN', 'FullName': 'Sequence Length', 'Description': 'Length of sequence', 'TermCount': '15029', 'IsDate': 'N', 'IsNumerical': 'Y', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}, {'Name': 'MLWT', 'FullName': 'Molecular Weight', 'Description': 'Molecular Weight', 'TermCount': '566834', 'IsDate': 'N', 'IsNumerical': 'Y', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}, {'Name': 'FKEY', 'FullName': 'Feature key', 'Description': 'Feature annotated on sequence', 'TermCount': '32', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}, {'Name': 'PORG', 'FullName': 'Primary Organism', 'Description': 'Scientific and common names of primary organism, and all higher levels of taxonomy', 'TermCount': '3668032', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'Y', 'IsHidden': 'N'}, {'Name': 'ASSM', 'FullName': 'Assembly', 'Description': 'Assembly', 'TermCount': '77', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}, {'Name': 'DIV', 'FullName': 'Division', 'Description': 'Division', 'TermCount': '13', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}, {'Name': 'STRN', 'FullName': 'Strain', 'Description': 'Strain', 'TermCount': '937942', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}, {'Name': 'ISOL', 'FullName': 'Isolate', 'Description': 'Isolate', 'TermCount': '2734730', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}, {'Name': 'CULT', 'FullName': 'Cultivar', 'Description': 'Cultivar', 'TermCount': '24108', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}, {'Name': 'BRD', 'FullName': 'Breed', 'Description': 'Breed', 'TermCount': '3224', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}], 'LinkList': [{'Name': 'nuccore_protein_wp', 'Menu': 'Links to autonomous proteins', 'Description': 'Autonomous protein records', 'DbTo': 'nuccore'}, {'Name': 'protein_biocollections', 'Menu': 'BioCollections', 'Description': 'BioCollections', 'DbTo': 'biocollections'}, {'Name': 'protein_bioproject', 'Menu': 'BioProject Links', 'Description': 'Proteins related to BioProjects', 'DbTo': 'bioproject'}, {'Name': 'protein_biosystems', 'Menu': 'BioSystem Links', 'Description': 'BioSystems', 'DbTo': 'biosystems'}, {'Name': 'protein_ccds', 'Menu': 'ccds', 'Description': 'Link to Consensus CDS', 'DbTo': 'ccds'}, {'Name': 'protein_cdd', 'Menu': 'Conserved Domain Links', 'Description': 'Link to conserved domains within a protein', 'DbTo': 'cdd'}, {'Name': 'protein_cdd_concise_2', 'Menu': 'Concise Conserved Domain Links', 'Description': 'Link to a concise list of conserved domains in a protein', 'DbTo': 'cdd'}, {'Name': 'protein_cdd_summary', 'Menu': '', 'Description': 'Display conserved domains within a protein', 'DbTo': 'cdd'}, {'Name': 'protein_gene', 'Menu': 'Gene Links', 'Description': 'Link to related Genes', 'DbTo': 'gene'}, {'Name': 'protein_genome', 'Menu': 'Genome Links', 'Description': 'Genome record encoding protein sequence', 'DbTo': 'genome'}, {'Name': 'protein_homologene', 'Menu': 'HomoloGene Links', 'Description': 'Related HomoloGene', 'DbTo': 'homologene'}, {'Name': 'protein_nuccore', 'Menu': 'Nucleotide Links', 'Description': 'Nucleotide sequence from coding region', 'DbTo': 'nuccore'}, {'Name': 'protein_nuccore_mat_peptide', 'Menu': 'Mature Peptides', 'Description': 'Link between nucleotide record and mature peptides contained within nucleotide record', 'DbTo': 'nuccore'}, {'Name': 'protein_nuccore_mgc', 'Menu': 'Nucleotide NIH cDNA clone links', 'Description': 'Nucleotide NIH cDNA clone', 'DbTo': 'nuccore'}, {'Name': 'protein_nuccore_mrna', 'Menu': 'Encoding mRNA', 'Description': 'Link from protein to encoding mRNA', 'DbTo': 'nuccore'}, {'Name': 'protein_nuccore_small_genome', 'Menu': 'Link to all nucleotide sequences from this genome', 'Description': 'All nucleotide sequences from this genome', 'DbTo': 'nuccore'}, {'Name': 'protein_nuccore_tsa', 'Menu': 'TSA master records', 'Description': 'Links to TSA master records', 'DbTo': 'nuccore'}, {'Name': 'protein_nuccore_wgs', 'Menu': '', 'Description': 'Links to WGS master sequences', 'DbTo': 'nuccore'}, {'Name': 'protein_nuccore_wp', 'Menu': 'Link to genomic records', 'Description': 'Genomic records', 'DbTo': 'nuccore'}, {'Name': 'protein_nucgss', 'Menu': 'GSS Links', 'Description': 'GSS sequence from coding region', 'DbTo': 'nucgss'}, {'Name': 'protein_nucleotide_mgc_url', 'Menu': '', 'Description': 'Order cDNA clone', 'DbTo': 'nucleotide'}, {'Name': 'protein_omim', 'Menu': 'OMIM Links', 'Description': 'OMIM records associated with protein sequence', 'DbTo': 'omim'}, {'Name': 'protein_pcassay_target', 'Menu': 'BioAssay by Target (List)', 'Description': 'PubChem BioAssays done on the protein target', 'DbTo': 'pcassay'}, {'Name': 'protein_pcassay_target_pig', 'Menu': 'BioAssay by Target (Identical Proteins, List)', 'Description': 'PubChem BioAssays done on the protein target PIG', 'DbTo': 'pcassay'}, {'Name': 'protein_pcassay_target_pig_summary', 'Menu': 'pcassay', 'Description': 'BioAssay by Target (Identical Proteins, Summary)', 'DbTo': 'pcassay'}, {'Name': 'protein_pcassay_target_summary', 'Menu': 'pcassay', 'Description': 'BioAssay by Target (Summary)', 'DbTo': 'pcassay'}, {'Name': 'protein_pccompound', 'Menu': 'PubChem Compound Links', 'Description': 'Related PubChem Compound', 'DbTo': 'pccompound'}, {'Name': 'protein_pcsubstance', 'Menu': 'PubChem Substance Links', 'Description': 'Related PubChem Substance', 'DbTo': 'pcsubstance'}, {'Name': 'protein_pmc', 'Menu': 'PMC Links', 'Description': 'Free full text articles in PMC', 'DbTo': 'pmc'}, {'Name': 'protein_popset', 'Menu': 'PopSet Links', 'Description': 'Population set containing protein sequence', 'DbTo': 'popset'}, {'Name': 'protein_protein_cdart_summary', 'Menu': '', 'Description': 'Display proteins with similar conserved domain architecture', 'DbTo': 'protein'}, {'Name': 'protein_protein_ref2wp', 'Menu': 'Links from references to autonomous proteins', 'Description': 'Autonomous protein records', 'DbTo': 'protein'}, {'Name': 'protein_protein_refseq2uniprot', 'Menu': 'Protein (UniProtKB)', 'Description': 'Protein gi (refseq) to protein gi (UniProtKB)', 'DbTo': 'protein'}, {'Name': 'protein_protein_small_genome', 'Menu': 'Link to all proteins from this genome', 'Description': 'All proteins from this genome', 'DbTo': 'protein'}, {'Name': 'protein_protein_uniprot2refseq', 'Menu': 'Protein (RefSeq)', 'Description': 'Protein gi (UniProtKB) to protein gi (refseq)', 'DbTo': 'protein'}, {'Name': 'protein_protein_wp2ref', 'Menu': 'Link to referencing proteins', 'Description': 'Referencing proteins', 'DbTo': 'protein'}, {'Name': 'protein_proteinclusters', 'Menu': 'Protein Cluster Links', 'Description': 'Related Protein Clusters', 'DbTo': 'proteinclusters'}, {'Name': 'protein_pubmed', 'Menu': 'PubMed Links', 'Description': 'PubMed articles cited by protein sequence record', 'DbTo': 'pubmed'}, {'Name': 'protein_pubmed_refseq', 'Menu': 'PubMed (RefSeq) Links', 'Description': 'Link to RefSeq PubMed articles', 'DbTo': 'pubmed'}, {'Name': 'protein_pubmed_weighted', 'Menu': 'PubMed (Weighted) Links', 'Description': 'Links to pubmed', 'DbTo': 'pubmed'}, {'Name': 'protein_snp', 'Menu': 'SNP Links', 'Description': 'Related SNPs', 'DbTo': 'snp'}, {'Name': 'protein_snp_geneview', 'Menu': '', 'Description': 'SNPs linked from GeneView', 'DbTo': 'snp'}, {'Name': 'protein_structure', 'Menu': 'Identical Structures', 'Description': '3D structures for identical protein sequences', 'DbTo': 'structure'}, {'Name': 'protein_structure_direct', 'Menu': 'Structures', 'Description': 'Experimentally determined 3D Structures', 'DbTo': 'structure'}, {'Name': 'protein_structure_related', 'Menu': '', 'Description': 'Related structures', 'DbTo': 'structure'}, {'Name': 'protein_structure_related_list', 'Menu': 'Related Structures (List)', 'Description': 'List of related structures', 'DbTo': 'structure'}, {'Name': 'protein_taxonomy', 'Menu': 'Taxonomy Links', 'Description': 'Taxonomy sequences associated with protein record', 'DbTo': 'taxonomy'}, {'Name': 'protein_taxonomy_wp2species', 'Menu': 'Links from autonomous proteins to species level organisms', 'Description': 'Species level organisms', 'DbTo': 'taxonomy'}]}}"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "record"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['DbInfo'])"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "record.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['DbName', 'MenuName', 'Description', 'DbBuild', 'Count', 'LastUpdate', 'FieldList', 'LinkList'])"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "record['DbInfo'].keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'protein'"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "record['DbInfo']['DbName']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Protein'"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "record['DbInfo']['MenuName']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Protein sequence record'"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "record['DbInfo']['Description']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Build200315-0206m.1'"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "record['DbInfo']['DbBuild']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'831488909'"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "record['DbInfo']['Count']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'2020/03/16 21:46'"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "record['DbInfo']['LastUpdate']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "33"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(record['DbInfo']['FieldList'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ALL, All Fields, All terms from all searchable fields\n",
      "UID, UID, Unique number assigned to each sequence\n",
      "FILT, Filter, Limits the records\n",
      "WORD, Text Word, Free text associated with record\n",
      "TITL, Title, Words in definition line\n",
      "KYWD, Keyword, Nonstandardized terms provided by submitter\n",
      "AUTH, Author, Author(s) of publication\n",
      "JOUR, Journal, Journal abbreviation of publication\n",
      "VOL, Volume, Volume number of publication\n",
      "ISS, Issue, Issue number of publication\n",
      "PAGE, Page Number, Page number(s) of publication\n",
      "ORGN, Organism, Scientific and common names of organism, and all higher levels of taxonomy\n",
      "ACCN, Accession, Accession number of sequence\n",
      "PACC, Primary Accession, Does not include retired secondary accessions\n",
      "GENE, Gene Name, Name of gene associated with sequence\n",
      "PROT, Protein Name, Name of protein associated with sequence\n",
      "ECNO, EC/RN Number, EC number for enzyme or CAS registry number\n",
      "PDAT, Publication Date, Date sequence added to GenBank\n",
      "MDAT, Modification Date, Date of last update\n",
      "SUBS, Substance Name, CAS chemical name or MEDLINE Substance Name\n",
      "PROP, Properties, Classification by source qualifiers and molecule type\n",
      "SQID, SeqID String, String identifier for sequence\n",
      "GPRJ, BioProject, BioProject\n",
      "SLEN, Sequence Length, Length of sequence\n",
      "MLWT, Molecular Weight, Molecular Weight\n",
      "FKEY, Feature key, Feature annotated on sequence\n",
      "PORG, Primary Organism, Scientific and common names of primary organism, and all higher levels of taxonomy\n",
      "ASSM, Assembly, Assembly\n",
      "DIV, Division, Division\n",
      "STRN, Strain, Strain\n",
      "ISOL, Isolate, Isolate\n",
      "CULT, Cultivar, Cultivar\n",
      "BRD, Breed, Breed\n"
     ]
    }
   ],
   "source": [
    "for field in record[\"DbInfo\"][\"FieldList\"]:\n",
    "    print(\"%(Name)s, %(FullName)s, %(Description)s\" % field)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'Name': 'ALL', 'FullName': 'All Fields', 'Description': 'All terms from all searchable fields', 'TermCount': '4427725646', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'N', 'Hierarchy': 'N', 'IsHidden': 'N'}\n",
      "\n",
      "\n",
      "{'Name': 'UID', 'FullName': 'UID', 'Description': 'Unique number assigned to each sequence', 'TermCount': '0', 'IsDate': 'N', 'IsNumerical': 'Y', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'Y'}\n",
      "\n",
      "\n",
      "{'Name': 'FILT', 'FullName': 'Filter', 'Description': 'Limits the records', 'TermCount': '260', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}\n",
      "\n",
      "\n",
      "{'Name': 'WORD', 'FullName': 'Text Word', 'Description': 'Free text associated with record', 'TermCount': '1159081284', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'N', 'Hierarchy': 'N', 'IsHidden': 'N'}\n",
      "\n",
      "\n",
      "{'Name': 'TITL', 'FullName': 'Title', 'Description': 'Words in definition line', 'TermCount': '134102182', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'N', 'Hierarchy': 'N', 'IsHidden': 'N'}\n",
      "\n",
      "\n",
      "{'Name': 'KYWD', 'FullName': 'Keyword', 'Description': 'Nonstandardized terms provided by submitter', 'TermCount': '32361', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}\n",
      "\n",
      "\n",
      "{'Name': 'AUTH', 'FullName': 'Author', 'Description': 'Author(s) of publication', 'TermCount': '2816926', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}\n",
      "\n",
      "\n",
      "{'Name': 'JOUR', 'FullName': 'Journal', 'Description': 'Journal abbreviation of publication', 'TermCount': '85573', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}\n",
      "\n",
      "\n",
      "{'Name': 'VOL', 'FullName': 'Volume', 'Description': 'Volume number of publication', 'TermCount': '4633', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}\n",
      "\n",
      "\n",
      "{'Name': 'ISS', 'FullName': 'Issue', 'Description': 'Issue number of publication', 'TermCount': '4639', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}\n",
      "\n",
      "\n",
      "{'Name': 'PAGE', 'FullName': 'Page Number', 'Description': 'Page number(s) of publication', 'TermCount': '204739', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}\n",
      "\n",
      "\n",
      "{'Name': 'ORGN', 'FullName': 'Organism', 'Description': 'Scientific and common names of organism, and all higher levels of taxonomy', 'TermCount': '3676060', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'Y', 'IsHidden': 'N'}\n",
      "\n",
      "\n",
      "{'Name': 'ACCN', 'FullName': 'Accession', 'Description': 'Accession number of sequence', 'TermCount': '1664479509', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}\n",
      "\n",
      "\n",
      "{'Name': 'PACC', 'FullName': 'Primary Accession', 'Description': 'Does not include retired secondary accessions', 'TermCount': '1662924655', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}\n",
      "\n",
      "\n",
      "{'Name': 'GENE', 'FullName': 'Gene Name', 'Description': 'Name of gene associated with sequence', 'TermCount': '1273647323', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}\n",
      "\n",
      "\n",
      "{'Name': 'PROT', 'FullName': 'Protein Name', 'Description': 'Name of protein associated with sequence', 'TermCount': '6847131', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}\n",
      "\n",
      "\n",
      "{'Name': 'ECNO', 'FullName': 'EC/RN Number', 'Description': 'EC number for enzyme or CAS registry number', 'TermCount': '6214', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}\n",
      "\n",
      "\n",
      "{'Name': 'PDAT', 'FullName': 'Publication Date', 'Description': 'Date sequence added to GenBank', 'TermCount': '11535', 'IsDate': 'Y', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}\n",
      "\n",
      "\n",
      "{'Name': 'MDAT', 'FullName': 'Modification Date', 'Description': 'Date of last update', 'TermCount': '6916', 'IsDate': 'Y', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}\n",
      "\n",
      "\n",
      "{'Name': 'SUBS', 'FullName': 'Substance Name', 'Description': 'CAS chemical name or MEDLINE Substance Name', 'TermCount': '913681', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}\n",
      "\n",
      "\n",
      "{'Name': 'PROP', 'FullName': 'Properties', 'Description': 'Classification by source qualifiers and molecule type', 'TermCount': '161343', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}\n",
      "\n",
      "\n",
      "{'Name': 'SQID', 'FullName': 'SeqID String', 'Description': 'String identifier for sequence', 'TermCount': '623643845', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}\n",
      "\n",
      "\n",
      "{'Name': 'GPRJ', 'FullName': 'BioProject', 'Description': 'BioProject', 'TermCount': '140917', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'Y', 'IsHidden': 'N'}\n",
      "\n",
      "\n",
      "{'Name': 'SLEN', 'FullName': 'Sequence Length', 'Description': 'Length of sequence', 'TermCount': '15029', 'IsDate': 'N', 'IsNumerical': 'Y', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}\n",
      "\n",
      "\n",
      "{'Name': 'MLWT', 'FullName': 'Molecular Weight', 'Description': 'Molecular Weight', 'TermCount': '566834', 'IsDate': 'N', 'IsNumerical': 'Y', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}\n",
      "\n",
      "\n",
      "{'Name': 'FKEY', 'FullName': 'Feature key', 'Description': 'Feature annotated on sequence', 'TermCount': '32', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}\n",
      "\n",
      "\n",
      "{'Name': 'PORG', 'FullName': 'Primary Organism', 'Description': 'Scientific and common names of primary organism, and all higher levels of taxonomy', 'TermCount': '3668032', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'Y', 'IsHidden': 'N'}\n",
      "\n",
      "\n",
      "{'Name': 'ASSM', 'FullName': 'Assembly', 'Description': 'Assembly', 'TermCount': '77', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}\n",
      "\n",
      "\n",
      "{'Name': 'DIV', 'FullName': 'Division', 'Description': 'Division', 'TermCount': '13', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}\n",
      "\n",
      "\n",
      "{'Name': 'STRN', 'FullName': 'Strain', 'Description': 'Strain', 'TermCount': '937942', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}\n",
      "\n",
      "\n",
      "{'Name': 'ISOL', 'FullName': 'Isolate', 'Description': 'Isolate', 'TermCount': '2734730', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}\n",
      "\n",
      "\n",
      "{'Name': 'CULT', 'FullName': 'Cultivar', 'Description': 'Cultivar', 'TermCount': '24108', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}\n",
      "\n",
      "\n",
      "{'Name': 'BRD', 'FullName': 'Breed', 'Description': 'Breed', 'TermCount': '3224', 'IsDate': 'N', 'IsNumerical': 'N', 'SingleToken': 'Y', 'Hierarchy': 'N', 'IsHidden': 'N'}\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for field in record['DbInfo']['FieldList']:\n",
    "    print(field)\n",
    "    print('\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "nuccore_protein_wp, Links to autonomous proteins, Autonomous protein records\n",
      "protein_biocollections, BioCollections, BioCollections\n",
      "protein_bioproject, BioProject Links, Proteins related to BioProjects\n",
      "protein_biosystems, BioSystem Links, BioSystems\n",
      "protein_ccds, ccds, Link to Consensus CDS\n",
      "protein_cdd, Conserved Domain Links, Link to conserved domains within a protein\n",
      "protein_cdd_concise_2, Concise Conserved Domain Links, Link to a concise list of conserved domains in a protein\n",
      "protein_cdd_summary, , Display conserved domains within a protein\n",
      "protein_gene, Gene Links, Link to related Genes\n",
      "protein_genome, Genome Links, Genome record encoding protein sequence\n",
      "protein_homologene, HomoloGene Links, Related HomoloGene\n",
      "protein_nuccore, Nucleotide Links, Nucleotide sequence from coding region\n",
      "protein_nuccore_mat_peptide, Mature Peptides, Link between nucleotide record and mature peptides contained within nucleotide record\n",
      "protein_nuccore_mgc, Nucleotide NIH cDNA clone links, Nucleotide NIH cDNA clone\n",
      "protein_nuccore_mrna, Encoding mRNA, Link from protein to encoding mRNA\n",
      "protein_nuccore_small_genome, Link to all nucleotide sequences from this genome, All nucleotide sequences from this genome\n",
      "protein_nuccore_tsa, TSA master records, Links to TSA master records\n",
      "protein_nuccore_wgs, , Links to WGS master sequences\n",
      "protein_nuccore_wp, Link to genomic records, Genomic records\n",
      "protein_nucgss, GSS Links, GSS sequence from coding region\n",
      "protein_nucleotide_mgc_url, , Order cDNA clone\n",
      "protein_omim, OMIM Links, OMIM records associated with protein sequence\n",
      "protein_pcassay_target, BioAssay by Target (List), PubChem BioAssays done on the protein target\n",
      "protein_pcassay_target_pig, BioAssay by Target (Identical Proteins, List), PubChem BioAssays done on the protein target PIG\n",
      "protein_pcassay_target_pig_summary, pcassay, BioAssay by Target (Identical Proteins, Summary)\n",
      "protein_pcassay_target_summary, pcassay, BioAssay by Target (Summary)\n",
      "protein_pccompound, PubChem Compound Links, Related PubChem Compound\n",
      "protein_pcsubstance, PubChem Substance Links, Related PubChem Substance\n",
      "protein_pmc, PMC Links, Free full text articles in PMC\n",
      "protein_popset, PopSet Links, Population set containing protein sequence\n",
      "protein_protein_cdart_summary, , Display proteins with similar conserved domain architecture\n",
      "protein_protein_ref2wp, Links from references to autonomous proteins, Autonomous protein records\n",
      "protein_protein_refseq2uniprot, Protein (UniProtKB), Protein gi (refseq) to protein gi (UniProtKB)\n",
      "protein_protein_small_genome, Link to all proteins from this genome, All proteins from this genome\n",
      "protein_protein_uniprot2refseq, Protein (RefSeq), Protein gi (UniProtKB) to protein gi (refseq)\n",
      "protein_protein_wp2ref, Link to referencing proteins, Referencing proteins\n",
      "protein_proteinclusters, Protein Cluster Links, Related Protein Clusters\n",
      "protein_pubmed, PubMed Links, PubMed articles cited by protein sequence record\n",
      "protein_pubmed_refseq, PubMed (RefSeq) Links, Link to RefSeq PubMed articles\n",
      "protein_pubmed_weighted, PubMed (Weighted) Links, Links to pubmed\n",
      "protein_snp, SNP Links, Related SNPs\n",
      "protein_snp_geneview, , SNPs linked from GeneView\n",
      "protein_structure, Identical Structures, 3D structures for identical protein sequences\n",
      "protein_structure_direct, Structures, Experimentally determined 3D Structures\n",
      "protein_structure_related, , Related structures\n",
      "protein_structure_related_list, Related Structures (List), List of related structures\n",
      "protein_taxonomy, Taxonomy Links, Taxonomy sequences associated with protein record\n",
      "protein_taxonomy_wp2species, Links from autonomous proteins to species level organisms, Species level organisms\n"
     ]
    }
   ],
   "source": [
    "for field in record[\"DbInfo\"][\"LinkList\"]:\n",
    "    print(\"%(Name)s, %(Menu)s, %(Description)s\" % field)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'Name': 'nuccore_protein_wp', 'Menu': 'Links to autonomous proteins', 'Description': 'Autonomous protein records', 'DbTo': 'nuccore'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_biocollections', 'Menu': 'BioCollections', 'Description': 'BioCollections', 'DbTo': 'biocollections'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_bioproject', 'Menu': 'BioProject Links', 'Description': 'Proteins related to BioProjects', 'DbTo': 'bioproject'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_biosystems', 'Menu': 'BioSystem Links', 'Description': 'BioSystems', 'DbTo': 'biosystems'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_ccds', 'Menu': 'ccds', 'Description': 'Link to Consensus CDS', 'DbTo': 'ccds'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_cdd', 'Menu': 'Conserved Domain Links', 'Description': 'Link to conserved domains within a protein', 'DbTo': 'cdd'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_cdd_concise_2', 'Menu': 'Concise Conserved Domain Links', 'Description': 'Link to a concise list of conserved domains in a protein', 'DbTo': 'cdd'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_cdd_summary', 'Menu': '', 'Description': 'Display conserved domains within a protein', 'DbTo': 'cdd'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_gene', 'Menu': 'Gene Links', 'Description': 'Link to related Genes', 'DbTo': 'gene'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_genome', 'Menu': 'Genome Links', 'Description': 'Genome record encoding protein sequence', 'DbTo': 'genome'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_homologene', 'Menu': 'HomoloGene Links', 'Description': 'Related HomoloGene', 'DbTo': 'homologene'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_nuccore', 'Menu': 'Nucleotide Links', 'Description': 'Nucleotide sequence from coding region', 'DbTo': 'nuccore'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_nuccore_mat_peptide', 'Menu': 'Mature Peptides', 'Description': 'Link between nucleotide record and mature peptides contained within nucleotide record', 'DbTo': 'nuccore'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_nuccore_mgc', 'Menu': 'Nucleotide NIH cDNA clone links', 'Description': 'Nucleotide NIH cDNA clone', 'DbTo': 'nuccore'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_nuccore_mrna', 'Menu': 'Encoding mRNA', 'Description': 'Link from protein to encoding mRNA', 'DbTo': 'nuccore'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_nuccore_small_genome', 'Menu': 'Link to all nucleotide sequences from this genome', 'Description': 'All nucleotide sequences from this genome', 'DbTo': 'nuccore'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_nuccore_tsa', 'Menu': 'TSA master records', 'Description': 'Links to TSA master records', 'DbTo': 'nuccore'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_nuccore_wgs', 'Menu': '', 'Description': 'Links to WGS master sequences', 'DbTo': 'nuccore'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_nuccore_wp', 'Menu': 'Link to genomic records', 'Description': 'Genomic records', 'DbTo': 'nuccore'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_nucgss', 'Menu': 'GSS Links', 'Description': 'GSS sequence from coding region', 'DbTo': 'nucgss'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_nucleotide_mgc_url', 'Menu': '', 'Description': 'Order cDNA clone', 'DbTo': 'nucleotide'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_omim', 'Menu': 'OMIM Links', 'Description': 'OMIM records associated with protein sequence', 'DbTo': 'omim'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_pcassay_target', 'Menu': 'BioAssay by Target (List)', 'Description': 'PubChem BioAssays done on the protein target', 'DbTo': 'pcassay'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_pcassay_target_pig', 'Menu': 'BioAssay by Target (Identical Proteins, List)', 'Description': 'PubChem BioAssays done on the protein target PIG', 'DbTo': 'pcassay'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_pcassay_target_pig_summary', 'Menu': 'pcassay', 'Description': 'BioAssay by Target (Identical Proteins, Summary)', 'DbTo': 'pcassay'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_pcassay_target_summary', 'Menu': 'pcassay', 'Description': 'BioAssay by Target (Summary)', 'DbTo': 'pcassay'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_pccompound', 'Menu': 'PubChem Compound Links', 'Description': 'Related PubChem Compound', 'DbTo': 'pccompound'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_pcsubstance', 'Menu': 'PubChem Substance Links', 'Description': 'Related PubChem Substance', 'DbTo': 'pcsubstance'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_pmc', 'Menu': 'PMC Links', 'Description': 'Free full text articles in PMC', 'DbTo': 'pmc'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_popset', 'Menu': 'PopSet Links', 'Description': 'Population set containing protein sequence', 'DbTo': 'popset'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_protein_cdart_summary', 'Menu': '', 'Description': 'Display proteins with similar conserved domain architecture', 'DbTo': 'protein'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_protein_ref2wp', 'Menu': 'Links from references to autonomous proteins', 'Description': 'Autonomous protein records', 'DbTo': 'protein'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_protein_refseq2uniprot', 'Menu': 'Protein (UniProtKB)', 'Description': 'Protein gi (refseq) to protein gi (UniProtKB)', 'DbTo': 'protein'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_protein_small_genome', 'Menu': 'Link to all proteins from this genome', 'Description': 'All proteins from this genome', 'DbTo': 'protein'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_protein_uniprot2refseq', 'Menu': 'Protein (RefSeq)', 'Description': 'Protein gi (UniProtKB) to protein gi (refseq)', 'DbTo': 'protein'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_protein_wp2ref', 'Menu': 'Link to referencing proteins', 'Description': 'Referencing proteins', 'DbTo': 'protein'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_proteinclusters', 'Menu': 'Protein Cluster Links', 'Description': 'Related Protein Clusters', 'DbTo': 'proteinclusters'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_pubmed', 'Menu': 'PubMed Links', 'Description': 'PubMed articles cited by protein sequence record', 'DbTo': 'pubmed'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_pubmed_refseq', 'Menu': 'PubMed (RefSeq) Links', 'Description': 'Link to RefSeq PubMed articles', 'DbTo': 'pubmed'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_pubmed_weighted', 'Menu': 'PubMed (Weighted) Links', 'Description': 'Links to pubmed', 'DbTo': 'pubmed'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_snp', 'Menu': 'SNP Links', 'Description': 'Related SNPs', 'DbTo': 'snp'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_snp_geneview', 'Menu': '', 'Description': 'SNPs linked from GeneView', 'DbTo': 'snp'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_structure', 'Menu': 'Identical Structures', 'Description': '3D structures for identical protein sequences', 'DbTo': 'structure'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_structure_direct', 'Menu': 'Structures', 'Description': 'Experimentally determined 3D Structures', 'DbTo': 'structure'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_structure_related', 'Menu': '', 'Description': 'Related structures', 'DbTo': 'structure'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_structure_related_list', 'Menu': 'Related Structures (List)', 'Description': 'List of related structures', 'DbTo': 'structure'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_taxonomy', 'Menu': 'Taxonomy Links', 'Description': 'Taxonomy sequences associated with protein record', 'DbTo': 'taxonomy'}\n",
      "\n",
      "\n",
      "{'Name': 'protein_taxonomy_wp2species', 'Menu': 'Links from autonomous proteins to species level organisms', 'Description': 'Species level organisms', 'DbTo': 'taxonomy'}\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for field in record['DbInfo']['LinkList']:\n",
    "    print(field)\n",
    "    print('\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<?xml version=\"1.0\" encoding=\"UTF-8\" ?>\n",
      "<!DOCTYPE ePostResult PUBLIC \"-//NLM//DTD epost 20090526//EN\" \"https://eutils.ncbi.nlm.nih.gov/eutils/dtd/20090526/epost.dtd\"><ePostResult>\n",
      "\t<QueryKey>1</QueryKey>\n",
      "\t<WebEnv>NCID_1_190607792_130.14.18.97_9001_1584415437_1741210555_0MetA0_S_MegaStore</WebEnv>\n",
      "</ePostResult>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "from Bio import Entrez\n",
    "Entrez.api_key = api_key\n",
    "Entrez.email = 'elizabethfawcett47@gmail.com'\n",
    "Entrez.tool = 'locus_by_amino_acid'\n",
    "id_list = ['NP_001164209.1', 'XP_016874800.1']\n",
    "print(Entrez.epost(\"protein\", id=\",\".join(id_list)).read())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Entrez\n",
    "Entrez.api_key = api_key\n",
    "Entrez.email = 'elizabethfawcett47@gmail.com'\n",
    "Entrez.tool = 'locus_by_amino_acid'\n",
    "id_list = ['NP_001164209.1', 'XP_016874800.1']\n",
    "search_results = Entrez.read(Entrez.epost(\"protein\", id=\",\".join(id_list)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NCID_1_190536794_130.14.22.76_9001_1584415437_1110160424_0MetA0_S_MegaStore\n",
      "1\n"
     ]
    }
   ],
   "source": [
    "webenv = search_results[\"WebEnv\"]\n",
    "query_key = search_results[\"QueryKey\"]\n",
    "print(webenv)\n",
    "print(query_key)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Entrez\n",
    "Entrez.api_key = api_key\n",
    "Entrez.email = 'elizabethfawcett47@gmail.com'\n",
    "Entrez.tool = 'locus_by_amino_acid'\n",
    "id_list = ['NP_001164209.1']\n",
    "search_results = Entrez.read(Entrez.efetch(db=\"protein\", id=id_list, retmode='xml'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['GBSeq_locus', 'GBSeq_length', 'GBSeq_moltype', 'GBSeq_topology', 'GBSeq_division', 'GBSeq_update-date', 'GBSeq_create-date', 'GBSeq_definition', 'GBSeq_primary-accession', 'GBSeq_accession-version', 'GBSeq_other-seqids', 'GBSeq_keywords', 'GBSeq_source', 'GBSeq_organism', 'GBSeq_taxonomy', 'GBSeq_references', 'GBSeq_comment', 'GBSeq_source-db', 'GBSeq_feature-table', 'GBSeq_sequence'])"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0].keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'NP_001164209'"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0]['GBSeq_locus']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'1182'"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0]['GBSeq_length']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'AA'"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0]['GBSeq_moltype']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'linear'"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0]['GBSeq_topology']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'PRI'"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0]['GBSeq_division']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'25-OCT-2019'"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0]['GBSeq_update-date']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'31-DEC-2009'"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0]['GBSeq_create-date']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'IQ motif and SEC7 domain-containing protein 3 isoform 1 [Homo sapiens]'"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0]['GBSeq_definition']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'NP_001164209'"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0]['GBSeq_primary-accession']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'NP_001164209.1'"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0]['GBSeq_accession-version']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'ref|NP_001164209.1|'"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0]['GBSeq_other-seqids'][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'gi|282721018'"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0]['GBSeq_other-seqids'][1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['RefSeq', 'MANE Select']"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0]['GBSeq_keywords']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Homo sapiens (human)'"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0]['GBSeq_source']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Homo sapiens'"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0]['GBSeq_organism']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi; Mammalia; Eutheria; Euarchontoglires; Primates; Haplorrhini; Catarrhini; Hominidae; Homo'"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0]['GBSeq_taxonomy']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'GBReference_reference': '1', 'GBReference_position': '1..1182', 'GBReference_authors': ['Um JW', 'Choii G', 'Park D', 'Kim D', 'Jeon S', 'Kang H', 'Mori T', 'Papadopoulos T', 'Yoo T', 'Lee Y', 'Kim E', 'Tabuchi K', 'Ko J'], 'GBReference_title': 'IQ Motif and SEC7 Domain-containing Protein 3 (IQSEC3) Interacts with Gephyrin to Promote Inhibitory Synapse Formation', 'GBReference_journal': 'J. Biol. Chem. 291 (19), 10119-10130 (2016)', 'GBReference_xref': [{'GBXref_dbname': 'doi', 'GBXref_id': '10.1074/jbc.M115.712893'}], 'GBReference_pubmed': '27002143', 'GBReference_remark': 'GeneRIF: these data reveal that IQSEC3 acts together with gephyrin to regulate inhibitory synapse development.'}"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0]['GBSeq_references'][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'GBReference_reference': '2', 'GBReference_position': '1..1182', 'GBReference_authors': ['Turner ST', 'Bailey KR', 'Schwartz GL', 'Chapman AB', 'Chai HS', 'Boerwinkle E'], 'GBReference_title': 'Genomic association analysis identifies multiple loci influencing antihypertensive response to an angiotensin II receptor blocker', 'GBReference_journal': 'Hypertension 59 (6), 1204-1211 (2012)', 'GBReference_xref': [{'GBXref_dbname': 'doi', 'GBXref_id': '10.1161/HYP.0b013e31825b30f8'}], 'GBReference_pubmed': '22566498'}"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0]['GBSeq_references'][1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'GBReference_reference': '3', 'GBReference_position': '1..1182', 'GBReference_authors': ['Hattori Y', 'Ohta S', 'Hamada K', 'Yamada-Okabe H', 'Kanemura Y', 'Matsuzaki Y', 'Okano H', 'Kawakami Y', 'Toda M'], 'GBReference_title': 'Identification of a neuron-specific human gene, KIAA1110, that is a guanine nucleotide exchange factor for ARF1', 'GBReference_journal': 'Biochem. Biophys. Res. Commun. 364 (4), 737-742 (2007)', 'GBReference_xref': [{'GBXref_dbname': 'doi', 'GBXref_id': '10.1016/j.bbrc.2007.10.041'}], 'GBReference_pubmed': '17981261', 'GBReference_remark': 'GeneRIF: These results suggest that KIAA1110 is expressed specifically in mature neurons and may play an important role in the secretion pathway as a GEF for ARF1.'}"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0]['GBSeq_references'][2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'VALIDATED REFSEQ: This record has undergone validation or preliminary review. The reference sequence was derived from AC026369.21 and AC007406.1.; ~Transcript Variant: This variant (1) represents the longer transcript and encodes the longer isoform (1).; ~Sequence Note: The RefSeq transcript and protein were derived from genomic sequence to make the sequence consistent with the reference genome assembly. The genomic coordinates used for the transcript record were based on transcript alignments and orthologous data.; ; ##Evidence-Data-START## ; RNAseq introns :: single sample supports all introns SAMEA2145743, SAMEA2154665 [ECO:0000348] ; ##Evidence-Data-END##; ; ##RefSeq-Attributes-START## ; MANE Ensembl match :: ENST00000538872.6/ ENSP00000437554.1 ; RefSeq Select criteria :: based on conservation, expression, longest protein ; ##RefSeq-Attributes-END##'"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0]['GBSeq_comment']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'REFSEQ: accession NM_001170738.2'"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0]['GBSeq_source-db']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "9"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(search_results[0]['GBSeq_feature-table'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'GBFeature_key': 'source', 'GBFeature_location': '1..1182', 'GBFeature_intervals': [{'GBInterval_from': '1', 'GBInterval_to': '1182', 'GBInterval_accession': 'NP_001164209.1'}], 'GBFeature_quals': [{'GBQualifier_name': 'organism', 'GBQualifier_value': 'Homo sapiens'}, {'GBQualifier_name': 'db_xref', 'GBQualifier_value': 'taxon:9606'}, {'GBQualifier_name': 'chromosome', 'GBQualifier_value': '12'}, {'GBQualifier_name': 'map', 'GBQualifier_value': '12p13.33'}]}\n",
      "\n",
      "\n",
      "{'GBFeature_key': 'Protein', 'GBFeature_location': '1..1182', 'GBFeature_intervals': [{'GBInterval_from': '1', 'GBInterval_to': '1182', 'GBInterval_accession': 'NP_001164209.1'}], 'GBFeature_quals': [{'GBQualifier_name': 'product', 'GBQualifier_value': 'IQ motif and SEC7 domain-containing protein 3 isoform 1'}, {'GBQualifier_name': 'note', 'GBQualifier_value': 'IQ motif and SEC7 domain-containing protein 3; IQ motif and Sec7 domain 3'}, {'GBQualifier_name': 'calculated_mol_wt', 'GBQualifier_value': '127491'}]}\n",
      "\n",
      "\n",
      "{'GBFeature_key': 'Region', 'GBFeature_location': '<32..63', 'GBFeature_intervals': [{'GBInterval_from': '32', 'GBInterval_to': '63', 'GBInterval_accession': 'NP_001164209.1'}], 'GBFeature_partial5': StringElement('', attributes={'value': 'true'}), 'GBFeature_quals': [{'GBQualifier_name': 'region_name', 'GBQualifier_value': 'bZIP'}, {'GBQualifier_name': 'note', 'GBQualifier_value': 'Basic leucine zipper (bZIP) domain of bZIP transcription factors: a DNA-binding and dimerization domain; cd14686'}, {'GBQualifier_name': 'db_xref', 'GBQualifier_value': 'CDD:269834'}]}\n",
      "\n",
      "\n",
      "{'GBFeature_key': 'Site', 'GBFeature_location': 'order(32,35..36,39..40,42..43,46..47,49..50,53..54,56..57,60..61,63)', 'GBFeature_intervals': [{'GBInterval_point': '32', 'GBInterval_accession': 'NP_001164209.1'}, {'GBInterval_from': '35', 'GBInterval_to': '36', 'GBInterval_accession': 'NP_001164209.1'}, {'GBInterval_from': '39', 'GBInterval_to': '40', 'GBInterval_accession': 'NP_001164209.1'}, {'GBInterval_from': '42', 'GBInterval_to': '43', 'GBInterval_accession': 'NP_001164209.1'}, {'GBInterval_from': '46', 'GBInterval_to': '47', 'GBInterval_accession': 'NP_001164209.1'}, {'GBInterval_from': '49', 'GBInterval_to': '50', 'GBInterval_accession': 'NP_001164209.1'}, {'GBInterval_from': '53', 'GBInterval_to': '54', 'GBInterval_accession': 'NP_001164209.1'}, {'GBInterval_from': '56', 'GBInterval_to': '57', 'GBInterval_accession': 'NP_001164209.1'}, {'GBInterval_from': '60', 'GBInterval_to': '61', 'GBInterval_accession': 'NP_001164209.1'}, {'GBInterval_point': '63', 'GBInterval_accession': 'NP_001164209.1'}], 'GBFeature_operator': 'order', 'GBFeature_quals': [{'GBQualifier_name': 'site_type', 'GBQualifier_value': 'other'}, {'GBQualifier_name': 'note', 'GBQualifier_value': 'dimer interface [polypeptide binding]'}, {'GBQualifier_name': 'db_xref', 'GBQualifier_value': 'CDD:269834'}]}\n",
      "\n",
      "\n",
      "{'GBFeature_key': 'Site', 'GBFeature_location': '259', 'GBFeature_intervals': [{'GBInterval_point': '259', 'GBInterval_accession': 'NP_001164209.1'}], 'GBFeature_quals': [{'GBQualifier_name': 'site_type', 'GBQualifier_value': 'phosphorylation'}, {'GBQualifier_name': 'experiment', 'GBQualifier_value': 'experimental evidence, no additional details recorded'}, {'GBQualifier_name': 'note', 'GBQualifier_value': 'Phosphoserine. {ECO:0000250|UniProtKB:Q3TES0}; propagated from UniProtKB/Swiss-Prot (Q9UPP2.3)'}]}\n",
      "\n",
      "\n",
      "{'GBFeature_key': 'Region', 'GBFeature_location': '651..839', 'GBFeature_intervals': [{'GBInterval_from': '651', 'GBInterval_to': '839', 'GBInterval_accession': 'NP_001164209.1'}], 'GBFeature_quals': [{'GBQualifier_name': 'region_name', 'GBQualifier_value': 'Sec7'}, {'GBQualifier_name': 'note', 'GBQualifier_value': 'Sec7 domain; pfam01369'}, {'GBQualifier_name': 'db_xref', 'GBQualifier_value': 'CDD:279680'}]}\n",
      "\n",
      "\n",
      "{'GBFeature_key': 'Site', 'GBFeature_location': 'order(742..749,781..794)', 'GBFeature_intervals': [{'GBInterval_from': '742', 'GBInterval_to': '749', 'GBInterval_accession': 'NP_001164209.1'}, {'GBInterval_from': '781', 'GBInterval_to': '794', 'GBInterval_accession': 'NP_001164209.1'}], 'GBFeature_operator': 'order', 'GBFeature_quals': [{'GBQualifier_name': 'site_type', 'GBQualifier_value': 'active'}, {'GBQualifier_name': 'note', 'GBQualifier_value': 'active site/putative ARF binding site [active]'}, {'GBQualifier_name': 'db_xref', 'GBQualifier_value': 'CDD:238100'}]}\n",
      "\n",
      "\n",
      "{'GBFeature_key': 'Region', 'GBFeature_location': '866..996', 'GBFeature_intervals': [{'GBInterval_from': '866', 'GBInterval_to': '996', 'GBInterval_accession': 'NP_001164209.1'}], 'GBFeature_quals': [{'GBQualifier_name': 'region_name', 'GBQualifier_value': 'PH_IQSEC'}, {'GBQualifier_name': 'note', 'GBQualifier_value': 'IQ motif and SEC7 domain-containing protein family Pleckstrin homology domain; cd13318'}, {'GBQualifier_name': 'db_xref', 'GBQualifier_value': 'CDD:270128'}]}\n",
      "\n",
      "\n",
      "{'GBFeature_key': 'CDS', 'GBFeature_location': '1..1182', 'GBFeature_intervals': [{'GBInterval_from': '1', 'GBInterval_to': '1182', 'GBInterval_accession': 'NP_001164209.1'}], 'GBFeature_quals': [{'GBQualifier_name': 'gene', 'GBQualifier_value': 'IQSEC3'}, {'GBQualifier_name': 'coded_by', 'GBQualifier_value': 'NM_001170738.2:117..3665'}, {'GBQualifier_name': 'note', 'GBQualifier_value': 'isoform 1 is encoded by transcript variant 1'}, {'GBQualifier_name': 'transl_table', 'GBQualifier_value': '1'}, {'GBQualifier_name': 'db_xref', 'GBQualifier_value': 'CCDS:CCDS53728.1'}, {'GBQualifier_name': 'db_xref', 'GBQualifier_value': 'GeneID:440073'}, {'GBQualifier_name': 'db_xref', 'GBQualifier_value': 'HGNC:HGNC:29193'}, {'GBQualifier_name': 'db_xref', 'GBQualifier_value': 'MIM:612118'}]}\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for feature in search_results[0]['GBSeq_feature-table']:\n",
    "    print(feature)\n",
    "    print('\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['GBFeature_key', 'GBFeature_location', 'GBFeature_intervals', 'GBFeature_quals'])"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0]['GBSeq_feature-table'][0].keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "source\n",
      "1..1182\n",
      "{'GBInterval_from': '1', 'GBInterval_to': '1182', 'GBInterval_accession': 'NP_001164209.1'}\n",
      "1\n",
      "1182\n",
      "{'GBQualifier_name': 'organism', 'GBQualifier_value': 'Homo sapiens'}\n",
      "{'GBQualifier_name': 'db_xref', 'GBQualifier_value': 'taxon:9606'}\n",
      "{'GBQualifier_name': 'chromosome', 'GBQualifier_value': '12'}\n",
      "12\n"
     ]
    }
   ],
   "source": [
    "print(search_results[0]['GBSeq_feature-table'][0]['GBFeature_key'])\n",
    "print(search_results[0]['GBSeq_feature-table'][0]['GBFeature_location'])\n",
    "print(search_results[0]['GBSeq_feature-table'][0]['GBFeature_intervals'][0])\n",
    "print(search_results[0]['GBSeq_feature-table'][0]['GBFeature_intervals'][0]['GBInterval_from'])\n",
    "print(search_results[0]['GBSeq_feature-table'][0]['GBFeature_intervals'][0]['GBInterval_to'])\n",
    "print(search_results[0]['GBSeq_feature-table'][0]['GBFeature_quals'][0])\n",
    "print(search_results[0]['GBSeq_feature-table'][0]['GBFeature_quals'][1])\n",
    "print(search_results[0]['GBSeq_feature-table'][0]['GBFeature_quals'][2])\n",
    "print(search_results[0]['GBSeq_feature-table'][0]['GBFeature_quals'][2]['GBQualifier_value'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['GBFeature_key', 'GBFeature_location', 'GBFeature_intervals', 'GBFeature_quals'])"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0]['GBSeq_feature-table'][1].keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Protein\n",
      "1..1182\n",
      "{'GBInterval_from': '1', 'GBInterval_to': '1182', 'GBInterval_accession': 'NP_001164209.1'}\n",
      "1\n",
      "1182\n",
      "{'GBQualifier_name': 'product', 'GBQualifier_value': 'IQ motif and SEC7 domain-containing protein 3 isoform 1'}\n",
      "{'GBQualifier_name': 'note', 'GBQualifier_value': 'IQ motif and SEC7 domain-containing protein 3; IQ motif and Sec7 domain 3'}\n",
      "{'GBQualifier_name': 'calculated_mol_wt', 'GBQualifier_value': '127491'}\n",
      "127491\n"
     ]
    }
   ],
   "source": [
    "print(search_results[0]['GBSeq_feature-table'][1]['GBFeature_key'])\n",
    "print(search_results[0]['GBSeq_feature-table'][1]['GBFeature_location'])\n",
    "print(search_results[0]['GBSeq_feature-table'][1]['GBFeature_intervals'][0])\n",
    "print(search_results[0]['GBSeq_feature-table'][1]['GBFeature_intervals'][0]['GBInterval_from'])\n",
    "print(search_results[0]['GBSeq_feature-table'][1]['GBFeature_intervals'][0]['GBInterval_to'])\n",
    "print(search_results[0]['GBSeq_feature-table'][1]['GBFeature_quals'][0])\n",
    "print(search_results[0]['GBSeq_feature-table'][1]['GBFeature_quals'][1])\n",
    "print(search_results[0]['GBSeq_feature-table'][1]['GBFeature_quals'][2])\n",
    "print(search_results[0]['GBSeq_feature-table'][1]['GBFeature_quals'][2]['GBQualifier_value'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['GBFeature_key', 'GBFeature_location', 'GBFeature_intervals', 'GBFeature_partial5', 'GBFeature_quals'])"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0]['GBSeq_feature-table'][2].keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Region\n",
      "<32..63\n",
      "{'GBInterval_from': '32', 'GBInterval_to': '63', 'GBInterval_accession': 'NP_001164209.1'}\n",
      "32\n",
      "63\n",
      "\n",
      "{'GBQualifier_name': 'region_name', 'GBQualifier_value': 'bZIP'}\n",
      "{'GBQualifier_name': 'note', 'GBQualifier_value': 'Basic leucine zipper (bZIP) domain of bZIP transcription factors: a DNA-binding and dimerization domain; cd14686'}\n",
      "{'GBQualifier_name': 'db_xref', 'GBQualifier_value': 'CDD:269834'}\n",
      "CDD:269834\n"
     ]
    }
   ],
   "source": [
    "print(search_results[0]['GBSeq_feature-table'][2]['GBFeature_key'])\n",
    "print(search_results[0]['GBSeq_feature-table'][2]['GBFeature_location'])\n",
    "print(search_results[0]['GBSeq_feature-table'][2]['GBFeature_intervals'][0])\n",
    "print(search_results[0]['GBSeq_feature-table'][2]['GBFeature_intervals'][0]['GBInterval_from'])\n",
    "print(search_results[0]['GBSeq_feature-table'][2]['GBFeature_intervals'][0]['GBInterval_to'])\n",
    "print(search_results[0]['GBSeq_feature-table'][2]['GBFeature_partial5'])\n",
    "print(search_results[0]['GBSeq_feature-table'][2]['GBFeature_quals'][0])\n",
    "print(search_results[0]['GBSeq_feature-table'][2]['GBFeature_quals'][1])\n",
    "print(search_results[0]['GBSeq_feature-table'][2]['GBFeature_quals'][2])\n",
    "print(search_results[0]['GBSeq_feature-table'][2]['GBFeature_quals'][2]['GBQualifier_value'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "StringElement('', attributes={'value': 'true'})"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0]['GBSeq_feature-table'][2]['GBFeature_partial5']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['GBFeature_key', 'GBFeature_location', 'GBFeature_intervals', 'GBFeature_operator', 'GBFeature_quals'])"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0]['GBSeq_feature-table'][3].keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Site\n",
      "order(32,35..36,39..40,42..43,46..47,49..50,53..54,56..57,60..61,63)\n",
      "{'GBInterval_point': '32', 'GBInterval_accession': 'NP_001164209.1'}\n",
      "{'GBInterval_from': '35', 'GBInterval_to': '36', 'GBInterval_accession': 'NP_001164209.1'}\n",
      "{'GBInterval_from': '39', 'GBInterval_to': '40', 'GBInterval_accession': 'NP_001164209.1'}\n",
      "{'GBInterval_from': '42', 'GBInterval_to': '43', 'GBInterval_accession': 'NP_001164209.1'}\n",
      "{'GBInterval_from': '46', 'GBInterval_to': '47', 'GBInterval_accession': 'NP_001164209.1'}\n",
      "{'GBInterval_from': '49', 'GBInterval_to': '50', 'GBInterval_accession': 'NP_001164209.1'}\n",
      "{'GBInterval_from': '53', 'GBInterval_to': '54', 'GBInterval_accession': 'NP_001164209.1'}\n",
      "{'GBInterval_from': '56', 'GBInterval_to': '57', 'GBInterval_accession': 'NP_001164209.1'}\n",
      "{'GBInterval_from': '60', 'GBInterval_to': '61', 'GBInterval_accession': 'NP_001164209.1'}\n",
      "{'GBInterval_point': '63', 'GBInterval_accession': 'NP_001164209.1'}\n",
      "order\n",
      "{'GBQualifier_name': 'site_type', 'GBQualifier_value': 'other'}\n",
      "{'GBQualifier_name': 'note', 'GBQualifier_value': 'dimer interface [polypeptide binding]'}\n",
      "{'GBQualifier_name': 'db_xref', 'GBQualifier_value': 'CDD:269834'}\n"
     ]
    }
   ],
   "source": [
    "print(search_results[0]['GBSeq_feature-table'][3]['GBFeature_key'])\n",
    "print(search_results[0]['GBSeq_feature-table'][3]['GBFeature_location'])\n",
    "for i in range(0, len(search_results[0]['GBSeq_feature-table'][3]['GBFeature_intervals'])):\n",
    "    print(search_results[0]['GBSeq_feature-table'][3]['GBFeature_intervals'][i])\n",
    "print(search_results[0]['GBSeq_feature-table'][3]['GBFeature_operator'])\n",
    "for i in search_results[0]['GBSeq_feature-table'][3]['GBFeature_quals']:\n",
    "    print(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['GBFeature_key', 'GBFeature_location', 'GBFeature_intervals', 'GBFeature_quals'])"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0]['GBSeq_feature-table'][4].keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Site\n",
      "259\n",
      "{'GBInterval_point': '259', 'GBInterval_accession': 'NP_001164209.1'}\n",
      "{'GBQualifier_name': 'site_type', 'GBQualifier_value': 'phosphorylation'}\n",
      "{'GBQualifier_name': 'experiment', 'GBQualifier_value': 'experimental evidence, no additional details recorded'}\n",
      "{'GBQualifier_name': 'note', 'GBQualifier_value': 'Phosphoserine. {ECO:0000250|UniProtKB:Q3TES0}; propagated from UniProtKB/Swiss-Prot (Q9UPP2.3)'}\n",
      "Phosphoserine. {ECO:0000250|UniProtKB:Q3TES0}; propagated from UniProtKB/Swiss-Prot (Q9UPP2.3)\n"
     ]
    }
   ],
   "source": [
    "print(search_results[0]['GBSeq_feature-table'][4]['GBFeature_key'])\n",
    "print(search_results[0]['GBSeq_feature-table'][4]['GBFeature_location'])\n",
    "print(search_results[0]['GBSeq_feature-table'][4]['GBFeature_intervals'][0])\n",
    "# print(search_results[0]['GBSeq_feature-table'][4]['GBFeature_intervals'][0]['GBInterval_from'])\n",
    "# print(search_results[0]['GBSeq_feature-table'][4]['GBFeature_intervals'][0]['GBInterval_to'])\n",
    "print(search_results[0]['GBSeq_feature-table'][4]['GBFeature_quals'][0])\n",
    "print(search_results[0]['GBSeq_feature-table'][4]['GBFeature_quals'][1])\n",
    "print(search_results[0]['GBSeq_feature-table'][4]['GBFeature_quals'][2])\n",
    "print(search_results[0]['GBSeq_feature-table'][4]['GBFeature_quals'][2]['GBQualifier_value'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['GBFeature_key', 'GBFeature_location', 'GBFeature_intervals', 'GBFeature_quals'])"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0]['GBSeq_feature-table'][5].keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Region\n",
      "651..839\n",
      "{'GBInterval_from': '651', 'GBInterval_to': '839', 'GBInterval_accession': 'NP_001164209.1'}\n",
      "651\n",
      "839\n",
      "{'GBQualifier_name': 'region_name', 'GBQualifier_value': 'Sec7'}\n",
      "{'GBQualifier_name': 'note', 'GBQualifier_value': 'Sec7 domain; pfam01369'}\n",
      "{'GBQualifier_name': 'db_xref', 'GBQualifier_value': 'CDD:279680'}\n",
      "CDD:279680\n"
     ]
    }
   ],
   "source": [
    "print(search_results[0]['GBSeq_feature-table'][5]['GBFeature_key'])\n",
    "print(search_results[0]['GBSeq_feature-table'][5]['GBFeature_location'])\n",
    "print(search_results[0]['GBSeq_feature-table'][5]['GBFeature_intervals'][0])\n",
    "print(search_results[0]['GBSeq_feature-table'][5]['GBFeature_intervals'][0]['GBInterval_from'])\n",
    "print(search_results[0]['GBSeq_feature-table'][5]['GBFeature_intervals'][0]['GBInterval_to'])\n",
    "print(search_results[0]['GBSeq_feature-table'][5]['GBFeature_quals'][0])\n",
    "print(search_results[0]['GBSeq_feature-table'][5]['GBFeature_quals'][1])\n",
    "print(search_results[0]['GBSeq_feature-table'][5]['GBFeature_quals'][2])\n",
    "print(search_results[0]['GBSeq_feature-table'][5]['GBFeature_quals'][2]['GBQualifier_value'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['GBFeature_key', 'GBFeature_location', 'GBFeature_intervals', 'GBFeature_operator', 'GBFeature_quals'])"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0]['GBSeq_feature-table'][6].keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Site\n",
      "order(742..749,781..794)\n",
      "{'GBInterval_from': '742', 'GBInterval_to': '749', 'GBInterval_accession': 'NP_001164209.1'}\n",
      "{'GBInterval_from': '781', 'GBInterval_to': '794', 'GBInterval_accession': 'NP_001164209.1'}\n",
      "order\n",
      "{'GBQualifier_name': 'site_type', 'GBQualifier_value': 'active'}\n",
      "{'GBQualifier_name': 'note', 'GBQualifier_value': 'active site/putative ARF binding site [active]'}\n",
      "{'GBQualifier_name': 'db_xref', 'GBQualifier_value': 'CDD:238100'}\n"
     ]
    }
   ],
   "source": [
    "print(search_results[0]['GBSeq_feature-table'][6]['GBFeature_key'])\n",
    "print(search_results[0]['GBSeq_feature-table'][6]['GBFeature_location'])\n",
    "for i in range(0, len(search_results[0]['GBSeq_feature-table'][6]['GBFeature_intervals'])):\n",
    "    print(search_results[0]['GBSeq_feature-table'][6]['GBFeature_intervals'][i])\n",
    "print(search_results[0]['GBSeq_feature-table'][6]['GBFeature_operator'])\n",
    "for i in search_results[0]['GBSeq_feature-table'][6]['GBFeature_quals']:\n",
    "    print(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['GBFeature_key', 'GBFeature_location', 'GBFeature_intervals', 'GBFeature_quals'])"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0]['GBSeq_feature-table'][7].keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Region\n",
      "866..996\n",
      "{'GBInterval_from': '866', 'GBInterval_to': '996', 'GBInterval_accession': 'NP_001164209.1'}\n",
      "866\n",
      "996\n",
      "{'GBQualifier_name': 'region_name', 'GBQualifier_value': 'PH_IQSEC'}\n",
      "{'GBQualifier_name': 'note', 'GBQualifier_value': 'IQ motif and SEC7 domain-containing protein family Pleckstrin homology domain; cd13318'}\n",
      "{'GBQualifier_name': 'db_xref', 'GBQualifier_value': 'CDD:270128'}\n",
      "CDD:270128\n"
     ]
    }
   ],
   "source": [
    "print(search_results[0]['GBSeq_feature-table'][7]['GBFeature_key'])\n",
    "print(search_results[0]['GBSeq_feature-table'][7]['GBFeature_location'])\n",
    "print(search_results[0]['GBSeq_feature-table'][7]['GBFeature_intervals'][0])\n",
    "print(search_results[0]['GBSeq_feature-table'][7]['GBFeature_intervals'][0]['GBInterval_from'])\n",
    "print(search_results[0]['GBSeq_feature-table'][7]['GBFeature_intervals'][0]['GBInterval_to'])\n",
    "print(search_results[0]['GBSeq_feature-table'][7]['GBFeature_quals'][0])\n",
    "print(search_results[0]['GBSeq_feature-table'][7]['GBFeature_quals'][1])\n",
    "print(search_results[0]['GBSeq_feature-table'][7]['GBFeature_quals'][2])\n",
    "print(search_results[0]['GBSeq_feature-table'][7]['GBFeature_quals'][2]['GBQualifier_value'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['GBFeature_key', 'GBFeature_location', 'GBFeature_intervals', 'GBFeature_quals'])"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0]['GBSeq_feature-table'][8].keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CDS\n",
      "1..1182\n",
      "{'GBInterval_from': '1', 'GBInterval_to': '1182', 'GBInterval_accession': 'NP_001164209.1'}\n",
      "1\n",
      "1182\n",
      "{'GBQualifier_name': 'gene', 'GBQualifier_value': 'IQSEC3'}\n",
      "{'GBQualifier_name': 'coded_by', 'GBQualifier_value': 'NM_001170738.2:117..3665'}\n",
      "{'GBQualifier_name': 'note', 'GBQualifier_value': 'isoform 1 is encoded by transcript variant 1'}\n",
      "isoform 1 is encoded by transcript variant 1\n"
     ]
    }
   ],
   "source": [
    "print(search_results[0]['GBSeq_feature-table'][8]['GBFeature_key'])\n",
    "print(search_results[0]['GBSeq_feature-table'][8]['GBFeature_location'])\n",
    "print(search_results[0]['GBSeq_feature-table'][8]['GBFeature_intervals'][0])\n",
    "print(search_results[0]['GBSeq_feature-table'][8]['GBFeature_intervals'][0]['GBInterval_from'])\n",
    "print(search_results[0]['GBSeq_feature-table'][8]['GBFeature_intervals'][0]['GBInterval_to'])\n",
    "print(search_results[0]['GBSeq_feature-table'][8]['GBFeature_quals'][0])\n",
    "print(search_results[0]['GBSeq_feature-table'][8]['GBFeature_quals'][1])\n",
    "print(search_results[0]['GBSeq_feature-table'][8]['GBFeature_quals'][2])\n",
    "print(search_results[0]['GBSeq_feature-table'][8]['GBFeature_quals'][2]['GBQualifier_value'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'mesllenpvravlylkeltaivqnqqslihtqreridelerrldelsaenrslwehqqllqaqpppglvppssaplpaapatapaaaaraqeplqdqgqrsaaaphpapdrpprqhhgqlleqpqrgpgsrahtpqsphkhlgtqgavtdkekerppsccaaagallqhkspsalgkgvlsrrpenetvlhqfccpaadacsdlasqsdgsctqagggmedsvvaaaavaagrpsahapkaqaqelqeeeerpgagaaspragpqhkaspgrqqpalatalcphapaasdyelsldlknkqiemlehkygghlvsrraactiqtafrqyqlsknfekirnsllesrlprrislrkvrsptaeslaaekalmegyglvglplvrspslpptfagtlteledsfteqvqslaksiddalstwslktmcslresgayqlhqalqaaagppgleaegrapesagpgpgddaaetpglppahsgtlmmafrdvtvqianqnisvssstalsvanclgaqtvqapaepaagkaeqgetsgreapeapavgredasaedscaeaaasgaadgatapkteeeeeeeetaevgrgaeaeagdleqlsssststksaksgseasasaskdalqamilslpryhcenpascksptlstdtlrkrlyriglnlfninpdkgiqflisrgfipdtpigvahfllqrkglsrqmigeflgnskkqfnrdvldcvvdemdfssmeldealrkfqahirvqgeaqkverlieafsqrycmcnpevvqqfhnpdtifilafaiillntdmyspnikpdrkmmledfirnlrgvddgadiprelvvgiyeriqqkelksnedhvtyvtkveksivgmktvlsvphrrlvccsrlfevtdvnklqkqaahqrevflfndllvilklcpkkkssstytfcksvgllgmqfqlfeneyyshgitlvtplsgsekkqvlhfcalgsdemqkfvedlkesiaevteleqiriewelekqqgtktlsfkpcgaqgdpqskqgsptakreaalrerpaestvevsihnrlqtsqhnsglgaergapvpppdlqpspprqqtpplppppptppgtlvqcqqivkvivldkpclarmepllsqalscytssssdscgstplggpgspvkvthqpplppppppynhphqfcppgsllhghryssgsrslv'"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_results[0]['GBSeq_sequence']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'NCID_1_190536794_130.14.22.76_9001_1584415437_1110160424_0MetA0_S_MegaStore'"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "webenv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'1'"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query_key"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [],
   "source": [
    "results = Entrez.read(Entrez.efetch(db=\"protein\", webenv=webenv, query_key=query_key, retmode='xml'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(results)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I need to pull out the following information:\n",
    "- GBSeq_sequence\n",
    "- GBSeq_references using line comprehension --> not in every entry\n",
    "- GBSeq_organism\n",
    "- GBSeq_accession-version\n",
    "- GBSeq_definition\n",
    "- GBSeq_division\n",
    "- GBSeq_topology\n",
    "- GBSeq_moltype\n",
    "- ['GBSeq_feature-table'][0]['GBFeature_key'] == 'source', then ['GBSeq_feature-table'][0]['GBFeature_quals'][2]['GBQualifier_value'] # for chromosome #\n",
    "- ['GBSeq_feature-table'][1]['GBFeature_key'] == 'Protein', ['GBSeq_feature-table'][1]['GBFeature_quals'][2]['GBQualifier_value'] # for molar weight\n",
    " - count 'GBSeq_feature-table[x]['GBFeature_key'] == 'Region'\n",
    " - count 'GBSeq_feature-table[x]['GBFeature_key'] == 'Site'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [],
   "source": [
    "protein_additional_data = pd.DataFrame()\n",
    "accession_version = []\n",
    "definition = []\n",
    "division = []\n",
    "topology = []\n",
    "moltype = []\n",
    "organism = []\n",
    "sequence = []\n",
    "chromosome = []\n",
    "molar_weight = []\n",
    "regions = []\n",
    "sites = []\n",
    "for i in results:\n",
    "    region = 0\n",
    "    site = 0\n",
    "    accession_version.append(i['GBSeq_accession-version'])\n",
    "    organism.append(i['GBSeq_organism'])\n",
    "    definition.append(i['GBSeq_definition'])\n",
    "    division.append(i['GBSeq_division'])\n",
    "    topology.append(i['GBSeq_topology'])\n",
    "    moltype.append(i['GBSeq_moltype'])\n",
    "    sequence.append(i['GBSeq_sequence'])\n",
    "    if i['GBSeq_feature-table'][0]['GBFeature_key'] == 'source':\n",
    "        chromosome.append(i['GBSeq_feature-table'][0]['GBFeature_quals'][2]['GBQualifier_value'])\n",
    "    else:\n",
    "        continue\n",
    "    if i['GBSeq_feature-table'][1]['GBFeature_key'] == 'Protein':\n",
    "        if i['GBSeq_feature-table'][1]['GBFeature_quals'][0]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][0]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][1]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][1]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][2]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][2]['GBQualifier_value'])\n",
    "        else:\n",
    "            print('add more elif statements')\n",
    "    else:\n",
    "        continue\n",
    "    for j in range(0, len(i['GBSeq_feature-table'])):\n",
    "        if i['GBSeq_feature-table'][j]['GBFeature_key'] == 'Region':\n",
    "            region += 1\n",
    "        else:\n",
    "            continue\n",
    "    for j in range(0, len(i['GBSeq_feature-table'])):\n",
    "        if i['GBSeq_feature-table'][j]['GBFeature_key'] == 'Site':\n",
    "            site += 1\n",
    "        else:\n",
    "            continue\n",
    "    regions.append(region)\n",
    "    sites.append(site)\n",
    "protein_additional_data['Chromosome'] = chromosome\n",
    "protein_additional_data['Accesion Version'] = accession_version\n",
    "protein_additional_data['Organism'] = organism\n",
    "protein_additional_data['Definition'] = definition\n",
    "protein_additional_data['Division'] = division\n",
    "protein_additional_data['Topology'] = topology\n",
    "protein_additional_data['Molecule Type'] = moltype\n",
    "protein_additional_data['Molecular Weight'] = molar_weight\n",
    "protein_additional_data['Number of Regions'] = regions\n",
    "protein_additional_data['Number of Binding Sites'] = sites\n",
    "protein_additional_data['Sequence'] = sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Accesion Version</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Definition</th>\n",
       "      <th>Division</th>\n",
       "      <th>Topology</th>\n",
       "      <th>Molecule Type</th>\n",
       "      <th>Molecular Weight</th>\n",
       "      <th>Number of Regions</th>\n",
       "      <th>Number of Binding Sites</th>\n",
       "      <th>Sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>NP_001164209.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>IQ motif and SEC7 domain-containing protein 3 ...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>127491</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>mesllenpvravlylkeltaivqnqqslihtqreridelerrldel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>XP_016874800.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>IQ motif and SEC7 domain-containing protein 3 ...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>112808</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>mesllenpvravlylkeltaivqnqqslihtqreridelerrldel...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Chromosome Accesion Version      Organism  \\\n",
       "0         12   NP_001164209.1  Homo sapiens   \n",
       "1         12   XP_016874800.1  Homo sapiens   \n",
       "\n",
       "                                          Definition Division Topology  \\\n",
       "0  IQ motif and SEC7 domain-containing protein 3 ...      PRI   linear   \n",
       "1  IQ motif and SEC7 domain-containing protein 3 ...      PRI   linear   \n",
       "\n",
       "  Molecule Type Molecular Weight  Number of Regions  Number of Binding Sites  \\\n",
       "0            AA           127491                  3                        3   \n",
       "1            AA           112808                  0                        0   \n",
       "\n",
       "                                            Sequence  \n",
       "0  mesllenpvravlylkeltaivqnqqslihtqreridelerrldel...  \n",
       "1  mesllenpvravlylkeltaivqnqqslihtqreridelerrldel...  "
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "protein_additional_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [],
   "source": [
    "reference_accession_numbers_dna = dna_data['RefSeq-Accn'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Accession Number</th>\n",
       "      <th>Number of Proteins</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>NC_000021.9</td>\n",
       "      <td>1286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>NC_000013.11</td>\n",
       "      <td>2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>NC_000018.10</td>\n",
       "      <td>2038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>NC_000022.11</td>\n",
       "      <td>2508</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>NC_000020.11</td>\n",
       "      <td>2805</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Accession Number  Number of Proteins\n",
       "20      NC_000021.9                1286\n",
       "12     NC_000013.11                2019\n",
       "17     NC_000018.10                2038\n",
       "21     NC_000022.11                2508\n",
       "19     NC_000020.11                2805"
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "decision_about_chromosome = pd.DataFrame()\n",
    "accession_numbers = dna_data['RefSeq-Accn'].unique()\n",
    "number_of_proteins = []\n",
    "for accession_number in reference_accession_numbers_dna:\n",
    "    number_of_proteins.append(len(protein_data[protein_data['Replicon Accession']==accession_number]))\n",
    "decision_about_chromosome['Accession Number'] = accession_numbers\n",
    "decision_about_chromosome['Number of Proteins'] = number_of_proteins\n",
    "decision_about_chromosome[decision_about_chromosome['Number of Proteins'] >= 1000].nsmallest(5, 'Number of Proteins')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Since Accession Number 'NC_000021.9' has more rows than required for the project, but is still a relatively managable amount of proteins to study, I have selected chromosome 21 to study."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'NC_000021.9'"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "accession_to_study = decision_about_chromosome[decision_about_chromosome['Number of Proteins'] >= 1000].nsmallest(1, 'Number of Proteins').loc[20, 'Accession Number']\n",
    "accession_to_study"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['XP_006723962.1',\n",
       " 'XP_011544381.1',\n",
       " 'XP_011544380.1',\n",
       " 'NP_001350699.1',\n",
       " 'NP_001350689.1',\n",
       " 'NP_001350690.1',\n",
       " 'NP_001350687.1',\n",
       " 'NP_001355167.1',\n",
       " 'XP_011544384.3',\n",
       " 'XP_016883695.1',\n",
       " 'XP_011544387.2',\n",
       " 'XP_024307804.1',\n",
       " 'XP_011544385.1',\n",
       " 'NP_001308973.1',\n",
       " 'XP_016883696.1',\n",
       " 'XP_011544386.1',\n",
       " 'XP_006723981.1',\n",
       " 'XP_011544388.1',\n",
       " 'XP_006723982.1',\n",
       " 'XP_016884017.1',\n",
       " 'XP_016883697.1',\n",
       " 'NP_001307572.1',\n",
       " 'XP_016883999.1',\n",
       " 'XP_016884000.1',\n",
       " 'XP_024307912.1',\n",
       " 'NP_001340935.1',\n",
       " 'NP_001340936.1',\n",
       " 'NP_001308002.1',\n",
       " 'XP_016883699.1',\n",
       " 'NP_001340944.1',\n",
       " 'NP_001340937.1',\n",
       " 'NP_001340943.1',\n",
       " 'NP_001340941.1',\n",
       " 'XP_016883698.1',\n",
       " 'NP_001340938.1',\n",
       " 'NP_001340939.1',\n",
       " 'XP_024307806.1',\n",
       " 'XP_024307805.1',\n",
       " 'NP_001307575.1',\n",
       " 'NP_001307580.1',\n",
       " 'XP_016883709.1',\n",
       " 'XP_016883708.1',\n",
       " 'NP_001307579.1',\n",
       " 'XP_024307807.1',\n",
       " 'NP_001307577.1',\n",
       " 'NP_001300979.1',\n",
       " 'NP_001307648.1',\n",
       " 'XP_024307809.1',\n",
       " 'XP_016883711.1',\n",
       " 'XP_024307808.1',\n",
       " 'XP_016883710.1',\n",
       " 'XP_011544404.1',\n",
       " 'XP_016883712.1',\n",
       " 'NP_001308976.1',\n",
       " 'XP_011544405.1',\n",
       " 'XP_016883713.1',\n",
       " 'XP_016883714.1',\n",
       " 'XP_024307810.1',\n",
       " 'XP_011544409.2',\n",
       " 'XP_016883715.1',\n",
       " 'XP_011544407.1',\n",
       " 'XP_011544408.1',\n",
       " 'NP_001308965.1',\n",
       " 'NP_001309904.1',\n",
       " 'XP_006724005.1',\n",
       " 'XP_006724007.1',\n",
       " 'XP_011544411.1',\n",
       " 'XP_024307801.1',\n",
       " 'NP_001300621.1',\n",
       " 'XP_016883694.1',\n",
       " 'XP_016883693.1',\n",
       " 'NP_001351640.1',\n",
       " 'NP_001356798.1',\n",
       " 'NP_001316994.1',\n",
       " 'XP_006724009.1',\n",
       " 'XP_024307911.1',\n",
       " 'XP_011542311.1',\n",
       " 'NP_872288.2',\n",
       " 'XP_016884009.1',\n",
       " 'XP_016884010.1',\n",
       " 'NP_954870.2',\n",
       " 'NP_954869.1',\n",
       " 'NP_954868.1',\n",
       " 'NP_001277153.1',\n",
       " 'XP_006724060.1',\n",
       " 'NP_778146.2',\n",
       " 'XP_011527852.1',\n",
       " 'NP_001289930.1',\n",
       " 'NP_001289929.1',\n",
       " 'NP_001289928.1',\n",
       " 'NP_945347.2',\n",
       " 'NP_001289927.1',\n",
       " 'XP_006724028.1',\n",
       " 'NP_001307531.1',\n",
       " 'XP_005261054.1',\n",
       " 'NP_658983.3',\n",
       " 'XP_016883873.1',\n",
       " 'XP_016883875.1',\n",
       " 'XP_016883874.1',\n",
       " 'XP_016883876.1',\n",
       " 'NP_008879.3',\n",
       " 'NP_001273452.1',\n",
       " 'NP_071419.3',\n",
       " 'XP_011527986.1',\n",
       " 'NP_001243299.1',\n",
       " 'XP_011527987.1',\n",
       " 'XP_016883916.1',\n",
       " 'XP_011527988.1',\n",
       " 'NP_001243508.2',\n",
       " 'XP_016883964.1',\n",
       " 'XP_005261122.1',\n",
       " 'XP_016883965.1',\n",
       " 'XP_011528051.1',\n",
       " 'XP_011528053.1',\n",
       " 'XP_016883963.1',\n",
       " 'XP_011528049.1',\n",
       " 'NP_003480.2',\n",
       " 'XP_011528050.1',\n",
       " 'XP_005261120.1',\n",
       " 'XP_016883962.1',\n",
       " 'XP_011528054.1',\n",
       " 'XP_016883814.1',\n",
       " 'NP_001269970.1',\n",
       " 'NP_001269971.1',\n",
       " 'NP_037528.3',\n",
       " 'XP_011527837.1',\n",
       " 'XP_011527831.1',\n",
       " 'XP_016883812.1',\n",
       " 'XP_016883811.1',\n",
       " 'NP_001339489.1',\n",
       " 'XP_024307831.1',\n",
       " 'XP_024307832.1',\n",
       " 'XP_016883816.1',\n",
       " 'XP_005261006.1',\n",
       " 'NP_001339490.1',\n",
       " 'XP_011527779.1',\n",
       " 'NP_001193993.1',\n",
       " 'NP_001329.1',\n",
       " 'NP_001193994.1',\n",
       " 'XP_011527781.1',\n",
       " 'XP_011527778.1',\n",
       " 'NP_001193992.1',\n",
       " 'NP_001193995.1',\n",
       " 'XP_011527780.1',\n",
       " 'NP_001124386.1',\n",
       " 'XP_016883754.1',\n",
       " 'XP_024307819.1',\n",
       " 'XP_011527743.1',\n",
       " 'NP_006797.3',\n",
       " 'XP_016883755.1',\n",
       " 'XP_024307820.1',\n",
       " 'NP_059143.3',\n",
       " 'NP_001093890.1',\n",
       " 'NP_001093891.1',\n",
       " 'XP_011527755.1',\n",
       " 'NP_079220.2',\n",
       " 'NP_001191103.1',\n",
       " 'XP_016883762.1',\n",
       " 'NP_001191106.1',\n",
       " 'NP_001191105.1',\n",
       " 'NP_001191104.1',\n",
       " 'XP_011527759.1',\n",
       " 'NP_001191107.1',\n",
       " 'XP_011527960.1',\n",
       " 'NP_002763.3',\n",
       " 'XP_011527958.1',\n",
       " 'XP_011527961.1',\n",
       " 'XP_011527959.1',\n",
       " 'XP_011527956.1',\n",
       " 'XP_011527957.1',\n",
       " 'NP_001339525.1',\n",
       " 'NP_001339522.1',\n",
       " 'NP_001339524.1',\n",
       " 'NP_004531.2',\n",
       " 'XP_024307849.1',\n",
       " 'NP_001339520.1',\n",
       " 'NP_001339523.1',\n",
       " 'XP_011527883.1',\n",
       " 'NP_001339521.1',\n",
       " 'XP_011527884.1',\n",
       " 'XP_011527878.1',\n",
       " 'XP_011527882.1',\n",
       " 'XP_011527877.1',\n",
       " 'NP_001339526.1',\n",
       " 'XP_016883845.1',\n",
       " 'XP_016883846.1',\n",
       " 'XP_011527887.1',\n",
       " 'XP_011527953.1',\n",
       " 'NP_059142.3',\n",
       " 'NP_542984.3',\n",
       " 'XP_006724089.1',\n",
       " 'NP_067042.1',\n",
       " 'NP_001257337.1',\n",
       " 'NP_001257336.1',\n",
       " 'XP_024307891.1',\n",
       " 'XP_024307892.1',\n",
       " 'NP_001003701.1',\n",
       " 'NP_001003697.1',\n",
       " 'NP_001003696.1',\n",
       " 'NP_001307196.1',\n",
       " 'NP_001676.2',\n",
       " 'NP_001307195.1',\n",
       " 'NP_001003703.1',\n",
       " 'XP_005260995.1',\n",
       " 'NP_001184226.1',\n",
       " 'XP_024307830.1',\n",
       " 'XP_011527822.1',\n",
       " 'XP_011527823.1',\n",
       " 'NP_002031.2',\n",
       " 'XP_016883802.1',\n",
       " 'NP_001129488.1',\n",
       " 'NP_001191230.1',\n",
       " 'NP_958817.1',\n",
       " 'NP_001191231.1',\n",
       " 'NP_958816.1',\n",
       " 'NP_001129603.1',\n",
       " 'NP_000475.1',\n",
       " 'NP_001129601.1',\n",
       " 'NP_001191232.1',\n",
       " 'NP_001129602.1',\n",
       " 'XP_024307843.1',\n",
       " 'NP_443186.3',\n",
       " 'NP_001307697.2',\n",
       " 'XP_011527752.1',\n",
       " 'NP_008919.3',\n",
       " 'XP_024307822.1',\n",
       " 'XP_024307821.1',\n",
       " 'NP_008969.2',\n",
       " 'NP_877426.4',\n",
       " 'NP_037372.4',\n",
       " 'XP_016883806.1',\n",
       " 'NP_001307695.2',\n",
       " 'XP_016883805.1',\n",
       " 'NP_056380.3',\n",
       " 'XP_006724050.1',\n",
       " 'NP_058636.1',\n",
       " 'NP_001307653.1',\n",
       " 'XP_016883746.1',\n",
       " 'XP_016883748.1',\n",
       " 'NP_001001992.1',\n",
       " 'XP_016883747.1',\n",
       " 'NP_006438.1',\n",
       " 'NP_001027582.1',\n",
       " 'XP_016883749.1',\n",
       " 'XP_016883750.1',\n",
       " 'XP_024307813.1',\n",
       " 'NP_001269836.1',\n",
       " 'NP_001269837.1',\n",
       " 'NP_001269838.1',\n",
       " 'NP_006576.2',\n",
       " 'NP_001273552.1',\n",
       " 'NP_001358298.1',\n",
       " 'NP_064537.1',\n",
       " 'NP_001273563.1',\n",
       " 'NP_001273551.1',\n",
       " 'NP_001358299.1',\n",
       " 'NP_001358300.1',\n",
       " 'NP_001273548.1',\n",
       " 'NP_001358301.1',\n",
       " 'NP_001273547.1',\n",
       " 'NP_001358303.1',\n",
       " 'NP_001273546.1',\n",
       " 'NP_001358302.1',\n",
       " 'NP_001273549.1',\n",
       " 'NP_001273553.1',\n",
       " 'NP_001358305.1',\n",
       " 'NP_996749.1',\n",
       " 'NP_001177.1',\n",
       " 'NP_001307550.1',\n",
       " 'NP_001317922.1',\n",
       " 'NP_783300.1',\n",
       " 'NP_001307545.1',\n",
       " 'NP_001317923.1',\n",
       " 'NP_001307547.1',\n",
       " 'NP_000821.1',\n",
       " 'XP_005261001.1',\n",
       " 'NP_001307559.1',\n",
       " 'NP_036263.1',\n",
       " 'NP_955360.1',\n",
       " 'NP_001078924.1',\n",
       " 'NP_001122070.1',\n",
       " 'NP_981950.1',\n",
       " 'NP_001071179.1',\n",
       " 'NP_853655.1',\n",
       " 'NP_853652.1',\n",
       " 'NP_853630.2',\n",
       " 'NP_853653.1',\n",
       " 'NP_853631.1',\n",
       " 'NP_853654.1',\n",
       " 'NP_853638.1',\n",
       " 'NP_853639.1',\n",
       " 'NP_853640.1',\n",
       " 'NP_853641.1',\n",
       " 'NP_853642.1',\n",
       " 'NP_853643.1',\n",
       " 'NP_853645.1',\n",
       " 'NP_001157906.1',\n",
       " 'NP_853636.3',\n",
       " 'NP_853635.1',\n",
       " 'NP_853651.1',\n",
       " 'NP_853633.1',\n",
       " 'NP_853646.1',\n",
       " 'NP_001337906.1',\n",
       " 'NP_853647.1',\n",
       " 'NP_001121549.1',\n",
       " 'NP_001157907.1',\n",
       " 'NP_853648.1',\n",
       " 'NP_853650.1',\n",
       " 'NP_787053.1',\n",
       " 'NP_853637.2',\n",
       " 'NP_787054.1',\n",
       " 'NP_001092689.1',\n",
       " 'XP_016883939.1',\n",
       " 'NP_001340614.1',\n",
       " 'NP_001340619.1',\n",
       " 'XP_005261095.1',\n",
       " 'NP_001340616.1',\n",
       " 'XP_024307895.1',\n",
       " 'NP_001340618.1',\n",
       " 'NP_001340623.1',\n",
       " 'NP_001340615.1',\n",
       " 'XP_011528014.1',\n",
       " 'XP_016883937.1',\n",
       " 'NP_001340620.1',\n",
       " 'XP_016883940.1',\n",
       " 'NP_001340622.1',\n",
       " 'NP_003244.2',\n",
       " 'NP_001340621.1',\n",
       " 'NP_001340613.1',\n",
       " 'XP_011528015.1',\n",
       " 'NP_001340617.1',\n",
       " 'XP_005261097.1',\n",
       " 'NP_001307952.1',\n",
       " 'NP_000445.1',\n",
       " 'NP_001138917.1',\n",
       " 'XP_006724099.1',\n",
       " 'XP_005261074.1',\n",
       " 'XP_016883905.1',\n",
       " 'XP_006724098.1',\n",
       " 'NP_001138916.1',\n",
       " 'XP_016883904.1',\n",
       " 'XP_016883906.1',\n",
       " 'NP_065757.1',\n",
       " 'XP_011527839.1',\n",
       " 'NP_055401.1',\n",
       " 'XP_016883889.1',\n",
       " 'XP_016883891.1',\n",
       " 'XP_016883890.1',\n",
       " 'NP_061817.1',\n",
       " 'NP_996781.1',\n",
       " 'NP_848932.1',\n",
       " 'XP_006724091.1',\n",
       " 'XP_016883896.1',\n",
       " 'NP_001272323.1',\n",
       " 'NP_055640.2',\n",
       " 'XP_016883911.1',\n",
       " 'NP_001273485.1',\n",
       " 'XP_016883907.1',\n",
       " 'XP_016883908.1',\n",
       " 'NP_001307673.1',\n",
       " 'XP_006724101.1',\n",
       " 'XP_005261076.1',\n",
       " 'XP_016883912.1',\n",
       " 'NP_478067.2',\n",
       " 'NP_001307674.1',\n",
       " 'XP_016883910.1',\n",
       " 'XP_016883909.1',\n",
       " 'XP_011527971.1',\n",
       " 'XP_024307894.1',\n",
       " 'XP_024307893.1',\n",
       " 'NP_653260.1',\n",
       " 'NP_001337267.1',\n",
       " 'NP_067077.1',\n",
       " 'NP_001337263.1',\n",
       " 'NP_001337265.1',\n",
       " 'NP_001337266.1',\n",
       " 'NP_001337264.1',\n",
       " 'NP_003886.3',\n",
       " 'XP_016883988.1',\n",
       " 'XP_016883987.1',\n",
       " 'XP_016883984.1',\n",
       " 'XP_016883983.1',\n",
       " 'NP_001153778.1',\n",
       " 'XP_016883986.1',\n",
       " 'XP_016883985.1',\n",
       " 'XP_016883994.1',\n",
       " 'XP_016883991.1',\n",
       " 'NP_982271.3',\n",
       " 'XP_016883990.1',\n",
       " 'XP_016883993.1',\n",
       " 'NP_001153774.1',\n",
       " 'XP_016883989.1',\n",
       " 'XP_016883992.1',\n",
       " 'NP_057715.2',\n",
       " 'XP_006724129.1',\n",
       " 'NP_037461.2',\n",
       " 'XP_011528106.1',\n",
       " 'NP_001155967.2',\n",
       " 'NP_062542.5',\n",
       " 'NP_001155968.2',\n",
       " 'XP_011527955.1',\n",
       " 'NP_005797.1',\n",
       " 'XP_005260965.1',\n",
       " 'NP_620450.2',\n",
       " 'NP_000865.2',\n",
       " 'NP_001276055.1',\n",
       " 'NP_001276054.1',\n",
       " 'NP_001276057.1',\n",
       " 'NP_997468.1',\n",
       " 'NP_997467.1',\n",
       " 'NP_000619.3',\n",
       " 'XP_011527854.1',\n",
       " 'NP_000620.2',\n",
       " 'XP_005261021.1',\n",
       " 'NP_001316057.1',\n",
       " 'NP_005525.2',\n",
       " 'XP_011527855.1',\n",
       " 'XP_011527856.1',\n",
       " 'XP_011528048.1',\n",
       " 'NP_006125.2',\n",
       " 'NP_060303.2',\n",
       " 'NP_001307675.1',\n",
       " 'NP_001035282.1',\n",
       " 'XP_005260998.1',\n",
       " 'NP_001129478.1',\n",
       " 'NP_000810.1',\n",
       " 'NP_001129477.1',\n",
       " 'XP_006724053.1',\n",
       " 'XP_006724052.1',\n",
       " 'XP_011527828.1',\n",
       " 'NP_780294.1',\n",
       " 'NP_001278341.1',\n",
       " 'NP_620305.2',\n",
       " 'NP_001278340.1',\n",
       " 'NP_115571.2',\n",
       " 'NP_060083.1',\n",
       " 'NP_665857.2',\n",
       " 'NP_001317939.1',\n",
       " 'XP_016883919.1',\n",
       " 'NP_003015.2',\n",
       " 'XP_016883929.1',\n",
       " 'NP_001317937.1',\n",
       " 'XP_016883922.1',\n",
       " 'XP_016883925.1',\n",
       " 'XP_016883918.1',\n",
       " 'XP_016883917.1',\n",
       " 'NP_001317941.1',\n",
       " 'XP_011527994.1',\n",
       " 'NP_001317938.1',\n",
       " 'XP_016883921.1',\n",
       " 'XP_016883924.1',\n",
       " 'XP_016883927.1',\n",
       " 'NP_001317940.1',\n",
       " 'XP_016883923.1',\n",
       " 'XP_016883930.1',\n",
       " 'NP_001001132.1',\n",
       " 'XP_011527995.1',\n",
       " 'XP_016883920.1',\n",
       " 'XP_016883926.1',\n",
       " 'NP_001688.1',\n",
       " 'NP_115865.1',\n",
       " 'NP_008864.4',\n",
       " 'NP_751951.1',\n",
       " 'NP_001363828.1',\n",
       " 'NP_478062.1',\n",
       " 'NP_001269466.1',\n",
       " 'NP_001354277.1',\n",
       " 'NP_001257331.1',\n",
       " 'NP_001121142.1',\n",
       " 'NP_001257334.1',\n",
       " 'NP_000210.2',\n",
       " 'NP_001121141.1',\n",
       " 'NP_001257333.1',\n",
       " 'XP_016883831.1',\n",
       " 'NP_001121140.1',\n",
       " 'NP_001257332.1',\n",
       " 'XP_016883772.1',\n",
       " 'NP_981963.1',\n",
       " 'NP_004405.3',\n",
       " 'NP_001272318.1',\n",
       " 'NP_001272320.2',\n",
       " 'NP_981962.1',\n",
       " 'NP_001272321.1',\n",
       " 'NP_001272322.1',\n",
       " 'XP_016883771.1',\n",
       " 'NP_001317945.1',\n",
       " 'XP_005260986.1',\n",
       " 'NP_444507.1',\n",
       " 'XP_016883895.1',\n",
       " 'NP_001303938.1',\n",
       " 'XP_016883976.1',\n",
       " 'NP_001745.2',\n",
       " 'XP_011528069.1',\n",
       " 'XP_005261125.1',\n",
       " 'NP_001001890.1',\n",
       " 'XP_011528068.1',\n",
       " 'XP_011528070.1',\n",
       " 'XP_005261126.1',\n",
       " 'NP_001116079.1',\n",
       " 'XP_011528072.1',\n",
       " 'XP_016883893.1',\n",
       " 'NP_001273681.1',\n",
       " 'XP_011527945.1',\n",
       " 'XP_011527939.1',\n",
       " 'XP_011527946.1',\n",
       " 'XP_011527942.1',\n",
       " 'XP_011527941.1',\n",
       " 'XP_011527938.1',\n",
       " 'XP_011527944.1',\n",
       " 'NP_059134.1',\n",
       " 'XP_011527940.1',\n",
       " 'NP_001007260.1',\n",
       " 'XP_016883894.1',\n",
       " 'NP_001007262.1',\n",
       " 'XP_024307890.1',\n",
       " 'XP_024307889.1',\n",
       " 'NP_001273718.1',\n",
       " 'NP_001748.1',\n",
       " 'XP_011528074.1',\n",
       " 'NP_001227.1',\n",
       " 'NP_005119.2',\n",
       " 'NP_001307643.1',\n",
       " 'XP_016883998.1',\n",
       " 'NP_056173.1',\n",
       " 'NP_001307374.1',\n",
       " 'XP_011527820.1',\n",
       " 'NP_001307375.1',\n",
       " 'XP_016883800.1',\n",
       " 'XP_016883801.1',\n",
       " 'XP_016883967.1',\n",
       " 'XP_016883966.1',\n",
       " 'NP_005432.1',\n",
       " 'XP_011528057.1',\n",
       " 'NP_001139550.1',\n",
       " 'NP_001139549.1',\n",
       " 'NP_036262.1',\n",
       " 'NP_652763.1',\n",
       " 'NP_001139551.1',\n",
       " 'NP_033664.2',\n",
       " 'NP_005060.1',\n",
       " 'XP_016883931.1',\n",
       " 'XP_011527996.1',\n",
       " 'XP_016883819.1',\n",
       " 'NP_001229713.1',\n",
       " 'NP_000402.3',\n",
       " 'NP_001229714.1',\n",
       " 'NP_001339447.1',\n",
       " 'NP_001339443.1',\n",
       " 'NP_001339446.1',\n",
       " 'XP_024307834.1',\n",
       " 'XP_024307833.1',\n",
       " 'XP_011527841.1',\n",
       " 'XP_006724058.1',\n",
       " 'NP_001339445.1',\n",
       " 'NP_001339444.1',\n",
       " 'XP_011527840.1',\n",
       " 'XP_011527842.1',\n",
       " 'NP_061835.1',\n",
       " 'NP_001304697.1',\n",
       " 'NP_001304706.1',\n",
       " 'NP_710148.1',\n",
       " 'NP_057514.2',\n",
       " 'NP_001307409.1',\n",
       " 'XP_016883854.1',\n",
       " 'NP_710149.1',\n",
       " 'XP_016883947.1',\n",
       " 'NP_001307632.1',\n",
       " 'XP_016883948.1',\n",
       " 'XP_016883949.1',\n",
       " 'XP_005261109.1',\n",
       " 'NP_001307633.1',\n",
       " 'NP_001317612.1',\n",
       " 'XP_005261105.1',\n",
       " 'XP_005261111.1',\n",
       " 'XP_016883951.1',\n",
       " 'NP_001001894.1',\n",
       " 'XP_005261107.1',\n",
       " 'XP_016883950.1',\n",
       " 'XP_005261106.1',\n",
       " 'NP_003307.3',\n",
       " 'XP_011528042.1',\n",
       " 'XP_011528041.1',\n",
       " 'NP_001340865.1',\n",
       " 'XP_005261112.1',\n",
       " 'NP_001317611.1',\n",
       " 'XP_016883954.1',\n",
       " 'NP_001317610.1',\n",
       " 'XP_005261113.1',\n",
       " 'NP_001340866.1',\n",
       " 'NP_001340867.1',\n",
       " 'NP_006043.1',\n",
       " 'NP_001317951.1',\n",
       " 'NP_001317950.1',\n",
       " 'NP_001317947.1',\n",
       " 'NP_001334651.1',\n",
       " 'XP_006724040.1',\n",
       " 'XP_016883773.1',\n",
       " 'NP_001334650.1',\n",
       " 'NP_001387.2',\n",
       " 'XP_006724039.1',\n",
       " 'XP_011527785.1',\n",
       " 'NP_569120.1',\n",
       " 'NP_567824.1',\n",
       " 'NP_569122.1',\n",
       " 'XP_006724041.1',\n",
       " 'XP_024307825.1',\n",
       " 'XP_016883775.1',\n",
       " 'NP_001334652.1',\n",
       " 'XP_005260990.1',\n",
       " 'XP_016884012.1',\n",
       " 'NP_002231.1',\n",
       " 'XP_011527863.1',\n",
       " 'NP_001263368.1',\n",
       " 'XP_016883834.1',\n",
       " 'NP_733933.1',\n",
       " 'NP_001263364.1',\n",
       " 'XP_016883832.1',\n",
       " 'XP_005261032.1',\n",
       " 'NP_001263367.1',\n",
       " 'NP_733932.1',\n",
       " 'XP_016883833.1',\n",
       " 'NP_001263366.1',\n",
       " 'NP_002234.2',\n",
       " 'XP_011527862.1',\n",
       " 'XP_006724065.1',\n",
       " 'NP_001263365.1',\n",
       " 'NP_001230361.1',\n",
       " 'NP_001230358.1',\n",
       " 'NP_001230357.1',\n",
       " 'NP_001129627.1',\n",
       " 'NP_891548.1',\n",
       " 'NP_001129626.1',\n",
       " 'NP_001317954.1',\n",
       " 'NP_004440.1',\n",
       " 'NP_001278320.1',\n",
       " 'NP_001243224.1',\n",
       " 'XP_005260992.1',\n",
       " 'XP_016883779.1',\n",
       " 'NP_005230.1',\n",
       " 'NP_001248753.1',\n",
       " 'NP_001307724.1',\n",
       " 'NP_003711.1',\n",
       " 'XP_024307903.1',\n",
       " 'NP_982257.1',\n",
       " 'NP_061836.2',\n",
       " 'XP_011527914.1',\n",
       " 'XP_016883862.1',\n",
       " 'XP_016883863.1',\n",
       " 'XP_011527915.1',\n",
       " 'XP_011527913.1',\n",
       " 'NP_387505.1',\n",
       " 'XP_016883864.1',\n",
       " 'NP_001007247.1',\n",
       " 'XP_016883822.1',\n",
       " 'XP_024307836.1',\n",
       " 'XP_011527845.1',\n",
       " 'XP_016883825.1',\n",
       " 'XP_024307839.1',\n",
       " 'XP_011527851.1',\n",
       " 'XP_024307835.1',\n",
       " 'XP_016883824.1',\n",
       " 'XP_011527846.1',\n",
       " 'XP_024307838.1',\n",
       " 'XP_016883823.1',\n",
       " 'XP_024307837.1',\n",
       " 'NP_004956.5',\n",
       " 'NP_001337229.1',\n",
       " 'NP_004618.2',\n",
       " 'NP_001304673.1',\n",
       " 'NP_001337222.1',\n",
       " 'NP_001337224.1',\n",
       " 'NP_001139690.1',\n",
       " 'NP_001337225.1',\n",
       " 'NP_001337223.1',\n",
       " 'XP_011527769.1',\n",
       " 'XP_024307823.1',\n",
       " 'XP_011527760.1',\n",
       " 'XP_011527772.1',\n",
       " 'XP_016883765.1',\n",
       " 'XP_011527768.1',\n",
       " 'XP_011527763.1',\n",
       " 'XP_006724030.1',\n",
       " 'XP_011527764.1',\n",
       " 'XP_016883764.1',\n",
       " 'XP_011527765.1',\n",
       " 'XP_011527770.1',\n",
       " 'XP_006724036.1',\n",
       " 'XP_011527773.1',\n",
       " 'XP_016883767.1',\n",
       " 'XP_005260983.1',\n",
       " 'XP_011527767.1',\n",
       " 'XP_011527766.1',\n",
       " 'XP_011527761.1',\n",
       " 'NP_689718.1',\n",
       " 'XP_024307824.1',\n",
       " 'XP_006724035.1',\n",
       " 'XP_016883766.1',\n",
       " 'XP_011527771.1',\n",
       " 'XP_016883763.1',\n",
       " 'XP_011527762.1',\n",
       " 'NP_001304669.1',\n",
       " 'NP_031367.1',\n",
       " 'NP_001304671.1',\n",
       " 'NP_001304670.1',\n",
       " 'NP_001001713.1',\n",
       " 'NP_149362.2',\n",
       " 'NP_001265579.1',\n",
       " 'NP_149360.1',\n",
       " 'NP_001343268.1',\n",
       " 'NP_001343265.1',\n",
       " 'NP_006048.1',\n",
       " 'NP_001343267.1',\n",
       " 'NP_149361.1',\n",
       " 'XP_011527774.1',\n",
       " 'NP_001073913.1',\n",
       " 'NP_006189.2',\n",
       " 'NP_001380.2',\n",
       " 'XP_016883770.1',\n",
       " 'NP_001258463.1',\n",
       " 'NP_036237.2',\n",
       " 'NP_620477.1',\n",
       " 'NP_620476.1',\n",
       " 'XP_016883803.1',\n",
       " 'XP_011527951.1',\n",
       " 'NP_996847.1',\n",
       " 'NP_478066.3',\n",
       " 'XP_011527950.2',\n",
       " 'XP_005261040.1',\n",
       " 'XP_011527873.1',\n",
       " 'XP_011527875.1',\n",
       " 'XP_011527874.1',\n",
       " 'XP_024307848.1',\n",
       " 'XP_005261041.1',\n",
       " 'NP_002454.1',\n",
       " 'XP_011527876.1',\n",
       " 'NP_001138397.1',\n",
       " 'XP_016883839.1',\n",
       " 'XP_011527872.1',\n",
       " 'XP_016883841.1',\n",
       " 'XP_016883838.1',\n",
       " 'XP_005261036.1',\n",
       " 'XP_005261039.1',\n",
       " 'NP_002453.2',\n",
       " 'XP_016883840.1',\n",
       " 'NP_001269849.1',\n",
       " 'XP_005261038.1',\n",
       " 'XP_011527870.1',\n",
       " 'XP_005261037.1',\n",
       " 'NP_001171517.1',\n",
       " 'XP_005261035.1',\n",
       " 'NP_001128571.1',\n",
       " 'NP_005647.3',\n",
       " 'NP_065690.2',\n",
       " 'XP_016883914.1',\n",
       " 'XP_011527978.1',\n",
       " 'XP_011527979.1',\n",
       " 'XP_011527980.1',\n",
       " 'XP_006724103.1',\n",
       " 'XP_011527976.1',\n",
       " 'NP_001269863.2',\n",
       " 'XP_011527977.1',\n",
       " 'XP_016883915.1',\n",
       " 'XP_006724102.1',\n",
       " 'NP_001035514.2',\n",
       " 'XP_011527985.1',\n",
       " 'XP_011527981.1',\n",
       " 'NP_071398.4',\n",
       " 'XP_011527983.1',\n",
       " 'XP_005261166.1',\n",
       " 'NP_950251.1',\n",
       " 'NP_056315.1',\n",
       " 'XP_011527825.1',\n",
       " 'XP_016883804.1',\n",
       " 'XP_011527824.1',\n",
       " 'XP_011527826.1',\n",
       " 'NP_001307660.1',\n",
       " 'NP_001091873.1',\n",
       " 'NP_065778.3',\n",
       " 'NP_001307658.1',\n",
       " 'XP_005261178.1',\n",
       " 'XP_016883849.1',\n",
       " 'NP_001091872.1',\n",
       " 'XP_011527892.1',\n",
       " 'XP_011527890.1',\n",
       " 'XP_016883850.1',\n",
       " 'NP_775839.3',\n",
       " 'NP_001004416.2',\n",
       " 'NP_001186457.3',\n",
       " 'XP_016883995.1',\n",
       " 'NP_001186456.2',\n",
       " 'XP_016883996.1',\n",
       " 'XP_011528099.1',\n",
       " 'XP_016883997.1',\n",
       " 'NP_997510.1',\n",
       " 'NP_997512.1',\n",
       " 'NP_004906.3',\n",
       " 'NP_058198.2',\n",
       " 'XP_011528109.1',\n",
       " 'NP_997057.1',\n",
       " 'XP_011528108.1',\n",
       " 'NP_997511.1',\n",
       " 'XP_024307909.1',\n",
       " 'NP_003217.3',\n",
       " 'NP_005414.1',\n",
       " 'NP_003216.1',\n",
       " 'NP_001243246.1',\n",
       " 'NP_115780.1',\n",
       " 'NP_076927.1',\n",
       " 'NP_115781.1',\n",
       " 'NP_061834.1',\n",
       " 'XP_011527911.1',\n",
       " 'NP_001001895.1',\n",
       " 'NP_001230396.1',\n",
       " 'XP_011527907.1',\n",
       " 'XP_006724076.1',\n",
       " 'XP_011527909.1',\n",
       " 'XP_011527908.1',\n",
       " 'XP_011527912.1',\n",
       " 'XP_011528088.1',\n",
       " 'NP_001273435.1',\n",
       " 'NP_543136.1',\n",
       " 'XP_005261265.1',\n",
       " 'XP_016883867.1',\n",
       " 'XP_016883870.1',\n",
       " 'NP_061837.3',\n",
       " 'NP_001307466.1',\n",
       " 'XP_016883866.1',\n",
       " 'XP_016883869.1',\n",
       " 'XP_016883872.1',\n",
       " 'XP_016883868.1',\n",
       " 'XP_011527916.1',\n",
       " 'XP_011527917.1',\n",
       " 'XP_016883865.1',\n",
       " 'XP_016883871.1',\n",
       " 'XP_016884013.1',\n",
       " 'XP_016884014.1',\n",
       " 'NP_001001569.1',\n",
       " 'NP_002597.1',\n",
       " 'NP_001001578.1',\n",
       " 'XP_011527902.1',\n",
       " 'XP_016883856.1',\n",
       " 'NP_001001574.1',\n",
       " 'NP_001001567.1',\n",
       " 'NP_001001570.1',\n",
       " 'NP_001001583.1',\n",
       " 'NP_001001575.1',\n",
       " 'NP_001001582.1',\n",
       " 'NP_001001568.1',\n",
       " 'NP_001001581.1',\n",
       " 'NP_001001571.1',\n",
       " 'NP_001001577.1',\n",
       " 'XP_024307852.1',\n",
       " 'NP_001302462.1',\n",
       " 'XP_011527900.1',\n",
       " 'NP_001001580.1',\n",
       " 'NP_001001576.1',\n",
       " 'NP_001001579.1',\n",
       " 'NP_001001585.1',\n",
       " 'NP_001001572.1',\n",
       " 'NP_001001584.1',\n",
       " 'NP_001001573.1',\n",
       " 'XP_016883855.1',\n",
       " 'XP_024307817.1',\n",
       " 'XP_011527735.1',\n",
       " 'XP_016883752.1',\n",
       " 'NP_387510.1',\n",
       " 'XP_024307816.1',\n",
       " 'NP_061139.2',\n",
       " 'XP_024307815.1',\n",
       " 'XP_016883751.1',\n",
       " 'NP_001247406.1',\n",
       " 'XP_024307818.1',\n",
       " 'NP_001247405.1',\n",
       " 'XP_016883753.1',\n",
       " 'NP_001247403.1',\n",
       " 'NP_001247404.1',\n",
       " 'NP_001001503.1',\n",
       " 'XP_011527888.1',\n",
       " 'NP_066553.3',\n",
       " 'XP_016883848.1',\n",
       " 'NP_001295420.1',\n",
       " 'NP_001307623.1',\n",
       " 'NP_004562.2',\n",
       " 'NP_001273187.1',\n",
       " 'XP_016883980.1',\n",
       " 'NP_001171480.1',\n",
       " 'NP_001307227.1',\n",
       " 'XP_024307908.1',\n",
       " 'XP_024307905.1',\n",
       " 'NP_001171479.1',\n",
       " 'XP_024307907.1',\n",
       " 'NP_000062.1',\n",
       " 'XP_011528079.1',\n",
       " 'XP_011528085.1',\n",
       " 'XP_011528076.1',\n",
       " 'NP_001308001.1',\n",
       " 'XP_024307904.1',\n",
       " 'XP_024307906.1',\n",
       " 'XP_011528045.1',\n",
       " 'NP_006749.1',\n",
       " 'XP_024307898.1',\n",
       " 'XP_016883957.1',\n",
       " 'NP_001020375.1',\n",
       " 'XP_024307897.1',\n",
       " 'XP_024307899.1',\n",
       " 'NP_001020374.1',\n",
       " 'NP_000385.1',\n",
       " 'NP_001350695.1',\n",
       " 'NP_775490.2',\n",
       " 'XP_011527776.1',\n",
       " 'XP_016883758.1',\n",
       " 'XP_016883757.1',\n",
       " 'XP_011527747.2',\n",
       " 'XP_016883756.1',\n",
       " 'XP_016883759.1',\n",
       " 'XP_011527745.2',\n",
       " 'XP_016883760.1',\n",
       " 'XP_011527748.1',\n",
       " 'NP_008962.1',\n",
       " 'XP_016883761.1',\n",
       " 'XP_011527749.1',\n",
       " 'NP_059141.1',\n",
       " 'NP_055871.1',\n",
       " 'NP_003672.1',\n",
       " 'XP_005261252.2',\n",
       " 'XP_011528064.1',\n",
       " 'XP_011528060.1',\n",
       " 'XP_005261253.1',\n",
       " 'XP_011528113.1',\n",
       " 'NP_001317959.1',\n",
       " 'XP_011528063.1',\n",
       " 'XP_011528062.1',\n",
       " 'XP_005261256.1',\n",
       " 'XP_016883973.1',\n",
       " 'XP_016883972.1',\n",
       " 'NP_000091.1',\n",
       " 'XP_016883974.1',\n",
       " 'NP_003674.1',\n",
       " 'XP_016883975.1',\n",
       " 'XP_024307902.1',\n",
       " 'XP_016883897.1',\n",
       " 'XP_011527965.1',\n",
       " 'XP_016883903.1',\n",
       " 'XP_011527963.1',\n",
       " 'XP_011527966.1',\n",
       " 'XP_011527962.1',\n",
       " 'XP_011527964.1',\n",
       " 'NP_064517.1',\n",
       " 'XP_006724093.1',\n",
       " 'NP_001356807.1',\n",
       " 'NP_001032642.1',\n",
       " 'XP_011527967.1',\n",
       " 'XP_005261217.1',\n",
       " 'NP_001356809.1',\n",
       " 'XP_006724092.1',\n",
       " 'NP_001356810.1',\n",
       " 'XP_006724094.1',\n",
       " 'XP_016883901.1',\n",
       " 'XP_016883900.1',\n",
       " 'XP_011528028.1',\n",
       " 'NP_003265.3',\n",
       " 'XP_011528016.1',\n",
       " 'XP_011528020.1',\n",
       " 'XP_011528018.1',\n",
       " 'XP_011528022.1',\n",
       " 'XP_011528021.1',\n",
       " 'XP_011528027.1',\n",
       " 'XP_005261225.1',\n",
       " 'XP_011528019.1',\n",
       " 'XP_011528032.1',\n",
       " 'XP_011528017.1',\n",
       " 'XP_016883943.1',\n",
       " 'XP_011528023.1',\n",
       " 'XP_024307896.1',\n",
       " 'XP_011528024.1',\n",
       " 'XP_011528025.1',\n",
       " 'XP_011528031.1',\n",
       " 'XP_011528029.1',\n",
       " 'XP_011528030.1',\n",
       " 'NP_001338638.1',\n",
       " 'NP_005040.2',\n",
       " 'XP_011527969.1',\n",
       " 'NP_004640.4',\n",
       " 'NP_001307313.1',\n",
       " 'NP_001307312.1',\n",
       " 'NP_937798.4',\n",
       " 'XP_016883968.1',\n",
       " 'XP_024307901.1',\n",
       " 'XP_011527818.1',\n",
       " 'XP_011527816.1',\n",
       " 'NP_001269979.1',\n",
       " 'NP_056074.1',\n",
       " 'NP_001269981.1',\n",
       " 'NP_001352688.1',\n",
       " 'NP_001269980.1',\n",
       " 'XP_024307828.1',\n",
       " 'XP_011527817.1',\n",
       " 'NP_037501.2',\n",
       " 'NP_787063.1',\n",
       " 'NP_000374.1',\n",
       " ...]"
      ]
     },
     "execution_count": 108,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "protein_accession_numbers = list(protein_data[protein_data['Replicon Accession']==accession_to_study]['Protein product'])\n",
    "protein_accession_numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Entrez\n",
    "Entrez.api_key = api_key\n",
    "Entrez.email = 'elizabethfawcett47@gmail.com'\n",
    "Entrez.tool = 'locus_by_amino_acid'\n",
    "id_list = protein_accession_numbers[0:200]\n",
    "search_results = Entrez.read(Entrez.epost(\"protein\", id=\",\".join(id_list)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [],
   "source": [
    "webenv = search_results[\"WebEnv\"]\n",
    "query_key = search_results[\"QueryKey\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [],
   "source": [
    "results = Entrez.read(Entrez.efetch(db=\"protein\", webenv=webenv, query_key=query_key, retmode='xml'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [],
   "source": [
    "protein_additional_data_1 = pd.DataFrame()\n",
    "accession_version = []\n",
    "definition = []\n",
    "division = []\n",
    "topology = []\n",
    "moltype = []\n",
    "organism = []\n",
    "sequence = []\n",
    "chromosome = []\n",
    "molar_weight = []\n",
    "regions = []\n",
    "sites = []\n",
    "for i in results:\n",
    "    region = 0\n",
    "    site = 0\n",
    "    accession_version.append(i['GBSeq_accession-version'])\n",
    "    organism.append(i['GBSeq_organism'])\n",
    "    definition.append(i['GBSeq_definition'])\n",
    "    division.append(i['GBSeq_division'])\n",
    "    topology.append(i['GBSeq_topology'])\n",
    "    moltype.append(i['GBSeq_moltype'])\n",
    "    sequence.append(i['GBSeq_sequence'])\n",
    "    if i['GBSeq_feature-table'][0]['GBFeature_key'] == 'source':\n",
    "        chromosome.append(i['GBSeq_feature-table'][0]['GBFeature_quals'][2]['GBQualifier_value'])\n",
    "    else:\n",
    "        continue\n",
    "    if i['GBSeq_feature-table'][1]['GBFeature_key'] == 'Protein':\n",
    "        if i['GBSeq_feature-table'][1]['GBFeature_quals'][0]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][0]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][1]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][1]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][2]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][2]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][3]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][3]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][4]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][4]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][5]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][5]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][6]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][6]['GBQualifier_value'])\n",
    "        else:\n",
    "            print('add more elif statements')\n",
    "    else:\n",
    "        continue\n",
    "    for j in range(0, len(i['GBSeq_feature-table'])):\n",
    "        if i['GBSeq_feature-table'][j]['GBFeature_key'] == 'Region':\n",
    "            region += 1\n",
    "        else:\n",
    "            continue\n",
    "    for j in range(0, len(i['GBSeq_feature-table'])):\n",
    "        if i['GBSeq_feature-table'][j]['GBFeature_key'] == 'Site':\n",
    "            site += 1\n",
    "        else:\n",
    "            continue\n",
    "    regions.append(region)\n",
    "    sites.append(site)\n",
    "protein_additional_data_1['Chromosome'] = chromosome\n",
    "protein_additional_data_1['Accession Version'] = accession_version\n",
    "protein_additional_data_1['Organism'] = organism\n",
    "protein_additional_data_1['Definition'] = definition\n",
    "protein_additional_data_1['Division'] = division\n",
    "protein_additional_data_1['Topology'] = topology\n",
    "protein_additional_data_1['Molecule Type'] = moltype\n",
    "protein_additional_data_1['Molecular Weight'] = molar_weight\n",
    "protein_additional_data_1['Number of Regions'] = regions\n",
    "protein_additional_data_1['Number of Binding Sites'] = sites\n",
    "protein_additional_data_1['Sequence'] = sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Accession Version</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Definition</th>\n",
       "      <th>Division</th>\n",
       "      <th>Topology</th>\n",
       "      <th>Molecule Type</th>\n",
       "      <th>Molecular Weight</th>\n",
       "      <th>Number of Regions</th>\n",
       "      <th>Number of Binding Sites</th>\n",
       "      <th>Sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_006723962.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ICOS ligand isoform X1 [Homo sapiens]</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>56181</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_011544381.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ICOS ligand isoform X3 [Homo sapiens]</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>33350</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_011544380.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ICOS ligand isoform X2 [Homo sapiens]</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>39507</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001350699.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ICOS ligand precursor [Homo sapiens]</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>31389</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001350689.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>glutamine amidotransferase-like class 1 domain...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>24598</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>maavralvasrlaaasaftslspggrtpsqraalhlsvprpaarva...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>195</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_024307892.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>junctional adhesion molecule B isoform X1 [Hom...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>22874</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>midfniriknvtrsdagkyrcevsapseqgqnleedtvtlevlvap...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>196</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001003701.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ATP synthase-coupling factor 6, mitochondrial ...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>13257</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>mhcdggismilqrlfrfssvirsavsvhlrrnigvtavafnkeldp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>197</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001003697.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ATP synthase-coupling factor 6, mitochondrial ...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>8960</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>milqrlfrfssvirsavsvhlrrnigvtavafnkeldpiqklfvdk...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>198</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001003696.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ATP synthase-coupling factor 6, mitochondrial ...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>8960</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>milqrlfrfssvirsavsvhlrrnigvtavafnkeldpiqklfvdk...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>199</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001307196.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ATP synthase-coupling factor 6, mitochondrial ...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>12456</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>milqrlfrfssvirsavsvhlrrnigvtavafnkeldpiqklfvdk...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>200 rows  11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Chromosome Accession Version      Organism  \\\n",
       "0           21    XP_006723962.1  Homo sapiens   \n",
       "1           21    XP_011544381.1  Homo sapiens   \n",
       "2           21    XP_011544380.1  Homo sapiens   \n",
       "3           21    NP_001350699.1  Homo sapiens   \n",
       "4           21    NP_001350689.1  Homo sapiens   \n",
       "..         ...               ...           ...   \n",
       "195         21    XP_024307892.1  Homo sapiens   \n",
       "196         21    NP_001003701.1  Homo sapiens   \n",
       "197         21    NP_001003697.1  Homo sapiens   \n",
       "198         21    NP_001003696.1  Homo sapiens   \n",
       "199         21    NP_001307196.1  Homo sapiens   \n",
       "\n",
       "                                            Definition Division Topology  \\\n",
       "0                ICOS ligand isoform X1 [Homo sapiens]      PRI   linear   \n",
       "1                ICOS ligand isoform X3 [Homo sapiens]      PRI   linear   \n",
       "2                ICOS ligand isoform X2 [Homo sapiens]      PRI   linear   \n",
       "3                 ICOS ligand precursor [Homo sapiens]      PRI   linear   \n",
       "4    glutamine amidotransferase-like class 1 domain...      PRI   linear   \n",
       "..                                                 ...      ...      ...   \n",
       "195  junctional adhesion molecule B isoform X1 [Hom...      PRI   linear   \n",
       "196  ATP synthase-coupling factor 6, mitochondrial ...      PRI   linear   \n",
       "197  ATP synthase-coupling factor 6, mitochondrial ...      PRI   linear   \n",
       "198  ATP synthase-coupling factor 6, mitochondrial ...      PRI   linear   \n",
       "199  ATP synthase-coupling factor 6, mitochondrial ...      PRI   linear   \n",
       "\n",
       "    Molecule Type Molecular Weight  Number of Regions  \\\n",
       "0              AA            56181                  2   \n",
       "1              AA            33350                  2   \n",
       "2              AA            39507                  2   \n",
       "3              AA            31389                  2   \n",
       "4              AA            24598                  1   \n",
       "..            ...              ...                ...   \n",
       "195            AA            22874                  2   \n",
       "196            AA            13257                  1   \n",
       "197            AA             8960                  1   \n",
       "198            AA             8960                  1   \n",
       "199            AA            12456                  1   \n",
       "\n",
       "     Number of Binding Sites  \\\n",
       "0                          0   \n",
       "1                          0   \n",
       "2                          0   \n",
       "3                          0   \n",
       "4                          1   \n",
       "..                       ...   \n",
       "195                        0   \n",
       "196                        0   \n",
       "197                        9   \n",
       "198                        9   \n",
       "199                        9   \n",
       "\n",
       "                                              Sequence  \n",
       "0    mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...  \n",
       "1    mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...  \n",
       "2    mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...  \n",
       "3    mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...  \n",
       "4    maavralvasrlaaasaftslspggrtpsqraalhlsvprpaarva...  \n",
       "..                                                 ...  \n",
       "195  midfniriknvtrsdagkyrcevsapseqgqnleedtvtlevlvap...  \n",
       "196  mhcdggismilqrlfrfssvirsavsvhlrrnigvtavafnkeldp...  \n",
       "197  milqrlfrfssvirsavsvhlrrnigvtavafnkeldpiqklfvdk...  \n",
       "198  milqrlfrfssvirsavsvhlrrnigvtavafnkeldpiqklfvdk...  \n",
       "199  milqrlfrfssvirsavsvhlrrnigvtavafnkeldpiqklfvdk...  \n",
       "\n",
       "[200 rows x 11 columns]"
      ]
     },
     "execution_count": 113,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "protein_additional_data_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [],
   "source": [
    "import time\n",
    "time.sleep(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Entrez\n",
    "Entrez.api_key = api_key\n",
    "Entrez.email = 'elizabethfawcett47@gmail.com'\n",
    "Entrez.tool = 'locus_by_amino_acid'\n",
    "id_list = protein_accession_numbers[200:400]\n",
    "search_results = Entrez.read(Entrez.epost(\"protein\", id=\",\".join(id_list)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [],
   "source": [
    "webenv = search_results[\"WebEnv\"]\n",
    "query_key = search_results[\"QueryKey\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [],
   "source": [
    "results = Entrez.read(Entrez.efetch(db=\"protein\", webenv=webenv, query_key=query_key, retmode='xml'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [],
   "source": [
    "protein_additional_data_2 = pd.DataFrame()\n",
    "accession_version = []\n",
    "definition = []\n",
    "division = []\n",
    "topology = []\n",
    "moltype = []\n",
    "organism = []\n",
    "sequence = []\n",
    "chromosome = []\n",
    "molar_weight = []\n",
    "regions = []\n",
    "sites = []\n",
    "for i in results:\n",
    "    region = 0\n",
    "    site = 0\n",
    "    accession_version.append(i['GBSeq_accession-version'])\n",
    "    organism.append(i['GBSeq_organism'])\n",
    "    definition.append(i['GBSeq_definition'])\n",
    "    division.append(i['GBSeq_division'])\n",
    "    topology.append(i['GBSeq_topology'])\n",
    "    moltype.append(i['GBSeq_moltype'])\n",
    "    sequence.append(i['GBSeq_sequence'])\n",
    "    if i['GBSeq_feature-table'][0]['GBFeature_key'] == 'source':\n",
    "        chromosome.append(i['GBSeq_feature-table'][0]['GBFeature_quals'][2]['GBQualifier_value'])\n",
    "    else:\n",
    "        continue\n",
    "    if i['GBSeq_feature-table'][1]['GBFeature_key'] == 'Protein':\n",
    "        if i['GBSeq_feature-table'][1]['GBFeature_quals'][0]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][0]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][1]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][1]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][2]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][2]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][3]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][3]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][4]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][4]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][5]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][5]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][6]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][6]['GBQualifier_value'])\n",
    "        else:\n",
    "            print('add more elif statements')\n",
    "    else:\n",
    "        continue\n",
    "    for j in range(0, len(i['GBSeq_feature-table'])):\n",
    "        if i['GBSeq_feature-table'][j]['GBFeature_key'] == 'Region':\n",
    "            region += 1\n",
    "        else:\n",
    "            continue\n",
    "    for j in range(0, len(i['GBSeq_feature-table'])):\n",
    "        if i['GBSeq_feature-table'][j]['GBFeature_key'] == 'Site':\n",
    "            site += 1\n",
    "        else:\n",
    "            continue\n",
    "    regions.append(region)\n",
    "    sites.append(site)\n",
    "protein_additional_data_2['Chromosome'] = chromosome\n",
    "protein_additional_data_2['Accession Version'] = accession_version\n",
    "protein_additional_data_2['Organism'] = organism\n",
    "protein_additional_data_2['Definition'] = definition\n",
    "protein_additional_data_2['Division'] = division\n",
    "protein_additional_data_2['Topology'] = topology\n",
    "protein_additional_data_2['Molecule Type'] = moltype\n",
    "protein_additional_data_2['Molecular Weight'] = molar_weight\n",
    "protein_additional_data_2['Number of Regions'] = regions\n",
    "protein_additional_data_2['Number of Binding Sites'] = sites\n",
    "protein_additional_data_2['Sequence'] = sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Accession Version</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Definition</th>\n",
       "      <th>Division</th>\n",
       "      <th>Topology</th>\n",
       "      <th>Molecule Type</th>\n",
       "      <th>Molecular Weight</th>\n",
       "      <th>Number of Regions</th>\n",
       "      <th>Number of Binding Sites</th>\n",
       "      <th>Sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001676.2</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ATP synthase-coupling factor 6, mitochondrial ...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>8960</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>milqrlfrfssvirsavsvhlrrnigvtavafnkeldpiqklfvdk...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001307195.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ATP synthase-coupling factor 6, mitochondrial ...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>12456</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>milqrlfrfssvirsavsvhlrrnigvtavafnkeldpiqklfvdk...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001003703.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ATP synthase-coupling factor 6, mitochondrial ...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>8960</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>milqrlfrfssvirsavsvhlrrnigvtavafnkeldpiqklfvdk...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_005260995.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GA-binding protein alpha chain isoform X1 [Hom...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>51164</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>mtkreaeelieieidgtekaecteesiveqtyapaecvsqaidine...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001184226.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GA-binding protein alpha chain [Homo sapiens]</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>51164</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>mtkreaeelieieidgtekaecteesiveqtyapaecvsqaidine...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>195</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_037461.2</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>PAX3- and PAX7-binding protein 1 isoform 2 [Ho...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>93040</td>\n",
       "      <td>2</td>\n",
       "      <td>10</td>\n",
       "      <td>mfrkarrvnvrkrndseeeererdeeqepppllpppgtgeeagpgg...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>196</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_011528106.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>PAX3- and PAX7-binding protein 1 isoform X2 [H...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>73726</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>mfrkarrvnvrkrndseeeererdeeqepppllpppgtgeeagpgg...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>197</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001155967.2</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>uncharacterized protein C21orf62 precursor [Ho...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>22843</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>mappsrhclllistlgvfalncftkgqknstliftrentirncscs...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>198</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_062542.5</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>uncharacterized protein C21orf62 precursor [Ho...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>22843</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>mappsrhclllistlgvfalncftkgqknstliftrentirncscs...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>199</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001155968.2</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>uncharacterized protein C21orf62 precursor [Ho...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>22843</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>mappsrhclllistlgvfalncftkgqknstliftrentirncscs...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>200 rows  11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Chromosome Accession Version      Organism  \\\n",
       "0           21       NP_001676.2  Homo sapiens   \n",
       "1           21    NP_001307195.1  Homo sapiens   \n",
       "2           21    NP_001003703.1  Homo sapiens   \n",
       "3           21    XP_005260995.1  Homo sapiens   \n",
       "4           21    NP_001184226.1  Homo sapiens   \n",
       "..         ...               ...           ...   \n",
       "195         21       NP_037461.2  Homo sapiens   \n",
       "196         21    XP_011528106.1  Homo sapiens   \n",
       "197         21    NP_001155967.2  Homo sapiens   \n",
       "198         21       NP_062542.5  Homo sapiens   \n",
       "199         21    NP_001155968.2  Homo sapiens   \n",
       "\n",
       "                                            Definition Division Topology  \\\n",
       "0    ATP synthase-coupling factor 6, mitochondrial ...      PRI   linear   \n",
       "1    ATP synthase-coupling factor 6, mitochondrial ...      PRI   linear   \n",
       "2    ATP synthase-coupling factor 6, mitochondrial ...      PRI   linear   \n",
       "3    GA-binding protein alpha chain isoform X1 [Hom...      PRI   linear   \n",
       "4        GA-binding protein alpha chain [Homo sapiens]      PRI   linear   \n",
       "..                                                 ...      ...      ...   \n",
       "195  PAX3- and PAX7-binding protein 1 isoform 2 [Ho...      PRI   linear   \n",
       "196  PAX3- and PAX7-binding protein 1 isoform X2 [H...      PRI   linear   \n",
       "197  uncharacterized protein C21orf62 precursor [Ho...      PRI   linear   \n",
       "198  uncharacterized protein C21orf62 precursor [Ho...      PRI   linear   \n",
       "199  uncharacterized protein C21orf62 precursor [Ho...      PRI   linear   \n",
       "\n",
       "    Molecule Type Molecular Weight  Number of Regions  \\\n",
       "0              AA             8960                  1   \n",
       "1              AA            12456                  1   \n",
       "2              AA             8960                  1   \n",
       "3              AA            51164                  3   \n",
       "4              AA            51164                  3   \n",
       "..            ...              ...                ...   \n",
       "195            AA            93040                  2   \n",
       "196            AA            73726                  1   \n",
       "197            AA            22843                  1   \n",
       "198            AA            22843                  1   \n",
       "199            AA            22843                  1   \n",
       "\n",
       "     Number of Binding Sites  \\\n",
       "0                          9   \n",
       "1                          9   \n",
       "2                          9   \n",
       "3                          0   \n",
       "4                          1   \n",
       "..                       ...   \n",
       "195                       10   \n",
       "196                        0   \n",
       "197                        0   \n",
       "198                        0   \n",
       "199                        0   \n",
       "\n",
       "                                              Sequence  \n",
       "0    milqrlfrfssvirsavsvhlrrnigvtavafnkeldpiqklfvdk...  \n",
       "1    milqrlfrfssvirsavsvhlrrnigvtavafnkeldpiqklfvdk...  \n",
       "2    milqrlfrfssvirsavsvhlrrnigvtavafnkeldpiqklfvdk...  \n",
       "3    mtkreaeelieieidgtekaecteesiveqtyapaecvsqaidine...  \n",
       "4    mtkreaeelieieidgtekaecteesiveqtyapaecvsqaidine...  \n",
       "..                                                 ...  \n",
       "195  mfrkarrvnvrkrndseeeererdeeqepppllpppgtgeeagpgg...  \n",
       "196  mfrkarrvnvrkrndseeeererdeeqepppllpppgtgeeagpgg...  \n",
       "197  mappsrhclllistlgvfalncftkgqknstliftrentirncscs...  \n",
       "198  mappsrhclllistlgvfalncftkgqknstliftrentirncscs...  \n",
       "199  mappsrhclllistlgvfalncftkgqknstliftrentirncscs...  \n",
       "\n",
       "[200 rows x 11 columns]"
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "protein_additional_data_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [],
   "source": [
    "time.sleep(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Entrez\n",
    "Entrez.api_key = api_key\n",
    "Entrez.email = 'elizabethfawcett47@gmail.com'\n",
    "Entrez.tool = 'locus_by_amino_acid'\n",
    "id_list = protein_accession_numbers[400:600]\n",
    "search_results = Entrez.read(Entrez.epost(\"protein\", id=\",\".join(id_list)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [],
   "source": [
    "webenv = search_results[\"WebEnv\"]\n",
    "query_key = search_results[\"QueryKey\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [],
   "source": [
    "results = Entrez.read(Entrez.efetch(db=\"protein\", webenv=webenv, query_key=query_key, retmode='xml'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [],
   "source": [
    "protein_additional_data_3 = pd.DataFrame()\n",
    "accession_version = []\n",
    "definition = []\n",
    "division = []\n",
    "topology = []\n",
    "moltype = []\n",
    "organism = []\n",
    "sequence = []\n",
    "chromosome = []\n",
    "molar_weight = []\n",
    "regions = []\n",
    "sites = []\n",
    "for i in results:\n",
    "    region = 0\n",
    "    site = 0\n",
    "    accession_version.append(i['GBSeq_accession-version'])\n",
    "    organism.append(i['GBSeq_organism'])\n",
    "    definition.append(i['GBSeq_definition'])\n",
    "    division.append(i['GBSeq_division'])\n",
    "    topology.append(i['GBSeq_topology'])\n",
    "    moltype.append(i['GBSeq_moltype'])\n",
    "    sequence.append(i['GBSeq_sequence'])\n",
    "    if i['GBSeq_feature-table'][0]['GBFeature_key'] == 'source':\n",
    "        chromosome.append(i['GBSeq_feature-table'][0]['GBFeature_quals'][2]['GBQualifier_value'])\n",
    "    else:\n",
    "        continue\n",
    "    if i['GBSeq_feature-table'][1]['GBFeature_key'] == 'Protein':\n",
    "        if i['GBSeq_feature-table'][1]['GBFeature_quals'][0]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][0]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][1]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][1]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][2]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][2]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][3]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][3]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][4]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][4]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][5]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][5]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][6]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][6]['GBQualifier_value'])\n",
    "        else:\n",
    "            print('add more elif statements')\n",
    "    else:\n",
    "        continue\n",
    "    for j in range(0, len(i['GBSeq_feature-table'])):\n",
    "        if i['GBSeq_feature-table'][j]['GBFeature_key'] == 'Region':\n",
    "            region += 1\n",
    "        else:\n",
    "            continue\n",
    "    for j in range(0, len(i['GBSeq_feature-table'])):\n",
    "        if i['GBSeq_feature-table'][j]['GBFeature_key'] == 'Site':\n",
    "            site += 1\n",
    "        else:\n",
    "            continue\n",
    "    regions.append(region)\n",
    "    sites.append(site)\n",
    "protein_additional_data_3['Chromosome'] = chromosome\n",
    "protein_additional_data_3['Accession Version'] = accession_version\n",
    "protein_additional_data_3['Organism'] = organism\n",
    "protein_additional_data_3['Definition'] = definition\n",
    "protein_additional_data_3['Division'] = division\n",
    "protein_additional_data_3['Topology'] = topology\n",
    "protein_additional_data_3['Molecule Type'] = moltype\n",
    "protein_additional_data_3['Molecular Weight'] = molar_weight\n",
    "protein_additional_data_3['Number of Regions'] = regions\n",
    "protein_additional_data_3['Number of Binding Sites'] = sites\n",
    "protein_additional_data_3['Sequence'] = sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Accession Version</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Definition</th>\n",
       "      <th>Division</th>\n",
       "      <th>Topology</th>\n",
       "      <th>Molecule Type</th>\n",
       "      <th>Molecular Weight</th>\n",
       "      <th>Number of Regions</th>\n",
       "      <th>Number of Binding Sites</th>\n",
       "      <th>Sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_011527955.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>uncharacterized protein C21orf62 isoform X1 [H...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>11529</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>mrlvgqqlppgchgndqaagppkteknskcktthstlacrlnqlss...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_005797.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>oligodendrocyte transcription factor 2 [Homo s...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>32254</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>mdsdaslvssrpsspepddlflparskgssgsaftggtvssstpsd...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_005260965.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>oligodendrocyte transcription factor 2 isoform...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>32254</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>mdsdaslvssrpsspepddlflparskgssgsaftggtvssstpsd...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_620450.2</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>oligodendrocyte transcription factor 1 [Homo s...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>27774</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>myyavsqarvnavpgtmlrpqrpgdlqlgaslyelvgyrqppssss...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_000865.2</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>interferon alpha/beta receptor 2 isoform b [Ho...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>37262</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>mllsqnafifrslnlvlmvyislvfgisydspdytdesctfkislr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>195</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_006724040.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>dual specificity tyrosine-phosphorylation-regu...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>85453</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>mhtggetsackpssvrlapsfsfhaaglqmagqmphshqysdrrqp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>196</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_016883773.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>dual specificity tyrosine-phosphorylation-regu...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>84426</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>mhtggetsackpssvrlapsfsfhaaglqmagqmphshqysdrrqp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>197</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001334650.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>dual specificity tyrosine-phosphorylation-regu...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>84426</td>\n",
       "      <td>2</td>\n",
       "      <td>20</td>\n",
       "      <td>mhtggetsackpssvrlapsfsfhaaglqmagqmphshqysdrrqp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>198</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001387.2</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>dual specificity tyrosine-phosphorylation-regu...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>85453</td>\n",
       "      <td>2</td>\n",
       "      <td>20</td>\n",
       "      <td>mhtggetsackpssvrlapsfsfhaaglqmagqmphshqysdrrqp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>199</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_006724039.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>dual specificity tyrosine-phosphorylation-regu...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>85453</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>mhtggetsackpssvrlapsfsfhaaglqmagqmphshqysdrrqp...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>200 rows  11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Chromosome Accession Version      Organism  \\\n",
       "0           21    XP_011527955.1  Homo sapiens   \n",
       "1           21       NP_005797.1  Homo sapiens   \n",
       "2           21    XP_005260965.1  Homo sapiens   \n",
       "3           21       NP_620450.2  Homo sapiens   \n",
       "4           21       NP_000865.2  Homo sapiens   \n",
       "..         ...               ...           ...   \n",
       "195         21    XP_006724040.1  Homo sapiens   \n",
       "196         21    XP_016883773.1  Homo sapiens   \n",
       "197         21    NP_001334650.1  Homo sapiens   \n",
       "198         21       NP_001387.2  Homo sapiens   \n",
       "199         21    XP_006724039.1  Homo sapiens   \n",
       "\n",
       "                                            Definition Division Topology  \\\n",
       "0    uncharacterized protein C21orf62 isoform X1 [H...      PRI   linear   \n",
       "1    oligodendrocyte transcription factor 2 [Homo s...      PRI   linear   \n",
       "2    oligodendrocyte transcription factor 2 isoform...      PRI   linear   \n",
       "3    oligodendrocyte transcription factor 1 [Homo s...      PRI   linear   \n",
       "4    interferon alpha/beta receptor 2 isoform b [Ho...      PRI   linear   \n",
       "..                                                 ...      ...      ...   \n",
       "195  dual specificity tyrosine-phosphorylation-regu...      PRI   linear   \n",
       "196  dual specificity tyrosine-phosphorylation-regu...      PRI   linear   \n",
       "197  dual specificity tyrosine-phosphorylation-regu...      PRI   linear   \n",
       "198  dual specificity tyrosine-phosphorylation-regu...      PRI   linear   \n",
       "199  dual specificity tyrosine-phosphorylation-regu...      PRI   linear   \n",
       "\n",
       "    Molecule Type Molecular Weight  Number of Regions  \\\n",
       "0              AA            11529                  0   \n",
       "1              AA            32254                  1   \n",
       "2              AA            32254                  1   \n",
       "3              AA            27774                  1   \n",
       "4              AA            37262                  3   \n",
       "..            ...              ...                ...   \n",
       "195            AA            85453                  1   \n",
       "196            AA            84426                  1   \n",
       "197            AA            84426                  2   \n",
       "198            AA            85453                  2   \n",
       "199            AA            85453                  1   \n",
       "\n",
       "     Number of Binding Sites  \\\n",
       "0                          0   \n",
       "1                          3   \n",
       "2                          3   \n",
       "3                          3   \n",
       "4                          0   \n",
       "..                       ...   \n",
       "195                        4   \n",
       "196                        4   \n",
       "197                       20   \n",
       "198                       20   \n",
       "199                        4   \n",
       "\n",
       "                                              Sequence  \n",
       "0    mrlvgqqlppgchgndqaagppkteknskcktthstlacrlnqlss...  \n",
       "1    mdsdaslvssrpsspepddlflparskgssgsaftggtvssstpsd...  \n",
       "2    mdsdaslvssrpsspepddlflparskgssgsaftggtvssstpsd...  \n",
       "3    myyavsqarvnavpgtmlrpqrpgdlqlgaslyelvgyrqppssss...  \n",
       "4    mllsqnafifrslnlvlmvyislvfgisydspdytdesctfkislr...  \n",
       "..                                                 ...  \n",
       "195  mhtggetsackpssvrlapsfsfhaaglqmagqmphshqysdrrqp...  \n",
       "196  mhtggetsackpssvrlapsfsfhaaglqmagqmphshqysdrrqp...  \n",
       "197  mhtggetsackpssvrlapsfsfhaaglqmagqmphshqysdrrqp...  \n",
       "198  mhtggetsackpssvrlapsfsfhaaglqmagqmphshqysdrrqp...  \n",
       "199  mhtggetsackpssvrlapsfsfhaaglqmagqmphshqysdrrqp...  \n",
       "\n",
       "[200 rows x 11 columns]"
      ]
     },
     "execution_count": 125,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "protein_additional_data_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [],
   "source": [
    "time.sleep(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Entrez\n",
    "Entrez.api_key = api_key\n",
    "Entrez.email = 'elizabethfawcett47@gmail.com'\n",
    "Entrez.tool = 'locus_by_amino_acid'\n",
    "id_list = protein_accession_numbers[600:800]\n",
    "search_results = Entrez.read(Entrez.epost(\"protein\", id=\",\".join(id_list)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [],
   "source": [
    "webenv = search_results[\"WebEnv\"]\n",
    "query_key = search_results[\"QueryKey\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [],
   "source": [
    "results = Entrez.read(Entrez.efetch(db=\"protein\", webenv=webenv, query_key=query_key, retmode='xml'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [],
   "source": [
    "protein_additional_data_4 = pd.DataFrame()\n",
    "accession_version = []\n",
    "definition = []\n",
    "division = []\n",
    "topology = []\n",
    "moltype = []\n",
    "organism = []\n",
    "sequence = []\n",
    "chromosome = []\n",
    "molar_weight = []\n",
    "regions = []\n",
    "sites = []\n",
    "for i in results:\n",
    "    region = 0\n",
    "    site = 0\n",
    "    accession_version.append(i['GBSeq_accession-version'])\n",
    "    organism.append(i['GBSeq_organism'])\n",
    "    definition.append(i['GBSeq_definition'])\n",
    "    division.append(i['GBSeq_division'])\n",
    "    topology.append(i['GBSeq_topology'])\n",
    "    moltype.append(i['GBSeq_moltype'])\n",
    "    sequence.append(i['GBSeq_sequence'])\n",
    "    if i['GBSeq_feature-table'][0]['GBFeature_key'] == 'source':\n",
    "        chromosome.append(i['GBSeq_feature-table'][0]['GBFeature_quals'][2]['GBQualifier_value'])\n",
    "    else:\n",
    "        continue\n",
    "    if i['GBSeq_feature-table'][1]['GBFeature_key'] == 'Protein':\n",
    "        if i['GBSeq_feature-table'][1]['GBFeature_quals'][0]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][0]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][1]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][1]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][2]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][2]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][3]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][3]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][4]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][4]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][5]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][5]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][6]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][6]['GBQualifier_value'])\n",
    "        else:\n",
    "            print('add more elif statements')\n",
    "    else:\n",
    "        continue\n",
    "    for j in range(0, len(i['GBSeq_feature-table'])):\n",
    "        if i['GBSeq_feature-table'][j]['GBFeature_key'] == 'Region':\n",
    "            region += 1\n",
    "        else:\n",
    "            continue\n",
    "    for j in range(0, len(i['GBSeq_feature-table'])):\n",
    "        if i['GBSeq_feature-table'][j]['GBFeature_key'] == 'Site':\n",
    "            site += 1\n",
    "        else:\n",
    "            continue\n",
    "    regions.append(region)\n",
    "    sites.append(site)\n",
    "protein_additional_data_4['Chromosome'] = chromosome\n",
    "protein_additional_data_4['Accession Version'] = accession_version\n",
    "protein_additional_data_4['Organism'] = organism\n",
    "protein_additional_data_4['Definition'] = definition\n",
    "protein_additional_data_4['Division'] = division\n",
    "protein_additional_data_4['Topology'] = topology\n",
    "protein_additional_data_4['Molecule Type'] = moltype\n",
    "protein_additional_data_4['Molecular Weight'] = molar_weight\n",
    "protein_additional_data_4['Number of Regions'] = regions\n",
    "protein_additional_data_4['Number of Binding Sites'] = sites\n",
    "protein_additional_data_4['Sequence'] = sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Accession Version</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Definition</th>\n",
       "      <th>Division</th>\n",
       "      <th>Topology</th>\n",
       "      <th>Molecule Type</th>\n",
       "      <th>Molecular Weight</th>\n",
       "      <th>Number of Regions</th>\n",
       "      <th>Number of Binding Sites</th>\n",
       "      <th>Sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_011527785.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>dual specificity tyrosine-phosphorylation-regu...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>85453</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>mhtggetsackpssvrlapsfsfhaaglqmagqmphshqysdrrqp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_569120.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>dual specificity tyrosine-phosphorylation-regu...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>84426</td>\n",
       "      <td>2</td>\n",
       "      <td>20</td>\n",
       "      <td>mhtggetsackpssvrlapsfsfhaaglqmagqmphshqysdrrqp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_567824.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>dual specificity tyrosine-phosphorylation-regu...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>65968</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>mhtggetsackpssvrlapsfsfhaaglqmagqmphshqysdrrqp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_569122.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>dual specificity tyrosine-phosphorylation-regu...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>60199</td>\n",
       "      <td>2</td>\n",
       "      <td>16</td>\n",
       "      <td>mhtggetsackpssvrlapsfsfhaaglqmagqmphshqysdrrqp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_006724041.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>dual specificity tyrosine-phosphorylation-regu...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>85453</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>mhtggetsackpssvrlapsfsfhaaglqmagqmphshqysdrrqp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>195</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_004906.3</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ATP-binding cassette sub-family G member 1 iso...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>75461</td>\n",
       "      <td>3</td>\n",
       "      <td>13</td>\n",
       "      <td>maclmaafsvgtamnassysaemtepksvcvsvdevvssnmeatet...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>196</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_058198.2</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ATP-binding cassette sub-family G member 1 iso...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>74024</td>\n",
       "      <td>3</td>\n",
       "      <td>13</td>\n",
       "      <td>maclmaafsvgtamnassysaemtepksvcvsvdevvssnmeatet...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>197</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_011528109.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ATP-binding cassette sub-family G member 1 iso...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>70746</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>mrislpraperdggvsasslldtvtnassysaemtepksvcvsvde...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>198</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_997057.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ATP-binding cassette sub-family G member 1 iso...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>75250</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>mrislpraperdggvsasslldtvtnassysaemtepksvcvsvde...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>199</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_011528108.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ATP-binding cassette sub-family G member 1 iso...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>76687</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>mrislpraperdggvsasslldtvtnassysaemtepksvcvsvde...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>200 rows  11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Chromosome Accession Version      Organism  \\\n",
       "0           21    XP_011527785.1  Homo sapiens   \n",
       "1           21       NP_569120.1  Homo sapiens   \n",
       "2           21       NP_567824.1  Homo sapiens   \n",
       "3           21       NP_569122.1  Homo sapiens   \n",
       "4           21    XP_006724041.1  Homo sapiens   \n",
       "..         ...               ...           ...   \n",
       "195         21       NP_004906.3  Homo sapiens   \n",
       "196         21       NP_058198.2  Homo sapiens   \n",
       "197         21    XP_011528109.1  Homo sapiens   \n",
       "198         21       NP_997057.1  Homo sapiens   \n",
       "199         21    XP_011528108.1  Homo sapiens   \n",
       "\n",
       "                                            Definition Division Topology  \\\n",
       "0    dual specificity tyrosine-phosphorylation-regu...      PRI   linear   \n",
       "1    dual specificity tyrosine-phosphorylation-regu...      PRI   linear   \n",
       "2    dual specificity tyrosine-phosphorylation-regu...      PRI   linear   \n",
       "3    dual specificity tyrosine-phosphorylation-regu...      PRI   linear   \n",
       "4    dual specificity tyrosine-phosphorylation-regu...      PRI   linear   \n",
       "..                                                 ...      ...      ...   \n",
       "195  ATP-binding cassette sub-family G member 1 iso...      PRI   linear   \n",
       "196  ATP-binding cassette sub-family G member 1 iso...      PRI   linear   \n",
       "197  ATP-binding cassette sub-family G member 1 iso...      PRI   linear   \n",
       "198  ATP-binding cassette sub-family G member 1 iso...      PRI   linear   \n",
       "199  ATP-binding cassette sub-family G member 1 iso...      PRI   linear   \n",
       "\n",
       "    Molecule Type Molecular Weight  Number of Regions  \\\n",
       "0              AA            85453                  1   \n",
       "1              AA            84426                  2   \n",
       "2              AA            65968                  2   \n",
       "3              AA            60199                  2   \n",
       "4              AA            85453                  1   \n",
       "..            ...              ...                ...   \n",
       "195            AA            75461                  3   \n",
       "196            AA            74024                  3   \n",
       "197            AA            70746                  3   \n",
       "198            AA            75250                  3   \n",
       "199            AA            76687                  3   \n",
       "\n",
       "     Number of Binding Sites  \\\n",
       "0                          4   \n",
       "1                         20   \n",
       "2                          4   \n",
       "3                         16   \n",
       "4                          4   \n",
       "..                       ...   \n",
       "195                       13   \n",
       "196                       13   \n",
       "197                        7   \n",
       "198                        7   \n",
       "199                        7   \n",
       "\n",
       "                                              Sequence  \n",
       "0    mhtggetsackpssvrlapsfsfhaaglqmagqmphshqysdrrqp...  \n",
       "1    mhtggetsackpssvrlapsfsfhaaglqmagqmphshqysdrrqp...  \n",
       "2    mhtggetsackpssvrlapsfsfhaaglqmagqmphshqysdrrqp...  \n",
       "3    mhtggetsackpssvrlapsfsfhaaglqmagqmphshqysdrrqp...  \n",
       "4    mhtggetsackpssvrlapsfsfhaaglqmagqmphshqysdrrqp...  \n",
       "..                                                 ...  \n",
       "195  maclmaafsvgtamnassysaemtepksvcvsvdevvssnmeatet...  \n",
       "196  maclmaafsvgtamnassysaemtepksvcvsvdevvssnmeatet...  \n",
       "197  mrislpraperdggvsasslldtvtnassysaemtepksvcvsvde...  \n",
       "198  mrislpraperdggvsasslldtvtnassysaemtepksvcvsvde...  \n",
       "199  mrislpraperdggvsasslldtvtnassysaemtepksvcvsvde...  \n",
       "\n",
       "[200 rows x 11 columns]"
      ]
     },
     "execution_count": 131,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "protein_additional_data_4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [],
   "source": [
    "time.sleep(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Entrez\n",
    "Entrez.api_key = api_key\n",
    "Entrez.email = 'elizabethfawcett47@gmail.com'\n",
    "Entrez.tool = 'locus_by_amino_acid'\n",
    "id_list = protein_accession_numbers[800:1000]\n",
    "search_results = Entrez.read(Entrez.epost(\"protein\", id=\",\".join(id_list)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [],
   "source": [
    "webenv = search_results[\"WebEnv\"]\n",
    "query_key = search_results[\"QueryKey\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [],
   "source": [
    "results = Entrez.read(Entrez.efetch(db=\"protein\", webenv=webenv, query_key=query_key, retmode='xml'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [],
   "source": [
    "protein_additional_data_5 = pd.DataFrame()\n",
    "accession_version = []\n",
    "definition = []\n",
    "division = []\n",
    "topology = []\n",
    "moltype = []\n",
    "organism = []\n",
    "sequence = []\n",
    "chromosome = []\n",
    "molar_weight = []\n",
    "regions = []\n",
    "sites = []\n",
    "for i in results:\n",
    "    region = 0\n",
    "    site = 0\n",
    "    accession_version.append(i['GBSeq_accession-version'])\n",
    "    organism.append(i['GBSeq_organism'])\n",
    "    definition.append(i['GBSeq_definition'])\n",
    "    division.append(i['GBSeq_division'])\n",
    "    topology.append(i['GBSeq_topology'])\n",
    "    moltype.append(i['GBSeq_moltype'])\n",
    "    sequence.append(i['GBSeq_sequence'])\n",
    "    if i['GBSeq_feature-table'][0]['GBFeature_key'] == 'source':\n",
    "        chromosome.append(i['GBSeq_feature-table'][0]['GBFeature_quals'][2]['GBQualifier_value'])\n",
    "    else:\n",
    "        continue\n",
    "    if i['GBSeq_feature-table'][1]['GBFeature_key'] == 'Protein':\n",
    "        if i['GBSeq_feature-table'][1]['GBFeature_quals'][0]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][0]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][1]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][1]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][2]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][2]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][3]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][3]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][4]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][4]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][5]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][5]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][6]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][6]['GBQualifier_value'])\n",
    "        else:\n",
    "            print('add more elif statements')\n",
    "    else:\n",
    "        continue\n",
    "    for j in range(0, len(i['GBSeq_feature-table'])):\n",
    "        if i['GBSeq_feature-table'][j]['GBFeature_key'] == 'Region':\n",
    "            region += 1\n",
    "        else:\n",
    "            continue\n",
    "    for j in range(0, len(i['GBSeq_feature-table'])):\n",
    "        if i['GBSeq_feature-table'][j]['GBFeature_key'] == 'Site':\n",
    "            site += 1\n",
    "        else:\n",
    "            continue\n",
    "    regions.append(region)\n",
    "    sites.append(site)\n",
    "protein_additional_data_5['Chromosome'] = chromosome\n",
    "protein_additional_data_5['Accession Version'] = accession_version\n",
    "protein_additional_data_5['Organism'] = organism\n",
    "protein_additional_data_5['Definition'] = definition\n",
    "protein_additional_data_5['Division'] = division\n",
    "protein_additional_data_5['Topology'] = topology\n",
    "protein_additional_data_5['Molecule Type'] = moltype\n",
    "protein_additional_data_5['Molecular Weight'] = molar_weight\n",
    "protein_additional_data_5['Number of Regions'] = regions\n",
    "protein_additional_data_5['Number of Binding Sites'] = sites\n",
    "protein_additional_data_5['Sequence'] = sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Accession Version</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Definition</th>\n",
       "      <th>Division</th>\n",
       "      <th>Topology</th>\n",
       "      <th>Molecule Type</th>\n",
       "      <th>Molecular Weight</th>\n",
       "      <th>Number of Regions</th>\n",
       "      <th>Number of Binding Sites</th>\n",
       "      <th>Sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_997511.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ATP-binding cassette sub-family G member 1 iso...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>71829</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>mtepksvcvsvdevvssnmeatetdllnghlkkvdnnlteaqrfss...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_024307909.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ATP-binding cassette sub-family G member 1 iso...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>86625</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>mvrrgwsvctaillarlwclvpthtflseypeaaeyphpgwvywlq...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_003217.3</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>trefoil factor 3 precursor [Homo sapiens]</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>6581</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>mkrvlscvpeptvvmaaralcmlglvlallssssaeeyvglsanqc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_005414.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>trefoil factor 2 precursor [Homo sapiens]</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>11990</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>mgrrdaqllaallvlglcalagsekpspcqcsrlsphnrtncgfpg...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_003216.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>trefoil factor 1 precursor [Homo sapiens]</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>6674</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>matmenkvicalvlvsmlalgtlaeaqtetctvaprerqncgfpgv...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>195</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_024307828.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ICOS ligand isoform X1 [Homo sapiens]</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>51794</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>196</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_011527817.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ICOS ligand isoform X3 [Homo sapiens]</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>33350</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>197</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_037501.2</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>DNA (cytosine-5)-methyltransferase 3-like isof...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>43539</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>maaipaldpeaepsmdvilvgsselsssvspgtgrdliayevkanq...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>198</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_787063.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>DNA (cytosine-5)-methyltransferase 3-like isof...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>43452</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>maaipaldpeaepsmdvilvgsselsssvspgtgrdliayevkanq...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>199</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_000374.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>autoimmune regulator [Homo sapiens]</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>57596</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>matdaalrrllrlhrteiavavdsafpllhaladhdvvpedkfqet...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>200 rows  11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Chromosome Accession Version      Organism  \\\n",
       "0           21       NP_997511.1  Homo sapiens   \n",
       "1           21    XP_024307909.1  Homo sapiens   \n",
       "2           21       NP_003217.3  Homo sapiens   \n",
       "3           21       NP_005414.1  Homo sapiens   \n",
       "4           21       NP_003216.1  Homo sapiens   \n",
       "..         ...               ...           ...   \n",
       "195         21    XP_024307828.1  Homo sapiens   \n",
       "196         21    XP_011527817.1  Homo sapiens   \n",
       "197         21       NP_037501.2  Homo sapiens   \n",
       "198         21       NP_787063.1  Homo sapiens   \n",
       "199         21       NP_000374.1  Homo sapiens   \n",
       "\n",
       "                                            Definition Division Topology  \\\n",
       "0    ATP-binding cassette sub-family G member 1 iso...      PRI   linear   \n",
       "1    ATP-binding cassette sub-family G member 1 iso...      PRI   linear   \n",
       "2            trefoil factor 3 precursor [Homo sapiens]      PRI   linear   \n",
       "3            trefoil factor 2 precursor [Homo sapiens]      PRI   linear   \n",
       "4            trefoil factor 1 precursor [Homo sapiens]      PRI   linear   \n",
       "..                                                 ...      ...      ...   \n",
       "195              ICOS ligand isoform X1 [Homo sapiens]      PRI   linear   \n",
       "196              ICOS ligand isoform X3 [Homo sapiens]      PRI   linear   \n",
       "197  DNA (cytosine-5)-methyltransferase 3-like isof...      PRI   linear   \n",
       "198  DNA (cytosine-5)-methyltransferase 3-like isof...      PRI   linear   \n",
       "199                autoimmune regulator [Homo sapiens]      PRI   linear   \n",
       "\n",
       "    Molecule Type Molecular Weight  Number of Regions  \\\n",
       "0              AA            71829                  3   \n",
       "1              AA            86625                  1   \n",
       "2              AA             6581                  1   \n",
       "3              AA            11990                  2   \n",
       "4              AA             6674                  1   \n",
       "..            ...              ...                ...   \n",
       "195            AA            51794                  2   \n",
       "196            AA            33350                  2   \n",
       "197            AA            43539                  1   \n",
       "198            AA            43452                  1   \n",
       "199            AA            57596                 12   \n",
       "\n",
       "     Number of Binding Sites  \\\n",
       "0                          7   \n",
       "1                          0   \n",
       "2                          2   \n",
       "3                          4   \n",
       "4                          2   \n",
       "..                       ...   \n",
       "195                        0   \n",
       "196                        0   \n",
       "197                        2   \n",
       "198                        2   \n",
       "199                        4   \n",
       "\n",
       "                                              Sequence  \n",
       "0    mtepksvcvsvdevvssnmeatetdllnghlkkvdnnlteaqrfss...  \n",
       "1    mvrrgwsvctaillarlwclvpthtflseypeaaeyphpgwvywlq...  \n",
       "2    mkrvlscvpeptvvmaaralcmlglvlallssssaeeyvglsanqc...  \n",
       "3    mgrrdaqllaallvlglcalagsekpspcqcsrlsphnrtncgfpg...  \n",
       "4    matmenkvicalvlvsmlalgtlaeaqtetctvaprerqncgfpgv...  \n",
       "..                                                 ...  \n",
       "195  mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...  \n",
       "196  mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...  \n",
       "197  maaipaldpeaepsmdvilvgsselsssvspgtgrdliayevkanq...  \n",
       "198  maaipaldpeaepsmdvilvgsselsssvspgtgrdliayevkanq...  \n",
       "199  matdaalrrllrlhrteiavavdsafpllhaladhdvvpedkfqet...  \n",
       "\n",
       "[200 rows x 11 columns]"
      ]
     },
     "execution_count": 137,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "protein_additional_data_5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [],
   "source": [
    "time.sleep(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Entrez\n",
    "Entrez.api_key = api_key\n",
    "Entrez.email = 'elizabethfawcett47@gmail.com'\n",
    "Entrez.tool = 'locus_by_amino_acid'\n",
    "id_list = protein_accession_numbers[1000:1200]\n",
    "search_results = Entrez.read(Entrez.epost(\"protein\", id=\",\".join(id_list)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {},
   "outputs": [],
   "source": [
    "webenv = search_results[\"WebEnv\"]\n",
    "query_key = search_results[\"QueryKey\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {},
   "outputs": [],
   "source": [
    "results = Entrez.read(Entrez.efetch(db=\"protein\", webenv=webenv, query_key=query_key, retmode='xml'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [],
   "source": [
    "protein_additional_data_6 = pd.DataFrame()\n",
    "accession_version = []\n",
    "definition = []\n",
    "division = []\n",
    "topology = []\n",
    "moltype = []\n",
    "organism = []\n",
    "sequence = []\n",
    "chromosome = []\n",
    "molar_weight = []\n",
    "regions = []\n",
    "sites = []\n",
    "for i in results:\n",
    "    region = 0\n",
    "    site = 0\n",
    "    accession_version.append(i['GBSeq_accession-version'])\n",
    "    organism.append(i['GBSeq_organism'])\n",
    "    definition.append(i['GBSeq_definition'])\n",
    "    division.append(i['GBSeq_division'])\n",
    "    topology.append(i['GBSeq_topology'])\n",
    "    moltype.append(i['GBSeq_moltype'])\n",
    "    sequence.append(i['GBSeq_sequence'])\n",
    "    if i['GBSeq_feature-table'][0]['GBFeature_key'] == 'source':\n",
    "        chromosome.append(i['GBSeq_feature-table'][0]['GBFeature_quals'][2]['GBQualifier_value'])\n",
    "    else:\n",
    "        continue\n",
    "    if i['GBSeq_feature-table'][1]['GBFeature_key'] == 'Protein':\n",
    "        if i['GBSeq_feature-table'][1]['GBFeature_quals'][0]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][0]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][1]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][1]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][2]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][2]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][3]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][3]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][4]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][4]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][5]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][5]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][6]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][6]['GBQualifier_value'])\n",
    "        else:\n",
    "            print('add more elif statements')\n",
    "    else:\n",
    "        continue\n",
    "    for j in range(0, len(i['GBSeq_feature-table'])):\n",
    "        if i['GBSeq_feature-table'][j]['GBFeature_key'] == 'Region':\n",
    "            region += 1\n",
    "        else:\n",
    "            continue\n",
    "    for j in range(0, len(i['GBSeq_feature-table'])):\n",
    "        if i['GBSeq_feature-table'][j]['GBFeature_key'] == 'Site':\n",
    "            site += 1\n",
    "        else:\n",
    "            continue\n",
    "    regions.append(region)\n",
    "    sites.append(site)\n",
    "protein_additional_data_6['Chromosome'] = chromosome\n",
    "protein_additional_data_6['Accession Version'] = accession_version\n",
    "protein_additional_data_6['Organism'] = organism\n",
    "protein_additional_data_6['Definition'] = definition\n",
    "protein_additional_data_6['Division'] = division\n",
    "protein_additional_data_6['Topology'] = topology\n",
    "protein_additional_data_6['Molecule Type'] = moltype\n",
    "protein_additional_data_6['Molecular Weight'] = molar_weight\n",
    "protein_additional_data_6['Number of Regions'] = regions\n",
    "protein_additional_data_6['Number of Binding Sites'] = sites\n",
    "protein_additional_data_6['Sequence'] = sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Accession Version</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Definition</th>\n",
       "      <th>Division</th>\n",
       "      <th>Topology</th>\n",
       "      <th>Molecule Type</th>\n",
       "      <th>Molecular Weight</th>\n",
       "      <th>Number of Regions</th>\n",
       "      <th>Number of Binding Sites</th>\n",
       "      <th>Sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_002617.3</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ATP-dependent 6-phosphofructokinase, liver typ...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>84888</td>\n",
       "      <td>10</td>\n",
       "      <td>5</td>\n",
       "      <td>maavdleklrasgagkaigvltsggdaqgmnaavravtrmgiyvga...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_005261192.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ATP-dependent 6-phosphofructokinase, liver typ...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>87546</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>maavdleklrasgagkaigvltsggdaqgmnaavravtrmgiyvga...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001002021.2</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ATP-dependent 6-phosphofructokinase, liver typ...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>90447</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>mrsmcnqgrgressrgglhvqgscrglsrspqqetgfakapagtdc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_016883857.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ATP-dependent 6-phosphofructokinase, liver typ...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>93105</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>mrsmcnqgrgressrgglhvqgscrglsrspqqetgfakapagtdc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_006724074.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ATP-dependent 6-phosphofructokinase, liver typ...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>93105</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>mrsmcnqgrgressrgglhvqgscrglsrspqqetgfakapagtdc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>195</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_005261245.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>speriolin-like protein isoform X1 [Homo sapiens]</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>37482</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>maeggelmsrllsenadlkkqvrllkenqmlrrllsqscqeggghd...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>196</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001138908.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>lanosterol synthase isoform 2 [Homo sapiens]</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>82068</td>\n",
       "      <td>8</td>\n",
       "      <td>3</td>\n",
       "      <td>mtegtclrrrggpyktepatdlgrwrlncergrqtwtylqderagr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>197</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_002331.3</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>lanosterol synthase isoform 1 [Homo sapiens]</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>83178</td>\n",
       "      <td>9</td>\n",
       "      <td>3</td>\n",
       "      <td>mtegtclrrrggpyktepatdlgrwrlncergrqtwtylqderagr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>198</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001001438.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>lanosterol synthase isoform 1 [Homo sapiens]</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>83178</td>\n",
       "      <td>9</td>\n",
       "      <td>3</td>\n",
       "      <td>mtegtclrrrggpyktepatdlgrwrlncergrqtwtylqderagr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>199</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001138909.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>lanosterol synthase isoform 3 [Homo sapiens]</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>74087</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>mtfyvglqaedghwtgdyggplfllpgllitchvariplpagyree...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>200 rows  11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Chromosome Accession Version      Organism  \\\n",
       "0           21       NP_002617.3  Homo sapiens   \n",
       "1           21    XP_005261192.1  Homo sapiens   \n",
       "2           21    NP_001002021.2  Homo sapiens   \n",
       "3           21    XP_016883857.1  Homo sapiens   \n",
       "4           21    XP_006724074.1  Homo sapiens   \n",
       "..         ...               ...           ...   \n",
       "195         21    XP_005261245.1  Homo sapiens   \n",
       "196         21    NP_001138908.1  Homo sapiens   \n",
       "197         21       NP_002331.3  Homo sapiens   \n",
       "198         21    NP_001001438.1  Homo sapiens   \n",
       "199         21    NP_001138909.1  Homo sapiens   \n",
       "\n",
       "                                            Definition Division Topology  \\\n",
       "0    ATP-dependent 6-phosphofructokinase, liver typ...      PRI   linear   \n",
       "1    ATP-dependent 6-phosphofructokinase, liver typ...      PRI   linear   \n",
       "2    ATP-dependent 6-phosphofructokinase, liver typ...      PRI   linear   \n",
       "3    ATP-dependent 6-phosphofructokinase, liver typ...      PRI   linear   \n",
       "4    ATP-dependent 6-phosphofructokinase, liver typ...      PRI   linear   \n",
       "..                                                 ...      ...      ...   \n",
       "195   speriolin-like protein isoform X1 [Homo sapiens]      PRI   linear   \n",
       "196       lanosterol synthase isoform 2 [Homo sapiens]      PRI   linear   \n",
       "197       lanosterol synthase isoform 1 [Homo sapiens]      PRI   linear   \n",
       "198       lanosterol synthase isoform 1 [Homo sapiens]      PRI   linear   \n",
       "199       lanosterol synthase isoform 3 [Homo sapiens]      PRI   linear   \n",
       "\n",
       "    Molecule Type Molecular Weight  Number of Regions  \\\n",
       "0              AA            84888                 10   \n",
       "1              AA            87546                  2   \n",
       "2              AA            90447                  1   \n",
       "3              AA            93105                  2   \n",
       "4              AA            93105                  2   \n",
       "..            ...              ...                ...   \n",
       "195            AA            37482                  2   \n",
       "196            AA            82068                  8   \n",
       "197            AA            83178                  9   \n",
       "198            AA            83178                  9   \n",
       "199            AA            74087                  2   \n",
       "\n",
       "     Number of Binding Sites  \\\n",
       "0                          5   \n",
       "1                          0   \n",
       "2                          0   \n",
       "3                          0   \n",
       "4                          0   \n",
       "..                       ...   \n",
       "195                        0   \n",
       "196                        3   \n",
       "197                        3   \n",
       "198                        3   \n",
       "199                        2   \n",
       "\n",
       "                                              Sequence  \n",
       "0    maavdleklrasgagkaigvltsggdaqgmnaavravtrmgiyvga...  \n",
       "1    maavdleklrasgagkaigvltsggdaqgmnaavravtrmgiyvga...  \n",
       "2    mrsmcnqgrgressrgglhvqgscrglsrspqqetgfakapagtdc...  \n",
       "3    mrsmcnqgrgressrgglhvqgscrglsrspqqetgfakapagtdc...  \n",
       "4    mrsmcnqgrgressrgglhvqgscrglsrspqqetgfakapagtdc...  \n",
       "..                                                 ...  \n",
       "195  maeggelmsrllsenadlkkqvrllkenqmlrrllsqscqeggghd...  \n",
       "196  mtegtclrrrggpyktepatdlgrwrlncergrqtwtylqderagr...  \n",
       "197  mtegtclrrrggpyktepatdlgrwrlncergrqtwtylqderagr...  \n",
       "198  mtegtclrrrggpyktepatdlgrwrlncergrqtwtylqderagr...  \n",
       "199  mtfyvglqaedghwtgdyggplfllpgllitchvariplpagyree...  \n",
       "\n",
       "[200 rows x 11 columns]"
      ]
     },
     "execution_count": 143,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "protein_additional_data_6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {},
   "outputs": [],
   "source": [
    "time.sleep(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Entrez\n",
    "Entrez.api_key = api_key\n",
    "Entrez.email = 'elizabethfawcett47@gmail.com'\n",
    "Entrez.tool = 'locus_by_amino_acid'\n",
    "id_list = protein_accession_numbers[1200:1287]\n",
    "search_results = Entrez.read(Entrez.epost(\"protein\", id=\",\".join(id_list)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {},
   "outputs": [],
   "source": [
    "webenv = search_results[\"WebEnv\"]\n",
    "query_key = search_results[\"QueryKey\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [],
   "source": [
    "results = Entrez.read(Entrez.efetch(db=\"protein\", webenv=webenv, query_key=query_key, retmode='xml'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {},
   "outputs": [],
   "source": [
    "protein_additional_data_7 = pd.DataFrame()\n",
    "accession_version = []\n",
    "definition = []\n",
    "division = []\n",
    "topology = []\n",
    "moltype = []\n",
    "organism = []\n",
    "sequence = []\n",
    "chromosome = []\n",
    "molar_weight = []\n",
    "regions = []\n",
    "sites = []\n",
    "for i in results:\n",
    "    region = 0\n",
    "    site = 0\n",
    "    accession_version.append(i['GBSeq_accession-version'])\n",
    "    organism.append(i['GBSeq_organism'])\n",
    "    definition.append(i['GBSeq_definition'])\n",
    "    division.append(i['GBSeq_division'])\n",
    "    topology.append(i['GBSeq_topology'])\n",
    "    moltype.append(i['GBSeq_moltype'])\n",
    "    sequence.append(i['GBSeq_sequence'])\n",
    "    if i['GBSeq_feature-table'][0]['GBFeature_key'] == 'source':\n",
    "        chromosome.append(i['GBSeq_feature-table'][0]['GBFeature_quals'][2]['GBQualifier_value'])\n",
    "    else:\n",
    "        continue\n",
    "    if i['GBSeq_feature-table'][1]['GBFeature_key'] == 'Protein':\n",
    "        if i['GBSeq_feature-table'][1]['GBFeature_quals'][0]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][0]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][1]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][1]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][2]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][2]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][3]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][3]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][4]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][4]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][5]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][5]['GBQualifier_value'])\n",
    "        elif i['GBSeq_feature-table'][1]['GBFeature_quals'][6]['GBQualifier_name'] == 'calculated_mol_wt':\n",
    "            molar_weight.append(i['GBSeq_feature-table'][1]['GBFeature_quals'][6]['GBQualifier_value'])\n",
    "        else:\n",
    "            print('add more elif statements')\n",
    "    else:\n",
    "        continue\n",
    "    for j in range(0, len(i['GBSeq_feature-table'])):\n",
    "        if i['GBSeq_feature-table'][j]['GBFeature_key'] == 'Region':\n",
    "            region += 1\n",
    "        else:\n",
    "            continue\n",
    "    for j in range(0, len(i['GBSeq_feature-table'])):\n",
    "        if i['GBSeq_feature-table'][j]['GBFeature_key'] == 'Site':\n",
    "            site += 1\n",
    "        else:\n",
    "            continue\n",
    "    regions.append(region)\n",
    "    sites.append(site)\n",
    "protein_additional_data_7['Chromosome'] = chromosome\n",
    "protein_additional_data_7['Accession Version'] = accession_version\n",
    "protein_additional_data_7['Organism'] = organism\n",
    "protein_additional_data_7['Definition'] = definition\n",
    "protein_additional_data_7['Division'] = division\n",
    "protein_additional_data_7['Topology'] = topology\n",
    "protein_additional_data_7['Molecule Type'] = moltype\n",
    "protein_additional_data_7['Molecular Weight'] = molar_weight\n",
    "protein_additional_data_7['Number of Regions'] = regions\n",
    "protein_additional_data_7['Number of Binding Sites'] = sites\n",
    "protein_additional_data_7['Sequence'] = sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Accession Version</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Definition</th>\n",
       "      <th>Division</th>\n",
       "      <th>Topology</th>\n",
       "      <th>Molecule Type</th>\n",
       "      <th>Molecular Weight</th>\n",
       "      <th>Number of Regions</th>\n",
       "      <th>Number of Binding Sites</th>\n",
       "      <th>Sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_005261261.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>germinal-center associated nuclear protein iso...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>218275</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>mnptnpfsgqqpsafsasssnvgtlpskppfrfgqpslfgqnstls...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_005261260.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>germinal-center associated nuclear protein iso...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>218275</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>mnptnpfsgqqpsafsasssnvgtlpskppfrfgqpslfgqnstls...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_003897.2</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>germinal-center associated nuclear protein [Ho...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>218275</td>\n",
       "      <td>12</td>\n",
       "      <td>7</td>\n",
       "      <td>mnptnpfsgqqpsafsasssnvgtlpskppfrfgqpslfgqnstls...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_005261262.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>germinal-center associated nuclear protein iso...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>218275</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>mnptnpfsgqqpsafsasssnvgtlpskppfrfgqpslfgqnstls...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_011527936.2</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>endoribonuclease YbeY isoform X4 [Homo sapiens]</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>33333</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>mwrpgraelrggrkrrpwrrpsphahrlapgprgprcplaprkdlr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>81</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001273607.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>protein arginine N-methyltransferase 2 isoform...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>25363</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>82</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001526.2</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>protein arginine N-methyltransferase 2 isoform...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>48911</td>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "      <td>matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>83</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_006724061.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>protein arginine N-methyltransferase 2 isoform...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>33412</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>84</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001229795.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>protein arginine N-methyltransferase 2 isoform...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>31062</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>85</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001229794.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>protein arginine N-methyltransferase 2 isoform...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>31031</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>86 rows  11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chromosome Accession Version      Organism  \\\n",
       "0          21    XP_005261261.1  Homo sapiens   \n",
       "1          21    XP_005261260.1  Homo sapiens   \n",
       "2          21       NP_003897.2  Homo sapiens   \n",
       "3          21    XP_005261262.1  Homo sapiens   \n",
       "4          21    XP_011527936.2  Homo sapiens   \n",
       "..        ...               ...           ...   \n",
       "81         21    NP_001273607.1  Homo sapiens   \n",
       "82         21       NP_001526.2  Homo sapiens   \n",
       "83         21    XP_006724061.1  Homo sapiens   \n",
       "84         21    NP_001229795.1  Homo sapiens   \n",
       "85         21    NP_001229794.1  Homo sapiens   \n",
       "\n",
       "                                           Definition Division Topology  \\\n",
       "0   germinal-center associated nuclear protein iso...      PRI   linear   \n",
       "1   germinal-center associated nuclear protein iso...      PRI   linear   \n",
       "2   germinal-center associated nuclear protein [Ho...      PRI   linear   \n",
       "3   germinal-center associated nuclear protein iso...      PRI   linear   \n",
       "4     endoribonuclease YbeY isoform X4 [Homo sapiens]      PRI   linear   \n",
       "..                                                ...      ...      ...   \n",
       "81  protein arginine N-methyltransferase 2 isoform...      PRI   linear   \n",
       "82  protein arginine N-methyltransferase 2 isoform...      PRI   linear   \n",
       "83  protein arginine N-methyltransferase 2 isoform...      PRI   linear   \n",
       "84  protein arginine N-methyltransferase 2 isoform...      PRI   linear   \n",
       "85  protein arginine N-methyltransferase 2 isoform...      PRI   linear   \n",
       "\n",
       "   Molecule Type Molecular Weight  Number of Regions  Number of Binding Sites  \\\n",
       "0             AA           218275                  5                        0   \n",
       "1             AA           218275                  5                        0   \n",
       "2             AA           218275                 12                        7   \n",
       "3             AA           218275                  5                        0   \n",
       "4             AA            33333                  0                        0   \n",
       "..           ...              ...                ...                      ...   \n",
       "81            AA            25363                  3                        4   \n",
       "82            AA            48911                  6                        4   \n",
       "83            AA            33412                  2                        2   \n",
       "84            AA            31062                  5                        4   \n",
       "85            AA            31031                  3                        4   \n",
       "\n",
       "                                             Sequence  \n",
       "0   mnptnpfsgqqpsafsasssnvgtlpskppfrfgqpslfgqnstls...  \n",
       "1   mnptnpfsgqqpsafsasssnvgtlpskppfrfgqpslfgqnstls...  \n",
       "2   mnptnpfsgqqpsafsasssnvgtlpskppfrfgqpslfgqnstls...  \n",
       "3   mnptnpfsgqqpsafsasssnvgtlpskppfrfgqpslfgqnstls...  \n",
       "4   mwrpgraelrggrkrrpwrrpsphahrlapgprgprcplaprkdlr...  \n",
       "..                                                ...  \n",
       "81  matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...  \n",
       "82  matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...  \n",
       "83  matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...  \n",
       "84  matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...  \n",
       "85  matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...  \n",
       "\n",
       "[86 rows x 11 columns]"
      ]
     },
     "execution_count": 149,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "protein_additional_data_7"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {},
   "outputs": [],
   "source": [
    "time.sleep(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {},
   "outputs": [],
   "source": [
    "to_concat = [protein_additional_data_1, protein_additional_data_2, protein_additional_data_3, protein_additional_data_4, \n",
    "           protein_additional_data_5, protein_additional_data_6, protein_additional_data_7]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {},
   "outputs": [],
   "source": [
    "protein_additional_data = pd.concat(to_concat)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Accession Version</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Definition</th>\n",
       "      <th>Division</th>\n",
       "      <th>Topology</th>\n",
       "      <th>Molecule Type</th>\n",
       "      <th>Molecular Weight</th>\n",
       "      <th>Number of Regions</th>\n",
       "      <th>Number of Binding Sites</th>\n",
       "      <th>Sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_006723962.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ICOS ligand isoform X1 [Homo sapiens]</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>56181</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_011544381.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ICOS ligand isoform X3 [Homo sapiens]</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>33350</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_011544380.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ICOS ligand isoform X2 [Homo sapiens]</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>39507</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001350699.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ICOS ligand precursor [Homo sapiens]</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>31389</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001350689.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>glutamine amidotransferase-like class 1 domain...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>24598</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>maavralvasrlaaasaftslspggrtpsqraalhlsvprpaarva...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>81</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001273607.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>protein arginine N-methyltransferase 2 isoform...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>25363</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>82</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001526.2</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>protein arginine N-methyltransferase 2 isoform...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>48911</td>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "      <td>matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>83</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_006724061.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>protein arginine N-methyltransferase 2 isoform...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>33412</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>84</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001229795.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>protein arginine N-methyltransferase 2 isoform...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>31062</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>85</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001229794.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>protein arginine N-methyltransferase 2 isoform...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>31031</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1286 rows  11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chromosome Accession Version      Organism  \\\n",
       "0          21    XP_006723962.1  Homo sapiens   \n",
       "1          21    XP_011544381.1  Homo sapiens   \n",
       "2          21    XP_011544380.1  Homo sapiens   \n",
       "3          21    NP_001350699.1  Homo sapiens   \n",
       "4          21    NP_001350689.1  Homo sapiens   \n",
       "..        ...               ...           ...   \n",
       "81         21    NP_001273607.1  Homo sapiens   \n",
       "82         21       NP_001526.2  Homo sapiens   \n",
       "83         21    XP_006724061.1  Homo sapiens   \n",
       "84         21    NP_001229795.1  Homo sapiens   \n",
       "85         21    NP_001229794.1  Homo sapiens   \n",
       "\n",
       "                                           Definition Division Topology  \\\n",
       "0               ICOS ligand isoform X1 [Homo sapiens]      PRI   linear   \n",
       "1               ICOS ligand isoform X3 [Homo sapiens]      PRI   linear   \n",
       "2               ICOS ligand isoform X2 [Homo sapiens]      PRI   linear   \n",
       "3                ICOS ligand precursor [Homo sapiens]      PRI   linear   \n",
       "4   glutamine amidotransferase-like class 1 domain...      PRI   linear   \n",
       "..                                                ...      ...      ...   \n",
       "81  protein arginine N-methyltransferase 2 isoform...      PRI   linear   \n",
       "82  protein arginine N-methyltransferase 2 isoform...      PRI   linear   \n",
       "83  protein arginine N-methyltransferase 2 isoform...      PRI   linear   \n",
       "84  protein arginine N-methyltransferase 2 isoform...      PRI   linear   \n",
       "85  protein arginine N-methyltransferase 2 isoform...      PRI   linear   \n",
       "\n",
       "   Molecule Type Molecular Weight  Number of Regions  Number of Binding Sites  \\\n",
       "0             AA            56181                  2                        0   \n",
       "1             AA            33350                  2                        0   \n",
       "2             AA            39507                  2                        0   \n",
       "3             AA            31389                  2                        0   \n",
       "4             AA            24598                  1                        1   \n",
       "..           ...              ...                ...                      ...   \n",
       "81            AA            25363                  3                        4   \n",
       "82            AA            48911                  6                        4   \n",
       "83            AA            33412                  2                        2   \n",
       "84            AA            31062                  5                        4   \n",
       "85            AA            31031                  3                        4   \n",
       "\n",
       "                                             Sequence  \n",
       "0   mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...  \n",
       "1   mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...  \n",
       "2   mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...  \n",
       "3   mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...  \n",
       "4   maavralvasrlaaasaftslspggrtpsqraalhlsvprpaarva...  \n",
       "..                                                ...  \n",
       "81  matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...  \n",
       "82  matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...  \n",
       "83  matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...  \n",
       "84  matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...  \n",
       "85  matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...  \n",
       "\n",
       "[1286 rows x 11 columns]"
      ]
     },
     "execution_count": 153,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "protein_additional_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th># Sequence-Name</th>\n",
       "      <th>Sequence-Role</th>\n",
       "      <th>Assigned-Molecule</th>\n",
       "      <th>Assigned-Molecule-Location/Type</th>\n",
       "      <th>GenBank-Accn</th>\n",
       "      <th>Relationship</th>\n",
       "      <th>RefSeq-Accn</th>\n",
       "      <th>Assembly-Unit</th>\n",
       "      <th>Sequence-Length</th>\n",
       "      <th>UCSC-style-name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>41</td>\n",
       "      <td>21</td>\n",
       "      <td>assembled-molecule</td>\n",
       "      <td>21</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>CM000683.2</td>\n",
       "      <td>=</td>\n",
       "      <td>NC_000021.9</td>\n",
       "      <td>Primary Assembly</td>\n",
       "      <td>46709983.0</td>\n",
       "      <td>chr21</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   # Sequence-Name       Sequence-Role Assigned-Molecule  \\\n",
       "41              21  assembled-molecule                21   \n",
       "\n",
       "   Assigned-Molecule-Location/Type GenBank-Accn Relationship  RefSeq-Accn  \\\n",
       "41                      Chromosome   CM000683.2            =  NC_000021.9   \n",
       "\n",
       "       Assembly-Unit  Sequence-Length UCSC-style-name  \n",
       "41  Primary Assembly       46709983.0           chr21  "
      ]
     },
     "execution_count": 154,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dna_data_21 = dna_data[dna_data['RefSeq-Accn']=='NC_000021.9']\n",
    "dna_data_21"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Replicon Name</th>\n",
       "      <th>Replicon Accession</th>\n",
       "      <th>Start</th>\n",
       "      <th>Stop</th>\n",
       "      <th>Strand</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>Locus</th>\n",
       "      <th>Protein product</th>\n",
       "      <th>Length</th>\n",
       "      <th>Protein name</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Protein product</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>XP_006723962.1</td>\n",
       "      <td>21</td>\n",
       "      <td>NC_000021.9</td>\n",
       "      <td>5022680</td>\n",
       "      <td>5034096</td>\n",
       "      <td>+</td>\n",
       "      <td>102723996</td>\n",
       "      <td>LOC102723996</td>\n",
       "      <td>XP_006723962.1</td>\n",
       "      <td>516</td>\n",
       "      <td>ICOS ligand isoform X1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XP_011544381.1</td>\n",
       "      <td>21</td>\n",
       "      <td>NC_000021.9</td>\n",
       "      <td>5022680</td>\n",
       "      <td>5034096</td>\n",
       "      <td>+</td>\n",
       "      <td>102723996</td>\n",
       "      <td>LOC102723996</td>\n",
       "      <td>XP_011544381.1</td>\n",
       "      <td>304</td>\n",
       "      <td>ICOS ligand isoform X3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XP_011544380.1</td>\n",
       "      <td>21</td>\n",
       "      <td>NC_000021.9</td>\n",
       "      <td>5022680</td>\n",
       "      <td>5045606</td>\n",
       "      <td>+</td>\n",
       "      <td>102723996</td>\n",
       "      <td>LOC102723996</td>\n",
       "      <td>XP_011544380.1</td>\n",
       "      <td>361</td>\n",
       "      <td>ICOS ligand isoform X2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NP_001350699.1</td>\n",
       "      <td>21</td>\n",
       "      <td>NC_000021.9</td>\n",
       "      <td>5022680</td>\n",
       "      <td>5034592</td>\n",
       "      <td>+</td>\n",
       "      <td>102723996</td>\n",
       "      <td>LOC102723996</td>\n",
       "      <td>NP_001350699.1</td>\n",
       "      <td>302</td>\n",
       "      <td>ICOS ligand precursor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NP_001350689.1</td>\n",
       "      <td>21</td>\n",
       "      <td>NC_000021.9</td>\n",
       "      <td>5117103</td>\n",
       "      <td>5128354</td>\n",
       "      <td>-</td>\n",
       "      <td>102724023</td>\n",
       "      <td>GATD3B</td>\n",
       "      <td>NP_001350689.1</td>\n",
       "      <td>237</td>\n",
       "      <td>glutamine amidotransferase-like class 1 domain...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NP_001273607.1</td>\n",
       "      <td>21</td>\n",
       "      <td>NC_000021.9</td>\n",
       "      <td>46636952</td>\n",
       "      <td>46664327</td>\n",
       "      <td>+</td>\n",
       "      <td>3275</td>\n",
       "      <td>PRMT2</td>\n",
       "      <td>NP_001273607.1</td>\n",
       "      <td>228</td>\n",
       "      <td>protein arginine N-methyltransferase 2 isoform 7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NP_001526.2</td>\n",
       "      <td>21</td>\n",
       "      <td>NC_000021.9</td>\n",
       "      <td>46636952</td>\n",
       "      <td>46664327</td>\n",
       "      <td>+</td>\n",
       "      <td>3275</td>\n",
       "      <td>PRMT2</td>\n",
       "      <td>NP_001526.2</td>\n",
       "      <td>433</td>\n",
       "      <td>protein arginine N-methyltransferase 2 isoform 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XP_006724061.1</td>\n",
       "      <td>21</td>\n",
       "      <td>NC_000021.9</td>\n",
       "      <td>46636952</td>\n",
       "      <td>46664374</td>\n",
       "      <td>+</td>\n",
       "      <td>3275</td>\n",
       "      <td>PRMT2</td>\n",
       "      <td>XP_006724061.1</td>\n",
       "      <td>301</td>\n",
       "      <td>protein arginine N-methyltransferase 2 isoform X2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NP_001229795.1</td>\n",
       "      <td>21</td>\n",
       "      <td>NC_000021.9</td>\n",
       "      <td>46636952</td>\n",
       "      <td>46658924</td>\n",
       "      <td>+</td>\n",
       "      <td>3275</td>\n",
       "      <td>PRMT2</td>\n",
       "      <td>NP_001229795.1</td>\n",
       "      <td>277</td>\n",
       "      <td>protein arginine N-methyltransferase 2 isoform 4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NP_001229794.1</td>\n",
       "      <td>21</td>\n",
       "      <td>NC_000021.9</td>\n",
       "      <td>46636952</td>\n",
       "      <td>46652006</td>\n",
       "      <td>+</td>\n",
       "      <td>3275</td>\n",
       "      <td>PRMT2</td>\n",
       "      <td>NP_001229794.1</td>\n",
       "      <td>284</td>\n",
       "      <td>protein arginine N-methyltransferase 2 isoform 3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1286 rows  10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                #Replicon Name Replicon Accession     Start      Stop Strand  \\\n",
       "Protein product                                                                \n",
       "XP_006723962.1              21        NC_000021.9   5022680   5034096      +   \n",
       "XP_011544381.1              21        NC_000021.9   5022680   5034096      +   \n",
       "XP_011544380.1              21        NC_000021.9   5022680   5045606      +   \n",
       "NP_001350699.1              21        NC_000021.9   5022680   5034592      +   \n",
       "NP_001350689.1              21        NC_000021.9   5117103   5128354      -   \n",
       "...                        ...                ...       ...       ...    ...   \n",
       "NP_001273607.1              21        NC_000021.9  46636952  46664327      +   \n",
       "NP_001526.2                 21        NC_000021.9  46636952  46664327      +   \n",
       "XP_006724061.1              21        NC_000021.9  46636952  46664374      +   \n",
       "NP_001229795.1              21        NC_000021.9  46636952  46658924      +   \n",
       "NP_001229794.1              21        NC_000021.9  46636952  46652006      +   \n",
       "\n",
       "                    GeneID         Locus Protein product  Length  \\\n",
       "Protein product                                                    \n",
       "XP_006723962.1   102723996  LOC102723996  XP_006723962.1     516   \n",
       "XP_011544381.1   102723996  LOC102723996  XP_011544381.1     304   \n",
       "XP_011544380.1   102723996  LOC102723996  XP_011544380.1     361   \n",
       "NP_001350699.1   102723996  LOC102723996  NP_001350699.1     302   \n",
       "NP_001350689.1   102724023        GATD3B  NP_001350689.1     237   \n",
       "...                    ...           ...             ...     ...   \n",
       "NP_001273607.1        3275         PRMT2  NP_001273607.1     228   \n",
       "NP_001526.2           3275         PRMT2     NP_001526.2     433   \n",
       "XP_006724061.1        3275         PRMT2  XP_006724061.1     301   \n",
       "NP_001229795.1        3275         PRMT2  NP_001229795.1     277   \n",
       "NP_001229794.1        3275         PRMT2  NP_001229794.1     284   \n",
       "\n",
       "                                                      Protein name  \n",
       "Protein product                                                     \n",
       "XP_006723962.1                              ICOS ligand isoform X1  \n",
       "XP_011544381.1                              ICOS ligand isoform X3  \n",
       "XP_011544380.1                              ICOS ligand isoform X2  \n",
       "NP_001350699.1                               ICOS ligand precursor  \n",
       "NP_001350689.1   glutamine amidotransferase-like class 1 domain...  \n",
       "...                                                            ...  \n",
       "NP_001273607.1    protein arginine N-methyltransferase 2 isoform 7  \n",
       "NP_001526.2       protein arginine N-methyltransferase 2 isoform 1  \n",
       "XP_006724061.1   protein arginine N-methyltransferase 2 isoform X2  \n",
       "NP_001229795.1    protein arginine N-methyltransferase 2 isoform 4  \n",
       "NP_001229794.1    protein arginine N-methyltransferase 2 isoform 3  \n",
       "\n",
       "[1286 rows x 10 columns]"
      ]
     },
     "execution_count": 159,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "protein_data_21 = protein_data[protein_data['Replicon Accession']=='NC_000021.9']\n",
    "protein_data_21.index=protein_data_21['Protein product']\n",
    "protein_data_21"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Accession Version</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Definition</th>\n",
       "      <th>Division</th>\n",
       "      <th>Topology</th>\n",
       "      <th>Molecule Type</th>\n",
       "      <th>Molecular Weight</th>\n",
       "      <th>Number of Regions</th>\n",
       "      <th>Number of Binding Sites</th>\n",
       "      <th>Sequence</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Accession Version</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>XP_006723962.1</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_006723962.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ICOS ligand isoform X1 [Homo sapiens]</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>56181</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XP_011544381.1</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_011544381.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ICOS ligand isoform X3 [Homo sapiens]</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>33350</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XP_011544380.1</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_011544380.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ICOS ligand isoform X2 [Homo sapiens]</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>39507</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NP_001350699.1</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001350699.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ICOS ligand precursor [Homo sapiens]</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>31389</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NP_001350689.1</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001350689.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>glutamine amidotransferase-like class 1 domain...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>24598</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>maavralvasrlaaasaftslspggrtpsqraalhlsvprpaarva...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NP_001273607.1</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001273607.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>protein arginine N-methyltransferase 2 isoform...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>25363</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NP_001526.2</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001526.2</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>protein arginine N-methyltransferase 2 isoform...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>48911</td>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "      <td>matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XP_006724061.1</td>\n",
       "      <td>21</td>\n",
       "      <td>XP_006724061.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>protein arginine N-methyltransferase 2 isoform...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>33412</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NP_001229795.1</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001229795.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>protein arginine N-methyltransferase 2 isoform...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>31062</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NP_001229794.1</td>\n",
       "      <td>21</td>\n",
       "      <td>NP_001229794.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>protein arginine N-methyltransferase 2 isoform...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>31031</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1286 rows  11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Chromosome Accession Version      Organism  \\\n",
       "Accession Version                                              \n",
       "XP_006723962.1            21    XP_006723962.1  Homo sapiens   \n",
       "XP_011544381.1            21    XP_011544381.1  Homo sapiens   \n",
       "XP_011544380.1            21    XP_011544380.1  Homo sapiens   \n",
       "NP_001350699.1            21    NP_001350699.1  Homo sapiens   \n",
       "NP_001350689.1            21    NP_001350689.1  Homo sapiens   \n",
       "...                      ...               ...           ...   \n",
       "NP_001273607.1            21    NP_001273607.1  Homo sapiens   \n",
       "NP_001526.2               21       NP_001526.2  Homo sapiens   \n",
       "XP_006724061.1            21    XP_006724061.1  Homo sapiens   \n",
       "NP_001229795.1            21    NP_001229795.1  Homo sapiens   \n",
       "NP_001229794.1            21    NP_001229794.1  Homo sapiens   \n",
       "\n",
       "                                                          Definition Division  \\\n",
       "Accession Version                                                               \n",
       "XP_006723962.1                 ICOS ligand isoform X1 [Homo sapiens]      PRI   \n",
       "XP_011544381.1                 ICOS ligand isoform X3 [Homo sapiens]      PRI   \n",
       "XP_011544380.1                 ICOS ligand isoform X2 [Homo sapiens]      PRI   \n",
       "NP_001350699.1                  ICOS ligand precursor [Homo sapiens]      PRI   \n",
       "NP_001350689.1     glutamine amidotransferase-like class 1 domain...      PRI   \n",
       "...                                                              ...      ...   \n",
       "NP_001273607.1     protein arginine N-methyltransferase 2 isoform...      PRI   \n",
       "NP_001526.2        protein arginine N-methyltransferase 2 isoform...      PRI   \n",
       "XP_006724061.1     protein arginine N-methyltransferase 2 isoform...      PRI   \n",
       "NP_001229795.1     protein arginine N-methyltransferase 2 isoform...      PRI   \n",
       "NP_001229794.1     protein arginine N-methyltransferase 2 isoform...      PRI   \n",
       "\n",
       "                  Topology Molecule Type Molecular Weight  Number of Regions  \\\n",
       "Accession Version                                                              \n",
       "XP_006723962.1      linear            AA            56181                  2   \n",
       "XP_011544381.1      linear            AA            33350                  2   \n",
       "XP_011544380.1      linear            AA            39507                  2   \n",
       "NP_001350699.1      linear            AA            31389                  2   \n",
       "NP_001350689.1      linear            AA            24598                  1   \n",
       "...                    ...           ...              ...                ...   \n",
       "NP_001273607.1      linear            AA            25363                  3   \n",
       "NP_001526.2         linear            AA            48911                  6   \n",
       "XP_006724061.1      linear            AA            33412                  2   \n",
       "NP_001229795.1      linear            AA            31062                  5   \n",
       "NP_001229794.1      linear            AA            31031                  3   \n",
       "\n",
       "                   Number of Binding Sites  \\\n",
       "Accession Version                            \n",
       "XP_006723962.1                           0   \n",
       "XP_011544381.1                           0   \n",
       "XP_011544380.1                           0   \n",
       "NP_001350699.1                           0   \n",
       "NP_001350689.1                           1   \n",
       "...                                    ...   \n",
       "NP_001273607.1                           4   \n",
       "NP_001526.2                              4   \n",
       "XP_006724061.1                           2   \n",
       "NP_001229795.1                           4   \n",
       "NP_001229794.1                           4   \n",
       "\n",
       "                                                            Sequence  \n",
       "Accession Version                                                     \n",
       "XP_006723962.1     mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...  \n",
       "XP_011544381.1     mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...  \n",
       "XP_011544380.1     mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...  \n",
       "NP_001350699.1     mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...  \n",
       "NP_001350689.1     maavralvasrlaaasaftslspggrtpsqraalhlsvprpaarva...  \n",
       "...                                                              ...  \n",
       "NP_001273607.1     matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...  \n",
       "NP_001526.2        matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...  \n",
       "XP_006724061.1     matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...  \n",
       "NP_001229795.1     matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...  \n",
       "NP_001229794.1     matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...  \n",
       "\n",
       "[1286 rows x 11 columns]"
      ]
     },
     "execution_count": 160,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "protein_additional_data_21 = protein_additional_data\n",
    "protein_additional_data_21.index = protein_additional_data_21['Accession Version']\n",
    "protein_additional_data_21"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Replicon Name</th>\n",
       "      <th>Replicon Accession</th>\n",
       "      <th>Start</th>\n",
       "      <th>Stop</th>\n",
       "      <th>Strand</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>Locus</th>\n",
       "      <th>Protein product</th>\n",
       "      <th>Length</th>\n",
       "      <th>Protein name</th>\n",
       "      <th>...</th>\n",
       "      <th>Accession Version</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Definition</th>\n",
       "      <th>Division</th>\n",
       "      <th>Topology</th>\n",
       "      <th>Molecule Type</th>\n",
       "      <th>Molecular Weight</th>\n",
       "      <th>Number of Regions</th>\n",
       "      <th>Number of Binding Sites</th>\n",
       "      <th>Sequence</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Protein product</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>XP_006723962.1</td>\n",
       "      <td>21</td>\n",
       "      <td>NC_000021.9</td>\n",
       "      <td>5022680</td>\n",
       "      <td>5034096</td>\n",
       "      <td>+</td>\n",
       "      <td>102723996</td>\n",
       "      <td>LOC102723996</td>\n",
       "      <td>XP_006723962.1</td>\n",
       "      <td>516</td>\n",
       "      <td>ICOS ligand isoform X1</td>\n",
       "      <td>...</td>\n",
       "      <td>XP_006723962.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ICOS ligand isoform X1 [Homo sapiens]</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>56181</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XP_011544381.1</td>\n",
       "      <td>21</td>\n",
       "      <td>NC_000021.9</td>\n",
       "      <td>5022680</td>\n",
       "      <td>5034096</td>\n",
       "      <td>+</td>\n",
       "      <td>102723996</td>\n",
       "      <td>LOC102723996</td>\n",
       "      <td>XP_011544381.1</td>\n",
       "      <td>304</td>\n",
       "      <td>ICOS ligand isoform X3</td>\n",
       "      <td>...</td>\n",
       "      <td>XP_011544381.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ICOS ligand isoform X3 [Homo sapiens]</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>33350</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XP_011544380.1</td>\n",
       "      <td>21</td>\n",
       "      <td>NC_000021.9</td>\n",
       "      <td>5022680</td>\n",
       "      <td>5045606</td>\n",
       "      <td>+</td>\n",
       "      <td>102723996</td>\n",
       "      <td>LOC102723996</td>\n",
       "      <td>XP_011544380.1</td>\n",
       "      <td>361</td>\n",
       "      <td>ICOS ligand isoform X2</td>\n",
       "      <td>...</td>\n",
       "      <td>XP_011544380.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ICOS ligand isoform X2 [Homo sapiens]</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>39507</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NP_001350699.1</td>\n",
       "      <td>21</td>\n",
       "      <td>NC_000021.9</td>\n",
       "      <td>5022680</td>\n",
       "      <td>5034592</td>\n",
       "      <td>+</td>\n",
       "      <td>102723996</td>\n",
       "      <td>LOC102723996</td>\n",
       "      <td>NP_001350699.1</td>\n",
       "      <td>302</td>\n",
       "      <td>ICOS ligand precursor</td>\n",
       "      <td>...</td>\n",
       "      <td>NP_001350699.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ICOS ligand precursor [Homo sapiens]</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>31389</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NP_001350689.1</td>\n",
       "      <td>21</td>\n",
       "      <td>NC_000021.9</td>\n",
       "      <td>5117103</td>\n",
       "      <td>5128354</td>\n",
       "      <td>-</td>\n",
       "      <td>102724023</td>\n",
       "      <td>GATD3B</td>\n",
       "      <td>NP_001350689.1</td>\n",
       "      <td>237</td>\n",
       "      <td>glutamine amidotransferase-like class 1 domain...</td>\n",
       "      <td>...</td>\n",
       "      <td>NP_001350689.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>glutamine amidotransferase-like class 1 domain...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>24598</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>maavralvasrlaaasaftslspggrtpsqraalhlsvprpaarva...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NP_001273607.1</td>\n",
       "      <td>21</td>\n",
       "      <td>NC_000021.9</td>\n",
       "      <td>46636952</td>\n",
       "      <td>46664327</td>\n",
       "      <td>+</td>\n",
       "      <td>3275</td>\n",
       "      <td>PRMT2</td>\n",
       "      <td>NP_001273607.1</td>\n",
       "      <td>228</td>\n",
       "      <td>protein arginine N-methyltransferase 2 isoform 7</td>\n",
       "      <td>...</td>\n",
       "      <td>NP_001273607.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>protein arginine N-methyltransferase 2 isoform...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>25363</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NP_001526.2</td>\n",
       "      <td>21</td>\n",
       "      <td>NC_000021.9</td>\n",
       "      <td>46636952</td>\n",
       "      <td>46664327</td>\n",
       "      <td>+</td>\n",
       "      <td>3275</td>\n",
       "      <td>PRMT2</td>\n",
       "      <td>NP_001526.2</td>\n",
       "      <td>433</td>\n",
       "      <td>protein arginine N-methyltransferase 2 isoform 1</td>\n",
       "      <td>...</td>\n",
       "      <td>NP_001526.2</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>protein arginine N-methyltransferase 2 isoform...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>48911</td>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "      <td>matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XP_006724061.1</td>\n",
       "      <td>21</td>\n",
       "      <td>NC_000021.9</td>\n",
       "      <td>46636952</td>\n",
       "      <td>46664374</td>\n",
       "      <td>+</td>\n",
       "      <td>3275</td>\n",
       "      <td>PRMT2</td>\n",
       "      <td>XP_006724061.1</td>\n",
       "      <td>301</td>\n",
       "      <td>protein arginine N-methyltransferase 2 isoform X2</td>\n",
       "      <td>...</td>\n",
       "      <td>XP_006724061.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>protein arginine N-methyltransferase 2 isoform...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>33412</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NP_001229795.1</td>\n",
       "      <td>21</td>\n",
       "      <td>NC_000021.9</td>\n",
       "      <td>46636952</td>\n",
       "      <td>46658924</td>\n",
       "      <td>+</td>\n",
       "      <td>3275</td>\n",
       "      <td>PRMT2</td>\n",
       "      <td>NP_001229795.1</td>\n",
       "      <td>277</td>\n",
       "      <td>protein arginine N-methyltransferase 2 isoform 4</td>\n",
       "      <td>...</td>\n",
       "      <td>NP_001229795.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>protein arginine N-methyltransferase 2 isoform...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>31062</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NP_001229794.1</td>\n",
       "      <td>21</td>\n",
       "      <td>NC_000021.9</td>\n",
       "      <td>46636952</td>\n",
       "      <td>46652006</td>\n",
       "      <td>+</td>\n",
       "      <td>3275</td>\n",
       "      <td>PRMT2</td>\n",
       "      <td>NP_001229794.1</td>\n",
       "      <td>284</td>\n",
       "      <td>protein arginine N-methyltransferase 2 isoform 3</td>\n",
       "      <td>...</td>\n",
       "      <td>NP_001229794.1</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>protein arginine N-methyltransferase 2 isoform...</td>\n",
       "      <td>PRI</td>\n",
       "      <td>linear</td>\n",
       "      <td>AA</td>\n",
       "      <td>31031</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1286 rows  21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                #Replicon Name Replicon Accession     Start      Stop Strand  \\\n",
       "Protein product                                                                \n",
       "XP_006723962.1              21        NC_000021.9   5022680   5034096      +   \n",
       "XP_011544381.1              21        NC_000021.9   5022680   5034096      +   \n",
       "XP_011544380.1              21        NC_000021.9   5022680   5045606      +   \n",
       "NP_001350699.1              21        NC_000021.9   5022680   5034592      +   \n",
       "NP_001350689.1              21        NC_000021.9   5117103   5128354      -   \n",
       "...                        ...                ...       ...       ...    ...   \n",
       "NP_001273607.1              21        NC_000021.9  46636952  46664327      +   \n",
       "NP_001526.2                 21        NC_000021.9  46636952  46664327      +   \n",
       "XP_006724061.1              21        NC_000021.9  46636952  46664374      +   \n",
       "NP_001229795.1              21        NC_000021.9  46636952  46658924      +   \n",
       "NP_001229794.1              21        NC_000021.9  46636952  46652006      +   \n",
       "\n",
       "                    GeneID         Locus Protein product  Length  \\\n",
       "Protein product                                                    \n",
       "XP_006723962.1   102723996  LOC102723996  XP_006723962.1     516   \n",
       "XP_011544381.1   102723996  LOC102723996  XP_011544381.1     304   \n",
       "XP_011544380.1   102723996  LOC102723996  XP_011544380.1     361   \n",
       "NP_001350699.1   102723996  LOC102723996  NP_001350699.1     302   \n",
       "NP_001350689.1   102724023        GATD3B  NP_001350689.1     237   \n",
       "...                    ...           ...             ...     ...   \n",
       "NP_001273607.1        3275         PRMT2  NP_001273607.1     228   \n",
       "NP_001526.2           3275         PRMT2     NP_001526.2     433   \n",
       "XP_006724061.1        3275         PRMT2  XP_006724061.1     301   \n",
       "NP_001229795.1        3275         PRMT2  NP_001229795.1     277   \n",
       "NP_001229794.1        3275         PRMT2  NP_001229794.1     284   \n",
       "\n",
       "                                                      Protein name  ...  \\\n",
       "Protein product                                                     ...   \n",
       "XP_006723962.1                              ICOS ligand isoform X1  ...   \n",
       "XP_011544381.1                              ICOS ligand isoform X3  ...   \n",
       "XP_011544380.1                              ICOS ligand isoform X2  ...   \n",
       "NP_001350699.1                               ICOS ligand precursor  ...   \n",
       "NP_001350689.1   glutamine amidotransferase-like class 1 domain...  ...   \n",
       "...                                                            ...  ...   \n",
       "NP_001273607.1    protein arginine N-methyltransferase 2 isoform 7  ...   \n",
       "NP_001526.2       protein arginine N-methyltransferase 2 isoform 1  ...   \n",
       "XP_006724061.1   protein arginine N-methyltransferase 2 isoform X2  ...   \n",
       "NP_001229795.1    protein arginine N-methyltransferase 2 isoform 4  ...   \n",
       "NP_001229794.1    protein arginine N-methyltransferase 2 isoform 3  ...   \n",
       "\n",
       "                Accession Version      Organism  \\\n",
       "Protein product                                   \n",
       "XP_006723962.1     XP_006723962.1  Homo sapiens   \n",
       "XP_011544381.1     XP_011544381.1  Homo sapiens   \n",
       "XP_011544380.1     XP_011544380.1  Homo sapiens   \n",
       "NP_001350699.1     NP_001350699.1  Homo sapiens   \n",
       "NP_001350689.1     NP_001350689.1  Homo sapiens   \n",
       "...                           ...           ...   \n",
       "NP_001273607.1     NP_001273607.1  Homo sapiens   \n",
       "NP_001526.2           NP_001526.2  Homo sapiens   \n",
       "XP_006724061.1     XP_006724061.1  Homo sapiens   \n",
       "NP_001229795.1     NP_001229795.1  Homo sapiens   \n",
       "NP_001229794.1     NP_001229794.1  Homo sapiens   \n",
       "\n",
       "                                                        Definition Division  \\\n",
       "Protein product                                                               \n",
       "XP_006723962.1               ICOS ligand isoform X1 [Homo sapiens]      PRI   \n",
       "XP_011544381.1               ICOS ligand isoform X3 [Homo sapiens]      PRI   \n",
       "XP_011544380.1               ICOS ligand isoform X2 [Homo sapiens]      PRI   \n",
       "NP_001350699.1                ICOS ligand precursor [Homo sapiens]      PRI   \n",
       "NP_001350689.1   glutamine amidotransferase-like class 1 domain...      PRI   \n",
       "...                                                            ...      ...   \n",
       "NP_001273607.1   protein arginine N-methyltransferase 2 isoform...      PRI   \n",
       "NP_001526.2      protein arginine N-methyltransferase 2 isoform...      PRI   \n",
       "XP_006724061.1   protein arginine N-methyltransferase 2 isoform...      PRI   \n",
       "NP_001229795.1   protein arginine N-methyltransferase 2 isoform...      PRI   \n",
       "NP_001229794.1   protein arginine N-methyltransferase 2 isoform...      PRI   \n",
       "\n",
       "                Topology Molecule Type Molecular Weight Number of Regions  \\\n",
       "Protein product                                                             \n",
       "XP_006723962.1    linear            AA            56181                 2   \n",
       "XP_011544381.1    linear            AA            33350                 2   \n",
       "XP_011544380.1    linear            AA            39507                 2   \n",
       "NP_001350699.1    linear            AA            31389                 2   \n",
       "NP_001350689.1    linear            AA            24598                 1   \n",
       "...                  ...           ...              ...               ...   \n",
       "NP_001273607.1    linear            AA            25363                 3   \n",
       "NP_001526.2       linear            AA            48911                 6   \n",
       "XP_006724061.1    linear            AA            33412                 2   \n",
       "NP_001229795.1    linear            AA            31062                 5   \n",
       "NP_001229794.1    linear            AA            31031                 3   \n",
       "\n",
       "                 Number of Binding Sites  \\\n",
       "Protein product                            \n",
       "XP_006723962.1                         0   \n",
       "XP_011544381.1                         0   \n",
       "XP_011544380.1                         0   \n",
       "NP_001350699.1                         0   \n",
       "NP_001350689.1                         1   \n",
       "...                                  ...   \n",
       "NP_001273607.1                         4   \n",
       "NP_001526.2                            4   \n",
       "XP_006724061.1                         2   \n",
       "NP_001229795.1                         4   \n",
       "NP_001229794.1                         4   \n",
       "\n",
       "                                                          Sequence  \n",
       "Protein product                                                     \n",
       "XP_006723962.1   mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...  \n",
       "XP_011544381.1   mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...  \n",
       "XP_011544380.1   mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...  \n",
       "NP_001350699.1   mrlgspgllfllfsslradtqekevramvgsdvelscacpegsrfd...  \n",
       "NP_001350689.1   maavralvasrlaaasaftslspggrtpsqraalhlsvprpaarva...  \n",
       "...                                                            ...  \n",
       "NP_001273607.1   matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...  \n",
       "NP_001526.2      matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...  \n",
       "XP_006724061.1   matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...  \n",
       "NP_001229795.1   matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...  \n",
       "NP_001229794.1   matsgdcprsesqgeepaecseagllqegvqpeefvaiadyaatde...  \n",
       "\n",
       "[1286 rows x 21 columns]"
      ]
     },
     "execution_count": 161,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_protein_data_21 = protein_data_21.join(protein_additional_data_21)\n",
    "all_protein_data_21"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pickle\n",
    "all_protein_data_21.to_csv('all_protein_data_21.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "References\n",
    "Stephen F. Altschul, Thomas L. Madden, Alejandro A. Schffer, Jinghui Zhang, Zheng Zhang, Webb Miller, and David J. Lipman (1997), \"Gapped BLAST and PSI-BLAST: a new generation of protein database search programs\", Nucleic Acids Res. 25:3389-3402.\n",
    "https://towardsdatascience.com/processing-xml-in-python-elementtree-c8992941efd2 # processing xml data\n",
    "https://stackoverflow.com/questions/10220412/convert-tab-delimited-txt-file-into-a-csv-file-using-python # import of a text file\n",
    "https://www.ncbi.nlm.nih.gov/assembly/model/\n",
    "ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/405/GCF_000001405.39_GRCh38.p13/GCF_000001405.39_GRCh38.p13_assembly_report.txt # dna data\n",
    "https://www.ncbi.nlm.nih.gov/genomes/Genome2BE/genome2srv.cgi?action=GetFeatures4Grid&amp;download=on&amp;type=Proteins&amp;genome_id=51&genome_assembly_id=582967&mode=2&is_locus=1&is_locus_tag=0&optcols=1,0,0,0,0 # protein data\n",
    "https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&PAGE_TYPE=BlastDocs&DOC_TYPE=References\n",
    "https://www.ncbi.nlm.nih.gov/pubmed/20003500?dopt=Citation # paper for the creation of blast+\n",
    "https://www.ncbi.nlm.nih.gov/books/NBK52637/ # stand alone blast+ set up for pc\n",
    "ftp://ftp.ncbi.nlm.nih.gov/blast/db/ # data base location\n",
    "http://seqanswers.com/forums/showthread.php?t=44370 # troubleshooting database set up\n",
    "https://widdowquinn.github.io/2018-03-06-ibioic/02-sequence_databases/03-programming_for_blast.html # python/blast+ integration\n",
    "http://biopython.org/DIST/docs/tutorial/Tutorial.pdf # use of entrez package for online searches using api\n",
    "https://www.ncbi.nlm.nih.gov/books/NBK25499/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "learn-env",
   "language": "python",
   "name": "learn-env"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
