{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AnnotationReports {}\n",
      "AssembliesReport {'target_set_accession': 'GCF_002925485.2'}\n",
      "BuildInfo {}\n",
      "FeatureCounts {}\n",
      "FeatureStats {}\n"
     ]
    }
   ],
   "source": [
    "import xml.etree.ElementTree as ET\n",
    "tree = ET.parse('Homo_sapiens_AR109.20190905_annotation_report.xml')\n",
    "root = tree.getroot()\n",
    "print(root.tag, root.attrib)\n",
    "for child in root:\n",
    "    print(child.tag, child.attrib)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4\n"
     ]
    }
   ],
   "source": [
    "# children\n",
    "count = 0 \n",
    "for child in root:\n",
    "    count += 1\n",
    "print(count)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Child:\n",
      "AssembliesReport {'target_set_accession': 'GCF_002925485.2'}\n",
      "Grandchildren:\n",
      "FullAssembly {}\n",
      "1\n",
      "\n",
      "\n",
      "\n",
      "Child:\n",
      "BuildInfo {}\n",
      "Grandchildren:\n",
      "Taxid {}\n",
      "ScientificName {}\n",
      "AnnotationReleaseId {}\n",
      "FreezeDate {}\n",
      "FreezeDateId {}\n",
      "IdLoadDate {}\n",
      "SoftwareMajorVersion {}\n",
      "SoftwareMinorVersion {}\n",
      "11\n",
      "\n",
      "\n",
      "\n",
      "Child:\n",
      "FeatureCounts {}\n",
      "Grandchildren:\n",
      "51\n",
      "\n",
      "\n",
      "\n",
      "Child:\n",
      "FeatureStats {}\n",
      "Grandchildren:\n",
      "91\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# grandchildren\n",
    "count = 0\n",
    "for child in root:\n",
    "    print('Child:')\n",
    "    print(child.tag, child.attrib)\n",
    "    print('Grandchildren:')\n",
    "    for grandchild in child:\n",
    "        count += 1\n",
    "        if count < 10:\n",
    "            print(grandchild.tag, grandchild.attrib)\n",
    "    print(count)\n",
    "    print('\\n\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Child:\n",
      "AssembliesReport {'target_set_accession': 'GCF_002925485.2'}\n",
      "Grandchildren:\n",
      "FullAssembly {}\n",
      "Great Grandchildren:  Name {}\n",
      "Great Grandchildren:  Accession {}\n",
      "Great Grandchildren:  IsReference {}\n",
      "Great Grandchildren:  SubmitterOrganization {}\n",
      "Great Grandchildren:  SubmitterDate {}\n",
      "Great Grandchildren:  AssemblyContents {}\n",
      "Great Great Grandchildren:  Chromosomes {}\n",
      "Great Great Grandchildren:  AlternateLoci {}\n",
      "203\n",
      "\n",
      "\n",
      "\n",
      "Child:\n",
      "BuildInfo {}\n",
      "Grandchildren:\n",
      "213\n",
      "\n",
      "\n",
      "\n",
      "Child:\n",
      "FeatureCounts {}\n",
      "Grandchildren:\n",
      "2621\n",
      "\n",
      "\n",
      "\n",
      "Child:\n",
      "FeatureStats {}\n",
      "Grandchildren:\n",
      "6391\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# great grandchildren\n",
    "count = 0\n",
    "for child in root:\n",
    "    print('Child:')\n",
    "    print(child.tag, child.attrib)\n",
    "    print('Grandchildren:')\n",
    "    for grandchild in child:\n",
    "        count += 1\n",
    "        if count < 10:\n",
    "            print(grandchild.tag, grandchild.attrib)\n",
    "        for greatgrandchild in grandchild:\n",
    "            count += 1\n",
    "            if count < 10:\n",
    "                print('Great Grandchildren: ', greatgrandchild.tag, greatgrandchild.attrib)\n",
    "            for greatgreatgrandchild in greatgrandchild:\n",
    "                count += 1\n",
    "                if count < 10:\n",
    "                    print('Great Great Grandchildren: ', greatgreatgrandchild.tag, greatgreatgrandchild.attrib)\n",
    "                for ggggchild in greatgreatgrandchild:\n",
    "                    count += 1\n",
    "                    if count < 10:\n",
    "                        print('Great Great Great Grandchildren: ', ggggchild.tag, ggggchild.attrib, '\\n')\n",
    "    print(count)\n",
    "    print('\\n\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6396\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['AnnotationReports',\n",
       " 'AssembliesReport',\n",
       " 'FullAssembly',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsReference',\n",
       " 'SubmitterOrganization',\n",
       " 'SubmitterDate',\n",
       " 'AssemblyContents',\n",
       " 'Chromosomes',\n",
       " 'AlternateLoci',\n",
       " 'AssemblyPatches',\n",
       " 'UnlocalizedScaffolds',\n",
       " 'UnplacedScaffolds',\n",
       " 'Units',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'AssemblyUnit',\n",
       " 'Name',\n",
       " 'Accession',\n",
       " 'IsPrimary',\n",
       " 'Type',\n",
       " 'BuildInfo',\n",
       " 'Taxid',\n",
       " 'ScientificName',\n",
       " 'AnnotationReleaseId',\n",
       " 'FreezeDate',\n",
       " 'FreezeDateId',\n",
       " 'IdLoadDate',\n",
       " 'SoftwareMajorVersion',\n",
       " 'SoftwareMinorVersion',\n",
       " 'MaskingSoftware',\n",
       " 'FtpSite',\n",
       " 'FeatureCounts',\n",
       " 'OverallCounts',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'AssemblyCounts',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'AssemblyCounts',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'AssemblyCounts',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'AssemblyCounts',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'AssemblyCounts',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'AssemblyCounts',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'AssemblyCounts',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'AssemblyCounts',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'AssemblyCounts',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'AssemblyCounts',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'AssemblyCounts',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'AssemblyCounts',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Counts',\n",
       " 'Total',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " 'Subcount',\n",
       " ...]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tags = []\n",
    "for element in root.iter():\n",
    "    tags.append(element.tag)\n",
    "print(len(tags))\n",
    "tags"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[[('Name', {}, 'GRCh38.p13')],\n",
       " [('Accession', {}, 'GCF_000001405.39')],\n",
       " [('IsReference', {}, 'Yes')],\n",
       " [('SubmitterOrganization', {}, 'Genome Reference Consortium')],\n",
       " [('SubmitterDate', {}, '02-28-2019')],\n",
       " [('AssemblyContents', {}, '\\n      '),\n",
       "  ('Chromosomes', {}, '25'),\n",
       "  ('AlternateLoci', {}, None),\n",
       "  ('AssemblyPatches', {}, None),\n",
       "  ('UnlocalizedScaffolds', {}, None),\n",
       "  ('UnplacedScaffolds', {}, None)],\n",
       " [('Units', {}, '\\n      '),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'Primary Assembly'),\n",
       "  ('Accession', {}, 'GCF_000001305.15'),\n",
       "  ('IsPrimary', {}, 'Yes'),\n",
       "  ('Type', {}, 'HaploidUnit'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'PATCHES'),\n",
       "  ('Accession', {}, 'GCF_000005045.25'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AssemblyPatches'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_1'),\n",
       "  ('Accession', {}, 'GCF_000001315.2'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_2'),\n",
       "  ('Accession', {}, 'GCF_000001325.3'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_3'),\n",
       "  ('Accession', {}, 'GCF_000001335.2'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_4'),\n",
       "  ('Accession', {}, 'GCF_000001345.2'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_5'),\n",
       "  ('Accession', {}, 'GCF_000001355.2'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_6'),\n",
       "  ('Accession', {}, 'GCF_000001365.3'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_7'),\n",
       "  ('Accession', {}, 'GCF_000001375.2'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_8'),\n",
       "  ('Accession', {}, 'GCF_000510225.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_9'),\n",
       "  ('Accession', {}, 'GCF_000510235.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_10'),\n",
       "  ('Accession', {}, 'GCF_000509915.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_11'),\n",
       "  ('Accession', {}, 'GCF_000509925.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_12'),\n",
       "  ('Accession', {}, 'GCF_000509935.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_13'),\n",
       "  ('Accession', {}, 'GCF_000509945.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_14'),\n",
       "  ('Accession', {}, 'GCF_000509955.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_15'),\n",
       "  ('Accession', {}, 'GCF_000509965.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_16'),\n",
       "  ('Accession', {}, 'GCF_000509975.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_17'),\n",
       "  ('Accession', {}, 'GCF_000509985.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_18'),\n",
       "  ('Accession', {}, 'GCF_000509995.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_19'),\n",
       "  ('Accession', {}, 'GCF_000510005.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_20'),\n",
       "  ('Accession', {}, 'GCF_000510015.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_21'),\n",
       "  ('Accession', {}, 'GCF_000510025.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_22'),\n",
       "  ('Accession', {}, 'GCF_000510035.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_23'),\n",
       "  ('Accession', {}, 'GCF_000510045.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_24'),\n",
       "  ('Accession', {}, 'GCF_000510055.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_25'),\n",
       "  ('Accession', {}, 'GCF_000510065.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_26'),\n",
       "  ('Accession', {}, 'GCF_000510075.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_27'),\n",
       "  ('Accession', {}, 'GCF_000510085.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_28'),\n",
       "  ('Accession', {}, 'GCF_000510095.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_29'),\n",
       "  ('Accession', {}, 'GCF_000510105.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_30'),\n",
       "  ('Accession', {}, 'GCF_000510125.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_31'),\n",
       "  ('Accession', {}, 'GCF_000510135.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_32'),\n",
       "  ('Accession', {}, 'GCF_000510145.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_33'),\n",
       "  ('Accession', {}, 'GCF_000510155.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_34'),\n",
       "  ('Accession', {}, 'GCF_000510165.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_35'),\n",
       "  ('Accession', {}, 'GCF_000510175.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci'),\n",
       "  ('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'non-nuclear'),\n",
       "  ('Accession', {}, 'GCF_000006015.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'HaploidUnit')]]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_assembly_collection = []\n",
    "for i in root[0][0]:\n",
    "    tags = []\n",
    "    text = []\n",
    "    attributes = []\n",
    "    for element in i.iter():\n",
    "        tags.append(element.tag)\n",
    "        text.append(element.text)\n",
    "        attributes.append(element.attrib)\n",
    "    thing = list(zip(tags,attributes,text))\n",
    "    full_assembly_collection.append(thing)\n",
    "full_assembly_collection"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[[('Chromosomes', {}, '25')],\n",
       " [('AlternateLoci', {}, None)],\n",
       " [('AssemblyPatches', {}, None)],\n",
       " [('UnlocalizedScaffolds', {}, None)],\n",
       " [('UnplacedScaffolds', {}, None)]]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "assembly_contents_collection = []\n",
    "for i in root[0][0][5]:\n",
    "    tags = []\n",
    "    text = []\n",
    "    attributes = []\n",
    "    for element in i.iter():\n",
    "        tags.append(element.tag)\n",
    "        text.append(element.text)\n",
    "        attributes.append(element.attrib)\n",
    "    thing = list(zip(tags,attributes,text))\n",
    "    assembly_contents_collection.append(thing)\n",
    "assembly_contents_collection"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "38\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[[('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'Primary Assembly'),\n",
       "  ('Accession', {}, 'GCF_000001305.15'),\n",
       "  ('IsPrimary', {}, 'Yes'),\n",
       "  ('Type', {}, 'HaploidUnit')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'PATCHES'),\n",
       "  ('Accession', {}, 'GCF_000005045.25'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AssemblyPatches')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_1'),\n",
       "  ('Accession', {}, 'GCF_000001315.2'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_2'),\n",
       "  ('Accession', {}, 'GCF_000001325.3'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_3'),\n",
       "  ('Accession', {}, 'GCF_000001335.2'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_4'),\n",
       "  ('Accession', {}, 'GCF_000001345.2'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_5'),\n",
       "  ('Accession', {}, 'GCF_000001355.2'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_6'),\n",
       "  ('Accession', {}, 'GCF_000001365.3'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_7'),\n",
       "  ('Accession', {}, 'GCF_000001375.2'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_8'),\n",
       "  ('Accession', {}, 'GCF_000510225.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_9'),\n",
       "  ('Accession', {}, 'GCF_000510235.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_10'),\n",
       "  ('Accession', {}, 'GCF_000509915.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_11'),\n",
       "  ('Accession', {}, 'GCF_000509925.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_12'),\n",
       "  ('Accession', {}, 'GCF_000509935.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_13'),\n",
       "  ('Accession', {}, 'GCF_000509945.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_14'),\n",
       "  ('Accession', {}, 'GCF_000509955.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_15'),\n",
       "  ('Accession', {}, 'GCF_000509965.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_16'),\n",
       "  ('Accession', {}, 'GCF_000509975.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_17'),\n",
       "  ('Accession', {}, 'GCF_000509985.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_18'),\n",
       "  ('Accession', {}, 'GCF_000509995.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_19'),\n",
       "  ('Accession', {}, 'GCF_000510005.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_20'),\n",
       "  ('Accession', {}, 'GCF_000510015.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_21'),\n",
       "  ('Accession', {}, 'GCF_000510025.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_22'),\n",
       "  ('Accession', {}, 'GCF_000510035.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_23'),\n",
       "  ('Accession', {}, 'GCF_000510045.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_24'),\n",
       "  ('Accession', {}, 'GCF_000510055.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_25'),\n",
       "  ('Accession', {}, 'GCF_000510065.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_26'),\n",
       "  ('Accession', {}, 'GCF_000510075.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_27'),\n",
       "  ('Accession', {}, 'GCF_000510085.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_28'),\n",
       "  ('Accession', {}, 'GCF_000510095.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_29'),\n",
       "  ('Accession', {}, 'GCF_000510105.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_30'),\n",
       "  ('Accession', {}, 'GCF_000510125.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_31'),\n",
       "  ('Accession', {}, 'GCF_000510135.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_32'),\n",
       "  ('Accession', {}, 'GCF_000510145.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_33'),\n",
       "  ('Accession', {}, 'GCF_000510155.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_34'),\n",
       "  ('Accession', {}, 'GCF_000510165.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'ALT_REF_LOCI_35'),\n",
       "  ('Accession', {}, 'GCF_000510175.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'AlternateLoci')],\n",
       " [('AssemblyUnit', {}, '\\n        '),\n",
       "  ('Name', {}, 'non-nuclear'),\n",
       "  ('Accession', {}, 'GCF_000006015.1'),\n",
       "  ('IsPrimary', {}, 'No'),\n",
       "  ('Type', {}, 'HaploidUnit')]]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "assembly_unit_collection = []\n",
    "for i in root[0][0][6]:\n",
    "    tags = []\n",
    "    text = []\n",
    "    attributes = []\n",
    "    for element in i.iter():\n",
    "        tags.append(element.tag)\n",
    "        text.append(element.text)\n",
    "        attributes.append(element.attrib)\n",
    "    thing = list(zip(tags, attributes, text))\n",
    "    assembly_unit_collection.append(thing)\n",
    "print(len(assembly_unit_collection))\n",
    "assembly_unit_collection"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[[('Taxid', {}, '9606')],\n",
       " [('ScientificName', {}, 'Homo sapiens')],\n",
       " [('AnnotationReleaseId', {}, '109')],\n",
       " [('FreezeDate', {}, 'Sep 5 2019')],\n",
       " [('FreezeDateId', {}, '20190905')],\n",
       " [('IdLoadDate', {}, 'Sep 9 2019')],\n",
       " [('SoftwareMajorVersion', {}, '8')],\n",
       " [('SoftwareMinorVersion', {}, '2')],\n",
       " [('MaskingSoftware', {}, 'repeatmasker')],\n",
       " [('FtpSite',\n",
       "   {},\n",
       "   'ftp://ftp.ncbi.nlm.nih.gov/genomes/all/annotation_releases/9606/109.20190905')]]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "build_info_collection = []\n",
    "for i in root[1]:\n",
    "    tags = []\n",
    "    text = []\n",
    "    attributes = []\n",
    "    for element in i.iter():\n",
    "        tags.append(element.tag)\n",
    "        text.append(element.text)\n",
    "        attributes.append(element.attrib)\n",
    "    thing = list(zip(tags,attributes,text))\n",
    "    build_info_collection.append(thing)\n",
    "print(len(build_info_collection))\n",
    "build_info_collection"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "40\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[[('OverallCounts', {}, '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '54639'),\n",
       "  ('Subcount', {'subtype': 'other'}, '47'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '15204'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1135'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '19965'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '364'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '397'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '18012'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '19844'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '112430'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '172'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '26'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '1542'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '58361'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '54069'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '1541'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '112283'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '88'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '1'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '47041'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '15'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '7'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '196'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '29863'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '15319'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '196'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '45168'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1442'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '-2'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '74'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '108'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1334'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '74'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1431'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '113017'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '2600'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '514'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '1109'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '58374'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '54069'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '534'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '112283'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '118'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '69'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '503'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '57'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '35')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000001405.39', 'Type': 'full assembly'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '54639'),\n",
       "  ('Subcount', {'subtype': 'other'}, '47'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '15204'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1135'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '19965'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '364'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '397'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '18012'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '19844'),\n",
       "  ('Subcount', {'subtype': 'multiply placed'}, '3433'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '112430'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '172'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '26'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '1542'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '58361'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '54069'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '1541'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '112283'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '88'),\n",
       "  ('Subcount', {'subtype': 'multiply placed'}, '2901'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '1'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '47041'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '15'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '7'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '196'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '29863'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '15319'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '196'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '45168'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'multiply placed'}, '695'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1442'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '-2'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '74'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '108'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1334'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '74'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1431'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'multiply placed'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '113017'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '2600'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '514'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '1109'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '58374'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '54069'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '534'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '112283'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '118'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '69'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '503'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '57'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '35')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000001305.15', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '54312'),\n",
       "  ('Subcount', {'subtype': 'other'}, '45'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '15137'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1127'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '19886'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '366'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '388'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '17860'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '19755'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '112169'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '172'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '31'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '1565'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '58188'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '53981'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '1564'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '112038'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '78'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '1'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '45167'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '14'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '4'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '195'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '28479'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '15310'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '195'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '43775'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1426'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '1'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '78'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '97'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1329'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '78'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1417'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '112558'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '2602'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '348'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '938'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '58188'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '53981'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '548'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '112038'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '104'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '66'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '493'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '61'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '45')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000001405.39', 'Type': 'grouped alt loci'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '2398'),\n",
       "  ('Subcount', {'subtype': 'other'}, '8'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '631'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '90'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '606'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '297'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '161'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '676'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '832'),\n",
       "  ('Subcount', {'subtype': 'multiply placed'}, '658'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '1917'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '267'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '818'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '136'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1781'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '818'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1906'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '7'),\n",
       "  ('Subcount', {'subtype': 'multiply placed'}, '638'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '1820'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '1'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '56'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '166'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '1013'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '506'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '170'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1519'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'multiply placed'}, '183'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '109'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '9'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '65'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '8'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '101'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '65'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '107'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'multiply placed'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '2077'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '28'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '364'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '654'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '136'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1778'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '491'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1906'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '10'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '61'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '544'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '26'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '28')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000001315.2', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '1942'),\n",
       "  ('Subcount', {'subtype': 'other'}, '8'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '515'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '71'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '486'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '305'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '161'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '529'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '658'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '1505'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '281'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '743'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '82'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1423'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '743'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1498'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '4'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '1038'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '1'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '71'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '162'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '582'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '417'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '166'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '999'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '85'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '6'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '53'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '6'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '79'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '53'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '84'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '1665'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '25'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '384'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '651'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '82'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1420'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '488'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1498'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '6'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '45'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '443'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '27'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '39')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000001325.3', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '784'),\n",
       "  ('Subcount', {'subtype': 'other'}, '2'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '182'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '31'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '210'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '24'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '241'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '328'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '700'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '44'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '428'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '19'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '681'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '428'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '697'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '2'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '470'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '8'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '103'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '215'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '208'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '104'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '423'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '38'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '3'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '27'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '36'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '27'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '37'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '700'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '9'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '38'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '207'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '19'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '681'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '207'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '697'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '3'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '24'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '183'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '17'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '17')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000001335.2', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '398'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '86'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '12'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '127'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '15'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '112'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '188'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '410'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '22'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '227'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '5'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '405'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '227'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '409'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '209'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '6'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '70'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '48'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '138'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '71'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '186'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '17'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '1'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '14'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '17'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '14'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '17'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '410'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '2'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '21'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '135'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '5'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '405'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '135'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '409'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '1'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '8'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '127'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '5'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '4')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000001345.2', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '340'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '83'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '10'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '111'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '13'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '92'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '155'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '366'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '29'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '213'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '7'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '359'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '213'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '366'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '176'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '1'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '58'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '35'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '113'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '60'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '148'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '15'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '11'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '15'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '11'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '15'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '366'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '3'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '24'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '128'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '7'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '359'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '128'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '366'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '13'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '115'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '6'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '8')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000001355.2', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '346'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '69'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '10'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '113'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '18'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '92'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '175'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '402'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '32'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '234'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '8'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '394'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '234'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '402'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '164'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '7'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '62'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '29'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '105'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '63'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '134'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '15'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '1'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '11'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '15'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '11'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '15'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '402'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '3'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '30'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '148'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '8'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '394'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '148'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '402'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '16'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '132'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '14'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '14')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000001365.3', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '367'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '75'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '9'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '123'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '7'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '104'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '179'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '408'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '12'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '276'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '6'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '402'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '276'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '408'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '198'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '2'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '71'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '36'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '130'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '72'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '166'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '13'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '10'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '13'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '10'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '13'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '408'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '3'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '12'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '134'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '6'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '402'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '134'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '408'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '8'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '126'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '4'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '6')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000001375.2', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '385'),\n",
       "  ('Subcount', {'subtype': 'other'}, '1'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '82'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '13'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '124'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '16'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '112'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '177'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '420'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '37'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '265'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '5'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '415'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '265'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '420'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '206'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '5'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '73'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '46'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '125'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '74'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '171'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '17'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '13'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '17'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '13'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '17'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '419'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '2'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '28'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '171'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '5'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '414'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '171'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '420'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '13'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '160'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '7'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '7')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510225.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '45'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '5'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '24'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '3'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '8'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '31'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '92'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '3'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '52'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '92'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '52'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '92'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '21'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '2'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '7'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '14'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '7'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '21'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '92'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '4'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '2'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '40'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '92'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '40'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '92'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '39'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '1')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510235.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '65'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '9'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '2'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '34'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '1'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '11'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '43'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '126'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '2'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '73'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '4'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '122'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '73'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '126'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '32'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '14'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '18'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '32'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '6'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '6'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '6'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '126'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '4'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '2'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '49'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '4'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '122'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '49'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '126'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '48'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '1')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000509915.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '15'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '2'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '5'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '11'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '19'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '10'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '19'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '10'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '19'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '19'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '9'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '19'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '9'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '19'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '9'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000509925.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '11'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '2'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '4'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '7'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '10'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '9'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '10'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '9'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '10'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '10'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '10'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '10'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000509935.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '8'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '0'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '4'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '0'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '7'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '14'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '9'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '14'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '9'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '14'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '0'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '0'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '14'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '9'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '14'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '9'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '14'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '9'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000509945.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '11'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '2'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '3'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '7'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '9'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '9'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '9'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '9'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '9'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '9'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000509955.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '14'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '2'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '5'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '1'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '10'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '17'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '2'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '17'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '17'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '17'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '2'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '11'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '17'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '11'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '17'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '11'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000509965.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '15'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '3'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '4'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '2'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '9'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '13'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '8'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '13'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '8'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '13'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '3'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '3'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '13'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '7'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '13'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '7'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '13'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '7'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000509975.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '8'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '0'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '3'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '0'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '7'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '13'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '13'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '13'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '0'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '0'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '13'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '13'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '13'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000509985.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '11'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '2'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '3'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '7'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '9'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '9'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '9'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '9'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '9'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '9'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000509995.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '10'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '0'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '4'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '0'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '9'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '17'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '17'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '17'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '0'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '0'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '17'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '17'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '17'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510005.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '11'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '2'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '3'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '7'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '9'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '7'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '9'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '7'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '9'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '9'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '9'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '9'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510015.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '14'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '2'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '5'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '1'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '10'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '17'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '2'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '17'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '17'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '17'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '2'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '17'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '17'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510025.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '11'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '2'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '3'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '7'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '9'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '9'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '9'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '9'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '9'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '9'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510035.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '10'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '0'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '4'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '0'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '9'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '17'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '7'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '17'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '7'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '17'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '0'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '0'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '17'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '7'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '17'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '7'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '17'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '7'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510045.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '13'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '3'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '5'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '1'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '9'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '22'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '5'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '22'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '22'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '0'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '0'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '22'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '5'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '9'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '22'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '9'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '22'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '7'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '2'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '2')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510055.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '9'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '2'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '2'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '1'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '5'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '6'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '1'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '6'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '6'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '6'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '1'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '6'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '6'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510065.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '11'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '2'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '4'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '7'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '10'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '10'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '10'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '10'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '10'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '10'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510075.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '13'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '2'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '5'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '2'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '9'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '22'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '5'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '14'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '22'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '14'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '22'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '1'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '22'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '5'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '13'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '22'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '13'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '22'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '11'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '2'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '2')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510085.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '19'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '3'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '7'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '1'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '14'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '33'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '5'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '15'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '33'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '15'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '33'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '33'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '5'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '14'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '33'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '14'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '33'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '2'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '2')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510095.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '13'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '3'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '6'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '1'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '9'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '23'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '5'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '13'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '23'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '13'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '23'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '0'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '0'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '23'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '5'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '11'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '23'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '11'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '23'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '9'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '2'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '2')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510105.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '10'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '0'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '6'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '2'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '0'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '9'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '27'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '5'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '27'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '27'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '0'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '0'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '1'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '1'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '27'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '5'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '27'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '27'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '4'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '2'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '2')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510125.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '13'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '3'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '5'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '2'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '9'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '22'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '6'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '22'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '22'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '0'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '0'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '22'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '6'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '11'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '22'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '11'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '22'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '9'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '2'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '2')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510135.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '12'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '4'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '9'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '12'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '12'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '12'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '12'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '10'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '12'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '10'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '12'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '10'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510145.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '12'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '3'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '3'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '7'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '9'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '8'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '9'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '8'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '9'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '9'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '9'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '9'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '5'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510155.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '15'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '3'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '5'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '10'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '17'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '17'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '17'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '17'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '17'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '17'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '12'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510165.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '11'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '2'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '4'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '1'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '7'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '10'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '10'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '10'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '10'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '10'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '2'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '2'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '2'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '2'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '10'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '8'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '10'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '8'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '10'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '8'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000510175.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '10'),\n",
       "  ('Subcount', {'subtype': 'other'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '0'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '1'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '4'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '1'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '0'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '0'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '9'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '17'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '2'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '10'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '17'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '10'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '17'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '0'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '0'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '1'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '0'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '17'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '2'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '0'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '17'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '17'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '6'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '0')],\n",
       " [('AssemblyCounts',\n",
       "   {'Accession': 'GCF_000005045.25', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('Counts', {'feature_type': 'genes'}, '\\n      '),\n",
       "  ('Total', {}, '1529'),\n",
       "  ('Subcount', {'subtype': 'other'}, '12'),\n",
       "  ('Subcount', {'subtype': 'non-transcribed pseudo'}, '431'),\n",
       "  ('Subcount', {'subtype': 'transcribed pseudo'}, '80'),\n",
       "  ('Subcount', {'subtype': 'has variants'}, '380'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '101'),\n",
       "  ('Subcount', {'subtype': 'Ig TCR segment'}, '22'),\n",
       "  ('Subcount', {'subtype': 'non coding'}, '386'),\n",
       "  ('Subcount', {'subtype': 'protein coding'}, '598'),\n",
       "  ('Counts', {'feature_type': 'mRNAs'}, '\\n      '),\n",
       "  ('Total', {}, '1382'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '176'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '492'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '37'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1345'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '492'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1377'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '3'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Counts', {'feature_type': 'non-coding RNAs'}, '\\n      '),\n",
       "  ('Total', {}, '758'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '39'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '104'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '371'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '359'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '105'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '730'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'pseudo transcripts'}, '\\n      '),\n",
       "  ('Total', {}, '90'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '0'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '6'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '48'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '3'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '87'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '48'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '90'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '0'),\n",
       "  ('Subcount', {'subtype': 'has gaps'}, '0'),\n",
       "  ('Counts', {'feature_type': 'CDSs'}, '\\n      '),\n",
       "  ('Total', {}, '1390'),\n",
       "  ('Subcount', {'subtype': 'exon <= 3nt'}, '23'),\n",
       "  ('Subcount', {'subtype': 'partial'}, '161'),\n",
       "  ('Subcount', {'subtype': 'correction'}, '274'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq'}, '37'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq'}, '1331'),\n",
       "  ('Subcount', {'subtype': 'known RefSeq with correction'}, '252'),\n",
       "  ('Subcount', {'subtype': 'fully supported'}, '1377'),\n",
       "  ('Subcount', {'subtype': 'ab initio > 5%'}, '4'),\n",
       "  ('Subcount', {'subtype': 'model RefSeq with correction'}, '0'),\n",
       "  ('Subcount', {'subtype': 'major correction'}, '57'),\n",
       "  ('Subcount', {'subtype': 'minor correction'}, '200'),\n",
       "  ('Subcount', {'subtype': 'premature stop'}, '36'),\n",
       "  ('Subcount', {'subtype': 'has frameshifts'}, '40')]]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "feature_counts_collection = []\n",
    "for i in root[2]:\n",
    "    tags = []\n",
    "    text = []\n",
    "    attributes = []\n",
    "    for element in i.iter():\n",
    "        tags.append(element.tag)\n",
    "        text.append(element.text)\n",
    "        attributes.append(element.attrib)\n",
    "    thing = list(zip(tags, attributes ,text))\n",
    "    feature_counts_collection.append(thing)\n",
    "print(len(feature_counts_collection))\n",
    "feature_counts_collection"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "40\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[[('OverallStats', {}, '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '113'),\n",
       "  ('Mean', {}, '4.181'),\n",
       "  ('Median', {}, '2'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '363'),\n",
       "  ('Mean', {}, '11.6507'),\n",
       "  ('Median', {}, '9'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '159471'),\n",
       "  ('Min', {}, '16'),\n",
       "  ('Max', {}, '109224'),\n",
       "  ('Mean', {}, '3565.83'),\n",
       "  ('Median', {}, '2847'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '112443'),\n",
       "  ('Min', {}, '75'),\n",
       "  ('Max', {}, '107976'),\n",
       "  ('Mean', {}, '1986.65'),\n",
       "  ('Median', {}, '1461'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '37904'),\n",
       "  ('Min', {}, '33'),\n",
       "  ('Max', {}, '2473592'),\n",
       "  ('Mean', {}, '45183.9'),\n",
       "  ('Median', {}, '14182'),\n",
       "  ('LengthStats', {'category': 'RNase_MRP_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '1'),\n",
       "  ('Min', {}, '277'),\n",
       "  ('Max', {}, '277'),\n",
       "  ('Mean', {}, '277'),\n",
       "  ('Median', {}, '277'),\n",
       "  ('LengthStats', {'category': 'RNase_P_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '1'),\n",
       "  ('Min', {}, '341'),\n",
       "  ('Max', {}, '341'),\n",
       "  ('Mean', {}, '341'),\n",
       "  ('Median', {}, '341'),\n",
       "  ('LengthStats', {'category': 'Single-exon transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '1800'),\n",
       "  ('Min', {}, '180'),\n",
       "  ('Max', {}, '91671'),\n",
       "  ('Mean', {}, '2035.75'),\n",
       "  ('Median', {}, '1432'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '1403'),\n",
       "  ('Min', {}, '180'),\n",
       "  ('Max', {}, '12344'),\n",
       "  ('Mean', {}, '1839.05'),\n",
       "  ('Median', {}, '1198'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in non-coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '397'),\n",
       "  ('Min', {}, '200'),\n",
       "  ('Max', {}, '91671'),\n",
       "  ('Mean', {}, '2730.89'),\n",
       "  ('Median', {}, '2064'),\n",
       "  ('LengthStats', {'category': 'Y_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '113'),\n",
       "  ('Mean', {}, '98.75'),\n",
       "  ('Median', {}, '102'),\n",
       "  ('LengthStats', {'category': 'antisense_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '22'),\n",
       "  ('Min', {}, '255'),\n",
       "  ('Max', {}, '37027'),\n",
       "  ('Mean', {}, '4169.59'),\n",
       "  ('Median', {}, '1662'),\n",
       "  ('LengthStats', {'category': 'guide_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '29'),\n",
       "  ('Min', {}, '125'),\n",
       "  ('Max', {}, '421'),\n",
       "  ('Mean', {}, '206.793'),\n",
       "  ('Median', {}, '148'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '27344'),\n",
       "  ('Min', {}, '99'),\n",
       "  ('Max', {}, '91671'),\n",
       "  ('Mean', {}, '2029.63'),\n",
       "  ('Median', {}, '1376'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '112430'),\n",
       "  ('Min', {}, '153'),\n",
       "  ('Max', {}, '109224'),\n",
       "  ('Mean', {}, '4072.68'),\n",
       "  ('Median', {}, '3313'),\n",
       "  ('LengthStats', {'category': 'miRNA'}, '\\n      '),\n",
       "  ('Count', {}, '3173'),\n",
       "  ('Min', {}, '16'),\n",
       "  ('Max', {}, '28'),\n",
       "  ('Mean', {}, '21.6306'),\n",
       "  ('Median', {}, '22'),\n",
       "  ('LengthStats', {'category': 'misc_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '15218'),\n",
       "  ('Min', {}, '158'),\n",
       "  ('Max', {}, '25390'),\n",
       "  ('Mean', {}, '3605.73'),\n",
       "  ('Median', {}, '2981'),\n",
       "  ('LengthStats', {'category': 'premsg'}, '\\n      '),\n",
       "  ('Count', {}, '1913'),\n",
       "  ('Min', {}, '41'),\n",
       "  ('Max', {}, '180'),\n",
       "  ('Mean', {}, '81.9226'),\n",
       "  ('Median', {}, '80'),\n",
       "  ('LengthStats', {'category': 'rRNA'}, '\\n      '),\n",
       "  ('Count', {}, '39'),\n",
       "  ('Min', {}, '121'),\n",
       "  ('Max', {}, '13373'),\n",
       "  ('Mean', {}, '2736.56'),\n",
       "  ('Median', {}, '157'),\n",
       "  ('LengthStats', {'category': 'scRNA'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '58'),\n",
       "  ('Max', {}, '299'),\n",
       "  ('Mean', {}, '238.75'),\n",
       "  ('Median', {}, '299'),\n",
       "  ('LengthStats', {'category': 'snRNA'}, '\\n      '),\n",
       "  ('Count', {}, '63'),\n",
       "  ('Min', {}, '63'),\n",
       "  ('Max', {}, '332'),\n",
       "  ('Mean', {}, '134.921'),\n",
       "  ('Median', {}, '121'),\n",
       "  ('LengthStats', {'category': 'snoRNA'}, '\\n      '),\n",
       "  ('Count', {}, '541'),\n",
       "  ('Min', {}, '33'),\n",
       "  ('Max', {}, '250'),\n",
       "  ('Mean', {}, '105.649'),\n",
       "  ('Median', {}, '94'),\n",
       "  ('LengthStats', {'category': 'tRNA'}, '\\n      '),\n",
       "  ('Count', {}, '597'),\n",
       "  ('Min', {}, '59'),\n",
       "  ('Max', {}, '84'),\n",
       "  ('Mean', {}, '73.7672'),\n",
       "  ('Median', {}, '73'),\n",
       "  ('LengthStats', {'category': 'telomerase_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '1'),\n",
       "  ('Min', {}, '451'),\n",
       "  ('Max', {}, '451'),\n",
       "  ('Mean', {}, '451'),\n",
       "  ('Median', {}, '451'),\n",
       "  ('LengthStats', {'category': 'vault_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '88'),\n",
       "  ('Max', {}, '108'),\n",
       "  ('Mean', {}, '95.5'),\n",
       "  ('Median', {}, '98')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000001405.39', 'Type': 'full assembly'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '113'),\n",
       "  ('Mean', {}, '4.181'),\n",
       "  ('Median', {}, '2'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '363'),\n",
       "  ('Mean', {}, '11.6507'),\n",
       "  ('Median', {}, '9'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '411273'),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '91671'),\n",
       "  ('Mean', {}, '441.791'),\n",
       "  ('Median', {}, '149'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '352207'),\n",
       "  ('Min', {}, '26'),\n",
       "  ('Max', {}, '1160411'),\n",
       "  ('Mean', {}, '8185.69'),\n",
       "  ('Median', {}, '1873'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '159471'),\n",
       "  ('Min', {}, '16'),\n",
       "  ('Max', {}, '109224'),\n",
       "  ('Mean', {}, '3565.83'),\n",
       "  ('Median', {}, '2847'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '112443'),\n",
       "  ('Min', {}, '75'),\n",
       "  ('Max', {}, '107976'),\n",
       "  ('Mean', {}, '1986.65'),\n",
       "  ('Median', {}, '1461'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '327126'),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '33290'),\n",
       "  ('Mean', {}, '413.104'),\n",
       "  ('Median', {}, '143'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '156519'),\n",
       "  ('Min', {}, '2'),\n",
       "  ('Max', {}, '91671'),\n",
       "  ('Mean', {}, '417.327'),\n",
       "  ('Median', {}, '147'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '37904'),\n",
       "  ('Min', {}, '33'),\n",
       "  ('Max', {}, '2473592'),\n",
       "  ('Mean', {}, '45183.9'),\n",
       "  ('Median', {}, '14182'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '288015'),\n",
       "  ('Min', {}, '26'),\n",
       "  ('Max', {}, '1160411'),\n",
       "  ('Mean', {}, '7752.87'),\n",
       "  ('Median', {}, '1759'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '132792'),\n",
       "  ('Min', {}, '28'),\n",
       "  ('Max', {}, '773192'),\n",
       "  ('Mean', {}, '7841.45'),\n",
       "  ('Median', {}, '1965'),\n",
       "  ('LengthStats', {'category': 'RNase_MRP_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '1'),\n",
       "  ('Min', {}, '277'),\n",
       "  ('Max', {}, '277'),\n",
       "  ('Mean', {}, '277'),\n",
       "  ('Median', {}, '277'),\n",
       "  ('LengthStats', {'category': 'RNase_P_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '1'),\n",
       "  ('Min', {}, '341'),\n",
       "  ('Max', {}, '341'),\n",
       "  ('Mean', {}, '341'),\n",
       "  ('Median', {}, '341'),\n",
       "  ('LengthStats', {'category': 'Single-exon transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '1800'),\n",
       "  ('Min', {}, '180'),\n",
       "  ('Max', {}, '91671'),\n",
       "  ('Mean', {}, '2035.75'),\n",
       "  ('Median', {}, '1432'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '1403'),\n",
       "  ('Min', {}, '180'),\n",
       "  ('Max', {}, '12344'),\n",
       "  ('Mean', {}, '1839.05'),\n",
       "  ('Median', {}, '1198'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in non-coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '397'),\n",
       "  ('Min', {}, '200'),\n",
       "  ('Max', {}, '91671'),\n",
       "  ('Mean', {}, '2730.89'),\n",
       "  ('Median', {}, '2064'),\n",
       "  ('LengthStats', {'category': 'Y_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '113'),\n",
       "  ('Mean', {}, '98.75'),\n",
       "  ('Median', {}, '102'),\n",
       "  ('LengthStats', {'category': 'antisense_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '22'),\n",
       "  ('Min', {}, '255'),\n",
       "  ('Max', {}, '37027'),\n",
       "  ('Mean', {}, '4169.59'),\n",
       "  ('Median', {}, '1662'),\n",
       "  ('LengthStats', {'category': 'guide_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '29'),\n",
       "  ('Min', {}, '125'),\n",
       "  ('Max', {}, '421'),\n",
       "  ('Mean', {}, '206.793'),\n",
       "  ('Median', {}, '148'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '27344'),\n",
       "  ('Min', {}, '99'),\n",
       "  ('Max', {}, '91671'),\n",
       "  ('Mean', {}, '2029.63'),\n",
       "  ('Median', {}, '1376'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '112430'),\n",
       "  ('Min', {}, '153'),\n",
       "  ('Max', {}, '109224'),\n",
       "  ('Mean', {}, '4072.68'),\n",
       "  ('Median', {}, '3313'),\n",
       "  ('LengthStats', {'category': 'miRNA'}, '\\n      '),\n",
       "  ('Count', {}, '3173'),\n",
       "  ('Min', {}, '16'),\n",
       "  ('Max', {}, '28'),\n",
       "  ('Mean', {}, '21.6306'),\n",
       "  ('Median', {}, '22'),\n",
       "  ('LengthStats', {'category': 'misc_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '15218'),\n",
       "  ('Min', {}, '158'),\n",
       "  ('Max', {}, '25390'),\n",
       "  ('Mean', {}, '3605.73'),\n",
       "  ('Median', {}, '2981'),\n",
       "  ('LengthStats', {'category': 'premsg'}, '\\n      '),\n",
       "  ('Count', {}, '1913'),\n",
       "  ('Min', {}, '41'),\n",
       "  ('Max', {}, '180'),\n",
       "  ('Mean', {}, '81.9226'),\n",
       "  ('Median', {}, '80'),\n",
       "  ('LengthStats', {'category': 'rRNA'}, '\\n      '),\n",
       "  ('Count', {}, '39'),\n",
       "  ('Min', {}, '121'),\n",
       "  ('Max', {}, '13373'),\n",
       "  ('Mean', {}, '2736.56'),\n",
       "  ('Median', {}, '157'),\n",
       "  ('LengthStats', {'category': 'scRNA'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '58'),\n",
       "  ('Max', {}, '299'),\n",
       "  ('Mean', {}, '238.75'),\n",
       "  ('Median', {}, '299'),\n",
       "  ('LengthStats', {'category': 'snRNA'}, '\\n      '),\n",
       "  ('Count', {}, '63'),\n",
       "  ('Min', {}, '63'),\n",
       "  ('Max', {}, '332'),\n",
       "  ('Mean', {}, '134.921'),\n",
       "  ('Median', {}, '121'),\n",
       "  ('LengthStats', {'category': 'snoRNA'}, '\\n      '),\n",
       "  ('Count', {}, '541'),\n",
       "  ('Min', {}, '33'),\n",
       "  ('Max', {}, '250'),\n",
       "  ('Mean', {}, '105.649'),\n",
       "  ('Median', {}, '94'),\n",
       "  ('LengthStats', {'category': 'tRNA'}, '\\n      '),\n",
       "  ('Count', {}, '597'),\n",
       "  ('Min', {}, '59'),\n",
       "  ('Max', {}, '84'),\n",
       "  ('Mean', {}, '73.7672'),\n",
       "  ('Median', {}, '73'),\n",
       "  ('LengthStats', {'category': 'telomerase_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '1'),\n",
       "  ('Min', {}, '451'),\n",
       "  ('Max', {}, '451'),\n",
       "  ('Mean', {}, '451'),\n",
       "  ('Median', {}, '451'),\n",
       "  ('LengthStats', {'category': 'vault_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '88'),\n",
       "  ('Max', {}, '108'),\n",
       "  ('Mean', {}, '95.5'),\n",
       "  ('Median', {}, '98')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000001305.15', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '113'),\n",
       "  ('Mean', {}, '4.18929'),\n",
       "  ('Median', {}, '2'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '363'),\n",
       "  ('Mean', {}, '11.7428'),\n",
       "  ('Median', {}, '9'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '378131'),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '91671'),\n",
       "  ('Mean', {}, '451.678'),\n",
       "  ('Median', {}, '149'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '323298'),\n",
       "  ('Min', {}, '26'),\n",
       "  ('Max', {}, '1160411'),\n",
       "  ('Mean', {}, '8626.91'),\n",
       "  ('Median', {}, '2004'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '157336'),\n",
       "  ('Min', {}, '16'),\n",
       "  ('Max', {}, '109224'),\n",
       "  ('Mean', {}, '3594.62'),\n",
       "  ('Median', {}, '2873'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '112169'),\n",
       "  ('Min', {}, '75'),\n",
       "  ('Max', {}, '107976'),\n",
       "  ('Mean', {}, '1988.61'),\n",
       "  ('Median', {}, '1461'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '299699'),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '33290'),\n",
       "  ('Mean', {}, '424.016'),\n",
       "  ('Median', {}, '144'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '147954'),\n",
       "  ('Min', {}, '2'),\n",
       "  ('Max', {}, '91671'),\n",
       "  ('Mean', {}, '418.11'),\n",
       "  ('Median', {}, '146'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '37660'),\n",
       "  ('Min', {}, '33'),\n",
       "  ('Max', {}, '2473592'),\n",
       "  ('Mean', {}, '45795.7'),\n",
       "  ('Median', {}, '14350'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '263421'),\n",
       "  ('Min', {}, '26'),\n",
       "  ('Max', {}, '1160411'),\n",
       "  ('Mean', {}, '8220.54'),\n",
       "  ('Median', {}, '1904'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '125972'),\n",
       "  ('Min', {}, '28'),\n",
       "  ('Max', {}, '773192'),\n",
       "  ('Mean', {}, '8006.62'),\n",
       "  ('Median', {}, '2015'),\n",
       "  ('LengthStats', {'category': 'RNase_MRP_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '1'),\n",
       "  ('Min', {}, '277'),\n",
       "  ('Max', {}, '277'),\n",
       "  ('Mean', {}, '277'),\n",
       "  ('Median', {}, '277'),\n",
       "  ('LengthStats', {'category': 'RNase_P_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '1'),\n",
       "  ('Min', {}, '341'),\n",
       "  ('Max', {}, '341'),\n",
       "  ('Mean', {}, '341'),\n",
       "  ('Median', {}, '341'),\n",
       "  ('LengthStats', {'category': 'Single-exon transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '1769'),\n",
       "  ('Min', {}, '180'),\n",
       "  ('Max', {}, '91671'),\n",
       "  ('Mean', {}, '2041.42'),\n",
       "  ('Median', {}, '1434'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '1376'),\n",
       "  ('Min', {}, '180'),\n",
       "  ('Max', {}, '12344'),\n",
       "  ('Mean', {}, '1844.77'),\n",
       "  ('Median', {}, '1201'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in non-coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '393'),\n",
       "  ('Min', {}, '200'),\n",
       "  ('Max', {}, '91671'),\n",
       "  ('Mean', {}, '2729.97'),\n",
       "  ('Median', {}, '2064'),\n",
       "  ('LengthStats', {'category': 'Y_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '113'),\n",
       "  ('Mean', {}, '98.75'),\n",
       "  ('Median', {}, '102'),\n",
       "  ('LengthStats', {'category': 'antisense_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '22'),\n",
       "  ('Min', {}, '255'),\n",
       "  ('Max', {}, '37027'),\n",
       "  ('Mean', {}, '4169.59'),\n",
       "  ('Median', {}, '1662'),\n",
       "  ('LengthStats', {'category': 'guide_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '29'),\n",
       "  ('Min', {}, '125'),\n",
       "  ('Max', {}, '421'),\n",
       "  ('Mean', {}, '206.793'),\n",
       "  ('Median', {}, '148'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '26085'),\n",
       "  ('Min', {}, '99'),\n",
       "  ('Max', {}, '91671'),\n",
       "  ('Mean', {}, '2044.77'),\n",
       "  ('Median', {}, '1388'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '112169'),\n",
       "  ('Min', {}, '153'),\n",
       "  ('Max', {}, '109224'),\n",
       "  ('Mean', {}, '4077.3'),\n",
       "  ('Median', {}, '3319'),\n",
       "  ('LengthStats', {'category': 'miRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2875'),\n",
       "  ('Min', {}, '16'),\n",
       "  ('Max', {}, '28'),\n",
       "  ('Mean', {}, '21.6303'),\n",
       "  ('Median', {}, '22'),\n",
       "  ('LengthStats', {'category': 'misc_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '15084'),\n",
       "  ('Min', {}, '158'),\n",
       "  ('Max', {}, '25390'),\n",
       "  ('Mean', {}, '3614.11'),\n",
       "  ('Median', {}, '2987'),\n",
       "  ('LengthStats', {'category': 'premsg'}, '\\n      '),\n",
       "  ('Count', {}, '1913'),\n",
       "  ('Min', {}, '41'),\n",
       "  ('Max', {}, '180'),\n",
       "  ('Mean', {}, '81.9226'),\n",
       "  ('Median', {}, '80'),\n",
       "  ('LengthStats', {'category': 'rRNA'}, '\\n      '),\n",
       "  ('Count', {}, '37'),\n",
       "  ('Min', {}, '121'),\n",
       "  ('Max', {}, '13373'),\n",
       "  ('Mean', {}, '2816.57'),\n",
       "  ('Median', {}, '157'),\n",
       "  ('LengthStats', {'category': 'scRNA'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '58'),\n",
       "  ('Max', {}, '299'),\n",
       "  ('Mean', {}, '238.75'),\n",
       "  ('Median', {}, '299'),\n",
       "  ('LengthStats', {'category': 'snRNA'}, '\\n      '),\n",
       "  ('Count', {}, '63'),\n",
       "  ('Min', {}, '63'),\n",
       "  ('Max', {}, '332'),\n",
       "  ('Mean', {}, '134.921'),\n",
       "  ('Median', {}, '121'),\n",
       "  ('LengthStats', {'category': 'snoRNA'}, '\\n      '),\n",
       "  ('Count', {}, '541'),\n",
       "  ('Min', {}, '33'),\n",
       "  ('Max', {}, '250'),\n",
       "  ('Mean', {}, '105.649'),\n",
       "  ('Median', {}, '94'),\n",
       "  ('LengthStats', {'category': 'tRNA'}, '\\n      '),\n",
       "  ('Count', {}, '416'),\n",
       "  ('Min', {}, '71'),\n",
       "  ('Max', {}, '84'),\n",
       "  ('Mean', {}, '73.9423'),\n",
       "  ('Median', {}, '73'),\n",
       "  ('LengthStats', {'category': 'telomerase_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '1'),\n",
       "  ('Min', {}, '451'),\n",
       "  ('Max', {}, '451'),\n",
       "  ('Mean', {}, '451'),\n",
       "  ('Median', {}, '451'),\n",
       "  ('LengthStats', {'category': 'vault_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '88'),\n",
       "  ('Max', {}, '108'),\n",
       "  ('Mean', {}, '95.5'),\n",
       "  ('Median', {}, '98')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000001405.39', 'Type': 'grouped alt loci'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '36'),\n",
       "  ('Mean', {}, '2.14198'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '66'),\n",
       "  ('Mean', {}, '7.05033'),\n",
       "  ('Median', {}, '5'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '24956'),\n",
       "  ('Min', {}, '2'),\n",
       "  ('Max', {}, '18726'),\n",
       "  ('Mean', {}, '316.512'),\n",
       "  ('Median', {}, '143'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '21748'),\n",
       "  ('Min', {}, '30'),\n",
       "  ('Max', {}, '232263'),\n",
       "  ('Mean', {}, '2901.07'),\n",
       "  ('Median', {}, '742'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '3737'),\n",
       "  ('Min', {}, '17'),\n",
       "  ('Max', {}, '22824'),\n",
       "  ('Mean', {}, '2135.54'),\n",
       "  ('Median', {}, '1649'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '1917'),\n",
       "  ('Min', {}, '75'),\n",
       "  ('Max', {}, '22257'),\n",
       "  ('Mean', {}, '1561.41'),\n",
       "  ('Median', {}, '1182'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '20697'),\n",
       "  ('Min', {}, '6'),\n",
       "  ('Max', {}, '18726'),\n",
       "  ('Mean', {}, '278.942'),\n",
       "  ('Median', {}, '135'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '6523'),\n",
       "  ('Min', {}, '2'),\n",
       "  ('Max', {}, '9896'),\n",
       "  ('Mean', {}, '396.419'),\n",
       "  ('Median', {}, '162'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '1517'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '1296601'),\n",
       "  ('Mean', {}, '22661.2'),\n",
       "  ('Median', {}, '7270'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '18560'),\n",
       "  ('Min', {}, '33'),\n",
       "  ('Max', {}, '197352'),\n",
       "  ('Mean', {}, '2376.67'),\n",
       "  ('Median', {}, '647'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '5178'),\n",
       "  ('Min', {}, '30'),\n",
       "  ('Max', {}, '232263'),\n",
       "  ('Mean', {}, '4504.26'),\n",
       "  ('Median', {}, '983'),\n",
       "  ('LengthStats', {'category': 'Single-exon transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '159'),\n",
       "  ('Min', {}, '252'),\n",
       "  ('Max', {}, '8155'),\n",
       "  ('Mean', {}, '1895.09'),\n",
       "  ('Median', {}, '1100'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '127'),\n",
       "  ('Min', {}, '445'),\n",
       "  ('Max', {}, '7528'),\n",
       "  ('Mean', {}, '1752.8'),\n",
       "  ('Median', {}, '1041'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in non-coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '32'),\n",
       "  ('Min', {}, '252'),\n",
       "  ('Max', {}, '8155'),\n",
       "  ('Mean', {}, '2459.81'),\n",
       "  ('Median', {}, '2175'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '1204'),\n",
       "  ('Min', {}, '99'),\n",
       "  ('Max', {}, '11661'),\n",
       "  ('Mean', {}, '1654.36'),\n",
       "  ('Median', {}, '1162'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '1917'),\n",
       "  ('Min', {}, '264'),\n",
       "  ('Max', {}, '22824'),\n",
       "  ('Mean', {}, '2828.26'),\n",
       "  ('Median', {}, '2252'),\n",
       "  ('LengthStats', {'category': 'miRNA'}, '\\n      '),\n",
       "  ('Count', {}, '215'),\n",
       "  ('Min', {}, '17'),\n",
       "  ('Max', {}, '25'),\n",
       "  ('Mean', {}, '21.6326'),\n",
       "  ('Median', {}, '22'),\n",
       "  ('LengthStats', {'category': 'misc_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '232'),\n",
       "  ('Min', {}, '207'),\n",
       "  ('Max', {}, '6314'),\n",
       "  ('Mean', {}, '2368.12'),\n",
       "  ('Median', {}, '2216'),\n",
       "  ('LengthStats', {'category': 'premsg'}, '\\n      '),\n",
       "  ('Count', {}, '71'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '153'),\n",
       "  ('Mean', {}, '80.7324'),\n",
       "  ('Median', {}, '77'),\n",
       "  ('LengthStats', {'category': 'snRNA'}, '\\n      '),\n",
       "  ('Count', {}, '1'),\n",
       "  ('Min', {}, '165'),\n",
       "  ('Max', {}, '165'),\n",
       "  ('Mean', {}, '165'),\n",
       "  ('Median', {}, '165'),\n",
       "  ('LengthStats', {'category': 'snoRNA'}, '\\n      '),\n",
       "  ('Count', {}, '11'),\n",
       "  ('Min', {}, '63'),\n",
       "  ('Max', {}, '175'),\n",
       "  ('Mean', {}, '96.9091'),\n",
       "  ('Median', {}, '78'),\n",
       "  ('LengthStats', {'category': 'tRNA'}, '\\n      '),\n",
       "  ('Count', {}, '157'),\n",
       "  ('Min', {}, '72'),\n",
       "  ('Max', {}, '83'),\n",
       "  ('Mean', {}, '73.9299'),\n",
       "  ('Median', {}, '73')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000001315.2', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '21'),\n",
       "  ('Mean', {}, '2.13567'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '60'),\n",
       "  ('Mean', {}, '7.20447'),\n",
       "  ('Median', {}, '5'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '8112'),\n",
       "  ('Min', {}, '2'),\n",
       "  ('Max', {}, '18726'),\n",
       "  ('Mean', {}, '360.229'),\n",
       "  ('Median', {}, '146'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '6903'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '232263'),\n",
       "  ('Mean', {}, '4463.71'),\n",
       "  ('Median', {}, '1276'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '2543'),\n",
       "  ('Min', {}, '17'),\n",
       "  ('Max', {}, '22824'),\n",
       "  ('Mean', {}, '2349.12'),\n",
       "  ('Median', {}, '1890'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '1505'),\n",
       "  ('Min', {}, '75'),\n",
       "  ('Max', {}, '22257'),\n",
       "  ('Mean', {}, '1550.77'),\n",
       "  ('Median', {}, '1233'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '6114'),\n",
       "  ('Min', {}, '6'),\n",
       "  ('Max', {}, '18726'),\n",
       "  ('Mean', {}, '308.109'),\n",
       "  ('Median', {}, '137'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '2695'),\n",
       "  ('Min', {}, '2'),\n",
       "  ('Max', {}, '9896'),\n",
       "  ('Mean', {}, '441.574'),\n",
       "  ('Median', {}, '171'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '1195'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '1296601'),\n",
       "  ('Mean', {}, '23813.6'),\n",
       "  ('Median', {}, '7853'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '5448'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '176607'),\n",
       "  ('Mean', {}, '4070.42'),\n",
       "  ('Median', {}, '1211'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '2067'),\n",
       "  ('Min', {}, '54'),\n",
       "  ('Max', {}, '232263'),\n",
       "  ('Mean', {}, '5248.83'),\n",
       "  ('Median', {}, '1413'),\n",
       "  ('LengthStats', {'category': 'Single-exon transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '143'),\n",
       "  ('Min', {}, '252'),\n",
       "  ('Max', {}, '8155'),\n",
       "  ('Mean', {}, '1922.1'),\n",
       "  ('Median', {}, '1096'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '113'),\n",
       "  ('Min', {}, '445'),\n",
       "  ('Max', {}, '7528'),\n",
       "  ('Mean', {}, '1757.2'),\n",
       "  ('Median', {}, '1039'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in non-coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '30'),\n",
       "  ('Min', {}, '252'),\n",
       "  ('Max', {}, '8155'),\n",
       "  ('Mean', {}, '2543.2'),\n",
       "  ('Median', {}, '2435'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '763'),\n",
       "  ('Min', {}, '128'),\n",
       "  ('Max', {}, '11661'),\n",
       "  ('Mean', {}, '1726.92'),\n",
       "  ('Median', {}, '1230'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '1505'),\n",
       "  ('Min', {}, '264'),\n",
       "  ('Max', {}, '22824'),\n",
       "  ('Mean', {}, '2853.48'),\n",
       "  ('Median', {}, '2304'),\n",
       "  ('LengthStats', {'category': 'miRNA'}, '\\n      '),\n",
       "  ('Count', {}, '80'),\n",
       "  ('Min', {}, '17'),\n",
       "  ('Max', {}, '25'),\n",
       "  ('Mean', {}, '21.7375'),\n",
       "  ('Median', {}, '22'),\n",
       "  ('LengthStats', {'category': 'misc_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '167'),\n",
       "  ('Min', {}, '207'),\n",
       "  ('Max', {}, '6314'),\n",
       "  ('Mean', {}, '2141.28'),\n",
       "  ('Median', {}, '2006'),\n",
       "  ('LengthStats', {'category': 'premsg'}, '\\n      '),\n",
       "  ('Count', {}, '59'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '153'),\n",
       "  ('Mean', {}, '79.4407'),\n",
       "  ('Median', {}, '77'),\n",
       "  ('LengthStats', {'category': 'snRNA'}, '\\n      '),\n",
       "  ('Count', {}, '1'),\n",
       "  ('Min', {}, '165'),\n",
       "  ('Max', {}, '165'),\n",
       "  ('Mean', {}, '165'),\n",
       "  ('Median', {}, '165'),\n",
       "  ('LengthStats', {'category': 'snoRNA'}, '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '63'),\n",
       "  ('Max', {}, '175'),\n",
       "  ('Mean', {}, '95.1111'),\n",
       "  ('Median', {}, '77'),\n",
       "  ('LengthStats', {'category': 'tRNA'}, '\\n      '),\n",
       "  ('Count', {}, '18'),\n",
       "  ('Min', {}, '72'),\n",
       "  ('Max', {}, '83'),\n",
       "  ('Mean', {}, '73.7222'),\n",
       "  ('Median', {}, '73')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000001325.3', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '36'),\n",
       "  ('Mean', {}, '2.10694'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '66'),\n",
       "  ('Mean', {}, '7.41941'),\n",
       "  ('Median', {}, '5'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '3876'),\n",
       "  ('Min', {}, '6'),\n",
       "  ('Max', {}, '14433'),\n",
       "  ('Mean', {}, '355.506'),\n",
       "  ('Median', {}, '142'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '3336'),\n",
       "  ('Min', {}, '33'),\n",
       "  ('Max', {}, '197352'),\n",
       "  ('Mean', {}, '3610.37'),\n",
       "  ('Median', {}, '656'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '1170'),\n",
       "  ('Min', {}, '18'),\n",
       "  ('Max', {}, '16756'),\n",
       "  ('Mean', {}, '2275.08'),\n",
       "  ('Median', {}, '1870'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '700'),\n",
       "  ('Min', {}, '180'),\n",
       "  ('Max', {}, '16239'),\n",
       "  ('Mean', {}, '1541.12'),\n",
       "  ('Median', {}, '1044'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3119'),\n",
       "  ('Min', {}, '6'),\n",
       "  ('Max', {}, '14433'),\n",
       "  ('Mean', {}, '305.422'),\n",
       "  ('Median', {}, '134'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '1087'),\n",
       "  ('Min', {}, '12'),\n",
       "  ('Max', {}, '7568'),\n",
       "  ('Mean', {}, '456.215'),\n",
       "  ('Median', {}, '165'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '571'),\n",
       "  ('Min', {}, '61'),\n",
       "  ('Max', {}, '460024'),\n",
       "  ('Mean', {}, '19216.8'),\n",
       "  ('Median', {}, '5361'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '2766'),\n",
       "  ('Min', {}, '33'),\n",
       "  ('Max', {}, '197352'),\n",
       "  ('Mean', {}, '2895.93'),\n",
       "  ('Median', {}, '565'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '862'),\n",
       "  ('Min', {}, '71'),\n",
       "  ('Max', {}, '124245'),\n",
       "  ('Mean', {}, '5756.43'),\n",
       "  ('Median', {}, '1182'),\n",
       "  ('LengthStats', {'category': 'Single-exon transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '65'),\n",
       "  ('Min', {}, '445'),\n",
       "  ('Max', {}, '4834'),\n",
       "  ('Mean', {}, '1556.38'),\n",
       "  ('Median', {}, '1075'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '56'),\n",
       "  ('Min', {}, '445'),\n",
       "  ('Max', {}, '4834'),\n",
       "  ('Mean', {}, '1451.48'),\n",
       "  ('Median', {}, '1048'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in non-coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '770'),\n",
       "  ('Max', {}, '4805'),\n",
       "  ('Mean', {}, '2209.11'),\n",
       "  ('Median', {}, '2083'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '312'),\n",
       "  ('Min', {}, '99'),\n",
       "  ('Max', {}, '9818'),\n",
       "  ('Mean', {}, '1756.58'),\n",
       "  ('Median', {}, '1219'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '700'),\n",
       "  ('Min', {}, '331'),\n",
       "  ('Max', {}, '16756'),\n",
       "  ('Mean', {}, '2663.46'),\n",
       "  ('Median', {}, '2131'),\n",
       "  ('LengthStats', {'category': 'miRNA'}, '\\n      '),\n",
       "  ('Count', {}, '30'),\n",
       "  ('Min', {}, '18'),\n",
       "  ('Max', {}, '23'),\n",
       "  ('Mean', {}, '21.4667'),\n",
       "  ('Median', {}, '22'),\n",
       "  ('LengthStats', {'category': 'misc_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '86'),\n",
       "  ('Min', {}, '310'),\n",
       "  ('Max', {}, '6314'),\n",
       "  ('Mean', {}, '2855.48'),\n",
       "  ('Median', {}, '2746'),\n",
       "  ('LengthStats', {'category': 'premsg'}, '\\n      '),\n",
       "  ('Count', {}, '19'),\n",
       "  ('Min', {}, '61'),\n",
       "  ('Max', {}, '110'),\n",
       "  ('Mean', {}, '83.7895'),\n",
       "  ('Median', {}, '86'),\n",
       "  ('LengthStats', {'category': 'snoRNA'}, '\\n      '),\n",
       "  ('Count', {}, '6'),\n",
       "  ('Min', {}, '63'),\n",
       "  ('Max', {}, '132'),\n",
       "  ('Mean', {}, '81.6667'),\n",
       "  ('Median', {}, '77'),\n",
       "  ('LengthStats', {'category': 'tRNA'}, '\\n      '),\n",
       "  ('Count', {}, '36'),\n",
       "  ('Min', {}, '72'),\n",
       "  ('Max', {}, '83'),\n",
       "  ('Mean', {}, '73.9444'),\n",
       "  ('Median', {}, '73')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000001335.2', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '13'),\n",
       "  ('Mean', {}, '2.08392'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '66'),\n",
       "  ('Mean', {}, '8.05705'),\n",
       "  ('Median', {}, '5'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '2281'),\n",
       "  ('Min', {}, '6'),\n",
       "  ('Max', {}, '8265'),\n",
       "  ('Mean', {}, '301.163'),\n",
       "  ('Median', {}, '143'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '2003'),\n",
       "  ('Min', {}, '48'),\n",
       "  ('Max', {}, '110063'),\n",
       "  ('Mean', {}, '1855.19'),\n",
       "  ('Median', {}, '433'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '619'),\n",
       "  ('Min', {}, '20'),\n",
       "  ('Max', {}, '16756'),\n",
       "  ('Mean', {}, '2027.55'),\n",
       "  ('Median', {}, '1675'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '410'),\n",
       "  ('Min', {}, '180'),\n",
       "  ('Max', {}, '16239'),\n",
       "  ('Mean', {}, '1595.51'),\n",
       "  ('Median', {}, '1071'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '1961'),\n",
       "  ('Min', {}, '6'),\n",
       "  ('Max', {}, '8265'),\n",
       "  ('Mean', {}, '273.348'),\n",
       "  ('Median', {}, '138'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '519'),\n",
       "  ('Min', {}, '11'),\n",
       "  ('Max', {}, '7568'),\n",
       "  ('Mean', {}, '366.651'),\n",
       "  ('Median', {}, '154'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '300'),\n",
       "  ('Min', {}, '63'),\n",
       "  ('Max', {}, '152246'),\n",
       "  ('Mean', {}, '11334.9'),\n",
       "  ('Median', {}, '5586'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '1751'),\n",
       "  ('Min', {}, '48'),\n",
       "  ('Max', {}, '35672'),\n",
       "  ('Mean', {}, '1161.36'),\n",
       "  ('Median', {}, '366'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '424'),\n",
       "  ('Min', {}, '71'),\n",
       "  ('Max', {}, '110063'),\n",
       "  ('Mean', {}, '4483.21'),\n",
       "  ('Median', {}, '964'),\n",
       "  ('LengthStats', {'category': 'Single-exon transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '24'),\n",
       "  ('Min', {}, '963'),\n",
       "  ('Max', {}, '2624'),\n",
       "  ('Mean', {}, '1634.71'),\n",
       "  ('Median', {}, '1348'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '18'),\n",
       "  ('Min', {}, '963'),\n",
       "  ('Max', {}, '2517'),\n",
       "  ('Mean', {}, '1531.33'),\n",
       "  ('Median', {}, '1260'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in non-coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '6'),\n",
       "  ('Min', {}, '1180'),\n",
       "  ('Max', {}, '2624'),\n",
       "  ('Mean', {}, '1944.83'),\n",
       "  ('Median', {}, '2083'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '139'),\n",
       "  ('Min', {}, '149'),\n",
       "  ('Max', {}, '8477'),\n",
       "  ('Mean', {}, '1434.51'),\n",
       "  ('Median', {}, '893'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '410'),\n",
       "  ('Min', {}, '402'),\n",
       "  ('Max', {}, '16756'),\n",
       "  ('Mean', {}, '2428.95'),\n",
       "  ('Median', {}, '2059'),\n",
       "  ('LengthStats', {'category': 'miRNA'}, '\\n      '),\n",
       "  ('Count', {}, '20'),\n",
       "  ('Min', {}, '20'),\n",
       "  ('Max', {}, '23'),\n",
       "  ('Mean', {}, '21.65'),\n",
       "  ('Median', {}, '22'),\n",
       "  ('LengthStats', {'category': 'misc_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '30'),\n",
       "  ('Min', {}, '505'),\n",
       "  ('Max', {}, '4543'),\n",
       "  ('Mean', {}, '1928.1'),\n",
       "  ('Median', {}, '1801'),\n",
       "  ('LengthStats', {'category': 'premsg'}, '\\n      '),\n",
       "  ('Count', {}, '11'),\n",
       "  ('Min', {}, '63'),\n",
       "  ('Max', {}, '110'),\n",
       "  ('Mean', {}, '81.8182'),\n",
       "  ('Median', {}, '81'),\n",
       "  ('LengthStats', {'category': 'snoRNA'}, '\\n      '),\n",
       "  ('Count', {}, '6'),\n",
       "  ('Min', {}, '63'),\n",
       "  ('Max', {}, '132'),\n",
       "  ('Mean', {}, '81.6667'),\n",
       "  ('Median', {}, '77'),\n",
       "  ('LengthStats', {'category': 'tRNA'}, '\\n      '),\n",
       "  ('Count', {}, '14'),\n",
       "  ('Min', {}, '72'),\n",
       "  ('Max', {}, '82'),\n",
       "  ('Mean', {}, '73.0714'),\n",
       "  ('Median', {}, '72')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000001345.2', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '13'),\n",
       "  ('Mean', {}, '2.26432'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '66'),\n",
       "  ('Mean', {}, '7.55642'),\n",
       "  ('Median', {}, '5'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '1708'),\n",
       "  ('Min', {}, '6'),\n",
       "  ('Max', {}, '12739'),\n",
       "  ('Mean', {}, '306.721'),\n",
       "  ('Median', {}, '149'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '1492'),\n",
       "  ('Min', {}, '68'),\n",
       "  ('Max', {}, '110356'),\n",
       "  ('Mean', {}, '1996.42'),\n",
       "  ('Median', {}, '535'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '542'),\n",
       "  ('Min', {}, '19'),\n",
       "  ('Max', {}, '16756'),\n",
       "  ('Mean', {}, '1938.63'),\n",
       "  ('Median', {}, '1603'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '366'),\n",
       "  ('Min', {}, '180'),\n",
       "  ('Max', {}, '16239'),\n",
       "  ('Mean', {}, '1458.86'),\n",
       "  ('Median', {}, '1002'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '1463'),\n",
       "  ('Min', {}, '6'),\n",
       "  ('Max', {}, '12739'),\n",
       "  ('Mean', {}, '279.636'),\n",
       "  ('Median', {}, '142'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '432'),\n",
       "  ('Min', {}, '23'),\n",
       "  ('Max', {}, '7568'),\n",
       "  ('Mean', {}, '343.002'),\n",
       "  ('Median', {}, '153'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '247'),\n",
       "  ('Min', {}, '61'),\n",
       "  ('Max', {}, '152594'),\n",
       "  ('Mean', {}, '10004.8'),\n",
       "  ('Median', {}, '5355'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '1293'),\n",
       "  ('Min', {}, '68'),\n",
       "  ('Max', {}, '53017'),\n",
       "  ('Mean', {}, '1237.51'),\n",
       "  ('Median', {}, '453'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '361'),\n",
       "  ('Min', {}, '71'),\n",
       "  ('Max', {}, '110356'),\n",
       "  ('Mean', {}, '4303.32'),\n",
       "  ('Median', {}, '861'),\n",
       "  ('LengthStats', {'category': 'Single-exon transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '20'),\n",
       "  ('Min', {}, '770'),\n",
       "  ('Max', {}, '2624'),\n",
       "  ('Mean', {}, '1417.75'),\n",
       "  ('Median', {}, '1136'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '15'),\n",
       "  ('Min', {}, '963'),\n",
       "  ('Max', {}, '2481'),\n",
       "  ('Mean', {}, '1310.47'),\n",
       "  ('Median', {}, '1075'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in non-coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '5'),\n",
       "  ('Min', {}, '770'),\n",
       "  ('Max', {}, '2624'),\n",
       "  ('Mean', {}, '1739.6'),\n",
       "  ('Median', {}, '1889'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '108'),\n",
       "  ('Min', {}, '99'),\n",
       "  ('Max', {}, '8477'),\n",
       "  ('Mean', {}, '1469.68'),\n",
       "  ('Median', {}, '955'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '366'),\n",
       "  ('Min', {}, '402'),\n",
       "  ('Max', {}, '16756'),\n",
       "  ('Mean', {}, '2290.78'),\n",
       "  ('Median', {}, '1970'),\n",
       "  ('LengthStats', {'category': 'miRNA'}, '\\n      '),\n",
       "  ('Count', {}, '18'),\n",
       "  ('Min', {}, '19'),\n",
       "  ('Max', {}, '23'),\n",
       "  ('Mean', {}, '21.5'),\n",
       "  ('Median', {}, '22'),\n",
       "  ('LengthStats', {'category': 'misc_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '26'),\n",
       "  ('Min', {}, '505'),\n",
       "  ('Max', {}, '6314'),\n",
       "  ('Mean', {}, '1975.19'),\n",
       "  ('Median', {}, '1571'),\n",
       "  ('LengthStats', {'category': 'premsg'}, '\\n      '),\n",
       "  ('Count', {}, '10'),\n",
       "  ('Min', {}, '61'),\n",
       "  ('Max', {}, '110'),\n",
       "  ('Mean', {}, '82.7'),\n",
       "  ('Median', {}, '87'),\n",
       "  ('LengthStats', {'category': 'snoRNA'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '76'),\n",
       "  ('Max', {}, '132'),\n",
       "  ('Mean', {}, '90.75'),\n",
       "  ('Median', {}, '78'),\n",
       "  ('LengthStats', {'category': 'tRNA'}, '\\n      '),\n",
       "  ('Count', {}, '20'),\n",
       "  ('Min', {}, '72'),\n",
       "  ('Max', {}, '83'),\n",
       "  ('Mean', {}, '73.95'),\n",
       "  ('Median', {}, '73')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000001355.2', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '13'),\n",
       "  ('Mean', {}, '2.17004'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '66'),\n",
       "  ('Mean', {}, '8.30224'),\n",
       "  ('Median', {}, '6'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '2015'),\n",
       "  ('Min', {}, '11'),\n",
       "  ('Max', {}, '7195'),\n",
       "  ('Mean', {}, '277.085'),\n",
       "  ('Median', {}, '141'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '1759'),\n",
       "  ('Min', {}, '30'),\n",
       "  ('Max', {}, '39615'),\n",
       "  ('Mean', {}, '1315.9'),\n",
       "  ('Median', {}, '398'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '566'),\n",
       "  ('Min', {}, '20'),\n",
       "  ('Max', {}, '16756'),\n",
       "  ('Mean', {}, '1938.92'),\n",
       "  ('Median', {}, '1632'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '402'),\n",
       "  ('Min', {}, '180'),\n",
       "  ('Max', {}, '16239'),\n",
       "  ('Mean', {}, '1513.62'),\n",
       "  ('Median', {}, '1044'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '1800'),\n",
       "  ('Min', {}, '12'),\n",
       "  ('Max', {}, '7195'),\n",
       "  ('Mean', {}, '261.909'),\n",
       "  ('Median', {}, '138'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '432'),\n",
       "  ('Min', {}, '11'),\n",
       "  ('Max', {}, '3210'),\n",
       "  ('Mean', {}, '295.051'),\n",
       "  ('Median', {}, '143'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '267'),\n",
       "  ('Min', {}, '61'),\n",
       "  ('Max', {}, '152244'),\n",
       "  ('Mean', {}, '9200.34'),\n",
       "  ('Median', {}, '5056'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '1605'),\n",
       "  ('Min', {}, '48'),\n",
       "  ('Max', {}, '36388'),\n",
       "  ('Mean', {}, '1140.18'),\n",
       "  ('Median', {}, '352'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '347'),\n",
       "  ('Min', {}, '30'),\n",
       "  ('Max', {}, '39615'),\n",
       "  ('Mean', {}, '2001.61'),\n",
       "  ('Median', {}, '594'),\n",
       "  ('LengthStats', {'category': 'Single-exon transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '24'),\n",
       "  ('Min', {}, '407'),\n",
       "  ('Max', {}, '2547'),\n",
       "  ('Mean', {}, '1351'),\n",
       "  ('Median', {}, '1136'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '16'),\n",
       "  ('Min', {}, '963'),\n",
       "  ('Max', {}, '2481'),\n",
       "  ('Mean', {}, '1335'),\n",
       "  ('Median', {}, '1106'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in non-coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '8'),\n",
       "  ('Min', {}, '407'),\n",
       "  ('Max', {}, '2547'),\n",
       "  ('Mean', {}, '1383'),\n",
       "  ('Median', {}, '1332'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '89'),\n",
       "  ('Min', {}, '333'),\n",
       "  ('Max', {}, '4070'),\n",
       "  ('Mean', {}, '1238.73'),\n",
       "  ('Median', {}, '940'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '402'),\n",
       "  ('Min', {}, '402'),\n",
       "  ('Max', {}, '16756'),\n",
       "  ('Mean', {}, '2303.87'),\n",
       "  ('Median', {}, '2005'),\n",
       "  ('LengthStats', {'category': 'miRNA'}, '\\n      '),\n",
       "  ('Count', {}, '22'),\n",
       "  ('Min', {}, '20'),\n",
       "  ('Max', {}, '23'),\n",
       "  ('Mean', {}, '21.6818'),\n",
       "  ('Median', {}, '22'),\n",
       "  ('LengthStats', {'category': 'misc_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '29'),\n",
       "  ('Min', {}, '505'),\n",
       "  ('Max', {}, '6314'),\n",
       "  ('Mean', {}, '2024.31'),\n",
       "  ('Median', {}, '1603'),\n",
       "  ('LengthStats', {'category': 'premsg'}, '\\n      '),\n",
       "  ('Count', {}, '12'),\n",
       "  ('Min', {}, '61'),\n",
       "  ('Max', {}, '110'),\n",
       "  ('Mean', {}, '81.6667'),\n",
       "  ('Median', {}, '86'),\n",
       "  ('LengthStats', {'category': 'snoRNA'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '76'),\n",
       "  ('Max', {}, '132'),\n",
       "  ('Mean', {}, '90.75'),\n",
       "  ('Median', {}, '78'),\n",
       "  ('LengthStats', {'category': 'tRNA'}, '\\n      '),\n",
       "  ('Count', {}, '20'),\n",
       "  ('Min', {}, '72'),\n",
       "  ('Max', {}, '83'),\n",
       "  ('Mean', {}, '74.1'),\n",
       "  ('Median', {}, '73')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000001365.3', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '13'),\n",
       "  ('Mean', {}, '2.19923'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '66'),\n",
       "  ('Mean', {}, '8.15505'),\n",
       "  ('Median', {}, '6'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '2083'),\n",
       "  ('Min', {}, '6'),\n",
       "  ('Max', {}, '12739'),\n",
       "  ('Mean', {}, '293.904'),\n",
       "  ('Median', {}, '144'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '1832'),\n",
       "  ('Min', {}, '45'),\n",
       "  ('Max', {}, '110042'),\n",
       "  ('Mean', {}, '1644.74'),\n",
       "  ('Median', {}, '437'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '606'),\n",
       "  ('Min', {}, '20'),\n",
       "  ('Max', {}, '16756'),\n",
       "  ('Mean', {}, '1950.13'),\n",
       "  ('Median', {}, '1621'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '408'),\n",
       "  ('Min', {}, '180'),\n",
       "  ('Max', {}, '16239'),\n",
       "  ('Mean', {}, '1524.63'),\n",
       "  ('Median', {}, '1071'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '1817'),\n",
       "  ('Min', {}, '6'),\n",
       "  ('Max', {}, '12739'),\n",
       "  ('Mean', {}, '266.216'),\n",
       "  ('Median', {}, '138'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '476'),\n",
       "  ('Min', {}, '23'),\n",
       "  ('Max', {}, '7568'),\n",
       "  ('Mean', {}, '347.704'),\n",
       "  ('Median', {}, '154'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '283'),\n",
       "  ('Min', {}, '61'),\n",
       "  ('Max', {}, '152236'),\n",
       "  ('Mean', {}, '9397.71'),\n",
       "  ('Median', {}, '5464'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '1624'),\n",
       "  ('Min', {}, '68'),\n",
       "  ('Max', {}, '35669'),\n",
       "  ('Mean', {}, '1151.79'),\n",
       "  ('Median', {}, '382'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '394'),\n",
       "  ('Min', {}, '45'),\n",
       "  ('Max', {}, '110042'),\n",
       "  ('Mean', {}, '3451.84'),\n",
       "  ('Median', {}, '791'),\n",
       "  ('LengthStats', {'category': 'Single-exon transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '20'),\n",
       "  ('Min', {}, '770'),\n",
       "  ('Max', {}, '2624'),\n",
       "  ('Mean', {}, '1561.95'),\n",
       "  ('Median', {}, '1332'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '14'),\n",
       "  ('Min', {}, '963'),\n",
       "  ('Max', {}, '2517'),\n",
       "  ('Mean', {}, '1525.79'),\n",
       "  ('Median', {}, '1260'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in non-coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '6'),\n",
       "  ('Min', {}, '770'),\n",
       "  ('Max', {}, '2624'),\n",
       "  ('Mean', {}, '1646.33'),\n",
       "  ('Median', {}, '1889'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '118'),\n",
       "  ('Min', {}, '149'),\n",
       "  ('Max', {}, '8477'),\n",
       "  ('Mean', {}, '1466.86'),\n",
       "  ('Median', {}, '940'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '408'),\n",
       "  ('Min', {}, '402'),\n",
       "  ('Max', {}, '16756'),\n",
       "  ('Mean', {}, '2322.43'),\n",
       "  ('Median', {}, '2028'),\n",
       "  ('LengthStats', {'category': 'miRNA'}, '\\n      '),\n",
       "  ('Count', {}, '22'),\n",
       "  ('Min', {}, '20'),\n",
       "  ('Max', {}, '23'),\n",
       "  ('Mean', {}, '21.6818'),\n",
       "  ('Median', {}, '22'),\n",
       "  ('LengthStats', {'category': 'misc_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '30'),\n",
       "  ('Min', {}, '505'),\n",
       "  ('Max', {}, '6314'),\n",
       "  ('Mean', {}, '1951.17'),\n",
       "  ('Median', {}, '1603'),\n",
       "  ('LengthStats', {'category': 'premsg'}, '\\n      '),\n",
       "  ('Count', {}, '12'),\n",
       "  ('Min', {}, '61'),\n",
       "  ('Max', {}, '110'),\n",
       "  ('Mean', {}, '81.6667'),\n",
       "  ('Median', {}, '86'),\n",
       "  ('LengthStats', {'category': 'snoRNA'}, '\\n      '),\n",
       "  ('Count', {}, '6'),\n",
       "  ('Min', {}, '63'),\n",
       "  ('Max', {}, '132'),\n",
       "  ('Mean', {}, '81.6667'),\n",
       "  ('Median', {}, '77'),\n",
       "  ('LengthStats', {'category': 'tRNA'}, '\\n      '),\n",
       "  ('Count', {}, '22'),\n",
       "  ('Min', {}, '72'),\n",
       "  ('Max', {}, '83'),\n",
       "  ('Mean', {}, '74.3182'),\n",
       "  ('Median', {}, '73')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000001375.2', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '13'),\n",
       "  ('Mean', {}, '2.24715'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '65'),\n",
       "  ('Mean', {}, '8.01861'),\n",
       "  ('Median', {}, '6'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '2142'),\n",
       "  ('Min', {}, '6'),\n",
       "  ('Max', {}, '12739'),\n",
       "  ('Mean', {}, '286.005'),\n",
       "  ('Median', {}, '145'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '1890'),\n",
       "  ('Min', {}, '30'),\n",
       "  ('Max', {}, '110010'),\n",
       "  ('Mean', {}, '1795.23'),\n",
       "  ('Median', {}, '425'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '626'),\n",
       "  ('Min', {}, '19'),\n",
       "  ('Max', {}, '16756'),\n",
       "  ('Mean', {}, '1974.92'),\n",
       "  ('Median', {}, '1678'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '420'),\n",
       "  ('Min', {}, '180'),\n",
       "  ('Max', {}, '16239'),\n",
       "  ('Mean', {}, '1455.1'),\n",
       "  ('Median', {}, '1038'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '1852'),\n",
       "  ('Min', {}, '6'),\n",
       "  ('Max', {}, '12739'),\n",
       "  ('Mean', {}, '257.789'),\n",
       "  ('Median', {}, '138'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '564'),\n",
       "  ('Min', {}, '13'),\n",
       "  ('Max', {}, '7568'),\n",
       "  ('Mean', {}, '328.339'),\n",
       "  ('Median', {}, '154'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '290'),\n",
       "  ('Min', {}, '61'),\n",
       "  ('Max', {}, '152236'),\n",
       "  ('Mean', {}, '10236'),\n",
       "  ('Median', {}, '5628'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '1668'),\n",
       "  ('Min', {}, '68'),\n",
       "  ('Max', {}, '52905'),\n",
       "  ('Mean', {}, '1189.78'),\n",
       "  ('Median', {}, '365'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '466'),\n",
       "  ('Min', {}, '30'),\n",
       "  ('Max', {}, '110010'),\n",
       "  ('Mean', {}, '3478.97'),\n",
       "  ('Median', {}, '568'),\n",
       "  ('LengthStats', {'category': 'Single-exon transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '23'),\n",
       "  ('Min', {}, '590'),\n",
       "  ('Max', {}, '4525'),\n",
       "  ('Mean', {}, '1549.43'),\n",
       "  ('Median', {}, '1180'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '17'),\n",
       "  ('Min', {}, '590'),\n",
       "  ('Max', {}, '4525'),\n",
       "  ('Mean', {}, '1515.24'),\n",
       "  ('Median', {}, '1106'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in non-coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '6'),\n",
       "  ('Min', {}, '770'),\n",
       "  ('Max', {}, '2624'),\n",
       "  ('Mean', {}, '1646.33'),\n",
       "  ('Median', {}, '1889'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '121'),\n",
       "  ('Min', {}, '99'),\n",
       "  ('Max', {}, '8477'),\n",
       "  ('Mean', {}, '1400.26'),\n",
       "  ('Median', {}, '940'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '420'),\n",
       "  ('Min', {}, '402'),\n",
       "  ('Max', {}, '16756'),\n",
       "  ('Mean', {}, '2337.2'),\n",
       "  ('Median', {}, '2042'),\n",
       "  ('LengthStats', {'category': 'miRNA'}, '\\n      '),\n",
       "  ('Count', {}, '19'),\n",
       "  ('Min', {}, '19'),\n",
       "  ('Max', {}, '23'),\n",
       "  ('Mean', {}, '21.4737'),\n",
       "  ('Median', {}, '22'),\n",
       "  ('LengthStats', {'category': 'misc_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '36'),\n",
       "  ('Min', {}, '505'),\n",
       "  ('Max', {}, '6314'),\n",
       "  ('Mean', {}, '2293.11'),\n",
       "  ('Median', {}, '1960'),\n",
       "  ('LengthStats', {'category': 'premsg'}, '\\n      '),\n",
       "  ('Count', {}, '11'),\n",
       "  ('Min', {}, '61'),\n",
       "  ('Max', {}, '107'),\n",
       "  ('Mean', {}, '82.8182'),\n",
       "  ('Median', {}, '86'),\n",
       "  ('LengthStats', {'category': 'snoRNA'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '76'),\n",
       "  ('Max', {}, '132'),\n",
       "  ('Mean', {}, '95.3333'),\n",
       "  ('Median', {}, '78'),\n",
       "  ('LengthStats', {'category': 'tRNA'}, '\\n      '),\n",
       "  ('Count', {}, '27'),\n",
       "  ('Min', {}, '72'),\n",
       "  ('Max', {}, '83'),\n",
       "  ('Mean', {}, '74.037'),\n",
       "  ('Median', {}, '73')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510225.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '8'),\n",
       "  ('Mean', {}, '2.89744'),\n",
       "  ('Median', {}, '2'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '18'),\n",
       "  ('Mean', {}, '7.90265'),\n",
       "  ('Median', {}, '7'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '358'),\n",
       "  ('Min', {}, '12'),\n",
       "  ('Max', {}, '4033'),\n",
       "  ('Mean', {}, '248.601'),\n",
       "  ('Median', {}, '154'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '324'),\n",
       "  ('Min', {}, '69'),\n",
       "  ('Max', {}, '36335'),\n",
       "  ('Mean', {}, '1632.85'),\n",
       "  ('Median', {}, '673'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '113'),\n",
       "  ('Min', {}, '21'),\n",
       "  ('Max', {}, '5116'),\n",
       "  ('Mean', {}, '2015.02'),\n",
       "  ('Median', {}, '1870'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '92'),\n",
       "  ('Min', {}, '240'),\n",
       "  ('Max', {}, '3189'),\n",
       "  ('Mean', {}, '1241.15'),\n",
       "  ('Median', {}, '933'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '325'),\n",
       "  ('Min', {}, '12'),\n",
       "  ('Max', {}, '4033'),\n",
       "  ('Mean', {}, '241.431'),\n",
       "  ('Median', {}, '148'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '93'),\n",
       "  ('Min', {}, '28'),\n",
       "  ('Max', {}, '4033'),\n",
       "  ('Mean', {}, '275.516'),\n",
       "  ('Median', {}, '162'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '39'),\n",
       "  ('Min', {}, '72'),\n",
       "  ('Max', {}, '36881'),\n",
       "  ('Mean', {}, '11086.5'),\n",
       "  ('Median', {}, '7776'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '299'),\n",
       "  ('Min', {}, '69'),\n",
       "  ('Max', {}, '21374'),\n",
       "  ('Mean', {}, '1558.13'),\n",
       "  ('Median', {}, '646'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '77'),\n",
       "  ('Min', {}, '71'),\n",
       "  ('Max', {}, '36335'),\n",
       "  ('Mean', {}, '1841.68'),\n",
       "  ('Median', {}, '539'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '419'),\n",
       "  ('Max', {}, '2020'),\n",
       "  ('Mean', {}, '970.889'),\n",
       "  ('Median', {}, '1044'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '92'),\n",
       "  ('Min', {}, '526'),\n",
       "  ('Max', {}, '4875'),\n",
       "  ('Mean', {}, '2096.65'),\n",
       "  ('Median', {}, '1956'),\n",
       "  ('LengthStats', {'category': 'miRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '21'),\n",
       "  ('Max', {}, '22'),\n",
       "  ('Mean', {}, '21.5'),\n",
       "  ('Median', {}, '22'),\n",
       "  ('LengthStats', {'category': 'misc_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '10'),\n",
       "  ('Min', {}, '885'),\n",
       "  ('Max', {}, '5116'),\n",
       "  ('Mean', {}, '2602.4'),\n",
       "  ('Median', {}, '2364'),\n",
       "  ('LengthStats', {'category': 'premsg'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '72'),\n",
       "  ('Max', {}, '75'),\n",
       "  ('Mean', {}, '73.5'),\n",
       "  ('Median', {}, '75')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510235.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '8'),\n",
       "  ('Mean', {}, '2.92593'),\n",
       "  ('Median', {}, '2'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '18'),\n",
       "  ('Mean', {}, '8.02532'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '503'),\n",
       "  ('Min', {}, '12'),\n",
       "  ('Max', {}, '4033'),\n",
       "  ('Mean', {}, '248.541'),\n",
       "  ('Median', {}, '155'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '460'),\n",
       "  ('Min', {}, '69'),\n",
       "  ('Max', {}, '36335'),\n",
       "  ('Mean', {}, '1682.65'),\n",
       "  ('Median', {}, '652'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '158'),\n",
       "  ('Min', {}, '21'),\n",
       "  ('Max', {}, '5116'),\n",
       "  ('Mean', {}, '2097.47'),\n",
       "  ('Median', {}, '2056'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '126'),\n",
       "  ('Min', {}, '240'),\n",
       "  ('Max', {}, '3189'),\n",
       "  ('Mean', {}, '1266.02'),\n",
       "  ('Median', {}, '1035'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '451'),\n",
       "  ('Min', {}, '12'),\n",
       "  ('Max', {}, '4033'),\n",
       "  ('Mean', {}, '243.96'),\n",
       "  ('Median', {}, '142'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '142'),\n",
       "  ('Min', {}, '28'),\n",
       "  ('Max', {}, '4033'),\n",
       "  ('Mean', {}, '264.437'),\n",
       "  ('Median', {}, '171'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '54'),\n",
       "  ('Min', {}, '72'),\n",
       "  ('Max', {}, '36881'),\n",
       "  ('Mean', {}, '11802.4'),\n",
       "  ('Median', {}, '9771'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '419'),\n",
       "  ('Min', {}, '69'),\n",
       "  ('Max', {}, '21374'),\n",
       "  ('Mean', {}, '1586.19'),\n",
       "  ('Median', {}, '619'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '118'),\n",
       "  ('Min', {}, '71'),\n",
       "  ('Max', {}, '36335'),\n",
       "  ('Mean', {}, '1686.7'),\n",
       "  ('Median', {}, '448'),\n",
       "  ('LengthStats', {'category': 'Single-exon transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '1'),\n",
       "  ('Min', {}, '2050'),\n",
       "  ('Max', {}, '2050'),\n",
       "  ('Mean', {}, '2050'),\n",
       "  ('Median', {}, '2050'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '1'),\n",
       "  ('Min', {}, '2050'),\n",
       "  ('Max', {}, '2050'),\n",
       "  ('Mean', {}, '2050'),\n",
       "  ('Median', {}, '2050'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '16'),\n",
       "  ('Min', {}, '419'),\n",
       "  ('Max', {}, '2184'),\n",
       "  ('Mean', {}, '1075.81'),\n",
       "  ('Median', {}, '1009'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '126'),\n",
       "  ('Min', {}, '526'),\n",
       "  ('Max', {}, '5097'),\n",
       "  ('Mean', {}, '2208.09'),\n",
       "  ('Median', {}, '2149'),\n",
       "  ('LengthStats', {'category': 'miRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '21'),\n",
       "  ('Max', {}, '22'),\n",
       "  ('Mean', {}, '21.5'),\n",
       "  ('Median', {}, '22'),\n",
       "  ('LengthStats', {'category': 'misc_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '14'),\n",
       "  ('Min', {}, '885'),\n",
       "  ('Max', {}, '5116'),\n",
       "  ('Mean', {}, '2566.14'),\n",
       "  ('Median', {}, '2364'),\n",
       "  ('LengthStats', {'category': 'premsg'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '72'),\n",
       "  ('Max', {}, '75'),\n",
       "  ('Mean', {}, '73.5'),\n",
       "  ('Median', {}, '75')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000509915.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '5'),\n",
       "  ('Mean', {}, '1.75'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.33333'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '95'),\n",
       "  ('Min', {}, '34'),\n",
       "  ('Max', {}, '774'),\n",
       "  ('Mean', {}, '197.316'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '90'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '9888'),\n",
       "  ('Mean', {}, '1889.01'),\n",
       "  ('Median', {}, '1273'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '21'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '1877'),\n",
       "  ('Mean', {}, '1436.57'),\n",
       "  ('Median', {}, '1522'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '19'),\n",
       "  ('Min', {}, '321'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '991.263'),\n",
       "  ('Median', {}, '1029'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '91'),\n",
       "  ('Min', {}, '34'),\n",
       "  ('Max', {}, '774'),\n",
       "  ('Mean', {}, '195.242'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '12'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '16787'),\n",
       "  ('Mean', {}, '12227.3'),\n",
       "  ('Median', {}, '14312'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '87'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9888'),\n",
       "  ('Mean', {}, '1934.48'),\n",
       "  ('Median', {}, '1304'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '19'),\n",
       "  ('Min', {}, '979'),\n",
       "  ('Max', {}, '1877'),\n",
       "  ('Mean', {}, '1509.32'),\n",
       "  ('Median', {}, '1573')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000509925.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '2'),\n",
       "  ('Mean', {}, '1.5'),\n",
       "  ('Median', {}, '2'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.16667'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '62'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '207.242'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '57'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '9892'),\n",
       "  ('Mean', {}, '1948.54'),\n",
       "  ('Median', {}, '1113'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '12'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1523.5'),\n",
       "  ('Median', {}, '1592'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '10'),\n",
       "  ('Min', {}, '669'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '1056.6'),\n",
       "  ('Median', {}, '1047'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '58'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '204.672'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '8'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '16787'),\n",
       "  ('Mean', {}, '12586'),\n",
       "  ('Median', {}, '14521'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '54'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9892'),\n",
       "  ('Mean', {}, '2025.11'),\n",
       "  ('Median', {}, '1464'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '10'),\n",
       "  ('Min', {}, '1477'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1679.1'),\n",
       "  ('Median', {}, '1614')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000509935.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '5'),\n",
       "  ('Mean', {}, '2'),\n",
       "  ('Median', {}, '2'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '6'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.71429'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '59'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '688'),\n",
       "  ('Mean', {}, '210.644'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '58'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9892'),\n",
       "  ('Mean', {}, '2216.9'),\n",
       "  ('Median', {}, '1514'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '14'),\n",
       "  ('Min', {}, '979'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1573.36'),\n",
       "  ('Median', {}, '1582'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '14'),\n",
       "  ('Min', {}, '321'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '970.286'),\n",
       "  ('Median', {}, '1029'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '59'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '688'),\n",
       "  ('Mean', {}, '210.644'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '7'),\n",
       "  ('Min', {}, '10911'),\n",
       "  ('Max', {}, '16733'),\n",
       "  ('Mean', {}, '14087.1'),\n",
       "  ('Median', {}, '14345'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '58'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9892'),\n",
       "  ('Mean', {}, '2216.9'),\n",
       "  ('Median', {}, '1514'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '14'),\n",
       "  ('Min', {}, '979'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1573.36'),\n",
       "  ('Median', {}, '1582')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000509945.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '2'),\n",
       "  ('Mean', {}, '1.375'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.18182'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '62'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '207.613'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '56'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '9890'),\n",
       "  ('Mean', {}, '1850.05'),\n",
       "  ('Median', {}, '1113'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '11'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1524.82'),\n",
       "  ('Median', {}, '1605'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '915'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '1156'),\n",
       "  ('Median', {}, '1134'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '58'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '205.069'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '8'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '16751'),\n",
       "  ('Mean', {}, '12591.4'),\n",
       "  ('Median', {}, '14530'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '53'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9890'),\n",
       "  ('Mean', {}, '1922.49'),\n",
       "  ('Median', {}, '1273'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '1477'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1698'),\n",
       "  ('Median', {}, '1614')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000509955.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '5'),\n",
       "  ('Mean', {}, '1.72727'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '6.89474'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '84'),\n",
       "  ('Min', {}, '34'),\n",
       "  ('Max', {}, '774'),\n",
       "  ('Mean', {}, '204.393'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '77'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '7715'),\n",
       "  ('Mean', {}, '1864.09'),\n",
       "  ('Median', {}, '1273'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '19'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1467.21'),\n",
       "  ('Median', {}, '1573'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '17'),\n",
       "  ('Min', {}, '321'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '980.824'),\n",
       "  ('Median', {}, '1029'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '80'),\n",
       "  ('Min', {}, '34'),\n",
       "  ('Max', {}, '774'),\n",
       "  ('Mean', {}, '202.362'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '245'),\n",
       "  ('Median', {}, '269'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '11'),\n",
       "  ('Min', {}, '1644'),\n",
       "  ('Max', {}, '15237'),\n",
       "  ('Mean', {}, '11532.4'),\n",
       "  ('Median', {}, '14312'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '74'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '7715'),\n",
       "  ('Mean', {}, '1916.54'),\n",
       "  ('Median', {}, '1464'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '17'),\n",
       "  ('Min', {}, '979'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1552.12'),\n",
       "  ('Median', {}, '1582')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000509965.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '3'),\n",
       "  ('Mean', {}, '1.45455'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.0625'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '81'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '191.099'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '74'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '9888'),\n",
       "  ('Mean', {}, '1841.31'),\n",
       "  ('Median', {}, '1273'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '16'),\n",
       "  ('Min', {}, '536'),\n",
       "  ('Max', {}, '1877'),\n",
       "  ('Mean', {}, '1388.94'),\n",
       "  ('Median', {}, '1582'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '13'),\n",
       "  ('Min', {}, '828'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '1073.31'),\n",
       "  ('Median', {}, '1047'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '74'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '188.716'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '7'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '216.286'),\n",
       "  ('Median', {}, '197'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '11'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '16787'),\n",
       "  ('Mean', {}, '11811.2'),\n",
       "  ('Median', {}, '14015'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '69'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9888'),\n",
       "  ('Mean', {}, '1856.09'),\n",
       "  ('Median', {}, '1304'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '5'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '5470'),\n",
       "  ('Mean', {}, '1637.4'),\n",
       "  ('Median', {}, '849'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '13'),\n",
       "  ('Min', {}, '1101'),\n",
       "  ('Max', {}, '1877'),\n",
       "  ('Mean', {}, '1553.54'),\n",
       "  ('Median', {}, '1592'),\n",
       "  ('LengthStats', {'category': 'misc_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '1'),\n",
       "  ('Min', {}, '536'),\n",
       "  ('Max', {}, '536'),\n",
       "  ('Mean', {}, '536'),\n",
       "  ('Median', {}, '536')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000509975.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '5'),\n",
       "  ('Mean', {}, '1.85714'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '6'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.76923'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '59'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '688'),\n",
       "  ('Mean', {}, '211.017'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '57'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9891'),\n",
       "  ('Mean', {}, '2103.23'),\n",
       "  ('Median', {}, '1273'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '13'),\n",
       "  ('Min', {}, '979'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1578.23'),\n",
       "  ('Median', {}, '1582'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '13'),\n",
       "  ('Min', {}, '321'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '1008.46'),\n",
       "  ('Median', {}, '1029'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '59'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '688'),\n",
       "  ('Mean', {}, '211.017'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '7'),\n",
       "  ('Min', {}, '10951'),\n",
       "  ('Max', {}, '16752'),\n",
       "  ('Mean', {}, '13916.7'),\n",
       "  ('Median', {}, '14344'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '57'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9891'),\n",
       "  ('Mean', {}, '2103.23'),\n",
       "  ('Median', {}, '1273'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '13'),\n",
       "  ('Min', {}, '979'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1578.23'),\n",
       "  ('Median', {}, '1582')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000509985.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '2'),\n",
       "  ('Mean', {}, '1.375'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.18182'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '62'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '207.613'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '56'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '9890'),\n",
       "  ('Mean', {}, '1849.8'),\n",
       "  ('Median', {}, '1113'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '11'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1524.82'),\n",
       "  ('Median', {}, '1605'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '915'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '1156'),\n",
       "  ('Median', {}, '1134'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '58'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '205.069'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '8'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '16751'),\n",
       "  ('Mean', {}, '12589.6'),\n",
       "  ('Median', {}, '14519'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '53'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9890'),\n",
       "  ('Mean', {}, '1922.23'),\n",
       "  ('Median', {}, '1273'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '1477'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1698'),\n",
       "  ('Median', {}, '1614')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000509995.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '5'),\n",
       "  ('Mean', {}, '1.88889'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '6'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.76471'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '75'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '694'),\n",
       "  ('Mean', {}, '200.12'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '73'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9888'),\n",
       "  ('Mean', {}, '2023.41'),\n",
       "  ('Median', {}, '1476'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '17'),\n",
       "  ('Min', {}, '979'),\n",
       "  ('Max', {}, '1877'),\n",
       "  ('Mean', {}, '1524.71'),\n",
       "  ('Median', {}, '1573'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '17'),\n",
       "  ('Min', {}, '321'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '979.941'),\n",
       "  ('Median', {}, '1017'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '75'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '694'),\n",
       "  ('Mean', {}, '200.12'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '9465'),\n",
       "  ('Max', {}, '16787'),\n",
       "  ('Mean', {}, '13526'),\n",
       "  ('Median', {}, '14313'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '73'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9888'),\n",
       "  ('Mean', {}, '2023.41'),\n",
       "  ('Median', {}, '1476'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '17'),\n",
       "  ('Min', {}, '979'),\n",
       "  ('Max', {}, '1877'),\n",
       "  ('Mean', {}, '1524.71'),\n",
       "  ('Median', {}, '1573')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510005.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '2'),\n",
       "  ('Mean', {}, '1.375'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.18182'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '62'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '207.613'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '56'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '9889'),\n",
       "  ('Mean', {}, '1849.36'),\n",
       "  ('Median', {}, '1113'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '11'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1524.82'),\n",
       "  ('Median', {}, '1605'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '915'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '1156'),\n",
       "  ('Median', {}, '1134'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '58'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '205.069'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '8'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '16765'),\n",
       "  ('Mean', {}, '12586.6'),\n",
       "  ('Median', {}, '14520'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '53'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9889'),\n",
       "  ('Mean', {}, '1921.75'),\n",
       "  ('Median', {}, '1273'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '1477'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1698'),\n",
       "  ('Median', {}, '1614')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510015.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '5'),\n",
       "  ('Mean', {}, '1.72727'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '6.89474'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '84'),\n",
       "  ('Min', {}, '34'),\n",
       "  ('Max', {}, '774'),\n",
       "  ('Mean', {}, '210.167'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '77'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '7715'),\n",
       "  ('Mean', {}, '1875.66'),\n",
       "  ('Median', {}, '1273'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '19'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1492.89'),\n",
       "  ('Median', {}, '1581'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '17'),\n",
       "  ('Min', {}, '321'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '962.471'),\n",
       "  ('Median', {}, '1017'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '80'),\n",
       "  ('Min', {}, '34'),\n",
       "  ('Max', {}, '774'),\n",
       "  ('Mean', {}, '208.45'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '11'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '15868'),\n",
       "  ('Mean', {}, '11657.5'),\n",
       "  ('Median', {}, '14312'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '74'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '7715'),\n",
       "  ('Mean', {}, '1928.58'),\n",
       "  ('Median', {}, '1464'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '17'),\n",
       "  ('Min', {}, '979'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1580.82'),\n",
       "  ('Median', {}, '1583')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510025.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '2'),\n",
       "  ('Mean', {}, '1.375'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.18182'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '62'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '207.613'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '56'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '9890'),\n",
       "  ('Mean', {}, '1849.16'),\n",
       "  ('Median', {}, '1113'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '11'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1524.82'),\n",
       "  ('Median', {}, '1605'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '915'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '1156'),\n",
       "  ('Median', {}, '1134'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '58'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '205.069'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '8'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '16751'),\n",
       "  ('Mean', {}, '12585.1'),\n",
       "  ('Median', {}, '14514'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '53'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9890'),\n",
       "  ('Mean', {}, '1921.55'),\n",
       "  ('Median', {}, '1273'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '1477'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1698'),\n",
       "  ('Median', {}, '1614')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510035.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '5'),\n",
       "  ('Mean', {}, '1.88889'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '6'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.76471'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '75'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '688'),\n",
       "  ('Mean', {}, '193.573'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '73'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9888'),\n",
       "  ('Mean', {}, '2020.36'),\n",
       "  ('Median', {}, '1464'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '17'),\n",
       "  ('Min', {}, '979'),\n",
       "  ('Max', {}, '1877'),\n",
       "  ('Mean', {}, '1495.94'),\n",
       "  ('Median', {}, '1522'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '17'),\n",
       "  ('Min', {}, '321'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '979.941'),\n",
       "  ('Median', {}, '1017'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '75'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '688'),\n",
       "  ('Mean', {}, '193.573'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '9461'),\n",
       "  ('Max', {}, '16787'),\n",
       "  ('Mean', {}, '13446.7'),\n",
       "  ('Median', {}, '14313'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '73'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9888'),\n",
       "  ('Mean', {}, '2020.36'),\n",
       "  ('Median', {}, '1464'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '17'),\n",
       "  ('Min', {}, '979'),\n",
       "  ('Max', {}, '1877'),\n",
       "  ('Mean', {}, '1495.94'),\n",
       "  ('Median', {}, '1522')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510045.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '8'),\n",
       "  ('Mean', {}, '2.4'),\n",
       "  ('Median', {}, '2'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '5.54167'),\n",
       "  ('Median', {}, '5'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '76'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '1772'),\n",
       "  ('Mean', {}, '221.921'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '70'),\n",
       "  ('Min', {}, '77'),\n",
       "  ('Max', {}, '13683'),\n",
       "  ('Mean', {}, '2227.69'),\n",
       "  ('Median', {}, '1420'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '24'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '2476'),\n",
       "  ('Mean', {}, '1684.92'),\n",
       "  ('Median', {}, '1723'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '22'),\n",
       "  ('Min', {}, '537'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '901.773'),\n",
       "  ('Median', {}, '912'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '72'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '1772'),\n",
       "  ('Mean', {}, '220.667'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '10'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '17186'),\n",
       "  ('Mean', {}, '12181.9'),\n",
       "  ('Median', {}, '14529'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '67'),\n",
       "  ('Min', {}, '77'),\n",
       "  ('Max', {}, '13683'),\n",
       "  ('Mean', {}, '2301.9'),\n",
       "  ('Median', {}, '1428'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '22'),\n",
       "  ('Min', {}, '821'),\n",
       "  ('Max', {}, '2476'),\n",
       "  ('Mean', {}, '1770.32'),\n",
       "  ('Median', {}, '1826')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510055.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '2'),\n",
       "  ('Mean', {}, '1.33333'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '8'),\n",
       "  ('Mean', {}, '6.125'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '40'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '205.225'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '35'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '5467'),\n",
       "  ('Mean', {}, '1581'),\n",
       "  ('Median', {}, '934'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '8'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1433.12'),\n",
       "  ('Median', {}, '1592'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '6'),\n",
       "  ('Min', {}, '1026'),\n",
       "  ('Max', {}, '1335'),\n",
       "  ('Mean', {}, '1134'),\n",
       "  ('Median', {}, '1134'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '36'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '200.861'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '6'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '14531'),\n",
       "  ('Mean', {}, '9614.83'),\n",
       "  ('Median', {}, '12048'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '32'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '5467'),\n",
       "  ('Mean', {}, '1675.75'),\n",
       "  ('Median', {}, '999'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '6'),\n",
       "  ('Min', {}, '1477'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1662.33'),\n",
       "  ('Median', {}, '1614')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510065.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '2'),\n",
       "  ('Mean', {}, '1.5'),\n",
       "  ('Median', {}, '2'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.16667'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '62'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '207.242'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '57'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '9890'),\n",
       "  ('Mean', {}, '1948.89'),\n",
       "  ('Median', {}, '1113'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '12'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1523.5'),\n",
       "  ('Median', {}, '1592'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '10'),\n",
       "  ('Min', {}, '669'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '1056.6'),\n",
       "  ('Median', {}, '1047'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '58'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '204.672'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '8'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '16789'),\n",
       "  ('Mean', {}, '12588.8'),\n",
       "  ('Median', {}, '14530'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '54'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9890'),\n",
       "  ('Mean', {}, '2025.48'),\n",
       "  ('Median', {}, '1464'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '10'),\n",
       "  ('Min', {}, '1477'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1679.1'),\n",
       "  ('Median', {}, '1614')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510075.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '8'),\n",
       "  ('Mean', {}, '2.4'),\n",
       "  ('Median', {}, '2'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '5.54167'),\n",
       "  ('Median', {}, '5'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '76'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '1772'),\n",
       "  ('Mean', {}, '221.921'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '70'),\n",
       "  ('Min', {}, '77'),\n",
       "  ('Max', {}, '13683'),\n",
       "  ('Mean', {}, '2216.06'),\n",
       "  ('Median', {}, '1401'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '24'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '2476'),\n",
       "  ('Mean', {}, '1684.92'),\n",
       "  ('Median', {}, '1723'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '22'),\n",
       "  ('Min', {}, '537'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '901.773'),\n",
       "  ('Median', {}, '912'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '72'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '1772'),\n",
       "  ('Mean', {}, '220.667'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '10'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '17186'),\n",
       "  ('Mean', {}, '12100.6'),\n",
       "  ('Median', {}, '14528'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '67'),\n",
       "  ('Min', {}, '77'),\n",
       "  ('Max', {}, '13683'),\n",
       "  ('Mean', {}, '2289.75'),\n",
       "  ('Median', {}, '1420'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '22'),\n",
       "  ('Min', {}, '821'),\n",
       "  ('Max', {}, '2476'),\n",
       "  ('Mean', {}, '1770.32'),\n",
       "  ('Median', {}, '1826')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510085.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '8'),\n",
       "  ('Mean', {}, '2.33333'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '6.17143'),\n",
       "  ('Median', {}, '7'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '117'),\n",
       "  ('Min', {}, '34'),\n",
       "  ('Max', {}, '1772'),\n",
       "  ('Mean', {}, '202.171'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '111'),\n",
       "  ('Min', {}, '77'),\n",
       "  ('Max', {}, '13683'),\n",
       "  ('Mean', {}, '2122.87'),\n",
       "  ('Median', {}, '1420'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '35'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '2476'),\n",
       "  ('Mean', {}, '1564.23'),\n",
       "  ('Median', {}, '1573'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '33'),\n",
       "  ('Min', {}, '321'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '884.364'),\n",
       "  ('Median', {}, '912'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '113'),\n",
       "  ('Min', {}, '34'),\n",
       "  ('Max', {}, '1772'),\n",
       "  ('Mean', {}, '200.673'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '15'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '17186'),\n",
       "  ('Mean', {}, '12135.1'),\n",
       "  ('Median', {}, '14159'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '108'),\n",
       "  ('Min', {}, '77'),\n",
       "  ('Max', {}, '13683'),\n",
       "  ('Mean', {}, '2166'),\n",
       "  ('Median', {}, '1464'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '33'),\n",
       "  ('Min', {}, '821'),\n",
       "  ('Max', {}, '2476'),\n",
       "  ('Mean', {}, '1613.85'),\n",
       "  ('Median', {}, '1582')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510095.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '8'),\n",
       "  ('Mean', {}, '2.5'),\n",
       "  ('Median', {}, '2'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '5.6'),\n",
       "  ('Median', {}, '5'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '76'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '1772'),\n",
       "  ('Mean', {}, '221.632'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '71'),\n",
       "  ('Min', {}, '77'),\n",
       "  ('Max', {}, '13683'),\n",
       "  ('Mean', {}, '2301.82'),\n",
       "  ('Median', {}, '1420'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '25'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '2476'),\n",
       "  ('Mean', {}, '1677.92'),\n",
       "  ('Median', {}, '1614'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '23'),\n",
       "  ('Min', {}, '537'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '883.174'),\n",
       "  ('Median', {}, '864'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '72'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '1772'),\n",
       "  ('Mean', {}, '220.361'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '10'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '17186'),\n",
       "  ('Mean', {}, '12180.8'),\n",
       "  ('Median', {}, '14519'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '68'),\n",
       "  ('Min', {}, '77'),\n",
       "  ('Max', {}, '13683'),\n",
       "  ('Mean', {}, '2378.21'),\n",
       "  ('Median', {}, '1464'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '23'),\n",
       "  ('Min', {}, '821'),\n",
       "  ('Max', {}, '2476'),\n",
       "  ('Mean', {}, '1759'),\n",
       "  ('Median', {}, '1723')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510105.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '8'),\n",
       "  ('Mean', {}, '3'),\n",
       "  ('Median', {}, '2'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '6'),\n",
       "  ('Median', {}, '6'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '73'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '1772'),\n",
       "  ('Mean', {}, '224.945'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '72'),\n",
       "  ('Min', {}, '77'),\n",
       "  ('Max', {}, '13683'),\n",
       "  ('Mean', {}, '2515.99'),\n",
       "  ('Median', {}, '1514'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '27'),\n",
       "  ('Min', {}, '821'),\n",
       "  ('Max', {}, '2476'),\n",
       "  ('Mean', {}, '1692.26'),\n",
       "  ('Median', {}, '1583'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '27'),\n",
       "  ('Min', {}, '321'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '841'),\n",
       "  ('Median', {}, '828'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '73'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '1772'),\n",
       "  ('Mean', {}, '224.945'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '3950'),\n",
       "  ('Max', {}, '17186'),\n",
       "  ('Mean', {}, '13325.7'),\n",
       "  ('Median', {}, '14341'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '72'),\n",
       "  ('Min', {}, '77'),\n",
       "  ('Max', {}, '13683'),\n",
       "  ('Mean', {}, '2515.99'),\n",
       "  ('Median', {}, '1514'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '27'),\n",
       "  ('Min', {}, '821'),\n",
       "  ('Max', {}, '2476'),\n",
       "  ('Mean', {}, '1692.26'),\n",
       "  ('Median', {}, '1583')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510125.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '8'),\n",
       "  ('Mean', {}, '2.4'),\n",
       "  ('Median', {}, '2'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '5.45833'),\n",
       "  ('Median', {}, '5'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '74'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '1772'),\n",
       "  ('Mean', {}, '226.311'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '68'),\n",
       "  ('Min', {}, '77'),\n",
       "  ('Max', {}, '13683'),\n",
       "  ('Mean', {}, '2272.18'),\n",
       "  ('Median', {}, '1428'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '24'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '2476'),\n",
       "  ('Mean', {}, '1684.92'),\n",
       "  ('Median', {}, '1723'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '22'),\n",
       "  ('Min', {}, '537'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '901.773'),\n",
       "  ('Median', {}, '912'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '70'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '1772'),\n",
       "  ('Mean', {}, '225.271'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '10'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '17186'),\n",
       "  ('Mean', {}, '12027.2'),\n",
       "  ('Median', {}, '14529'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '65'),\n",
       "  ('Min', {}, '77'),\n",
       "  ('Max', {}, '13683'),\n",
       "  ('Mean', {}, '2350.72'),\n",
       "  ('Median', {}, '1464'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '22'),\n",
       "  ('Min', {}, '821'),\n",
       "  ('Max', {}, '2476'),\n",
       "  ('Mean', {}, '1770.32'),\n",
       "  ('Median', {}, '1826')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510135.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '2'),\n",
       "  ('Mean', {}, '1.4'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.28571'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '78'),\n",
       "  ('Min', {}, '34'),\n",
       "  ('Max', {}, '774'),\n",
       "  ('Mean', {}, '206.141'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '71'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '9889'),\n",
       "  ('Mean', {}, '1863.48'),\n",
       "  ('Median', {}, '1113'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '14'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1536.86'),\n",
       "  ('Median', {}, '1602'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '12'),\n",
       "  ('Min', {}, '669'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '1050.75'),\n",
       "  ('Median', {}, '1047'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '74'),\n",
       "  ('Min', {}, '34'),\n",
       "  ('Max', {}, '774'),\n",
       "  ('Mean', {}, '204.068'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '10'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '16769'),\n",
       "  ('Mean', {}, '12516.1'),\n",
       "  ('Median', {}, '14535'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '68'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9889'),\n",
       "  ('Mean', {}, '1920.53'),\n",
       "  ('Median', {}, '1464'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '12'),\n",
       "  ('Min', {}, '1477'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1668.75'),\n",
       "  ('Median', {}, '1614')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510145.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '2'),\n",
       "  ('Mean', {}, '1.375'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.18182'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '62'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '207.613'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '56'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '9892'),\n",
       "  ('Mean', {}, '1833.68'),\n",
       "  ('Median', {}, '1113'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '11'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1524.82'),\n",
       "  ('Median', {}, '1605'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '915'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '1156'),\n",
       "  ('Median', {}, '1134'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '58'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '205.069'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '8'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '16768'),\n",
       "  ('Mean', {}, '12476.5'),\n",
       "  ('Median', {}, '14514'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '53'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9892'),\n",
       "  ('Mean', {}, '1905.19'),\n",
       "  ('Median', {}, '1273'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '1477'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1698'),\n",
       "  ('Median', {}, '1614')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510155.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '5'),\n",
       "  ('Mean', {}, '1.72727'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.26316'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '87'),\n",
       "  ('Min', {}, '34'),\n",
       "  ('Max', {}, '774'),\n",
       "  ('Mean', {}, '209.793'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '82'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '9888'),\n",
       "  ('Mean', {}, '2009.87'),\n",
       "  ('Median', {}, '1464'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '19'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1492.89'),\n",
       "  ('Median', {}, '1581'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '17'),\n",
       "  ('Min', {}, '321'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '962.471'),\n",
       "  ('Median', {}, '1017'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '83'),\n",
       "  ('Min', {}, '34'),\n",
       "  ('Max', {}, '774'),\n",
       "  ('Mean', {}, '208.12'),\n",
       "  ('Median', {}, '104'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '11'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '16787'),\n",
       "  ('Mean', {}, '12692.8'),\n",
       "  ('Median', {}, '14342'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '79'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9888'),\n",
       "  ('Mean', {}, '2064.53'),\n",
       "  ('Median', {}, '1476'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '17'),\n",
       "  ('Min', {}, '979'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1580.82'),\n",
       "  ('Median', {}, '1583')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510165.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '2'),\n",
       "  ('Mean', {}, '1.5'),\n",
       "  ('Median', {}, '2'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '3'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.16667'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '62'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '207.242'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '57'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '9889'),\n",
       "  ('Mean', {}, '1945.98'),\n",
       "  ('Median', {}, '1113'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '12'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1523.5'),\n",
       "  ('Median', {}, '1592'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '10'),\n",
       "  ('Min', {}, '669'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '1056.6'),\n",
       "  ('Median', {}, '1047'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '58'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '686'),\n",
       "  ('Mean', {}, '204.672'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '4'),\n",
       "  ('Min', {}, '43'),\n",
       "  ('Max', {}, '470'),\n",
       "  ('Mean', {}, '244.5'),\n",
       "  ('Median', {}, '268'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '8'),\n",
       "  ('Min', {}, '1643'),\n",
       "  ('Max', {}, '16765'),\n",
       "  ('Mean', {}, '12568'),\n",
       "  ('Median', {}, '14527'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '54'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '9889'),\n",
       "  ('Mean', {}, '2022.41'),\n",
       "  ('Median', {}, '1464'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '3'),\n",
       "  ('Min', {}, '84'),\n",
       "  ('Max', {}, '849'),\n",
       "  ('Mean', {}, '570.333'),\n",
       "  ('Median', {}, '778'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '710'),\n",
       "  ('Max', {}, '781'),\n",
       "  ('Mean', {}, '745.5'),\n",
       "  ('Median', {}, '781'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '10'),\n",
       "  ('Min', {}, '1477'),\n",
       "  ('Max', {}, '1986'),\n",
       "  ('Mean', {}, '1679.1'),\n",
       "  ('Median', {}, '1614')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000510175.1', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '5'),\n",
       "  ('Mean', {}, '1.88889'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '5'),\n",
       "  ('Max', {}, '9'),\n",
       "  ('Mean', {}, '7.41176'),\n",
       "  ('Median', {}, '8'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '73'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '694'),\n",
       "  ('Mean', {}, '198.123'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '69'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '7716'),\n",
       "  ('Mean', {}, '1891.99'),\n",
       "  ('Median', {}, '1476'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '17'),\n",
       "  ('Min', {}, '979'),\n",
       "  ('Max', {}, '1877'),\n",
       "  ('Mean', {}, '1524.71'),\n",
       "  ('Median', {}, '1573'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '17'),\n",
       "  ('Min', {}, '321'),\n",
       "  ('Max', {}, '1368'),\n",
       "  ('Mean', {}, '979.941'),\n",
       "  ('Median', {}, '1017'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '73'),\n",
       "  ('Min', {}, '36'),\n",
       "  ('Max', {}, '694'),\n",
       "  ('Mean', {}, '198.123'),\n",
       "  ('Median', {}, '105'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '9'),\n",
       "  ('Min', {}, '8713'),\n",
       "  ('Max', {}, '15021'),\n",
       "  ('Mean', {}, '12624.4'),\n",
       "  ('Median', {}, '14015'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '69'),\n",
       "  ('Min', {}, '98'),\n",
       "  ('Max', {}, '7716'),\n",
       "  ('Mean', {}, '1891.99'),\n",
       "  ('Median', {}, '1476'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '17'),\n",
       "  ('Min', {}, '979'),\n",
       "  ('Max', {}, '1877'),\n",
       "  ('Mean', {}, '1524.71'),\n",
       "  ('Median', {}, '1573')],\n",
       " [('AssemblyStats',\n",
       "   {'Accession': 'GCF_000005045.25', 'Type': 'assembly unit'},\n",
       "   '\\n    '),\n",
       "  ('TranscriptsPerGene', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '33'),\n",
       "  ('Mean', {}, '2.14199'),\n",
       "  ('Median', {}, '1'),\n",
       "  ('ExonsPerTranscript', {}, '\\n      '),\n",
       "  ('Min', {}, '1'),\n",
       "  ('Max', {}, '93'),\n",
       "  ('Mean', {}, '7.87595'),\n",
       "  ('Median', {}, '5'),\n",
       "  ('LengthStats', {'category': 'All exons'}, '\\n      '),\n",
       "  ('Count', {}, '8186'),\n",
       "  ('Min', {}, '7'),\n",
       "  ('Max', {}, '14433'),\n",
       "  ('Mean', {}, '366.997'),\n",
       "  ('Median', {}, '138'),\n",
       "  ('LengthStats', {'category': 'All introns'}, '\\n      '),\n",
       "  ('Count', {}, '7161'),\n",
       "  ('Min', {}, '30'),\n",
       "  ('Max', {}, '334517'),\n",
       "  ('Mean', {}, '4315.23'),\n",
       "  ('Median', {}, '1113'),\n",
       "  ('LengthStats', {'category': 'All transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '2140'),\n",
       "  ('Min', {}, '17'),\n",
       "  ('Max', {}, '21046'),\n",
       "  ('Mean', {}, '2885.79'),\n",
       "  ('Median', {}, '2234'),\n",
       "  ('LengthStats', {'category': 'CDSs'}, '\\n      '),\n",
       "  ('Count', {}, '1382'),\n",
       "  ('Min', {}, '132'),\n",
       "  ('Max', {}, '16965'),\n",
       "  ('Mean', {}, '1728.69'),\n",
       "  ('Median', {}, '1284'),\n",
       "  ('LengthStats', {'category': 'Exons in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '6730'),\n",
       "  ('Min', {}, '7'),\n",
       "  ('Max', {}, '14433'),\n",
       "  ('Mean', {}, '339.759'),\n",
       "  ('Median', {}, '133'),\n",
       "  ('LengthStats', {'category': 'Exons in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '2042'),\n",
       "  ('Min', {}, '9'),\n",
       "  ('Max', {}, '10558'),\n",
       "  ('Mean', {}, '427.424'),\n",
       "  ('Median', {}, '153'),\n",
       "  ('LengthStats', {'category': 'Genes'}, '\\n      '),\n",
       "  ('Count', {}, '996'),\n",
       "  ('Min', {}, '51'),\n",
       "  ('Max', {}, '650936'),\n",
       "  ('Mean', {}, '25456.7'),\n",
       "  ('Median', {}, '8349'),\n",
       "  ('LengthStats', {'category': 'Introns in coding models'}, '\\n      '),\n",
       "  ('Count', {}, '6034'),\n",
       "  ('Min', {}, '30'),\n",
       "  ('Max', {}, '334517'),\n",
       "  ('Mean', {}, '3872.72'),\n",
       "  ('Median', {}, '1015'),\n",
       "  ('LengthStats', {'category': 'Introns in non-coding models'}, '\\n      '),\n",
       "  ('Count', {}, '1642'),\n",
       "  ('Min', {}, '38'),\n",
       "  ('Max', {}, '112483'),\n",
       "  ('Mean', {}, '5694.09'),\n",
       "  ('Median', {}, '1571'),\n",
       "  ('LengthStats', {'category': 'Single-exon transcripts'}, '\\n      '),\n",
       "  ('Count', {}, '118'),\n",
       "  ('Min', {}, '452'),\n",
       "  ('Max', {}, '18485'),\n",
       "  ('Mean', {}, '3601.76'),\n",
       "  ('Median', {}, '2485'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '104'),\n",
       "  ('Min', {}, '452'),\n",
       "  ('Max', {}, '18485'),\n",
       "  ('Mean', {}, '3517.5'),\n",
       "  ('Median', {}, '2462'),\n",
       "  ('LengthStats',\n",
       "   {'category': 'Single-exon transcripts in non-coding models'},\n",
       "   '\\n      '),\n",
       "  ('Count', {}, '14'),\n",
       "  ('Min', {}, '829'),\n",
       "  ('Max', {}, '12682'),\n",
       "  ('Mean', {}, '4227.71'),\n",
       "  ('Median', {}, '2879'),\n",
       "  ('LengthStats', {'category': 'guide_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '266'),\n",
       "  ('Max', {}, '278'),\n",
       "  ('Mean', {}, '272'),\n",
       "  ('Median', {}, '278'),\n",
       "  ('LengthStats', {'category': 'lncRNA'}, '\\n      '),\n",
       "  ('Count', {}, '475'),\n",
       "  ('Min', {}, '137'),\n",
       "  ('Max', {}, '13054'),\n",
       "  ('Mean', {}, '1834.45'),\n",
       "  ('Median', {}, '1308'),\n",
       "  ('LengthStats', {'category': 'mRNA'}, '\\n      '),\n",
       "  ('Count', {}, '1382'),\n",
       "  ('Min', {}, '234'),\n",
       "  ('Max', {}, '21046'),\n",
       "  ('Mean', {}, '3470.96'),\n",
       "  ('Median', {}, '2585'),\n",
       "  ('LengthStats', {'category': 'miRNA'}, '\\n      '),\n",
       "  ('Count', {}, '83'),\n",
       "  ('Min', {}, '17'),\n",
       "  ('Max', {}, '26'),\n",
       "  ('Mean', {}, '21.6386'),\n",
       "  ('Median', {}, '22'),\n",
       "  ('LengthStats', {'category': 'misc_RNA'}, '\\n      '),\n",
       "  ('Count', {}, '161'),\n",
       "  ('Min', {}, '173'),\n",
       "  ('Max', {}, '12682'),\n",
       "  ('Mean', {}, '2863.29'),\n",
       "  ('Median', {}, '2760'),\n",
       "  ('LengthStats', {'category': 'premsg'}, '\\n      '),\n",
       "  ('Count', {}, '57'),\n",
       "  ('Min', {}, '51'),\n",
       "  ('Max', {}, '138'),\n",
       "  ('Mean', {}, '83.7368'),\n",
       "  ('Median', {}, '80'),\n",
       "  ('LengthStats', {'category': 'rRNA'}, '\\n      '),\n",
       "  ('Count', {}, '25'),\n",
       "  ('Min', {}, '121'),\n",
       "  ('Max', {}, '13351'),\n",
       "  ('Mean', {}, '1715.6'),\n",
       "  ('Median', {}, '121'),\n",
       "  ('LengthStats', {'category': 'snoRNA'}, '\\n      '),\n",
       "  ('Count', {}, '10'),\n",
       "  ('Min', {}, '59'),\n",
       "  ('Max', {}, '197'),\n",
       "  ('Mean', {}, '98.2'),\n",
       "  ('Median', {}, '75'),\n",
       "  ('LengthStats', {'category': 'tRNA'}, '\\n      '),\n",
       "  ('Count', {}, '2'),\n",
       "  ('Min', {}, '82'),\n",
       "  ('Max', {}, '82'),\n",
       "  ('Mean', {}, '82'),\n",
       "  ('Median', {}, '82')]]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "feature_stats_collection = []\n",
    "for i in root[3]:\n",
    "    tags = []\n",
    "    text = []\n",
    "    attributes = []\n",
    "    for element in i.iter():\n",
    "        tags.append(element.tag)\n",
    "        text.append(element.text)\n",
    "        attributes.append(element.attrib)\n",
    "    thing = list(zip(tags,attributes,text))\n",
    "    feature_stats_collection.append(thing)\n",
    "print(len(feature_stats_collection))\n",
    "feature_stats_collection"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'GCF_000001405.39'"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_accession = full_assembly_collection[1][0][2]\n",
    "human_accession"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "From this 'human_accession', at the ncbi online database located at https://www.ncbi.nlm.nih.gov/assembly/GCF_000001405.39/, the full sequence report can be pulled down into a text file seen below. This information contains information about genetic information located on chromosomes, in other words, it's a type of dna data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "import csv\n",
    "txt_file_dna = r\"GCF_000001405.39_GRCh38.p13_assembly_report.txt\"\n",
    "csv_file_dna = r\"GCF_000001405.39_GRCh38.p13_assembly_report.csv\"\n",
    "in_txt_dna = csv.reader(open(txt_file_dna, \"r\", encoding='utf-8'), delimiter = '\\t')\n",
    "out_csv_dna = csv.writer(open(csv_file_dna, 'w'))\n",
    "out_csv_dna.writerows(in_txt_dna)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Assembly name:  GRCh38.p13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Description:    Genome Reference Consortium Hu...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Organism name:  Homo sapiens (human)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Taxid:          9606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>BioProject:     PRJNA31257</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>Submitter:      Genome Reference Consortium</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>Date:           2019-02-28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>Assembly type:  haploid-with-alt-loci</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>Release type:   patch</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>Assembly level: Chromosome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>Genome representation: full</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>RefSeq category: Reference Genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>GenBank assembly accession: GCA_000001405.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>RefSeq assembly accession: GCF_000001405.39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>RefSeq assembly and GenBank assemblies identic...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>Assembly-Units:</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           Assembly name:  GRCh38.p13\n",
       "1   Description:    Genome Reference Consortium Hu...\n",
       "3                Organism name:  Homo sapiens (human)\n",
       "5                                Taxid:          9606\n",
       "7                          BioProject:     PRJNA31257\n",
       "9         Submitter:      Genome Reference Consortium\n",
       "11                         Date:           2019-02-28\n",
       "13              Assembly type:  haploid-with-alt-loci\n",
       "15                              Release type:   patch\n",
       "17                         Assembly level: Chromosome\n",
       "19                        Genome representation: full\n",
       "21                  RefSeq category: Reference Genome\n",
       "23       GenBank assembly accession: GCA_000001405.28\n",
       "25        RefSeq assembly accession: GCF_000001405.39\n",
       "27  RefSeq assembly and GenBank assemblies identic...\n",
       "31                                    Assembly-Units:"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "to_skip_info = []\n",
    "for i in range(34,1406):\n",
    "    to_skip_info.append(i)\n",
    "information_about_dna_data = pd.read_csv('GCF_000001405.39_GRCh38.p13_assembly_report_fixed.csv', skiprows=(to_skip_info), header=0)\n",
    "pd.DataFrame(information_about_dna_data['Assembly name:  GRCh38.p13'].dropna(axis=0))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "dna_data = pd.read_csv('GCF_000001405.39_GRCh38.p13_assembly_report_fixed.csv', header=124)\n",
    "dna_data = dna_data.dropna(axis=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th># Sequence-Name</th>\n",
       "      <th>Sequence-Role</th>\n",
       "      <th>Assigned-Molecule</th>\n",
       "      <th>Assigned-Molecule-Location/Type</th>\n",
       "      <th>GenBank-Accn</th>\n",
       "      <th>Relationship</th>\n",
       "      <th>RefSeq-Accn</th>\n",
       "      <th>Assembly-Unit</th>\n",
       "      <th>Sequence-Length</th>\n",
       "      <th>UCSC-style-name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>781</td>\n",
       "      <td>HSCHR2_2_CTG7_2</td>\n",
       "      <td>alt-scaffold</td>\n",
       "      <td>2</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>GL582966.2</td>\n",
       "      <td>=</td>\n",
       "      <td>NW_003571033.2</td>\n",
       "      <td>ALT_REF_LOCI_1</td>\n",
       "      <td>96131.0</td>\n",
       "      <td>chr2_GL582966v2_alt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1151</td>\n",
       "      <td>HSCHR11_2_CTG1</td>\n",
       "      <td>alt-scaffold</td>\n",
       "      <td>11</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>KI270902.1</td>\n",
       "      <td>=</td>\n",
       "      <td>NT_187656.1</td>\n",
       "      <td>ALT_REF_LOCI_2</td>\n",
       "      <td>106711.0</td>\n",
       "      <td>chr11_KI270902v1_alt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1103</td>\n",
       "      <td>HSCHR21_4_CTG1_1</td>\n",
       "      <td>alt-scaffold</td>\n",
       "      <td>21</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>GL383581.2</td>\n",
       "      <td>=</td>\n",
       "      <td>NW_003315970.2</td>\n",
       "      <td>ALT_REF_LOCI_1</td>\n",
       "      <td>116689.0</td>\n",
       "      <td>chr21_GL383581v2_alt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>119</td>\n",
       "      <td>HSCHR22_UNLOCALIZED_CTG4</td>\n",
       "      <td>unlocalized-scaffold</td>\n",
       "      <td>22</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>KI270734.1</td>\n",
       "      <td>=</td>\n",
       "      <td>NT_187389.1</td>\n",
       "      <td>Primary Assembly</td>\n",
       "      <td>165050.0</td>\n",
       "      <td>chr22_KI270734v1_random</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>257</td>\n",
       "      <td>HSCHRUN_RANDOM_116</td>\n",
       "      <td>unplaced-scaffold</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>KI270418.1</td>\n",
       "      <td>=</td>\n",
       "      <td>NT_187412.1</td>\n",
       "      <td>Primary Assembly</td>\n",
       "      <td>2145.0</td>\n",
       "      <td>chrUn_KI270418v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>397</td>\n",
       "      <td>HG2002_PATCH</td>\n",
       "      <td>fix-patch</td>\n",
       "      <td>1</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>KZ208906.1</td>\n",
       "      <td>=</td>\n",
       "      <td>NW_018654708.1</td>\n",
       "      <td>PATCHES</td>\n",
       "      <td>330031.0</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>303</td>\n",
       "      <td>HSCHRUN_RANDOM_135</td>\n",
       "      <td>unplaced-scaffold</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>KI270516.1</td>\n",
       "      <td>=</td>\n",
       "      <td>NT_187431.1</td>\n",
       "      <td>Primary Assembly</td>\n",
       "      <td>1300.0</td>\n",
       "      <td>chrUn_KI270516v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>61</td>\n",
       "      <td>HSCHR1_CTG7_UNLOCALIZED</td>\n",
       "      <td>unlocalized-scaffold</td>\n",
       "      <td>1</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>KI270712.1</td>\n",
       "      <td>=</td>\n",
       "      <td>NT_187367.1</td>\n",
       "      <td>Primary Assembly</td>\n",
       "      <td>176043.0</td>\n",
       "      <td>chr1_KI270712v1_random</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>321</td>\n",
       "      <td>HSCHRUN_RANDOM_147</td>\n",
       "      <td>unplaced-scaffold</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>KI270538.1</td>\n",
       "      <td>=</td>\n",
       "      <td>NT_187443.1</td>\n",
       "      <td>Primary Assembly</td>\n",
       "      <td>91309.0</td>\n",
       "      <td>chrUn_KI270538v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>823</td>\n",
       "      <td>HSCHR4_1_CTG9</td>\n",
       "      <td>alt-scaffold</td>\n",
       "      <td>4</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>GL000257.2</td>\n",
       "      <td>=</td>\n",
       "      <td>NT_167250.2</td>\n",
       "      <td>ALT_REF_LOCI_1</td>\n",
       "      <td>586476.0</td>\n",
       "      <td>chr4_GL000257v2_alt</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               # Sequence-Name         Sequence-Role Assigned-Molecule  \\\n",
       "781            HSCHR2_2_CTG7_2          alt-scaffold                 2   \n",
       "1151            HSCHR11_2_CTG1          alt-scaffold                11   \n",
       "1103          HSCHR21_4_CTG1_1          alt-scaffold                21   \n",
       "119   HSCHR22_UNLOCALIZED_CTG4  unlocalized-scaffold                22   \n",
       "257         HSCHRUN_RANDOM_116     unplaced-scaffold                na   \n",
       "397               HG2002_PATCH             fix-patch                 1   \n",
       "303         HSCHRUN_RANDOM_135     unplaced-scaffold                na   \n",
       "61     HSCHR1_CTG7_UNLOCALIZED  unlocalized-scaffold                 1   \n",
       "321         HSCHRUN_RANDOM_147     unplaced-scaffold                na   \n",
       "823              HSCHR4_1_CTG9          alt-scaffold                 4   \n",
       "\n",
       "     Assigned-Molecule-Location/Type GenBank-Accn Relationship  \\\n",
       "781                       Chromosome   GL582966.2            =   \n",
       "1151                      Chromosome   KI270902.1            =   \n",
       "1103                      Chromosome   GL383581.2            =   \n",
       "119                       Chromosome   KI270734.1            =   \n",
       "257                               na   KI270418.1            =   \n",
       "397                       Chromosome   KZ208906.1            =   \n",
       "303                               na   KI270516.1            =   \n",
       "61                        Chromosome   KI270712.1            =   \n",
       "321                               na   KI270538.1            =   \n",
       "823                       Chromosome   GL000257.2            =   \n",
       "\n",
       "         RefSeq-Accn     Assembly-Unit  Sequence-Length  \\\n",
       "781   NW_003571033.2    ALT_REF_LOCI_1          96131.0   \n",
       "1151     NT_187656.1    ALT_REF_LOCI_2         106711.0   \n",
       "1103  NW_003315970.2    ALT_REF_LOCI_1         116689.0   \n",
       "119      NT_187389.1  Primary Assembly         165050.0   \n",
       "257      NT_187412.1  Primary Assembly           2145.0   \n",
       "397   NW_018654708.1           PATCHES         330031.0   \n",
       "303      NT_187431.1  Primary Assembly           1300.0   \n",
       "61       NT_187367.1  Primary Assembly         176043.0   \n",
       "321      NT_187443.1  Primary Assembly          91309.0   \n",
       "823      NT_167250.2    ALT_REF_LOCI_1         586476.0   \n",
       "\n",
       "              UCSC-style-name  \n",
       "781       chr2_GL582966v2_alt  \n",
       "1151     chr11_KI270902v1_alt  \n",
       "1103     chr21_GL383581v2_alt  \n",
       "119   chr22_KI270734v1_random  \n",
       "257          chrUn_KI270418v1  \n",
       "397                        na  \n",
       "303          chrUn_KI270516v1  \n",
       "61     chr1_KI270712v1_random  \n",
       "321          chrUn_KI270538v1  \n",
       "823       chr4_GL000257v2_alt  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dna_data.sample(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "Int64Index: 640 entries, 1 to 1279\n",
      "Data columns (total 10 columns):\n",
      "# Sequence-Name                    640 non-null object\n",
      "Sequence-Role                      640 non-null object\n",
      "Assigned-Molecule                  640 non-null object\n",
      "Assigned-Molecule-Location/Type    640 non-null object\n",
      "GenBank-Accn                       640 non-null object\n",
      "Relationship                       640 non-null object\n",
      "RefSeq-Accn                        640 non-null object\n",
      "Assembly-Unit                      640 non-null object\n",
      "Sequence-Length                    640 non-null float64\n",
      "UCSC-style-name                    640 non-null object\n",
      "dtypes: float64(1), object(9)\n",
      "memory usage: 55.0+ KB\n"
     ]
    }
   ],
   "source": [
    "dna_data.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "640"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(dna_data)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "From the same https://www.ncbi.nlm.nih.gov/assembly/GCF_000001405.39/ page, under genome information, https://www.ncbi.nlm.nih.gov/genome/?term=txid9606[orgn], and in the Genome Assembly and Annotation report, https://www.ncbi.nlm.nih.gov/genome/genomes/51?, in the protein section at https://www.ncbi.nlm.nih.gov/genome/proteins/51?genome_assembly_id=582967, a comprehensive report of human proteins can be found. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "txt_file_protein = r\"ProteinTable51_582967.txt\"\n",
    "csv_file_protein = r\"ProteinTable51_582967.csv\"\n",
    "in_txt_protein = csv.reader(open(txt_file_protein, \"r\"), delimiter = '\\t')\n",
    "out_csv_protein = csv.writer(open(csv_file_protein, 'w'))\n",
    "out_csv_protein.writerows(in_txt_protein)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Replicon Name</th>\n",
       "      <th>Replicon Accession</th>\n",
       "      <th>Start</th>\n",
       "      <th>Stop</th>\n",
       "      <th>Strand</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>Locus</th>\n",
       "      <th>Protein product</th>\n",
       "      <th>Length</th>\n",
       "      <th>Protein name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>112076</td>\n",
       "      <td>21</td>\n",
       "      <td>NC_000021.9</td>\n",
       "      <td>33718829</td>\n",
       "      <td>33836988</td>\n",
       "      <td>+</td>\n",
       "      <td>6453</td>\n",
       "      <td>ITSN1</td>\n",
       "      <td>NP_001001132.1</td>\n",
       "      <td>1220</td>\n",
       "      <td>intersectin-1 isoform ITSN-s</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>44605</td>\n",
       "      <td>6</td>\n",
       "      <td>NC_000006.12</td>\n",
       "      <td>39821984</td>\n",
       "      <td>39902037</td>\n",
       "      <td>+</td>\n",
       "      <td>23500</td>\n",
       "      <td>DAAM2</td>\n",
       "      <td>XP_006715102.3</td>\n",
       "      <td>1176</td>\n",
       "      <td>disheveled-associated activator of morphogenes...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>76489</td>\n",
       "      <td>12</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>49089859</td>\n",
       "      <td>49094512</td>\n",
       "      <td>-</td>\n",
       "      <td>50846</td>\n",
       "      <td>DHH</td>\n",
       "      <td>NP_066382.1</td>\n",
       "      <td>396</td>\n",
       "      <td>desert hedgehog protein preproprotein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22994</td>\n",
       "      <td>2</td>\n",
       "      <td>NC_000002.12</td>\n",
       "      <td>169529825</td>\n",
       "      <td>169558923</td>\n",
       "      <td>-</td>\n",
       "      <td>79675</td>\n",
       "      <td>FASTKD1</td>\n",
       "      <td>XP_006712814.1</td>\n",
       "      <td>527</td>\n",
       "      <td>FAST kinase domain-containing protein 1, mitoc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>91047</td>\n",
       "      <td>16</td>\n",
       "      <td>NC_000016.10</td>\n",
       "      <td>21161183</td>\n",
       "      <td>21179556</td>\n",
       "      <td>+</td>\n",
       "      <td>57146</td>\n",
       "      <td>TMEM159</td>\n",
       "      <td>NP_001288700.1</td>\n",
       "      <td>161</td>\n",
       "      <td>promethin isoform 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>65433</td>\n",
       "      <td>10</td>\n",
       "      <td>NC_000010.11</td>\n",
       "      <td>86876019</td>\n",
       "      <td>86923719</td>\n",
       "      <td>+</td>\n",
       "      <td>657</td>\n",
       "      <td>BMPR1A</td>\n",
       "      <td>NP_004320.2</td>\n",
       "      <td>532</td>\n",
       "      <td>bone morphogenetic protein receptor type-1A pr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>90729</td>\n",
       "      <td>16</td>\n",
       "      <td>NC_000016.10</td>\n",
       "      <td>14997777</td>\n",
       "      <td>15036275</td>\n",
       "      <td>+</td>\n",
       "      <td>23042</td>\n",
       "      <td>PDXDC1</td>\n",
       "      <td>XP_016878553.1</td>\n",
       "      <td>746</td>\n",
       "      <td>pyridoxal-dependent decarboxylase domain-conta...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10289</td>\n",
       "      <td>1</td>\n",
       "      <td>NC_000001.11</td>\n",
       "      <td>51288157</td>\n",
       "      <td>51322161</td>\n",
       "      <td>-</td>\n",
       "      <td>22996</td>\n",
       "      <td>TTC39A</td>\n",
       "      <td>NP_001073963.1</td>\n",
       "      <td>578</td>\n",
       "      <td>tetratricopeptide repeat protein 39A isoform 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>117298</td>\n",
       "      <td>X</td>\n",
       "      <td>NC_000023.11</td>\n",
       "      <td>75293494</td>\n",
       "      <td>75303511</td>\n",
       "      <td>+</td>\n",
       "      <td>139596</td>\n",
       "      <td>UPRT</td>\n",
       "      <td>NP_001350750.1</td>\n",
       "      <td>173</td>\n",
       "      <td>uracil phosphoribosyltransferase homolog isofo...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>52846</td>\n",
       "      <td>7</td>\n",
       "      <td>NC_000007.14</td>\n",
       "      <td>143356550</td>\n",
       "      <td>143359010</td>\n",
       "      <td>-</td>\n",
       "      <td>9715</td>\n",
       "      <td>FAM131B</td>\n",
       "      <td>NP_001265226.1</td>\n",
       "      <td>266</td>\n",
       "      <td>protein FAM131B isoform c</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       #Replicon Name Replicon Accession      Start       Stop Strand  GeneID  \\\n",
       "112076             21        NC_000021.9   33718829   33836988      +    6453   \n",
       "44605               6       NC_000006.12   39821984   39902037      +   23500   \n",
       "76489              12       NC_000012.12   49089859   49094512      -   50846   \n",
       "22994               2       NC_000002.12  169529825  169558923      -   79675   \n",
       "91047              16       NC_000016.10   21161183   21179556      +   57146   \n",
       "65433              10       NC_000010.11   86876019   86923719      +     657   \n",
       "90729              16       NC_000016.10   14997777   15036275      +   23042   \n",
       "10289               1       NC_000001.11   51288157   51322161      -   22996   \n",
       "117298              X       NC_000023.11   75293494   75303511      +  139596   \n",
       "52846               7       NC_000007.14  143356550  143359010      -    9715   \n",
       "\n",
       "          Locus Protein product  Length  \\\n",
       "112076    ITSN1  NP_001001132.1    1220   \n",
       "44605     DAAM2  XP_006715102.3    1176   \n",
       "76489       DHH     NP_066382.1     396   \n",
       "22994   FASTKD1  XP_006712814.1     527   \n",
       "91047   TMEM159  NP_001288700.1     161   \n",
       "65433    BMPR1A     NP_004320.2     532   \n",
       "90729    PDXDC1  XP_016878553.1     746   \n",
       "10289    TTC39A  NP_001073963.1     578   \n",
       "117298     UPRT  NP_001350750.1     173   \n",
       "52846   FAM131B  NP_001265226.1     266   \n",
       "\n",
       "                                             Protein name  \n",
       "112076                       intersectin-1 isoform ITSN-s  \n",
       "44605   disheveled-associated activator of morphogenes...  \n",
       "76489               desert hedgehog protein preproprotein  \n",
       "22994   FAST kinase domain-containing protein 1, mitoc...  \n",
       "91047                                 promethin isoform 2  \n",
       "65433   bone morphogenetic protein receptor type-1A pr...  \n",
       "90729   pyridoxal-dependent decarboxylase domain-conta...  \n",
       "10289      tetratricopeptide repeat protein 39A isoform 2  \n",
       "117298  uracil phosphoribosyltransferase homolog isofo...  \n",
       "52846                           protein FAM131B isoform c  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "protein_data = pd.read_csv('ProteinTable51_582967.csv')\n",
    "protein_data.sample(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 119489 entries, 0 to 119488\n",
      "Data columns (total 10 columns):\n",
      "#Replicon Name        119489 non-null object\n",
      "Replicon Accession    119489 non-null object\n",
      "Start                 119489 non-null int64\n",
      "Stop                  119489 non-null int64\n",
      "Strand                119489 non-null object\n",
      "GeneID                119489 non-null int64\n",
      "Locus                 119489 non-null object\n",
      "Protein product       119489 non-null object\n",
      "Length                119489 non-null int64\n",
      "Protein name          119489 non-null object\n",
      "dtypes: int64(4), object(6)\n",
      "memory usage: 9.1+ MB\n"
     ]
    }
   ],
   "source": [
    "protein_data.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "119489"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(protein_data)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The protein and dna information is related through the RefSeq-Accn column on the dna_data table and the Replicon Accession column on the protein_data table. The two cells below show that many types of protein information is stored in chromosomes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th># Sequence-Name</th>\n",
       "      <th>Sequence-Role</th>\n",
       "      <th>Assigned-Molecule</th>\n",
       "      <th>Assigned-Molecule-Location/Type</th>\n",
       "      <th>GenBank-Accn</th>\n",
       "      <th>Relationship</th>\n",
       "      <th>RefSeq-Accn</th>\n",
       "      <th>Assembly-Unit</th>\n",
       "      <th>Sequence-Length</th>\n",
       "      <th>UCSC-style-name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>12</td>\n",
       "      <td>assembled-molecule</td>\n",
       "      <td>12</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>CM000674.2</td>\n",
       "      <td>=</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>Primary Assembly</td>\n",
       "      <td>133275309.0</td>\n",
       "      <td>chr12</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   # Sequence-Name       Sequence-Role Assigned-Molecule  \\\n",
       "23              12  assembled-molecule                12   \n",
       "\n",
       "   Assigned-Molecule-Location/Type GenBank-Accn Relationship   RefSeq-Accn  \\\n",
       "23                      Chromosome   CM000674.2            =  NC_000012.12   \n",
       "\n",
       "       Assembly-Unit  Sequence-Length UCSC-style-name  \n",
       "23  Primary Assembly      133275309.0           chr12  "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dna_data[dna_data['RefSeq-Accn'] == 'NC_000012.12']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Replicon Name</th>\n",
       "      <th>Replicon Accession</th>\n",
       "      <th>Start</th>\n",
       "      <th>Stop</th>\n",
       "      <th>Strand</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>Locus</th>\n",
       "      <th>Protein product</th>\n",
       "      <th>Length</th>\n",
       "      <th>Protein name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>74388</td>\n",
       "      <td>12</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>66883</td>\n",
       "      <td>175033</td>\n",
       "      <td>+</td>\n",
       "      <td>440073</td>\n",
       "      <td>IQSEC3</td>\n",
       "      <td>NP_001164209.1</td>\n",
       "      <td>1182</td>\n",
       "      <td>IQ motif and SEC7 domain-containing protein 3 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74389</td>\n",
       "      <td>12</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>66883</td>\n",
       "      <td>171308</td>\n",
       "      <td>+</td>\n",
       "      <td>440073</td>\n",
       "      <td>IQSEC3</td>\n",
       "      <td>XP_016874800.1</td>\n",
       "      <td>1041</td>\n",
       "      <td>IQ motif and SEC7 domain-containing protein 3 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74390</td>\n",
       "      <td>12</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>66883</td>\n",
       "      <td>171321</td>\n",
       "      <td>+</td>\n",
       "      <td>440073</td>\n",
       "      <td>IQSEC3</td>\n",
       "      <td>XP_011519262.1</td>\n",
       "      <td>1062</td>\n",
       "      <td>IQ motif and SEC7 domain-containing protein 3 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74391</td>\n",
       "      <td>12</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>66883</td>\n",
       "      <td>171308</td>\n",
       "      <td>+</td>\n",
       "      <td>440073</td>\n",
       "      <td>IQSEC3</td>\n",
       "      <td>XP_016874799.1</td>\n",
       "      <td>1041</td>\n",
       "      <td>IQ motif and SEC7 domain-containing protein 3 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74392</td>\n",
       "      <td>12</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>66883</td>\n",
       "      <td>174693</td>\n",
       "      <td>+</td>\n",
       "      <td>440073</td>\n",
       "      <td>IQSEC3</td>\n",
       "      <td>XP_011519260.1</td>\n",
       "      <td>1063</td>\n",
       "      <td>IQ motif and SEC7 domain-containing protein 3 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>80340</td>\n",
       "      <td>12</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>133218885</td>\n",
       "      <td>133234356</td>\n",
       "      <td>-</td>\n",
       "      <td>647589</td>\n",
       "      <td>ANHX</td>\n",
       "      <td>XP_006719806.1</td>\n",
       "      <td>492</td>\n",
       "      <td>anomalous homeobox protein isoform X1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>80341</td>\n",
       "      <td>12</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>133218885</td>\n",
       "      <td>133234356</td>\n",
       "      <td>-</td>\n",
       "      <td>647589</td>\n",
       "      <td>ANHX</td>\n",
       "      <td>XP_011533124.1</td>\n",
       "      <td>336</td>\n",
       "      <td>anomalous homeobox protein isoform X3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>80342</td>\n",
       "      <td>12</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>133221223</td>\n",
       "      <td>133234356</td>\n",
       "      <td>-</td>\n",
       "      <td>647589</td>\n",
       "      <td>ANHX</td>\n",
       "      <td>XP_011533127.1</td>\n",
       "      <td>322</td>\n",
       "      <td>anomalous homeobox protein isoform X5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>80343</td>\n",
       "      <td>12</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>133221223</td>\n",
       "      <td>133234356</td>\n",
       "      <td>-</td>\n",
       "      <td>647589</td>\n",
       "      <td>ANHX</td>\n",
       "      <td>XP_011533126.1</td>\n",
       "      <td>331</td>\n",
       "      <td>anomalous homeobox protein isoform X4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>80344</td>\n",
       "      <td>12</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>133221223</td>\n",
       "      <td>133234356</td>\n",
       "      <td>-</td>\n",
       "      <td>647589</td>\n",
       "      <td>ANHX</td>\n",
       "      <td>XP_011533125.1</td>\n",
       "      <td>331</td>\n",
       "      <td>anomalous homeobox protein isoform X4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5957 rows  10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #Replicon Name Replicon Accession      Start       Stop Strand  GeneID  \\\n",
       "74388             12       NC_000012.12      66883     175033      +  440073   \n",
       "74389             12       NC_000012.12      66883     171308      +  440073   \n",
       "74390             12       NC_000012.12      66883     171321      +  440073   \n",
       "74391             12       NC_000012.12      66883     171308      +  440073   \n",
       "74392             12       NC_000012.12      66883     174693      +  440073   \n",
       "...              ...                ...        ...        ...    ...     ...   \n",
       "80340             12       NC_000012.12  133218885  133234356      -  647589   \n",
       "80341             12       NC_000012.12  133218885  133234356      -  647589   \n",
       "80342             12       NC_000012.12  133221223  133234356      -  647589   \n",
       "80343             12       NC_000012.12  133221223  133234356      -  647589   \n",
       "80344             12       NC_000012.12  133221223  133234356      -  647589   \n",
       "\n",
       "        Locus Protein product  Length  \\\n",
       "74388  IQSEC3  NP_001164209.1    1182   \n",
       "74389  IQSEC3  XP_016874800.1    1041   \n",
       "74390  IQSEC3  XP_011519262.1    1062   \n",
       "74391  IQSEC3  XP_016874799.1    1041   \n",
       "74392  IQSEC3  XP_011519260.1    1063   \n",
       "...       ...             ...     ...   \n",
       "80340    ANHX  XP_006719806.1     492   \n",
       "80341    ANHX  XP_011533124.1     336   \n",
       "80342    ANHX  XP_011533127.1     322   \n",
       "80343    ANHX  XP_011533126.1     331   \n",
       "80344    ANHX  XP_011533125.1     331   \n",
       "\n",
       "                                            Protein name  \n",
       "74388  IQ motif and SEC7 domain-containing protein 3 ...  \n",
       "74389  IQ motif and SEC7 domain-containing protein 3 ...  \n",
       "74390  IQ motif and SEC7 domain-containing protein 3 ...  \n",
       "74391  IQ motif and SEC7 domain-containing protein 3 ...  \n",
       "74392  IQ motif and SEC7 domain-containing protein 3 ...  \n",
       "...                                                  ...  \n",
       "80340              anomalous homeobox protein isoform X1  \n",
       "80341              anomalous homeobox protein isoform X3  \n",
       "80342              anomalous homeobox protein isoform X5  \n",
       "80343              anomalous homeobox protein isoform X4  \n",
       "80344              anomalous homeobox protein isoform X4  \n",
       "\n",
       "[5957 rows x 10 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "protein_data[protein_data['Replicon Accession']== 'NC_000012.12']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "from bs4 import BeautifulSoup\n",
    "import requests\n",
    "import time\n",
    "base_url = 'https://eutils.ncbi.nlm.nih.gov/'\n",
    "first_url_extension = 'protein/'\n",
    "protein_url_extension = protein_data['Protein product']\n",
    "report_url_extension = '?report=fasta'\n",
    "urls = []\n",
    "for url in range(0, len(protein_url)):\n",
    "    urls.append(base_url+first_url_extension+protein_url_extension[url]+report_url_extension)\n",
    "# html_page = requests.get(first_url)\n",
    "# soup = BeautifulSoup(html_page.content, 'html.parser')\n",
    "# soup.prettify"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'https://www.ncbi.nlm.nih.gov/protein/NP_874360.1?report=fasta'"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "urls[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<bound method Tag.prettify of <?xml version=\"1.0\" encoding=\"utf-8\"?>\n",
       "<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Transitional//EN\" \"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\">\n",
       "\n",
       "<html lang=\"en\" xml:lang=\"en\" xmlns=\"http://www.w3.org/1999/xhtml\">\n",
       "<head xmlns:xi=\"http://www.w3.org/2001/XInclude\"><meta content=\"text/html; charset=utf-8\" http-equiv=\"Content-Type\"/>\n",
       "<!-- meta -->\n",
       "<meta content=\"index,nofollow,noarchive\" name=\"robots\"/>\n",
       "<meta content=\"entrez\" name=\"ncbi_app\"/><meta content=\"protein\" name=\"ncbi_db\"/><meta content=\"fasta\" name=\"ncbi_report\"/><meta content=\"html\" name=\"ncbi_format\"/><meta content=\"20\" name=\"ncbi_pagesize\"/><meta content=\"default\" name=\"ncbi_sortorder\"/><meta content=\"1\" name=\"ncbi_pageno\"/><meta content=\"1\" name=\"ncbi_resultcount\"/><meta content=\"retrieve\" name=\"ncbi_op\"/><meta content=\"fasta\" name=\"ncbi_pdid\"/><meta content=\"CE884FD5E6AA2281_1606SID\" name=\"ncbi_sessionid\"/><meta content=\"33186887\" name=\"ncbi_uidlist\"/><meta content=\"all\" name=\"ncbi_filter\"/><meta content=\"false\" name=\"ncbi_stat\"/><meta content=\"false\" name=\"ncbi_hitstat\"/>\n",
       "<!-- title -->\n",
       "<title>glutathione peroxidase 6 precursor [Homo sapiens] - Protein - NCBI</title>\n",
       "<!-- Common JS and CSS -->\n",
       "<script type=\"text/javascript\">\n",
       "\t\t    var ncbi_startTime = new Date();\n",
       "\t\t</script>\n",
       "<style>.async-hide { opacity: 0 !important} </style><script src=\"/core/assets/kis/dist/kis_ga_nuc_protein.js\" type=\"text/javascript\"></script><script src=\"https://static.pubmed.gov/core/jig/1.14.8/js/jig.min.js\" type=\"text/javascript\"></script>\n",
       "<script src=\"/core/ajax_loader/2.1/js/loadingbar.js\" type=\"text/javascript\"></script>\n",
       "<script src=\"/core/ajax_loader/2.1/js/contentLoader.js\" type=\"text/javascript\"></script>\n",
       "<link href=\"/core/ajax_loader/2.1/css/loadingbar.css\" rel=\"stylesheet\" type=\"text/css\"/>\n",
       "<link href=\"//static.pubmed.gov/portal/portal3rc.fcgi/4189540/css/3881636/3579733.css\" rel=\"stylesheet\" type=\"text/css\" xml:base=\"http://127.0.0.1/sites/static/header_footer/\" xmlns=\"http://www.w3.org/1999/xhtml\"/>\n",
       "<link href=\"//www.ncbi.nlm.nih.gov/favicon.ico\" rel=\"shortcut icon\"/><meta content=\"CE884FD5E6A966D10000000006460349.m_33\" name=\"ncbi_phid\"/><script type=\"text/javascript\"><!--\n",
       "var ScriptPath = '/portal/';\n",
       "var objHierarchy = {\"name\":\"EntrezSystem2\",\"type\":\"Layout\",\"realname\":\"EntrezSystem2\",\n",
       "\"children\":[{\"name\":\"EntrezSystem2.PEntrez\",\"type\":\"Cluster\",\"realname\":\"EntrezSystem2.PEntrez\",\n",
       "\"children\":[{\"name\":\"EntrezSystem2.PEntrez.DbConnector\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.PEntrez.DbConnector\",\"shortname\":\"DbConnector\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.ParamContainer\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.PEntrez.ParamContainer\",\"shortname\":\"ParamContainer\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.MyNcbi\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.PEntrez.MyNcbi\",\"shortname\":\"MyNcbi\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.UserPreferenceUrlParamContainer\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.PEntrez.UserPreferenceUrlParamContainer\",\"shortname\":\"UserPreferenceUrlParamContainer\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.GridProperty\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.PEntrez.GridProperty\",\"shortname\":\"GridProperty\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein\",\"type\":\"Cluster\",\"realname\":\"EntrezSystem2.PEntrez.Protein\",\n",
       "\"children\":[{\"name\":\"EntrezSystem2.PEntrez.Protein.NoPortlet\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Entrez_Database.NoPortlet\",\"shortname\":\"NoPortlet\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_PageController\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_PageController\",\"shortname\":\"Sequence_PageController\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Entrez_SearchBar\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Entrez_Database.Entrez_SearchBar\",\"shortname\":\"Entrez_SearchBar\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Entrez_BotRequest\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Entrez_Database.Entrez_BotRequest\",\"shortname\":\"Entrez_BotRequest\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_LimitsTab\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_LimitsTab\",\"shortname\":\"Sequence_LimitsTab\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel\",\"type\":\"Cluster\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel\",\n",
       "\"children\":[{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.blankToolPanel\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Entrez_ResultsPanel.blankToolPanel\",\"shortname\":\"blankToolPanel\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_ResultsController\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Sequence_ResultsController\",\"shortname\":\"Sequence_ResultsController\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Entrez_Filters\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Entrez_ResultsPanel.Entrez_Filters\",\"shortname\":\"Entrez_Filters\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Entrez_Pager\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Entrez_ResultsPanel.Entrez_Pager\",\"shortname\":\"Entrez_Pager\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Sequence_DisplayBar\",\"shortname\":\"Sequence_DisplayBar\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.HelpFormAttributes\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Entrez_ResultsPanel.HelpFormAttributes\",\"shortname\":\"HelpFormAttributes\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Entrez_Collections\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Entrez_ResultsPanel.Entrez_Collections\",\"shortname\":\"Entrez_Collections\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.SpellCheck\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Entrez_ResultsPanel.SpellCheck\",\"shortname\":\"SpellCheck\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.SearchEngineReferralCheck\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Entrez_ResultsPanel.SearchEngineReferralCheck\",\"shortname\":\"SearchEngineReferralCheck\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.KnowledgePanel\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Entrez_ResultsPanel.KnowledgePanel\",\"shortname\":\"KnowledgePanel\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.HistoryDisplay\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Entrez_ResultsPanel.HistoryDisplay\",\"shortname\":\"HistoryDisplay\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Discovery_SearchDetails\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Entrez_ResultsPanel.Discovery_SearchDetails\",\"shortname\":\"Discovery_SearchDetails\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.KISSensor\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.KISSensor\",\"shortname\":\"KISSensor\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.MultiSensorPortlet\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.MultiSensorPortlet\",\"shortname\":\"MultiSensorPortlet\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.WrongDbSensor\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.WrongDbSensor\",\"shortname\":\"WrongDbSensor\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DiscoveryExptChooser\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Sequence_DiscoveryExptChooser\",\"shortname\":\"Sequence_DiscoveryExptChooser\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.SequenceViewer\",\"type\":\"Cluster\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.SequenceViewer\",\n",
       "\"children\":[{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.SequenceViewer.Sequence_ViewerTitle\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.SequenceViewer.Sequence_ViewerTitle\",\"shortname\":\"Sequence_ViewerTitle\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.SequenceViewer.Sequence_ViewerReport\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.SequenceViewer.Sequence_ViewerReport\",\"shortname\":\"Sequence_ViewerReport\"}]},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl\",\"type\":\"Cluster\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Sequence_SingleItemSupl\",\n",
       "\"children\":[{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_DiscoveryDbLinks\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_DiscoveryDbLinks\",\"shortname\":\"Sequence_DiscoveryDbLinks\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel\",\"type\":\"Cluster\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel\",\n",
       "\"children\":[{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_ToolsList\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_ToolsList\",\"shortname\":\"Sequence_ToolsList\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_SearchTool\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_SearchTool\",\"shortname\":\"Sequence_SearchTool\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_ViewerSidePanelInfo\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_ViewerSidePanelInfo\",\"shortname\":\"Sequence_ViewerSidePanelInfo\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_DiscoveryPubMedRA\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_DiscoveryPubMedRA\",\"shortname\":\"Sequence_DiscoveryPubMedRA\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_DiscoveryGeneInfo\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_DiscoveryGeneInfo\",\"shortname\":\"Sequence_DiscoveryGeneInfo\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_DiscoveryViralResource\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_DiscoveryViralResource\",\"shortname\":\"Sequence_DiscoveryViralResource\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_DiscoveryRefSeqs\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_DiscoveryRefSeqs\",\"shortname\":\"Sequence_DiscoveryRefSeqs\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_DiscoveryHomologs\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_DiscoveryHomologs\",\"shortname\":\"Sequence_DiscoveryHomologs\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.LinkOutPortlet\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.LinkOutPortlet\",\"shortname\":\"LinkOutPortlet\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_ViewerChangeRegion\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_ViewerChangeRegion\",\"shortname\":\"Sequence_ViewerChangeRegion\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_ViewerCustomizer\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_ViewerCustomizer\",\"shortname\":\"Sequence_ViewerCustomizer\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Discovery_Structure\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Discovery_Structure\",\"shortname\":\"Discovery_Structure\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.blank\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.blank\",\"shortname\":\"blank\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_DiscoveryIdenticalProteins\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_DiscoveryIdenticalProteins\",\"shortname\":\"Sequence_DiscoveryIdenticalProteins\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_DiscoveryProteinClusters\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_DiscoveryProteinClusters\",\"shortname\":\"Sequence_DiscoveryProteinClusters\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_DiscoveryBioSystems\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_DiscoveryBioSystems\",\"shortname\":\"Sequence_DiscoveryBioSystems\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.blank4\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.blank4\",\"shortname\":\"blank4\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.blank5\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.blank5\",\"shortname\":\"blank5\"}]}]}]},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_Facets\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence_Facets\",\"shortname\":\"Sequence_Facets\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Entrez_Clipboard\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Entrez_Database.Entrez_Clipboard\",\"shortname\":\"Entrez_Clipboard\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Sequence_StaticParts\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Sequence_StaticParts\",\"shortname\":\"Sequence_StaticParts\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.Entrez_Messages\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Entrez_Database.Entrez_Messages\",\"shortname\":\"Entrez_Messages\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.NcbiJSCheck\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Entrez_Database.NcbiJSCheck\",\"shortname\":\"NcbiJSCheck\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.NCBIFooter_dynamic\",\"type\":\"Cluster\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Entrez_Database.NCBIFooter_dynamic\",\n",
       "\"children\":[{\"name\":\"EntrezSystem2.PEntrez.Protein.NCBIFooter_dynamic.Footer_ExtraData\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Entrez_Database.NCBIFooter_dynamic.Footer_ExtraData\",\"shortname\":\"Footer_ExtraData\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.NCBIFooter_dynamic.NCBIFooter_dynamic\",\"type\":\"Cluster\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Entrez_Database.NCBIFooter_dynamic.NCBIFooter_dynamic\",\n",
       "\"children\":[{\"name\":\"EntrezSystem2.PEntrez.Protein.NCBIFooter_dynamic.NCBIFooter_dynamic.NCBIBreadcrumbs\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Entrez_Database.NCBIFooter_dynamic.NCBIFooter_dynamic.NCBIBreadcrumbs\",\"shortname\":\"NCBIBreadcrumbs\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.NCBIFooter_dynamic.NCBIFooter_dynamic.NCBIHelpDesk\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Entrez_Database.NCBIFooter_dynamic.NCBIFooter_dynamic.NCBIHelpDesk\",\"shortname\":\"NCBIHelpDesk\"},\n",
       "{\"name\":\"EntrezSystem2.PEntrez.Protein.NCBIFooter_dynamic.NCBIFooter_dynamic.NCBIApplog_NoScript_Ping\",\"type\":\"Portlet\",\"realname\":\"EntrezSystem2.PEntrez.Protein.Sequence.Entrez_Database.NCBIFooter_dynamic.NCBIFooter_dynamic.NCBIApplog_NoScript_Ping\",\"shortname\":\"NCBIApplog_NoScript_Ping\"}]}]}]}]}]};\n",
       "--></script>\n",
       "<meta content=\"origin-when-cross-origin\" name=\"referrer\"/><link href=\"//static.pubmed.gov/portal/portal3rc.fcgi/4189134/css/3808861/3917732/3974050/3751656/3395415/14534/4062871/4186458/4016739/4075711/12930/4033350/4128070/3861632/4161386/4091728/3257261/3325043/3325044/9685/35809/36029/3394192/2279/3395586/2744335.css\" rel=\"stylesheet\" type=\"text/css\"/><link href=\"//static.pubmed.gov/portal/portal3rc.fcgi/4189134/css/3501913/1303451/3529739.css\" media=\"print\" rel=\"stylesheet\" type=\"text/css\"/><script type=\"text/javascript\">\n",
       "\n",
       "var ObjectLinks=[{i:0, ename: \"p$ExL\", esid:\"*\", sname: \"p$ExL\", ssid:\"*\", dname:\"p$el\", dsid:\"0\",m:\"CopyValue\",p:[],f: function(src, dst) {fn_CopyValue(src, dst);}}]\n",
       "\n",
       "\n",
       "var ActiveNames = {\"p$ExL\":1, \"EntrezSystem2.PEntrez.DbConnector.Cmd\":0, \"EntrezSystem2.PEntrez.DbConnector.Db\":0, \"EntrezSystem2.PEntrez.DbConnector.IdsFromResult\":0, \"EntrezSystem2.PEntrez.DbConnector.LastDb\":0, \"EntrezSystem2.PEntrez.DbConnector.LastIdsFromResult\":0, \"EntrezSystem2.PEntrez.DbConnector.LastQueryKey\":0, \"EntrezSystem2.PEntrez.DbConnector.LastTabCmd\":0, \"EntrezSystem2.PEntrez.DbConnector.LinkName\":0, \"EntrezSystem2.PEntrez.DbConnector.LinkReadableName\":0, \"EntrezSystem2.PEntrez.DbConnector.LinkSrcDb\":0, \"EntrezSystem2.PEntrez.DbConnector.QueryKey\":0, \"EntrezSystem2.PEntrez.DbConnector.TabCmd\":0, \"EntrezSystem2.PEntrez.DbConnector.Term\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_PageController.PreviousPageName\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.HistoryDisplay.ClearHistory\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.HistoryDisplay.Cmd\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.HistoryDisplay.HistoryOn\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.HistoryDisplay.HistoryToggle\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.HistoryDisplay.Shutter\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.SequenceViewer.Sequence_ViewerReport.ExpandGaps\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.SequenceViewer.Sequence_ViewerReport.InUse\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.SequenceViewer.Sequence_ViewerReport.ItemCount\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.SequenceViewer.Sequence_ViewerReport.db\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.SequenceViewer.Sequence_ViewerReport.display_type\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.SequenceViewer.Sequence_ViewerReport.maxdownloadsize\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.SequenceViewer.Sequence_ViewerReport.report\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.SequenceViewer.Sequence_ViewerTitle.Database\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.SequenceViewer.Sequence_ViewerTitle.Report\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.SequenceViewer.Sequence_ViewerTitle.ReportShortCut\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.SequenceViewer.Sequence_ViewerTitle.ResultCount\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.CurrFilter\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.Db\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.Display\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.FFormat\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.FShowgi\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.FileFormat\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.FileSort\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.Format\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.Id\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.LastFormat\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.LastPageSize\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.LastPresentation\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.LastSort\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.PageSize\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.Presentation\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.PrevPageSize\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.PrevPresentation\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.PrevSort\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.QueryKey\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.ResultCount\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.SendTo\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.SendToSubmit\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.SetDisplay\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.Sort\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.ViewerParams\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.radio_analysis_tool\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.sPresentation\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.uid\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_ResultsController.AccnsFromResult\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_ResultsController.ResultCount\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_ResultsController.RunLastQuery\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_DiscoveryDbLinks.Shutter\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.LinkOutPortlet.Shutter\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_DiscoveryBioSystems.Shutter\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_DiscoveryGeneInfo.Shutter\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_DiscoveryHomologs.Shutter\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_DiscoveryPubMedRA.Shutter\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_DiscoveryRefSeqs.Shutter\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_ToolsList.Shutter\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_ViewerChangeRegion.ChangeRegion\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_ViewerChangeRegion.Contig\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_ViewerChangeRegion.From\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_ViewerChangeRegion.Max\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_ViewerChangeRegion.SetSelRegion\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_ViewerChangeRegion.Shutter\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_ViewerChangeRegion.To\":0, \"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_ViewerChangeRegion.Topology\":0};\n",
       "</script></head>\n",
       "<body>\n",
       "<div class=\"grid\">\n",
       "<div class=\"col twelve_col nomargin shadow\">\n",
       "<form action=\"/protein\" enctype=\"application/x-www-form-urlencoded\" id=\"EntrezForm\" method=\"post\" name=\"EntrezForm\" onsubmit=\"return false;\">\n",
       "<div xmlns:xi=\"http://www.w3.org/2001/XInclude\">\n",
       "<!-- no javascript message -->\n",
       "<noscript>\n",
       "<p class=\"nojs\">\n",
       "<strong>Warning:</strong>\n",
       "\tThe NCBI web site requires JavaScript to function. \n",
       "\t<a href=\"/guide/browsers/#enablejs\" target=\"_blank\" title=\"Learn how to enable JavaScript\">more...</a>\n",
       "</p>\n",
       "</noscript>\n",
       "<div class=\"universal_header\" id=\"universal_header\" xml:base=\"http://127.0.0.1/sites/static/header_footer/\" xmlns=\"http://www.w3.org/1999/xhtml\"><ul class=\"inline_list jig-ncbimenu ui-ncbimenu resources_list\" id=\"navcontent\"><li class=\"ui-ncbimenu-item-leaf ui-ncbimenu-item-first ui-helper-reset ui-ncbimenu-item-no-hlt\"><a accesskey=\"1\" class=\"ui-ncbimenu-link-first\" href=\"/\" id=\"ncbihome\" role=\"banner\" title=\"NCBI Home\"><span class=\"offscreen_noflow\">NCBI</span><img alt=\"NCBI Logo\" class=\"ncbi_logo\" src=\"//static.pubmed.gov/portal/portal3rc.fcgi/4189540/img/28977\" title=\"NCBI\"/></a></li><li class=\"offscreen_noflow ui-ncbimenu-item-skip access\"><a accesskey=\"3\" href=\"#maincontent\" tabindex=\"0\" title=\"Skip to the content\">Skip to main\n",
       "                        content</a></li><li class=\"offscreen_noflow ui-ncbimenu-item-skip access\"><a accesskey=\"4\" href=\"#navcontent\" tabindex=\"0\" title=\"Skip to the navigation\">Skip to\n",
       "                        navigation</a></li><li class=\"topmenu ui-helper-reset ui-ncbimenu-item-first ui-helper-reset\" id=\"resource-menu\"><a class=\"ui-ncbimenu-first-link-has-submenu ui-ncbimenu-link-first topanchor expandDown\" href=\"/static/header_footer_ajax/submenu/#resources\">Resources</a><ul id=\"resources-submenu\"><li><a href=\"/guide/all/\">All Resources</a></li><li><a href=\"#chemicals-bioassays\">Chemicals &amp; Bioassays</a><ul class=\"actual-resources-submenu chemicals-bioassays\"><li><a href=\"/biosystems/\" title=\" Database that groups biomedical literature, small molecules, and sequence data in terms of biological relationships. \">BioSystems</a></li><li><a href=\"/pcassay\" title=\" Contributed bioactivity data for PubChem Substances \">PubChem BioAssay</a></li><li><a href=\"/pccompound\" title=\" Unique, standardized chemical structures \">PubChem Compound</a></li><li><a href=\"https://pubchem.ncbi.nlm.nih.gov/search/search.cgi\" title=\" Searches PubChem Compound by chemical structure \">PubChem Structure Search</a></li><li><a href=\"/pcsubstance\" title=\" Provides information on the biological activities of small molecules. \">PubChem Substance</a></li><li><a href=\"/guide/chemicals-bioassays/\">All Chemicals &amp; Bioassays Resources...</a></li></ul></li><li><a href=\"#dna-rna\">DNA &amp; RNA</a><ul class=\"actual-resources-submenu dna-rna\"><li><a href=\"https://blast.ncbi.nlm.nih.gov/Blast.cgi\" title=\" Tool for finding local regions of sequence similarity \">BLAST (Basic Local Alignment Search Tool)</a></li><li><a href=\"https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs&amp;DOC_TYPE=Download\" title=\" BLAST programs for a local computer \">BLAST (Stand-alone)</a></li><li><a href=\"/books/NBK25501/ \" title=\" URL application programming interface for searching, linking and downloading data within NCBI's Entrez system \">E-Utilities</a></li><li><a href=\"/genbank/\" title=\" DNA and RNA sequences submitted to NCBI \">GenBank</a></li><li><a href=\"/WebSub/?tool=genbank\" title=\" Web tool for simple submissions of nucleotide sequences \">GenBank: BankIt</a></li><li><a href=\"/projects/Sequin/\" title=\" Software tool for complex submissions of nucleotide sequences \">GenBank: Sequin</a></li><li><a href=\"/genbank/tbl2asn2/\" title=\" Command-line program for submitting sequence data \">GenBank: tbl2asn</a></li><li><a href=\"/tools/gbench/\" title=\" Stand-alone application for analyzing sequence data \">Genome Workbench</a></li><li><a href=\"/genomes/FLU/\" title=\" Data from the NIAID Influenza Genome Sequencing Project \">Influenza Virus</a></li><li><a href=\"/nuccore\" title=\" Nucleotide sequence data in GenBank and RefSeq \">Nucleotide Database</a></li><li><a href=\"/popset\" title=\" Sets of nucleotide sequences representing a population \">PopSet</a></li><li><a href=\"/tools/primer-blast/\" title=\" Designs PCR primers and analyzes them for target specificity using BLAST. \">Primer-BLAST</a></li><li><a href=\"/sutils/static/prosplign/prosplign.html\" title=\" Tool for aligning proteins to genomic DNA \">ProSplign</a></li><li><a href=\"/refseq/\" title=\" Curated nucleotide and protein sequences produced by NCBI \">Reference Sequence (RefSeq)</a></li><li><a href=\"/refseq/rsg/\" title=\" RefSeqGene reference standards for genomic sequences of well-characterized genes \">RefSeqGene</a></li><li><a href=\"/Traces/sra/sra.cgi?\" title=\" Database of sequences reads from next generation sequencing technologies \">Sequence Read Archive (SRA)</a></li><li><a href=\"/sutils/splign/\" title=\" Tool for aligning transcripts to genomic DNA \">Splign</a></li><li><a href=\"/Traces/trace.cgi\" title=\" DNA chromatograms, base calls, and quality estimates for single-pass reads. \">Trace Archive</a></li><li><a href=\"/guide/dna-rna/\">All DNA &amp; RNA Resources...</a></li></ul></li><li><a href=\"#data-software\">Data &amp; Software</a><ul class=\"actual-resources-submenu data-software\"><li><a href=\"https://blast.ncbi.nlm.nih.gov/Blast.cgi\" title=\" Tool for finding local regions of sequence similarity \">BLAST (Basic Local Alignment Search Tool)</a></li><li><a href=\"https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs&amp;DOC_TYPE=Download\" title=\" BLAST programs for a local computer \">BLAST (Stand-alone)</a></li><li><a href=\"/Structure/CN3D/cn3d.shtml\" title=\" Program for viewing 3D structures and alignments \">Cn3D</a></li><li><a href=\"/Structure/cdd/wrpsb.cgi\" title=\" Locates conserved domains within a protein sequence \">Conserved Domain Search Service (CD Search)</a></li><li><a href=\"/books/NBK25501/ \" title=\" URL application programming interface for searching, linking and downloading data within NCBI's Entrez system \">E-Utilities</a></li><li><a href=\"/WebSub/?tool=genbank\" title=\" Web tool for simple submissions of nucleotide sequences \">GenBank: BankIt</a></li><li><a href=\"/projects/Sequin/\" title=\" Software tool for complex submissions of nucleotide sequences \">GenBank: Sequin</a></li><li><a href=\"/genbank/tbl2asn2/\" title=\" Command-line program for submitting sequence data \">GenBank: tbl2asn</a></li><li><a href=\"/sutils/protmap.cgi\" title=\" Maps proteins from a COG/VOG back to its genome \">Genome ProtMap</a></li><li><a href=\"/tools/gbench/\" title=\" Stand-alone application for analyzing sequence data \">Genome Workbench</a></li><li><a href=\"/tools/primer-blast/\" title=\" Designs PCR primers and analyzes them for target specificity using BLAST. \">Primer-BLAST</a></li><li><a href=\"/sutils/static/prosplign/prosplign.html\" title=\" Tool for aligning proteins to genomic DNA \">ProSplign</a></li><li><a href=\"https://pubchem.ncbi.nlm.nih.gov/search/search.cgi\" title=\" Searches PubChem Compound by chemical structure \">PubChem Structure Search</a></li><li><a href=\"/projects/SNP/\" title=\" The SNP database tools page provides links to the general submission guidelines and to the submission handle request. \">SNP Submission Tool</a></li><li><a href=\"/sutils/splign/\" title=\" Tool for aligning transcripts to genomic DNA \">Splign</a></li><li><a href=\"/Structure/VAST/vast.html\" title=\" Tool that generates 3D structural alignments \">Vector Alignment Search Tool (VAST)</a></li><li><a href=\"/guide/data-software/\">All Data &amp; Software Resources...</a></li></ul></li><li><a href=\"#domains-structures\">Domains &amp; Structures</a><ul class=\"actual-resources-submenu domains-structures\"><li><a href=\"/biosystems/\" title=\" Database that groups biomedical literature, small molecules, and sequence data in terms of biological relationships. \">BioSystems</a></li><li><a href=\"/Structure/CN3D/cn3d.shtml\" title=\" Program for viewing 3D structures and alignments \">Cn3D</a></li><li><a href=\"/cdd\" title=\" Database of conserved protein domains represented by multiple sequence alignments \">Conserved Domain Database (CDD)</a></li><li><a href=\"/Structure/cdd/wrpsb.cgi\" title=\" Locates conserved domains within a protein sequence \">Conserved Domain Search Service (CD Search)</a></li><li><a href=\"/structure/\" title=\" Database of 3D macromolecular structures derived from PDB \">Structure (Molecular Modeling Database)</a></li><li><a href=\"/Structure/VAST/vast.html\" title=\" Tool that generates 3D structural alignments \">Vector Alignment Search Tool (VAST)</a></li><li><a href=\"/guide/domains-structures/\">All Domains &amp; Structures Resources...</a></li></ul></li><li><a href=\"#genes-expression\">Genes &amp; Expression</a><ul class=\"actual-resources-submenu genes-expression\"><li><a href=\"/biosystems/\" title=\" Database that groups biomedical literature, small molecules, and sequence data in terms of biological relationships. \">BioSystems</a></li><li><a href=\"/gap\" title=\" Genotype/Phenotype association studies \">Database of Genotypes and Phenotypes (dbGaP)</a></li><li><a href=\"/books/NBK25501/ \" title=\" URL application programming interface for searching, linking and downloading data within NCBI's Entrez system \">E-Utilities</a></li><li><a href=\"/gene\" title=\" Database of gene loci and associated transcripts and proteins \">Gene</a></li><li><a href=\"/geo/\" title=\" Gene Expression Omnibus (GEO) Database \">Gene Expression Omnibus (GEO) Database </a></li><li><a href=\"/gds\" title=\" Curated collections of gene expression data sets \">Gene Expression Omnibus (GEO) Datasets</a></li><li><a href=\"/geoprofiles/\" title=\" Database of gene expression Profiles for individual sequences \">Gene Expression Omnibus (GEO) Profiles</a></li><li><a href=\"/tools/gbench/\" title=\" Stand-alone application for analyzing sequence data \">Genome Workbench</a></li><li><a href=\"/homologene\" title=\" Sets of homologous eukaryotic genes \">HomoloGene</a></li><li><a href=\"/omim\" title=\" Database of human genes and genetic disorders \">Online Mendelian Inheritance in Man (OMIM)</a></li><li><a href=\"/refseq/rsg/\" title=\" RefSeqGene reference standards for genomic sequences of well-characterized genes \">RefSeqGene</a></li><li><a href=\"/guide/genes-expression/\">All Genes &amp; Expression Resources...</a></li></ul></li><li><a href=\"#genetics-medicine\">Genetics &amp; Medicine</a><ul class=\"actual-resources-submenu genetics-medicine\"><li><a href=\"/books\" title=\" Online biomedical books \">Bookshelf</a></li><li><a href=\"/gap\" title=\" Genotype/Phenotype association studies \">Database of Genotypes and Phenotypes (dbGaP)</a></li><li><a href=\"/gtr/\" title=\" Detailed information about genetic tests and labs \">Genetic Testing Registry</a></li><li><a href=\"/genomes/FLU/\" title=\" Data from the NIAID Influenza Genome Sequencing Project \">Influenza Virus</a></li><li><a href=\"/omim\" title=\" Database of human genes and genetic disorders \">Online Mendelian Inheritance in Man (OMIM)</a></li><li><a href=\"/pubmed\" title=\" Biomedical literature abstracts \">PubMed</a></li><li><a href=\"/pmc/\" title=\" Free, full-text articles \">PubMed Central (PMC)</a></li><li><a href=\"/pubmed/clinical \" title=\" Directory of PubMed queries about clinical topics \">PubMed Clinical Queries</a></li><li><a href=\"/refseq/rsg/\" title=\" RefSeqGene reference standards for genomic sequences of well-characterized genes \">RefSeqGene</a></li><li><a href=\"/guide/genetics-medicine/\">All Genetics &amp; Medicine Resources...</a></li></ul></li><li><a href=\"#genomes-maps\">Genomes &amp; Maps</a><ul class=\"actual-resources-submenu genomes-maps\"><li><a href=\"/dbvar\" title=\" A database of genomic structural variation \">Database of Genomic Structural Variation (dbVar)</a></li><li><a href=\"/genbank/tbl2asn2/\" title=\" Command-line program for submitting sequence data \">GenBank: tbl2asn</a></li><li><a href=\"/sites/genome\" title=\" Curated genomic DNA sequences \">Genome</a></li><li><a href=\"/bioproject/\" title=\" Central status pages for genome sequencing projects \">Genome Project</a></li><li><a href=\"/genome/gdv/\" title=\"GDV is a genome browser supporting the exploration and analysis of eukaryotic RefSeq Genome assemblies. \">Genome Data Viewer (GDV)</a></li><li><a href=\"/sutils/protmap.cgi\" title=\" Maps proteins from a COG/VOG back to its genome \">Genome ProtMap</a></li><li><a href=\"/tools/gbench/\" title=\" Stand-alone application for analyzing sequence data \">Genome Workbench</a></li><li><a href=\"/genomes/FLU/\" title=\" Data from the NIAID Influenza Genome Sequencing Project \">Influenza Virus</a></li><li><a href=\"/nuccore\" title=\" Nucleotide sequence data in GenBank and RefSeq \">Nucleotide Database</a></li><li><a href=\"/popset\" title=\" Sets of nucleotide sequences representing a population \">PopSet</a></li><li><a href=\"/sutils/static/prosplign/prosplign.html\" title=\" Tool for aligning proteins to genomic DNA \">ProSplign</a></li><li><a href=\"/Traces/sra/sra.cgi?\" title=\" Database of sequences reads from next generation sequencing technologies \">Sequence Read Archive (SRA)</a></li><li><a href=\"/sutils/splign/\" title=\" Tool for aligning transcripts to genomic DNA \">Splign</a></li><li><a href=\"/Traces/trace.cgi\" title=\" DNA chromatograms, base calls, and quality estimates for single-pass reads. \">Trace Archive</a></li><li><a href=\"/guide/genomes-maps/\">All Genomes &amp; Maps Resources...</a></li></ul></li><li><a href=\"#homology\">Homology</a><ul class=\"actual-resources-submenu homology\"><li><a href=\"https://blast.ncbi.nlm.nih.gov/Blast.cgi\" title=\" Tool for finding local regions of sequence similarity \">BLAST (Basic Local Alignment Search Tool)</a></li><li><a href=\"https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs&amp;DOC_TYPE=Download\" title=\" BLAST programs for a local computer \">BLAST (Stand-alone)</a></li><li><a href=\"/sutils/blink.cgi?mode=query\" title=\" Pre-computed reports showing proteins with similar sequences \">BLAST Link (BLink)</a></li><li><a href=\"/cdd\" title=\" Database of conserved protein domains represented by multiple sequence alignments \">Conserved Domain Database (CDD)</a></li><li><a href=\"/Structure/cdd/wrpsb.cgi\" title=\" Locates conserved domains within a protein sequence \">Conserved Domain Search Service (CD Search)</a></li><li><a href=\"/sutils/protmap.cgi\" title=\" Maps proteins from a COG/VOG back to its genome \">Genome ProtMap</a></li><li><a href=\"/homologene\" title=\" Sets of homologous eukaryotic genes \">HomoloGene</a></li><li><a href=\"/proteinclusters\" title=\" Clusters of related proteins from prokaryotes and organelles \">Protein Clusters</a></li><li><a href=\"/guide/homology/\">All Homology Resources...</a></li></ul></li><li><a href=\"#literature\">Literature</a><ul class=\"actual-resources-submenu literature\"><li><a href=\"/books\" title=\" Online biomedical books \">Bookshelf</a></li><li><a href=\"/books/NBK25501/ \" title=\" URL application programming interface for searching, linking and downloading data within NCBI's Entrez system \">E-Utilities</a></li><li><a href=\"/nlmcatalog/journals\" title=\" Information on journals cited by NCBI records \">Journals in NCBI Databases</a></li><li><a href=\"/mesh\" title=\" Controlled vocabulary for MEDLINE \">MeSH Database</a></li><li><a href=\"/books/NBK143764/\" title=\" Introductory articles about NCBI resources \">NCBI Handbook</a></li><li><a href=\"/books/NBK3831/\" title=\" Online, searchable help documentation \">NCBI Help Manual</a></li><li><a href=\"https://ncbiinsights.ncbi.nlm.nih.gov\" title=\" NCBI's monthly newsletter \">NCBI News &amp; Blog</a></li><li><a href=\"/pubmed\" title=\" Biomedical literature abstracts \">PubMed</a></li><li><a href=\"/pmc/\" title=\" Free, full-text articles \">PubMed Central (PMC)</a></li><li><a href=\"/pubmed/clinical \" title=\" Directory of PubMed queries about clinical topics \">PubMed Clinical Queries</a></li><li><a href=\"/guide/literature/\">All Literature Resources...</a></li></ul></li><li><a href=\"#proteins\">Proteins</a><ul class=\"actual-resources-submenu proteins\"><li><a href=\"/biosystems/\" title=\" Database that groups biomedical literature, small molecules, and sequence data in terms of biological relationships. \">BioSystems</a></li><li><a href=\"https://blast.ncbi.nlm.nih.gov/Blast.cgi\" title=\" Tool for finding local regions of sequence similarity \">BLAST (Basic Local Alignment Search Tool)</a></li><li><a href=\"https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs&amp;DOC_TYPE=Download\" title=\" BLAST programs for a local computer \">BLAST (Stand-alone)</a></li><li><a href=\"/sutils/blink.cgi?mode=query\" title=\" Pre-computed reports showing proteins with similar sequences \">BLAST Link (BLink)</a></li><li><a href=\"/cdd\" title=\" Database of conserved protein domains represented by multiple sequence alignments \">Conserved Domain Database (CDD)</a></li><li><a href=\"/Structure/cdd/wrpsb.cgi\" title=\" Locates conserved domains within a protein sequence \">Conserved Domain Search Service (CD Search)</a></li><li><a href=\"/books/NBK25501/ \" title=\" URL application programming interface for searching, linking and downloading data within NCBI's Entrez system \">E-Utilities</a></li><li><a href=\"/sutils/static/prosplign/prosplign.html\" title=\" Tool for aligning proteins to genomic DNA \">ProSplign</a></li><li><a href=\"/proteinclusters\" title=\" Clusters of related proteins from prokaryotes and organelles \">Protein Clusters</a></li><li><a href=\"/protein\" title=\" Protein sequence data \">Protein Database</a></li><li><a href=\"/refseq/\" title=\" Curated nucleotide and protein sequences produced by NCBI \">Reference Sequence (RefSeq)</a></li><li><a href=\"/guide/proteins/\">All Proteins Resources...</a></li></ul></li><li><a href=\"#sequence-analysis\">Sequence Analysis</a><ul class=\"actual-resources-submenu sequence-analysis\"><li><a href=\"https://blast.ncbi.nlm.nih.gov/Blast.cgi\" title=\" Tool for finding local regions of sequence similarity \">BLAST (Basic Local Alignment Search Tool)</a></li><li><a href=\"https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs&amp;DOC_TYPE=Download\" title=\" BLAST programs for a local computer \">BLAST (Stand-alone)</a></li><li><a href=\"/sutils/blink.cgi?mode=query\" title=\" Pre-computed reports showing proteins with similar sequences \">BLAST Link (BLink)</a></li><li><a href=\"/Structure/cdd/wrpsb.cgi\" title=\" Locates conserved domains within a protein sequence \">Conserved Domain Search Service (CD Search)</a></li><li><a href=\"/sutils/protmap.cgi\" title=\" Maps proteins from a COG/VOG back to its genome \">Genome ProtMap</a></li><li><a href=\"/tools/gbench/\" title=\" Stand-alone application for analyzing sequence data \">Genome Workbench</a></li><li><a href=\"/genomes/FLU/\" title=\" Data from the NIAID Influenza Genome Sequencing Project \">Influenza Virus</a></li><li><a href=\"/tools/primer-blast/\" title=\" Designs PCR primers and analyzes them for target specificity using BLAST. \">Primer-BLAST</a></li><li><a href=\"/sutils/static/prosplign/prosplign.html\" title=\" Tool for aligning proteins to genomic DNA \">ProSplign</a></li><li><a href=\"/sutils/splign/\" title=\" Tool for aligning transcripts to genomic DNA \">Splign</a></li><li><a href=\"/guide/sequence-analysis/\">All Sequence Analysis Resources...</a></li></ul></li><li><a href=\"#taxonomy\">Taxonomy</a><ul class=\"actual-resources-submenu taxonomy\"><li><a href=\"/taxonomy\" title=\" The NCBI taxonomy tree in Entrez \">Taxonomy</a></li><li><a href=\"/Taxonomy/Browser/wwwtax.cgi?mode=Root\" title=\" Displays and links Entrez data to each taxonomic node \">Taxonomy Browser</a></li><li><a href=\"/Taxonomy/CommonTree/wwwcmt.cgi\" title=\" Creates custom taxonomic trees \">Taxonomy Common Tree</a></li><li><a href=\"/guide/taxonomy/\">All Taxonomy Resources...</a></li></ul></li><li><a href=\"#training-tutorials\">Training &amp; Tutorials</a><ul class=\"actual-resources-submenu training-tutorials\"><li><a href=\"/education/\" title=\" Central page providing links to tutorials and training materials \">NCBI Education Page</a></li><li><a href=\"/books/NBK143764/\" title=\" Introductory articles about NCBI resources \">NCBI Handbook</a></li><li><a href=\"/books/NBK3831/\" title=\" Online, searchable help documentation \">NCBI Help Manual</a></li><li><a href=\"https://ncbiinsights.ncbi.nlm.nih.gov\" title=\" NCBI's monthly newsletter \">NCBI News &amp; Blog</a></li><li><a href=\"/guide/training-tutorials/\">All Training &amp; Tutorials Resources...</a></li></ul></li><li><a href=\"#variation\">Variation</a><ul class=\"actual-resources-submenu variation\"><li><a href=\"/dbvar\" title=\" A database of genomic structural variation \">Database of Genomic Structural Variation (dbVar)</a></li><li><a href=\"/gap\" title=\" Genotype/Phenotype association studies \">Database of Genotypes and Phenotypes (dbGaP)</a></li><li><a href=\"/snp\" title=\" Database of nucleotide sequence polymorphisms \">Database of Single Nucleotide Polymorphisms (dbSNP)</a></li><li><a href=\"/projects/SNP/\" title=\" The SNP database tools page provides links to the general submission guidelines and to the submission handle request. \">SNP Submission Tool</a></li><li><a href=\"/guide/variation/\">All Variation Resources...</a></li></ul></li></ul></li><li class=\"topmenu ui-helper-reset ui-ncbimenu-item-first\" id=\"all-howtos-menu\"><a class=\"ui-ncbimenu-first-link-has-submenu ui-ncbimenu-link-first topanchor expandDown\" href=\"/static/header_footer_ajax/submenu/#howto\">How To</a><ul id=\"howto-submenu\"><li><a href=\"/guide/all/#howtos\">All How To</a></li><li><a href=\"/guide/chemicals-bioassays/#howtos\">Chemicals &amp; Bioassays</a></li><li><a href=\"/guide/dna-rna/#howtos\">DNA &amp; RNA</a></li><li><a href=\"/guide/data-software/#howtos\">Data &amp; Software</a></li><li><a href=\"/guide/domains-structures/#howtos\">Domains &amp; Structures</a></li><li><a href=\"/guide/genes-expression/#howtos\">Genes &amp; Expression</a></li><li><a href=\"/guide/genetics-medicine/#howtos\">Genetics &amp; Medicine</a></li><li><a href=\"/guide/genomes-maps/#howtos\">Genomes &amp; Maps</a></li><li><a href=\"/guide/homology/#howtos\">Homology</a></li><li><a href=\"/guide/literature/#howtos\">Literature</a></li><li><a href=\"/guide/proteins/#howtos\">Proteins</a></li><li><a href=\"/guide/sequence-analysis/#howtos\">Sequence Analysis</a></li><li><a href=\"/guide/taxonomy/#howtos\">Taxonomy</a></li><li><a href=\"/guide/training-tutorials/#howtos\">Training &amp; Tutorials</a></li><li><a href=\"/guide/variation/#howtos\">Variation</a></li></ul></li><li class=\"offscreen_noflow ui-ncbimenu-item-skip access\"><a accesskey=\"0\" href=\"/guide/browsers/#accesskeys\" tabindex=\"0\" title=\"About My NCBI Accesskeys\">About NCBI Accesskeys</a></li></ul><div class=\"myncbi\"><span id=\"myncbiusername\" style=\"display:none\"><a href=\"/account/settings/\" id=\"mnu\" title=\"Edit account settings\"></a></span><a accesskey=\"2\" href=\"/myncbi/\" id=\"myncbi\" style=\"display:none\">My NCBI</a><a href=\"/account/\" id=\"sign_in\">Sign in to NCBI</a><a href=\"/account/signout/\" id=\"sign_out\" style=\"display:none\">Sign Out</a></div></div>\n",
       "<div class=\"header\">\n",
       "<!-- logo -->\n",
       "<div class=\"res_logo\"><h1 class=\"res_name\"><a href=\"/protein\">Protein</a></h1></div>\n",
       "<!-- SearchBar -->\n",
       "<div class=\"search\"><div class=\"search_form\"><label class=\"offscreen_noflow\" for=\"database\">Search database</label><select id=\"database\"><option value=\"gquery\">All Databases</option><option value=\"assembly\">Assembly</option><option value=\"biocollections\">Biocollections</option><option value=\"bioproject\">BioProject</option><option value=\"biosample\">BioSample</option><option value=\"biosystems\">BioSystems</option><option value=\"books\">Books</option><option value=\"clinvar\">ClinVar</option><option value=\"cdd\">Conserved Domains</option><option value=\"gap\">dbGaP</option><option value=\"dbvar\">dbVar</option><option value=\"gene\">Gene</option><option value=\"genome\">Genome</option><option value=\"gds\">GEO DataSets</option><option value=\"geoprofiles\">GEO Profiles</option><option value=\"gtr\">GTR</option><option value=\"homologene\">HomoloGene</option><option value=\"ipg\">Identical Protein Groups</option><option data-ac_dict=\"medgen_disease_name\" value=\"medgen\">MedGen</option><option data-ac_dict=\"mesh_suggestions\" value=\"mesh\">MeSH</option><option value=\"ncbisearch\">NCBI Web Site</option><option value=\"nlmcatalog\">NLM Catalog</option><option value=\"nuccore\">Nucleotide</option><option value=\"omim\">OMIM</option><option value=\"pmc\">PMC</option><option value=\"popset\">PopSet</option><option value=\"probe\">Probe</option><option selected=\"selected\" value=\"protein\">Protein</option><option value=\"proteinclusters\">Protein Clusters</option><option value=\"pcassay\">PubChem BioAssay</option><option value=\"pccompound\">PubChem Compound</option><option value=\"pcsubstance\">PubChem Substance</option><option data-ac_dict=\"pm_related_queries_2\" value=\"pubmed\">PubMed</option><option value=\"snp\">SNP</option><option value=\"sparcle\">Sparcle</option><option value=\"sra\">SRA</option><option value=\"structure\">Structure</option><option value=\"taxonomy\">Taxonomy</option><option value=\"toolkit\">ToolKit</option><option value=\"toolkitall\">ToolKitAll</option><option value=\"toolkitbookgh\">ToolKitBookgh</option></select><div class=\"nowrap\"><label accesskey=\"/\" class=\"offscreen_noflow\" for=\"term\">Search term</label><div class=\"nowrap\"><input autocomplete=\"off\" class=\"jig-ncbiclearbutton jig-ncbiautocomplete\" data-jigconfig=\"isEnabled:false,disableUrl:'NcbiSearchBarAutoComplCtrl'\" data-sbconfig=\"ds:'no',pjs:'yes',afs:'yes'\" id=\"term\" name=\"term\" title=\"Search Protein\" type=\"text\" value=\"\"/></div><button class=\"button_search nowrap\" cmd=\"go\" id=\"search\" type=\"submit\">Search</button></div></div><ul class=\"searchlinks inline_list\"><set></set><li><a href=\"/protein/advanced\">Advanced</a></li><li class=\"help\"><a class=\"jig-ncbihelpwindow\" href=\"/books/NBK44864/\" id=\"help\" name=\"help\" target=\"ncbihelp\">Help</a></li></ul></div>\n",
       "</div>\n",
       "<input name=\"EntrezSystem2.PEntrez.Protein.Sequence_PageController.PreviousPageName\" sid=\"1\" type=\"hidden\" value=\"results\"/>\n",
       "<div class=\"\" id=\"faceted_search\">\n",
       "<h2 class=\"offscreen_noflow\">Result Filters</h2>\n",
       "</div>\n",
       "<div class=\"col nine_col\" id=\"maincontent\">\n",
       "<div class=\"content\">\n",
       "<div class=\"results_settings one_setting long_list\"><ul class=\"inline_list left display_settings\"><li><a class=\"jig-ncbipopper\" data-jigconfig=\"triggerPosition : 'bottom center',destPosition : 'top center',destSelector : '#display_settings_menu_report',                     hasArrow : false,openEvent : 'click',closeEvent : 'click',isTriggerElementCloseClick: false,addCloseButton : false,                      groupName: 'entrez_pg'\" href=\"#\" id=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.Display\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.Display\" sid=\"0\">FASTA<span class=\"tgt_dark\" href=\"#\"></span></a></li></ul><div class=\"disp_settings tabPopper\" id=\"display_settings_menu_report\"><fieldset class=\"format\"><legend>Format</legend><ul class=\"column_list\"><li><input format=\"\" id=\"docsum\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.sPresentation\" sid=\"1\" type=\"radio\" value=\"docsum\"/><label for=\"docsum\">Summary</label></li><li><input format=\"\" id=\"genpept\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.sPresentation\" sid=\"2\" type=\"radio\" value=\"genpept\"/><label for=\"genpept\">GenPept</label></li><li><input format=\"\" id=\"gpwithparts\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.sPresentation\" sid=\"3\" type=\"radio\" value=\"gpwithparts\"/><label for=\"gpwithparts\">GenPept (full)</label></li><li><input checked=\"true\" format=\"\" id=\"fasta\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.sPresentation\" sid=\"4\" type=\"radio\" value=\"fasta\"/><label for=\"fasta\">FASTA</label></li><li><input format=\"text\" id=\"fastatext\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.sPresentation\" sid=\"5\" type=\"radio\" value=\"fasta\"/><label for=\"fastatext\">FASTA (text)</label></li><li><input format=\"\" id=\"graph\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.sPresentation\" sid=\"6\" type=\"radio\" value=\"graph\"/><label for=\"graph\">Graphics</label></li><li><input format=\"\" id=\"asn1\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.sPresentation\" sid=\"7\" type=\"radio\" value=\"asn1\"/><label for=\"asn1\">ASN.1</label></li><li><input format=\"\" id=\"girevhist\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.sPresentation\" sid=\"8\" type=\"radio\" value=\"girevhist\"/><label for=\"girevhist\">Revision History</label></li><li><input format=\"text\" id=\"accnlisttext\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.sPresentation\" sid=\"9\" type=\"radio\" value=\"accnlist\"/><label for=\"accnlisttext\">Accession List</label></li><li><input format=\"text\" id=\"gilisttext\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.sPresentation\" sid=\"10\" type=\"radio\" value=\"gilist\"/><label for=\"gilisttext\">GI List</label></li></ul></fieldset></div><button class=\"button_apply ncbipopper-close-button\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.SetDisplay\" sid=\"2\" style=\"display:none\">Apply</button><h4 class=\"content_header send_to align_right jig-ncbipopper\" data-jigconfig=\"triggerPosition:'bottom center', destPosition : 'top center',destSelector : '#send_to_menu', hasArrow : false,                 openEvent : 'click',closeEvent : 'click', isTriggerElementCloseClick: false, addCloseButton:true, groupName: 'entrez_pg', adjustFit:'none'\" id=\"sendto\"><a class=\"tgt_dark\" href=\"#\" sourcecontent=\"send_to_menu\">Send to:</a><script type=\"text/javascript\">\n",
       "                        jQuery(document).ready( function () {\n",
       "                            jQuery(\"#send_to_menu input[type='radio']\").click( function () {\n",
       "                                var selectedValue = jQuery(this).val().toLowerCase();\n",
       "                                var selectedDiv = jQuery(\"#send_to_menu div.\" + selectedValue);\n",
       "                                \n",
       "                                if(selectedDiv.is(\":hidden\")){ \n",
       "                                jQuery(\"#send_to_menu div.submenu:visible\").slideUp();\n",
       "                                selectedDiv.slideDown();\n",
       "                                }\n",
       "                            });\n",
       "                        \n",
       "                        });\n",
       "                        \n",
       "                        jQuery(\"#sendto\").bind(\"ncbipopperclose\", function(){\n",
       "                            jQuery(\"#send_to_menu div.submenu:visible\").css(\"display\",\"none\");\n",
       "                            jQuery(\"#send_to_menu input[type='radio']:checked\").attr(\"checked\",false);\n",
       "                        });\n",
       "                        \n",
       "                    </script></h4><div class=\"tabPopper send_to\" id=\"send_to_menu\"><fieldset><legend>Choose Destination</legend><ul class=\"column_list\"><li><input id=\"dest_File\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.SendTo\" sid=\"1\" type=\"radio\" value=\"File\"/><label for=\"dest_File\">File</label></li><li><input id=\"dest_AddToClipboard\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.SendTo\" sid=\"2\" type=\"radio\" value=\"AddToClipboard\"/><label for=\"dest_AddToClipboard\">Clipboard</label></li><li><input id=\"dest_AddToCollections\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.SendTo\" sid=\"3\" type=\"radio\" value=\"AddToCollections\"/><label for=\"dest_AddToCollections\">Collections</label></li><li><input id=\"dest_AnalysisTool\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.SendTo\" sid=\"4\" type=\"radio\" value=\"AnalysisTool\"/><label for=\"dest_AnalysisTool\">Analysis Tool</label></li></ul></fieldset><div class=\"submenu file\" id=\"submenu_File\" style=\"display: none;\"><p class=\"hidden\" id=\"submenu_File_hint\"></p><ul><li><label for=\"file_format\">Format</label><select id=\"file_format\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.FFormat\" onchange=\"x_showgi(this)\" sid=\"1\"><option value=\"docsum\">Summary</option><option showgi=\"true\" value=\"genpept\">GenPept</option><option showgi=\"true\" value=\"gpwithparts\">GenPept (full)</option><option selected=\"selected\" showgi=\"true\" value=\"fasta\">FASTA</option><option value=\"asn1\">ASN.1</option><option value=\"xml\">XML</option><option value=\"gpc_xml\">INSDSeq XML</option><option showgi=\"true\" value=\"fasta_xml\">TinySeq XML</option><option value=\"ft\">Feature Table</option><option value=\"fasta_cds_na\">FASTA CDS</option><option format=\"text\" value=\"accnlist\">Accession List</option><option format=\"text\" value=\"gilist\">GI List</option><option format=\"text\" value=\"gff3\">GFF3</option></select></li><script type=\"text/javascript\">\n",
       "                    function x_showgi(el) {\n",
       "                    // ID-4455\n",
       "                    var elSeq = document.getElementById(\"file_showgi_section\");\n",
       "                        if (el.options[el.selectedIndex].getAttribute(\"showgi\") === \"true\") {\n",
       "                            elSeq.style.display = \"\";\n",
       "                        } else {\n",
       "                            elSeq.style.display = \"none\";\n",
       "                        }\n",
       "                    }\n",
       "                    setTimeout(function() {\n",
       "                        x_showgi(document.getElementById(\"file_format\"));\n",
       "                    }, 1000);\n",
       "                </script><li id=\"file_showgi_section\"><label for=\"file_showgi\" style=\"display:inline\">Show GI</label><input id=\"file_showgi\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.FShowgi\" sid=\"1\" type=\"checkbox\"/></li></ul><button class=\"button_apply file ncbipopper-close-button\" cmd=\"File\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.SendToSubmit\" sid=\"1\" type=\"submit\">Create File</button></div><div class=\"submenu addtoclipboard\" id=\"submenu_AddToClipboard\" style=\"display: none;\"><p class=\"hidden\" id=\"submenu_AddToClipboard_hint\"></p><button class=\"button_apply clipboard ncbipopper-close-button\" cmd=\"AddToClipboard\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.SendToSubmit\" sid=\"2\" type=\"submit\">Add to Clipboard</button></div><div class=\"submenu addtocollections\" id=\"submenu_AddToCollections\" style=\"display: none;\"><p class=\"hidden\" id=\"submenu_AddToCollections_hint\"></p><button class=\"button_apply collections ncbipopper-close-button\" cmd=\"AddToCollections\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.SendToSubmit\" sid=\"3\" type=\"submit\">Add to Collections</button></div><script type=\"text/javascript\">\n",
       "            function x_analysistool_evt(el) {\n",
       "            var elBtn = document.getElementById(\"analysistool_submit_btn\");\n",
       "            console.info(\" x_analysistool_evt, el=\", el);\n",
       "            elBtn.setAttribute(\"submit_url\", el.getAttribute(\"submit_url\"));\n",
       "            }\n",
       "        </script><div class=\"submenu analysistool\" id=\"submenu_AnalysisTool\" style=\"display: none;\"><fieldset><legend>Choose Sequence Analysis Tool</legend><ul class=\"column_list\"><li><input checked=\"checked\" id=\"analysistool_BLAST\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.radio_analysis_tool\" onclick=\"x_analysistool_evt(this)\" sid=\"1\" submit_url=\"//blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE=Proteins&amp;PROGRAM=blastp&amp;BLAST_PROGRAMS=blastp&amp;LINK_LOC=protein&amp;PAGE_TYPE=BlastSearch&amp;QUERY=NP_874360.1\" type=\"radio\" value=\"BLAST\"/><label for=\"analysistool_BLAST\">BLAST</label></li><li><input id=\"analysistool_CDD\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.radio_analysis_tool\" onclick=\"x_analysistool_evt(this)\" sid=\"4\" submit_url=\"/Structure/cdd/wrpsb.cgi?INPUT_TYPE=live&amp;SEQUENCE=NP_874360.1\" type=\"radio\" value=\"CDD\"/><label for=\"analysistool_CDD\">Conserved Domains Search</label></li></ul></fieldset><p class=\"submenu_analysistool_hint hidden\" id=\"submenu_AnalysisTool_hintNaN\"></p><button class=\"button_apply analysistool ncbipopper-close-button\" cmd=\"AnalysisTool\" id=\"analysistool_submit_btn\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.SendToSubmit\" sid=\"4\" submit_url=\"//blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE=Proteins&amp;PROGRAM=blastp&amp;BLAST_PROGRAMS=blastp&amp;LINK_LOC=protein&amp;PAGE_TYPE=BlastSearch&amp;QUERY=NP_874360.1\" type=\"submit\">Submit</button></div></div><div><input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.Db\" sid=\"2\" type=\"hidden\" value=\"protein\"/><input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.QueryKey\" sid=\"2\" type=\"hidden\" value=\"1\"/><input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.CurrFilter\" sid=\"2\" type=\"hidden\" value=\"all\"/><input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.ResultCount\" sid=\"2\" type=\"hidden\" value=\"1\"/><input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.ViewerParams\" sid=\"2\" type=\"hidden\"/><input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.Id\" sid=\"2\" type=\"hidden\" value=\"NP_874360.1\"/><input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.uid\" sid=\"2\" type=\"hidden\" value=\"33186887\"/><input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.FileFormat\" sid=\"2\" type=\"hidden\" value=\"fasta\"/><input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.LastPresentation\" sid=\"2\" type=\"hidden\" value=\"fasta\"/><input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.Presentation\" sid=\"2\" type=\"hidden\" value=\"fasta\"/><input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.PageSize\" sid=\"2\" type=\"hidden\" value=\"20\"/><input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.LastPageSize\" sid=\"2\" type=\"hidden\" value=\"20\"/><input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.Sort\" sid=\"2\" type=\"hidden\" value=\"\"/><input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.LastSort\" sid=\"2\" type=\"hidden\" value=\"\"/><input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.FileSort\" sid=\"2\" type=\"hidden\" value=\"\"/><input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.Format\" sid=\"2\" type=\"hidden\" value=\"\"/><input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.LastFormat\" sid=\"2\" type=\"hidden\" value=\"\"/><input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.PrevPageSize\" sid=\"2\" type=\"hidden\" value=\"20\"/><input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.PrevPresentation\" sid=\"2\" type=\"hidden\" value=\"fasta\"/><input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_DisplayBar.PrevSort\" sid=\"2\" type=\"hidden\" value=\"\"/><input id=\"coll_startindex\" name=\"CollectionStartIndex\" type=\"hidden\" value=\"1\"/></div></div>\n",
       "<div>\n",
       "</div>\n",
       "<div class=\"\">\n",
       "<div><span class=\"nowrap\" id=\"result_sel\"></span><input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_ResultsController.ResultCount\" sid=\"1\" type=\"hidden\" value=\"1\"/><input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_ResultsController.RunLastQuery\" sid=\"1\" type=\"hidden\"/><input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_ResultsController.AccnsFromResult\" sid=\"1\" type=\"hidden\" value=\"\"/></div>\n",
       "</div>\n",
       "<div class=\"empty\" id=\"messagearea\">\n",
       "</div>\n",
       "<div class=\"seqrprt seqviewer\">\n",
       "<div class=\"rprtheader\">\n",
       "<h1>glutathione peroxidase 6 precursor [Homo sapiens]</h1>\n",
       "<p class=\"itemid\">NCBI Reference Sequence: NP_874360.1</p>\n",
       "<p class=\"aux\">\n",
       "<span class=\"rprtlinks\">\n",
       "<a class=\"dblinks\" href=\"/protein/NP_874360.1?report=genpept\" id=\"ReportShortCut4\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.SequenceViewer.Sequence_ViewerTitle.ReportShortCut\" ref=\"ordinalpos=1&amp;ncbi_uid=33186887&amp;link_uid=33186887&amp;linksrc=fasta_genpept\" sid=\"14\">GenPept</a>\n",
       "<a class=\"dblinks\" href=\"/ipg/NP_874360.1\" ref=\"ordinalpos=1&amp;ncbi_uid=33186887&amp;link_uid=33186887&amp;linksrc=fasta_ipg\">Identical Proteins</a>\n",
       "<a class=\"dblinks\" href=\"/protein/NP_874360.1?report=graph\" id=\"ReportShortCut7\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.SequenceViewer.Sequence_ViewerTitle.ReportShortCut\" ref=\"ordinalpos=1&amp;ncbi_uid=33186887&amp;link_uid=33186887&amp;linksrc=fasta_graph\" sid=\"17\">Graphics</a>\n",
       "</span>\n",
       "</p>\n",
       "<div class=\"hidden\">\n",
       "<input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.SequenceViewer.Sequence_ViewerTitle.Database\" sid=\"1\" type=\"hidden\" value=\"protein\"/>\n",
       "<input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.SequenceViewer.Sequence_ViewerTitle.ResultCount\" sid=\"1\" type=\"hidden\" value=\"1\"/>\n",
       "<input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.SequenceViewer.Sequence_ViewerTitle.Report\" sid=\"1\" type=\"hidden\" value=\"fasta\"/>\n",
       "</div>\n",
       "</div>\n",
       "<div>\n",
       "<div class=\"seq gbff\" id=\"viewercontent1\" sequencesize=\"14852\" val=\"33186887\" virtualsequence=\"true\"></div>\n",
       "<div class=\"hidden\">\n",
       "<input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.SequenceViewer.Sequence_ViewerReport.db\" sid=\"1\" type=\"hidden\" value=\"protein\"/>\n",
       "<input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.SequenceViewer.Sequence_ViewerReport.report\" sid=\"1\" type=\"hidden\" value=\"fasta\"/>\n",
       "<input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.SequenceViewer.Sequence_ViewerReport.maxdownloadsize\" sid=\"1\" type=\"hidden\" value=\"1000000\"/>\n",
       "<input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.SequenceViewer.Sequence_ViewerReport.display_type\" sid=\"1\" type=\"hidden\" value=\"single\"/>\n",
       "<input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.SequenceViewer.Sequence_ViewerReport.ItemCount\" sid=\"1\" type=\"hidden\" value=\"1\"/>\n",
       "<input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.SequenceViewer.Sequence_ViewerReport.InUse\" sid=\"1\" type=\"hidden\" value=\"\"/>\n",
       "<input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.SequenceViewer.Sequence_ViewerReport.ExpandGaps\" sid=\"1\" type=\"hidden\" value=\"\"/>\n",
       "</div>\n",
       "<div id=\"fh_bar\">\n",
       "<div id=\"fh_bar_details_body\"></div>\n",
       "<div id=\"fh_bar_warning\"></div>\n",
       "<div class=\"fh_bar_left\">\n",
       "<select id=\"fh_bar_select\" ref=\"$ref\">\n",
       "<option>no features</option>\n",
       "</select>\n",
       "<a href=\"#\" id=\"fh_bar_to_feature\" ref=\"discoid=featurehighlight&amp;log$=featurehighlight\" title=\"Jump to the feature\">Feature</a>\n",
       "<div class=\"hide nav_control\" id=\"feat_nav_control\">\n",
       "<button href=\"#\" id=\"fh_bar_first\" ref=\"discoid=featurehighlight&amp;log$=featurehighlight\">First</button>\n",
       "<button id=\"fh_bar_previous\" ref=\"discoid=featurehighlight&amp;log$=featurehighlight\">Previous</button>\n",
       "<span id=\"fh_bar_total\" ref=\"discoid=featurehighlight&amp;log$=featurehighlight\"></span>\n",
       "<button id=\"fh_bar_next\" ref=\"discoid=featurehighlight&amp;log$=featurehighlight\">Next</button>\n",
       "<button id=\"fh_bar_last\" ref=\"discoid=featurehighlight&amp;log$=featurehighlight\">Last</button>\n",
       "</div>\n",
       "<span id=\"fh_bar_message\"></span>\n",
       "</div>\n",
       "<div class=\"fh_bar_right\">\n",
       "<span>Display: </span>\n",
       "<a href=\"#\" id=\"fh_bar_fasta\" ref=\"discoid=featurehighlight&amp;log$=featurehighlight\">FASTA</a>\n",
       "<a href=\"#\" id=\"fh_bar_genbank\" ref=\"discoid=featurehighlight&amp;log$=featurehighlight\">GenBank</a>\n",
       "<a href=\"/books/NBK44863/#sequencesquickstart.How_can_I_locate_and\" ref=\"discoid=featurehighlight&amp;log$=featurehighlight\">Help</a>\n",
       "<button id=\"fh_bar_hide\" ref=\"discoid=featurehighlight&amp;log$=featurehighlight\">Close</button>\n",
       "</div>\n",
       "<span class=\"fh_bar_details_ph\">\n",
       "<a class=\"tgt_dark\" href=\"#\" id=\"fh_bar_details\">Details</a>\n",
       "</span>\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<div id=\"messagearea_bottom\">\n",
       "</div>\n",
       "<div class=\"bottom\">\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"supplemental col three_col last\">\n",
       "<h2 class=\"offscreen_noflow\">Supplemental Content</h2>\n",
       "<div>\n",
       "<!-- ID-4171  GenBank Sammies ad on nucleotide docsum and record pages -->\n",
       "<div class=\"portlet ctrl\">\n",
       "<div class=\"portlet_head\">\n",
       "<div class=\"portlet_title\">\n",
       "<h3>Change region shown</h3>\n",
       "</div>\n",
       "<a class=\"portlet_shutter shutter_closed\" href=\"#\" id=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_ViewerChangeRegion.Shutter\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_ViewerChangeRegion.Shutter\" sid=\"1\" title=\"Show/hide content\"></a>\n",
       "</div>\n",
       "<div class=\"portlet_content pagectrl\" style=\"display: none;\">\n",
       "<p>\n",
       "<input checked=\"checked\" id=\"crwholeregion\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_ViewerChangeRegion.ChangeRegion\" sid=\"1\" type=\"radio\" value=\"whole\"/>\n",
       "<label for=\"crwholeregion\">Whole sequence</label>\n",
       "<br/>\n",
       "<input id=\"crselregion\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_ViewerChangeRegion.ChangeRegion\" sid=\"2\" type=\"radio\" value=\"sel\"/>\n",
       "<label for=\"crselregion\">Selected region</label>\n",
       "<br/>\n",
       "<input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_ViewerChangeRegion.Topology\" sid=\"1\" type=\"hidden\" value=\"linear\"/>\n",
       "<label for=\"crfrom\"> from: </label>\n",
       "<input class=\"text\" id=\"crfrom\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_ViewerChangeRegion.From\" sid=\"1\" size=\"8\" type=\"text\" value=\"\"/>\n",
       "<label for=\"crto\"> to: </label>\n",
       "<input class=\"text\" id=\"crto\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_ViewerChangeRegion.To\" sid=\"1\" size=\"8\" type=\"text\" value=\"\"/>\n",
       "</p>\n",
       "<div class=\"ctrlbtn\">\n",
       "<button id=\"updateselregion\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_ViewerChangeRegion.SetSelRegion\" sid=\"1\">Update View</button>\n",
       "<input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_ViewerChangeRegion.Max\" sid=\"1\" type=\"hidden\" value=\"221\"/>\n",
       "<input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_ViewerChangeRegion.Contig\" sid=\"1\" type=\"hidden\" value=\"\"/>\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"empty\"></div>\n",
       "<div class=\"portlet brieflink\">\n",
       "<div class=\"portlet_head\">\n",
       "<div class=\"portlet_title\">\n",
       "<h3>Analyze this sequence</h3>\n",
       "</div>\n",
       "<a class=\"portlet_shutter\" href=\"#\" id=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_ToolsList.Shutter\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_ToolsList.Shutter\" pgsec_name=\"Sequence_ToolsList\" remembercollapsed=\"true\" sid=\"1\" title=\"Show/hide content\"></a>\n",
       "</div>\n",
       "<div class=\"portlet_content Sequence_ToolsList\">\n",
       "<ul>\n",
       "<li class=\"brieflinkpopper\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"//blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE=Proteins&amp;PROGRAM=blastp&amp;BLAST_PROGRAMS=blastp&amp;QUERY=NP_874360.1&amp;LINK_LOC=protein&amp;PAGE_TYPE=BlastSearch\" ref=\"log$=seqview_list_blast&amp;discoId=seqtools\">Run BLAST</a>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">Find regions of similarity between this sequence and other sequences\n",
       "                    using BLAST.</div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"/Structure/cdd/wrpsb.cgi?INPUT_TYPE=live&amp;SEQUENCE=NP_874360.1\" ref=\"log$=seqview_list_cdsearch&amp;discoId=seqtools\">Identify Conserved Domains</a>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">View conserved domains detected in this protein sequence using\n",
       "                            CD-search.</div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"/nuccore/NP_874360.1?&amp;feature=any\" ref=\"log$=seqview_list_highlight&amp;highlightsource=any&amp;discoId=seqtools\">Highlight Sequence Features</a>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">Opens the Highlight Feature Bar and highlights feature annotations\n",
       "                        from the FEATURES table of the record. The Highlight Feature Bar can be used\n",
       "                        to navigate to and highlight other features and provides links to display\n",
       "                        the highlighted region separately. Links in the FEATURES table will also\n",
       "                        highlight the corresponding region of the sequence. <a href=\"/books/NBK44863/#sequencesquickstart.How_can_I_locate_and\">More...</a></div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"javascript:Portal.$send('SeqSearchClicked')\" ref=\"log$=seqview_list_find&amp;discoId=seqtools\">Find in this Sequence</a>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">Finds sub-sequences or patterns in the sequence and highlights the\n",
       "                        matching regions. The tool works with standard single letter nucleotide or\n",
       "                        protein codes including ambiguities and can match Prosite patterns in\n",
       "                        protein sequences. <a href=\"/books/NBK44863/#sequencesquickstart.How_can_I_search_for\">More...</a></div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"/genome/gdv/browser/?context=Protein&amp;acc=NP_874360.1\">Show in Genome Data Viewer</a>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">Opens the Genome Data Viewer, NCBI's genome browser, at the genomic\n",
       "                        location at which this sequence is annotated. View this sequence in the\n",
       "                        context of other features annotated at the same location and browse to other\n",
       "                        genome regions. </div>\n",
       "</li>\n",
       "</ul>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"portlet brieflink\">\n",
       "<div class=\"portlet_head\">\n",
       "<div class=\"portlet_title\">\n",
       "<h3>Articles about the GPX6 gene</h3>\n",
       "</div>\n",
       "<a class=\"portlet_shutter\" href=\"#\" id=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_DiscoveryPubMedRA.Shutter\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_DiscoveryPubMedRA.Shutter\" pgsec_name=\"Sequence_DiscoveryPubMedRA\" remembercollapsed=\"true\" sid=\"1\" title=\"Show/hide content\"></a>\n",
       "</div>\n",
       "<div class=\"portlet_content\">\n",
       "<ul>\n",
       "<li class=\"brieflinkpopper two_line\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"/pubmed/15910762\" ref=\"ordinalpos=1&amp;linkpos=1&amp;log$=seqview_pubmed&amp;ncbi_uid=33186887&amp;link_uid=15910762\">HIV-1 viral proteins gp120 and Tat induce oxidative stress in brain endothelial cells.</a>\n",
       "<span class=\"source\">[Brain Res. 2005]</span>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">HIV-1 viral proteins gp120 and Tat induce oxidative stress in brain endothelial cells.<div class=\"brieflinkpopdesc\"><em class=\"author\">Price TO, Ercal N, Nakaoke R, Banks WA. </em><em class=\"cit\">Brain Res. 2005 May 31; 1045(1-2):57-63. Epub 2005 Apr 19.</em></div></div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper two_line\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"/pubmed/11368344\" ref=\"ordinalpos=1&amp;linkpos=2&amp;log$=seqview_pubmed&amp;ncbi_uid=33186887&amp;link_uid=11368344\">Human immunodeficiency virus type 1 Tat protein impairs selenoglutathione peroxidase expression and activity by a mechanism independent of cellular selenium uptake: consequences on cellular resistance to UV-A radiation.</a>\n",
       "<span class=\"source\">[Arch Biochem Biophys. 2001]</span>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">Human immunodeficiency virus type 1 Tat protein impairs selenoglutathione peroxidase expression and activity by a mechanism independent of cellular selenium uptake: consequences on cellular resistance to UV-A radiation.<div class=\"brieflinkpopdesc\"><em class=\"author\">Richard MJ, Guiraud P, Didier C, Seve M, Flores SC, Favier A. </em><em class=\"cit\">Arch Biochem Biophys. 2001 Feb 15; 386(2):213-20. </em></div></div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper two_line\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"/pubmed/10652368\" ref=\"ordinalpos=1&amp;linkpos=3&amp;log$=seqview_pubmed&amp;ncbi_uid=33186887&amp;link_uid=10652368\">Molecular mechanism of decreased glutathione content in human immunodeficiency virus type 1 Tat-transgenic mice.</a>\n",
       "<span class=\"source\">[J Biol Chem. 2000]</span>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">Molecular mechanism of decreased glutathione content in human immunodeficiency virus type 1 Tat-transgenic mice.<div class=\"brieflinkpopdesc\"><em class=\"author\">Choi J, Liu RM, Kundu RK, Sangiorgi F, Wu W, Maxson R, Forman HJ. </em><em class=\"cit\">J Biol Chem. 2000 Feb 4; 275(5):3693-8. </em></div></div>\n",
       "</li>\n",
       "</ul>\n",
       "<a class=\"seemore\" href=\"/pubmed?linkname=protein_pubmed_weighted&amp;from_uid=33186887\" ref=\"ordinalpos=1&amp;log$=seqview_pubmed&amp;ncbi_uid=33186887\">See all...</a>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"portlet brieflink\">\n",
       "<div class=\"portlet_head\">\n",
       "<div class=\"portlet_title\">\n",
       "<h3>Pathways for the GPX6 gene</h3>\n",
       "</div>\n",
       "<a class=\"portlet_shutter\" href=\"#\" id=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_DiscoveryBioSystems.Shutter\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_DiscoveryBioSystems.Shutter\" pgsec_name=\"Sequence_DiscoveryBioSystems\" remembercollapsed=\"true\" sid=\"1\" title=\"Show/hide content\"></a>\n",
       "</div>\n",
       "<div class=\"portlet_content Sequence_DiscoveryBioSystems\">\n",
       "<ul>\n",
       "<li class=\"brieflinkpopper\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"/biosystems/835410?Sel=geneid:257202#show=genes\" ref=\"linkpos=1&amp;log$=seqview_biosystems&amp;ncbi_uid=33186887&amp;link_uid=835410\">Thyroid hormone synthesis</a>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">Thyroid hormone synthesis<em class=\"desc\">Thyroid hormones triiodothyronine (T3) and thyroxine (T4) are essential for normal development, growth and metabolic homeostasis in all vertebrates, and synthesized in the thyroid gland. The functiona ...\n",
       "                            </em><em class=\"note\">[bsid835410]</em></div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"/biosystems/1270420?Sel=geneid:257202#show=genes\" ref=\"linkpos=2&amp;log$=seqview_biosystems&amp;ncbi_uid=33186887&amp;link_uid=1270420\">Detoxification of Reactive Oxygen Species</a>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">Detoxification of Reactive Oxygen Species<em class=\"desc\">Reactive oxygen species such as superoxide (O2.-), peroxides (ROOR), singlet oxygen, peroxynitrite (ONOO-), and hydroxyl radical (OH.) are generated by cellular processes such as respiration (reviewed ...\n",
       "                            </em><em class=\"note\">[bsid1270420]</em></div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"/biosystems/1108784?Sel=geneid:257202#show=genes\" ref=\"linkpos=3&amp;log$=seqview_biosystems&amp;ncbi_uid=33186887&amp;link_uid=1108784\">reactive oxygen species degradation</a>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">reactive oxygen species degradation<em class=\"desc\">General Background  All organisms living in an aerobic environment are exposed to reactive oxygen species (ROS) that are formed through metabolic processes and various environmental stresses . ROS are ...\n",
       "                            </em><em class=\"note\">[bsid1108784]</em></div>\n",
       "</li>\n",
       "</ul>\n",
       "<a class=\"seemore\" href=\"/protein?linkname=protein_biosystems&amp;from_uid=33186887\" ref=\"log$=seqview_biosystems&amp;ncbi_uid=33186887\">See all...</a>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"portlet\">\n",
       "<div class=\"portlet_head\">\n",
       "<div class=\"portlet_title\">\n",
       "<h3>Reference sequence information</h3>\n",
       "</div>\n",
       "<a class=\"portlet_shutter\" href=\"#\" id=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_DiscoveryRefSeqs.Shutter\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_DiscoveryRefSeqs.Shutter\" pgsec_name=\"Sequence_DiscoveryRefSeqs\" remembercollapsed=\"true\" sid=\"1\" title=\"Show/hide content\"></a>\n",
       "</div>\n",
       "<div class=\"portlet_content\">\n",
       "<ul>\n",
       "<li class=\"\">\n",
       "<a href=\"/nuccore/33186886\" ref=\"log$=seqview_refseq_mRNA&amp;ncbi_uid=33186887&amp;link_uid=33186886\">RefSeq mRNA</a>\n",
       "<div class=\"long_info\">See reference mRNA sequence for the GPX6 gene (NM_182701.1).</div>\n",
       "</li>\n",
       "</ul>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"portlet\">\n",
       "<div class=\"portlet_head\">\n",
       "<div class=\"portlet_title\">\n",
       "<h3>More about the GPX6 gene</h3>\n",
       "</div>\n",
       "<a class=\"portlet_shutter\" href=\"#\" id=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_DiscoveryGeneInfo.Shutter\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_DiscoveryGeneInfo.Shutter\" pgsec_name=\"Sequence_DiscoveryGeneInfo\" remembercollapsed=\"true\" sid=\"1\" title=\"Show/hide content\"></a>\n",
       "</div>\n",
       "<div class=\"portlet_content\">\n",
       "<ul>\n",
       "<li class=\"\">\n",
       "<a href=\"/sites/entrez?db=gene&amp;cmd=retrieve&amp;list_uids=257202\" ref=\"log$=seqview_gene&amp;ncbi_uid=33186887&amp;link_uid=257202\">The protein encoded by this gene belongs to the glutathione peroxidase family, members of which catalyze the reduction of hydrogen peroxide,...</a>\n",
       "<div class=\"long_info\">Also Known As: GPX5p, GPXP3, GPx-6, GS...</div>\n",
       "</li>\n",
       "</ul>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"portlet\">\n",
       "<div class=\"portlet_head\">\n",
       "<div class=\"portlet_title\">\n",
       "<h3>Homologs of the GPX6 gene</h3>\n",
       "</div>\n",
       "<a class=\"portlet_shutter\" href=\"#\" id=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_DiscoveryHomologs.Shutter\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.Sequence_DiscoveryHomologs.Shutter\" pgsec_name=\"Sequence_DiscoveryHomologs\" remembercollapsed=\"true\" sid=\"1\" title=\"Show/hide content\"></a>\n",
       "</div>\n",
       "<div class=\"portlet_content\">\n",
       "<ul>\n",
       "<li class=\"\">\n",
       "<a href=\"/protein?Db=homologene&amp;Cmd=Retrieve&amp;list_uids=130008\" ref=\"log$=seqview_homolog&amp;ncbi_uid=33186887&amp;link_uid=130008\">The GPX6 gene is conserved in chimpanzee, Rhesus monkey, dog, cow, mouse, and rat.</a>\n",
       "</li>\n",
       "</ul>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"portlet brieflink\">\n",
       "<div class=\"portlet_head\">\n",
       "<div class=\"portlet_title\">\n",
       "<h3>Related information</h3>\n",
       "</div>\n",
       "<a class=\"portlet_shutter\" href=\"#\" id=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_DiscoveryDbLinks.Shutter\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_DiscoveryDbLinks.Shutter\" pgsec_name=\"discovery_db_links\" remembercollapsed=\"true\" sid=\"1\" title=\"Show/hide content\"></a>\n",
       "</div>\n",
       "<div class=\"portlet_content DiscoveryDbLinks\">\n",
       "<ul>\n",
       "<li class=\"brieflinkpopper\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"/bioproject?LinkName=protein_bioproject&amp;from_uid=33186887\" ref=\"log$=recordlinks&amp;logdbfrom=protein&amp;ncbi_uid=33186887&amp;ordinalpos=1&amp;linkpos=1\">BioProject</a>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">Proteins related to BioProjects</div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"/nuccore?LinkName=protein_nuccore&amp;from_uid=33186887\" ref=\"log$=recordlinks&amp;logdbfrom=protein&amp;ncbi_uid=33186887&amp;ordinalpos=1&amp;linkpos=2\">Nucleotide</a>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">Nucleotide sequence from coding region</div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"/pubmed?LinkName=protein_pubmed&amp;from_uid=33186887\" ref=\"log$=recordlinks&amp;logdbfrom=protein&amp;ncbi_uid=33186887&amp;ordinalpos=1&amp;linkpos=3\">PubMed</a>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">PubMed articles cited by protein sequence record</div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"/taxonomy?LinkName=protein_taxonomy&amp;from_uid=33186887\" ref=\"log$=recordlinks&amp;logdbfrom=protein&amp;ncbi_uid=33186887&amp;ordinalpos=1&amp;linkpos=4\">Taxonomy</a>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">Taxonomy sequences associated with protein record</div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"/biosystems?LinkName=protein_biosystems&amp;from_uid=33186887\" ref=\"log$=recordlinks&amp;logdbfrom=protein&amp;ncbi_uid=33186887&amp;ordinalpos=1&amp;linkpos=5\">BioSystems</a>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">BioSystems</div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"/CCDS/CcdsBrowse.cgi?REQUEST=PROTID&amp;DATA=33186887\" ref=\"log$=recordlinks&amp;logdbfrom=protein&amp;ncbi_uid=33186887&amp;ordinalpos=1&amp;linkpos=6\">CCDS</a>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">Link to Consensus CDS</div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"https://www.ncbi.nlm.nih.gov/Structure/cdd/wrpsb.cgi?INPUT_TYPE=precalc&amp;SEQUENCE=33186887\" ref=\"log$=recordlinks&amp;logdbfrom=protein&amp;ncbi_uid=33186887&amp;ordinalpos=1&amp;linkpos=7\">CDD Search Results</a>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">Display conserved domains within a protein</div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"/cdd?LinkName=protein_cdd_concise_2&amp;from_uid=33186887\" ref=\"log$=recordlinks&amp;logdbfrom=protein&amp;ncbi_uid=33186887&amp;ordinalpos=1&amp;linkpos=8\">Conserved Domains (Concise)</a>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">Link to a concise list of conserved domains in a protein</div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"/cdd?LinkName=protein_cdd&amp;from_uid=33186887\" ref=\"log$=recordlinks&amp;logdbfrom=protein&amp;ncbi_uid=33186887&amp;ordinalpos=1&amp;linkpos=9\">Conserved Domains (Full)</a>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">Link to conserved domains within a protein</div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"https://www.ncbi.nlm.nih.gov/Structure/lexington/lexington.cgi?cmd=prot&amp;uid=33186887\" ref=\"log$=recordlinks&amp;logdbfrom=protein&amp;ncbi_uid=33186887&amp;ordinalpos=1&amp;linkpos=10\">Domain Relatives</a>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">Display proteins with similar conserved domain architecture</div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"/nuccore?LinkName=protein_nuccore_mrna&amp;from_uid=33186887\" ref=\"log$=recordlinks&amp;logdbfrom=protein&amp;ncbi_uid=33186887&amp;ordinalpos=1&amp;linkpos=11\">Encoding mRNA</a>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">Link from protein to encoding mRNA</div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"/pmc?LinkName=protein_pmc&amp;from_uid=33186887\" ref=\"log$=recordlinks&amp;logdbfrom=protein&amp;ncbi_uid=33186887&amp;ordinalpos=1&amp;linkpos=12\">Full text in PMC</a>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">Free full text articles in PMC</div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"/gene?LinkName=protein_gene&amp;from_uid=33186887\" ref=\"log$=recordlinks&amp;logdbfrom=protein&amp;ncbi_uid=33186887&amp;ordinalpos=1&amp;linkpos=13\">Gene</a>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">Link to related Genes</div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"https://www.ncbi.nlm.nih.gov/SNP/CGI/genesnpview.cgi?fromdb=protein&amp;id=33186887\" ref=\"log$=recordlinks&amp;logdbfrom=protein&amp;ncbi_uid=33186887&amp;ordinalpos=1&amp;linkpos=14\">GeneView in dbSNP</a>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">SNPs linked from GeneView</div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"/genome?LinkName=protein_genome&amp;from_uid=33186887\" ref=\"log$=recordlinks&amp;logdbfrom=protein&amp;ncbi_uid=33186887&amp;ordinalpos=1&amp;linkpos=15\">Genome</a>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">Genome record encoding protein sequence</div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"/homologene?LinkName=protein_homologene&amp;from_uid=33186887\" ref=\"log$=recordlinks&amp;logdbfrom=protein&amp;ncbi_uid=33186887&amp;ordinalpos=1&amp;linkpos=16\">HomoloGene</a>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">Related HomoloGene</div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"/omim?LinkName=protein_omim&amp;from_uid=33186887\" ref=\"log$=recordlinks&amp;logdbfrom=protein&amp;ncbi_uid=33186887&amp;ordinalpos=1&amp;linkpos=17\">OMIM</a>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">OMIM records associated with protein sequence</div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"/protein?LinkName=protein_protein_refseq2uniprot&amp;from_uid=33186887\" ref=\"log$=recordlinks&amp;logdbfrom=protein&amp;ncbi_uid=33186887&amp;ordinalpos=1&amp;linkpos=18\">Protein (UniProtKB)</a>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">Protein gi (refseq) to protein gi (UniProtKB)</div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"/pubmed?LinkName=protein_pubmed_refseq&amp;from_uid=33186887\" ref=\"log$=recordlinks&amp;logdbfrom=protein&amp;ncbi_uid=33186887&amp;ordinalpos=1&amp;linkpos=19\">PubMed (RefSeq)</a>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">Link to RefSeq PubMed articles</div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"/pubmed?LinkName=protein_pubmed_weighted&amp;from_uid=33186887\" ref=\"log$=recordlinks&amp;logdbfrom=protein&amp;ncbi_uid=33186887&amp;ordinalpos=1&amp;linkpos=20\">PubMed (Weighted)</a>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">Links to pubmed</div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"/structure?LinkName=protein_structure_related_list&amp;from_uid=33186887\" ref=\"log$=recordlinks&amp;logdbfrom=protein&amp;ncbi_uid=33186887&amp;ordinalpos=1&amp;linkpos=21\">Related Structures (List)</a>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">List of related structures</div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"https://structure.ncbi.nlm.nih.gov/Structure/cblast/cblast.cgi?client=entrez&amp;query_gi=33186887\" ref=\"log$=recordlinks&amp;logdbfrom=protein&amp;ncbi_uid=33186887&amp;ordinalpos=1&amp;linkpos=22\">Related Structures (Summary)</a>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">Related structures</div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"/snp?LinkName=protein_snp&amp;from_uid=33186887\" ref=\"log$=recordlinks&amp;logdbfrom=protein&amp;ncbi_uid=33186887&amp;ordinalpos=1&amp;linkpos=23\">SNP</a>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">Related SNPs</div>\n",
       "</li>\n",
       "</ul>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"portlet brieflink\">\n",
       "<div class=\"portlet_head\">\n",
       "<div class=\"portlet_title\">\n",
       "<h3>LinkOut to external resources</h3>\n",
       "</div>\n",
       "<a class=\"portlet_shutter\" href=\"#\" id=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.LinkOutPortlet.Shutter\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.Sequence_SingleItemSupl.Sequence_ViewerGenbankSidePanel.LinkOutPortlet.Shutter\" pgsec_name=\"linkout_portlet\" remembercollapsed=\"true\" sid=\"1\" title=\"Show/hide content\"></a>\n",
       "</div>\n",
       "<div class=\"portlet_content\">\n",
       "<ul>\n",
       "<li class=\"brieflinkpopper two_line\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"http://bioinf.umbc.edu/DMDM/gene_prot_page.php?search_type=protein&amp;id=33186887\" ref=\"itool=LinkOutPortlet&amp;PrId=7642&amp;uid=33186887&amp;db=protein&amp;log$=linkoutlink\">NP_874360</a>\n",
       "<span class=\"source\">[Domain Mapping of Disease Mut...]</span>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">NP_874360<div class=\"brieflinkpopdesc\">Domain Mapping of Disease Mutations</div></div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper two_line\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"http://www.ensembl.org/id/ENSP00000354581\" ref=\"itool=LinkOutPortlet&amp;PrId=7853&amp;uid=33186887&amp;db=protein&amp;log$=linkoutlink\">Ensembl</a>\n",
       "<span class=\"source\">[Ensembl]</span>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">Ensembl</div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper two_line\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"http://www.ensembl.org/id/ENSP00000486277\" ref=\"itool=LinkOutPortlet&amp;PrId=7853&amp;uid=33186887&amp;db=protein&amp;log$=linkoutlink\">Ensembl</a>\n",
       "<span class=\"source\">[Ensembl]</span>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">Ensembl</div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper two_line\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"https://oncomx.org/searchview/?gene=GPX6\" ref=\"itool=LinkOutPortlet&amp;PrId=10163&amp;uid=33186887&amp;db=protein&amp;log$=linkoutlink\">OncoMX</a>\n",
       "<span class=\"source\">[OncoMX]</span>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">OncoMX</div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper two_line\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"https://www.orthodb.org/?query=NP_874360\" ref=\"itool=LinkOutPortlet&amp;PrId=8178&amp;uid=33186887&amp;db=protein&amp;log$=linkoutlink\">The Hierarchical Catalog of Orthologs</a>\n",
       "<span class=\"source\">[OrthoDB catalog of orthologs]</span>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">The Hierarchical Catalog of Orthologs<div class=\"brieflinkpopdesc\">OrthoDB catalog of orthologs</div></div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper two_line\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"http://www.pantherdb.org/genes/transcriptDetail.do?tranAcc=P59796\" ref=\"itool=LinkOutPortlet&amp;PrId=4118&amp;uid=33186887&amp;db=protein&amp;log$=linkoutlink\">Transcript/Protein Information</a>\n",
       "<span class=\"source\">[PANTHER Classification System]</span>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">Transcript/Protein Information<div class=\"brieflinkpopdesc\">PANTHER Classification System</div></div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper two_line\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"http://purl.obolibrary.org/obo/PR_P59796\" ref=\"itool=LinkOutPortlet&amp;PrId=7576&amp;uid=33186887&amp;db=protein&amp;log$=linkoutlink\">Protein Ontology Consortium</a>\n",
       "<span class=\"source\">[Protein Ontology Consortium]</span>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">Protein Ontology Consortium</div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper two_line\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"https://www.labome.com/gene/human/GPX6-antibody.html\" ref=\"itool=LinkOutPortlet&amp;PrId=5753&amp;uid=33186887&amp;db=protein&amp;log$=linkoutlink\">reagents</a>\n",
       "<span class=\"source\">[ExactAntigen/Labome]</span>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">reagents<div class=\"brieflinkpopdesc\">ExactAntigen/Labome</div></div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper two_line\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"https://www.lsbio.com/targets/gpx6-gpx-6/g200315?prefix=a#SearchTable_div\" ref=\"itool=LinkOutPortlet&amp;PrId=8607&amp;uid=33186887&amp;db=protein&amp;log$=linkoutlink\">LSBio GPX6 / GPX-6 Antibodies</a>\n",
       "<span class=\"source\">[LifeSpan BioSciences, Inc.]</span>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">LSBio GPX6 / GPX-6 Antibodies<div class=\"brieflinkpopdesc\">LifeSpan BioSciences, Inc.</div></div>\n",
       "</li>\n",
       "<li class=\"brieflinkpopper two_line\">\n",
       "<a class=\"brieflinkpopperctrl\" href=\"https://www.lsbio.com/targets/gpx6-gpx-6/g200315?prefix=r#SearchTable_div\" ref=\"itool=LinkOutPortlet&amp;PrId=8607&amp;uid=33186887&amp;db=protein&amp;log$=linkoutlink\">LSBio GPX6 / GPX-6 Proteins</a>\n",
       "<span class=\"source\">[LifeSpan BioSciences, Inc.]</span>\n",
       "<div class=\"brieflinkpop offscreen_noflow\">LSBio GPX6 / GPX-6 Proteins<div class=\"brieflinkpopdesc\">LifeSpan BioSciences, Inc.</div></div>\n",
       "</li>\n",
       "</ul>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"portlet\">\n",
       "<div class=\"portlet_head\">\n",
       "<div class=\"portlet_title\">\n",
       "<h3>Recent activity</h3>\n",
       "</div>\n",
       "<a class=\"portlet_shutter\" href=\"#\" id=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.HistoryDisplay.Shutter\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.HistoryDisplay.Shutter\" pgsec_name=\"recent_activity\" remembercollapsed=\"true\" sid=\"1\" title=\"Show/hide content\"></a>\n",
       "</div>\n",
       "<div class=\"portlet_content\">\n",
       "<div class=\"HTOn\" id=\"HTDisplay\">\n",
       "<input name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.HistoryDisplay.Cmd\" sid=\"1\" type=\"hidden\"/>\n",
       "<div class=\"action\">\n",
       "<a cmd=\"ClearHT\" href=\"?cmd=ClearHT&amp;\" id=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.HistoryDisplay.ClearHistory\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.HistoryDisplay.ClearHistory\" onclick=\"return false;\" realname=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.HistoryDisplay.ClearHistory\" sid=\"1\">\n",
       "                                Clear\n",
       "                            </a>\n",
       "<a class=\"HTOn\" cmd=\"HTOff\" href=\"?cmd=HTOff&amp;\" id=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.HistoryDisplay.HistoryToggle\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.HistoryDisplay.HistoryToggle\" onclick=\"return false;\" realname=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.HistoryDisplay.HistoryToggle\" sid=\"1\">\n",
       "                                Turn Off                        \n",
       "                            </a>\n",
       "<a class=\"HTOff\" cmd=\"HTOn\" href=\"?cmd=HTOn&amp;\" id=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.HistoryDisplay.HistoryToggle\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.HistoryDisplay.HistoryToggle\" onclick=\"return false;\" realname=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.HistoryDisplay.HistoryToggle\" sid=\"2\">\n",
       "                                Turn On                        \n",
       "                            </a>\n",
       "</div>\n",
       "<ul class=\"hide\" id=\"activity\">\n",
       "<li></li>\n",
       "</ul>\n",
       "<p class=\"HTOn\">Your browsing activity is empty.</p>\n",
       "<p class=\"HTOff\">Activity recording is turned off.</p>\n",
       "<p class=\"HTOff\" id=\"turnOn\">\n",
       "<a cmd=\"HTOn\" href=\"?cmd=HTOn&amp;\" id=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.HistoryDisplay.HistoryOn\" name=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.HistoryDisplay.HistoryOn\" onclick=\"return false;\" realname=\"EntrezSystem2.PEntrez.Protein.Sequence_ResultsPanel.HistoryDisplay.HistoryOn\" sid=\"1\">Turn recording back on</a>\n",
       "</p>\n",
       "<a class=\"seemore\" href=\"/sites/myncbi/recentactivity\">See more...</a>\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<div db=\"protein\" id=\"seq_search_bar\"><div id=\"seq_search_bar_warning\"></div><div class=\"seq_search_bar_left\"><div class=\"search_bar_box\"><label for=\"seq_search_bar_text\"></label><input id=\"seq_search_bar_text\" type=\"text\" value=\"\"/><button id=\"seq_search_bar_clear\">Clear</button></div><button id=\"seq_search_bar_find\">Find</button><div class=\"hide\" id=\"seq_nav_control\"><button id=\"seq_search_bar_first\">First</button><button id=\"seq_search_bar_previous\">Previous</button><span id=\"seq_search_bar_total\"></span><button id=\"seq_search_bar_next\">Next</button><button id=\"seq_search_bar_last\">Last</button></div><span id=\"seq_search_bar_message\"></span></div><div class=\"seq_search_bar_right\"><a href=\"/books/NBK44863/#sequencesquickstart.How_can_I_search_for\" target=\"_blank\" xclass=\"jig-ncbipopper\" xconfig=\"isTriggerElementCloseClick: false, destSelector:'#how-to-search'\">Help</a><button id=\"seq_search_bar_hide\">Close</button></div><div id=\"how-to-search\" style=\"display:none;\">\n",
       "                Finds sub-sequence or patterns in the sequence and highlights the matching region.\n",
       "                The tool works with standard single letter nucleotide \n",
       "                or protein codes including ambiguities and can match \n",
       "                Prosite patterns in protein sequences.\n",
       "                <br/><a href=\"/books/NBK44863/#sequencesquickstart.How_can_I_search_for\">More...</a><br/><br/></div></div>\n",
       "</div>\n",
       "</div>\n",
       "<div id=\"NCBIFooter_dynamic\">\n",
       "<div class=\"breadcrumbs\">You are here: \n",
       "            <span id=\"breadcrumb_text\"><a href=\"/guide/\">NCBI</a> &gt; <a href=\"https://www.ncbi.nlm.nih.gov/guide/proteins/\">Proteins</a> &gt; Protein Database</span></div>\n",
       "<a class=\"help_desk\" href=\"https://support.ncbi.nlm.nih.gov/ics/support/default.asp?Time=2020-03-12T16:57:12-04:00&amp;Snapshot=%2Fprojects%2FSequences%2FSeqDbRelease@1.123&amp;Host=portal101&amp;ncbi_phid=CE884FD5E6A966D10000000006460349&amp;ncbi_session=CE884FD5E6AA2281_1606SID&amp;from=https%3A%2F%2Fwww.ncbi.nlm.nih.gov%2Fprotein%2FNP_874360.1&amp;db=protein&amp;Ncbi_App=entrez&amp;Page=fasta&amp;style=classic&amp;deptID=28049&amp;report=fasta&amp;querykey=1&amp;query=1%20selected%20item:%2033186887;%20Page%20number:%201;%20Page%20size:%2020&amp;term=\" id=\"help-desk-link\" target=\"_blank\">Support Center</a>\n",
       "<noscript><img alt=\"\" src=\"/stat?jsdisabled=true&amp;ncbi_app=entrez&amp;ncbi_db=protein&amp;ncbi_pdid=fasta&amp;ncbi_phid=CE884FD5E6A966D10000000006460349\"/></noscript>\n",
       "</div>\n",
       "<div class=\"footer\" id=\"footer\" xml:base=\"http://127.0.0.1/sites/static/header_footer/\" xmlns=\"http://www.w3.org/1999/xhtml\">\n",
       "<!--    <div class=\"breadcrumbs\">You are here: <span id=\"breadcrumb_text\"><a href=\"#\">NCBI</a></span></div>\n",
       "        <a id=\"help-desk-link\" class=\"help_desk\" target=\"_blank\">Help Desk</a>-->\n",
       "<div class=\"subfooter\">\n",
       "<h2 class=\"offscreen_noflow\">Simple NCBI Directory</h2>\n",
       "<ul class=\"foot_list\" id=\"getting_started\">\n",
       "<li><h3>Getting Started</h3></li>\n",
       "<!-- removed; hp-250           <li><a href=\"/guide/all/\">Site Map</a></li>-->\n",
       "<li><a href=\"/home/learn.shtml\">NCBI Education</a></li>\n",
       "<li><a href=\"/books/NBK3831/\">NCBI Help Manual</a></li>\n",
       "<li><a href=\"/books/NBK143764/\">NCBI Handbook</a></li>\n",
       "<li><a href=\"/home/learn.shtml\">Training &amp; Tutorials</a></li>\n",
       "<li><a href=\"/home/submit.shtml\">Submit Data</a></li>\n",
       "</ul>\n",
       "<!-- this section is re-written by an XSLT before output -->\n",
       "<ul class=\"foot_list\" id=\"resources\">\n",
       "<li><h3>Resources</h3></li>\n",
       "<!-- fill this in with categories -->\n",
       "<li><a href=\"/guide/chemicals-bioassays/\">Chemicals &amp; Bioassays</a></li><li><a href=\"/guide/data-software/\">Data &amp; Software</a></li><li><a href=\"/guide/dna-rna/\">DNA &amp; RNA</a></li><li><a href=\"/guide/domains-structures/\">Domains &amp; Structures</a></li><li><a href=\"/guide/genes-expression/\">Genes &amp; Expression</a></li><li><a href=\"/guide/genetics-medicine/\">Genetics &amp; Medicine</a></li><li><a href=\"/guide/genomes-maps/\">Genomes &amp; Maps</a></li><li><a href=\"/guide/homology/\">Homology</a></li><li><a href=\"/guide/literature/\">Literature</a></li><li><a href=\"/guide/proteins/\">Proteins</a></li><li><a href=\"/guide/sequence-analysis/\">Sequence Analysis</a></li><li><a href=\"/guide/taxonomy/\">Taxonomy</a></li><li><a href=\"/guide/variation/\">Variation</a></li></ul>\n",
       "<ul class=\"foot_list\" id=\"popular\">\n",
       "<li><h3>Popular</h3></li>\n",
       "<li><a href=\"/pubmed/\">PubMed</a></li>\n",
       "<li><a href=\"/books/\">Bookshelf</a></li>\n",
       "<li><a href=\"/pmc/\">PubMed Central</a></li>\n",
       "<li><a class=\"newdomain\" href=\"https://blast.ncbi.nlm.nih.gov/Blast.cgi\">BLAST</a></li>\n",
       "<li><a href=\"/nucleotide/\">Nucleotide</a></li>\n",
       "<li><a href=\"/genome/\">Genome</a></li>\n",
       "<li><a href=\"/snp/\">SNP</a></li>\n",
       "<li><a href=\"/gene/\">Gene</a></li>\n",
       "<li><a href=\"/protein/\">Protein</a></li>\n",
       "<li><a class=\"newdomain\" href=\"https://pubchem.ncbi.nlm.nih.gov/\">PubChem</a></li>\n",
       "</ul>\n",
       "<ul class=\"foot_list\" id=\"featured\">\n",
       "<li><h3>Featured</h3></li>\n",
       "<li><a href=\"/gtr/\">Genetic Testing Registry</a></li>\n",
       "<li><a href=\"/genbank/\">GenBank</a></li>\n",
       "<li><a href=\"/refseq/\">Reference Sequences</a></li>\n",
       "<li><a href=\"/geo/\">Gene Expression Omnibus</a></li>\n",
       "<li><a href=\"/genome/gdv/\">Genome Data Viewer</a></li>\n",
       "<li><a href=\"/genome/guide/human/\">Human Genome</a></li>\n",
       "<li><a href=\"/genome/guide/mouse/\">Mouse Genome</a></li>\n",
       "<li><a href=\"/genomes/FLU/\">Influenza Virus</a></li>\n",
       "<li><a href=\"/tools/primer-blast/\">Primer-BLAST</a></li>\n",
       "<li><a href=\"/Traces/sra/\">Sequence Read Archive</a></li>\n",
       "</ul>\n",
       "<ul class=\"foot_list\" id=\"info\">\n",
       "<li><h3>NCBI Information</h3></li>\n",
       "<li><a href=\"/home/about/\">About NCBI</a></li>\n",
       "<li><a href=\"/research/\">Research at NCBI</a></li>\n",
       "<li><a href=\"https://ncbiinsights.ncbi.nlm.nih.gov\">NCBI News &amp; Blog</a></li>\n",
       "<li><a href=\"ftp://ftp.ncbi.nlm.nih.gov/\">NCBI FTP Site</a></li>\n",
       "<li><a class=\"external\" href=\"https://www.facebook.com/ncbi.nlm\">NCBI on Facebook</a></li>\n",
       "<li><a class=\"external\" href=\"https://twitter.com/ncbi\">NCBI on Twitter</a></li>\n",
       "<li><a class=\"external\" href=\"https://www.youtube.com/ncbinlm\">NCBI on YouTube</a></li>\n",
       "<li><a href=\"https://www.nlm.nih.gov/privacy.html\">Privacy Policy</a></li>\n",
       "</ul>\n",
       "</div>\n",
       "<div class=\"offscreen_noflow\" id=\"external-disclaimer\">\n",
       "        External link. Please review our <a href=\"https://www.nlm.nih.gov/privacy.html\">privacy policy</a>.\n",
       "    </div>\n",
       "<div class=\"contact_info\" id=\"ncbifooter\">\n",
       "<div id=\"footer-contents-right\">\n",
       "<div id=\"nlm_thumb_logo\">\n",
       "<a href=\"https://www.nlm.nih.gov\" title=\"NLM\">NLM</a>\n",
       "</div>\n",
       "<div id=\"nih_thumb_logo\">\n",
       "<a href=\"https://www.nih.gov\" title=\"NIH\">NIH</a>\n",
       "</div>\n",
       "<div id=\"hhs_thumb_logo\">\n",
       "<a href=\"https://www.hhs.gov\" title=\"DHHS\">DHHS</a>\n",
       "</div>\n",
       "<div id=\"usagov_thumb_logo\">\n",
       "<a href=\"https://www.usa.gov\" title=\"USA.gov\">USA.gov</a>\n",
       "</div>\n",
       "</div>\n",
       "<div id=\"footer-contents-left\">\n",
       "<p class=\"address vcard\">\n",
       "<span class=\"url\">\n",
       "<a class=\"fn url newdomain\" href=\"https://www.ncbi.nlm.nih.gov\">National Center for\n",
       "                        Biotechnology Information</a>,\n",
       "                </span> <span class=\"org url newdomain\"><a href=\"https://www.nlm.nih.gov/\">U.S. National Library of Medicine</a></span>\n",
       "<span class=\"adr\">\n",
       "<span class=\"street-address\">8600 Rockville Pike</span>, <span class=\"locality\">Bethesda</span>\n",
       "<span class=\"region\">MD</span>, <span class=\"postal-code\">20894</span>\n",
       "<span class=\"country-name\">USA</span>\n",
       "</span>\n",
       "</p>\n",
       "<a href=\"/home/about/policies.shtml\">Policies and Guidelines</a> | <a href=\"/home/about/contact.shtml\">Contact</a>\n",
       "</div>\n",
       "</div>\n",
       "<script src=\"/portal/portal3rc.fcgi/rlib/js/InstrumentOmnitureBaseJS/InstrumentNCBIConfigJS/InstrumentNCBIBaseJS/InstrumentPageStarterJS.js?v=1\" type=\"text/javascript\"> </script>\n",
       "<script src=\"/portal/portal3rc.fcgi/static/js/hfjs2.js\" type=\"text/javascript\"> </script>\n",
       "</div>\n",
       "</div>\n",
       "<div><input name=\"EntrezSystem2.PEntrez.DbConnector.Db\" sid=\"1\" type=\"hidden\" value=\"protein\"/><input name=\"EntrezSystem2.PEntrez.DbConnector.LastDb\" sid=\"1\" type=\"hidden\" value=\"protein\"/><input name=\"EntrezSystem2.PEntrez.DbConnector.Term\" sid=\"1\" type=\"hidden\" value=\"\"/><input name=\"EntrezSystem2.PEntrez.DbConnector.LastTabCmd\" sid=\"1\" type=\"hidden\" value=\"\"/><input name=\"EntrezSystem2.PEntrez.DbConnector.LastQueryKey\" sid=\"1\" type=\"hidden\" value=\"1\"/><input name=\"EntrezSystem2.PEntrez.DbConnector.IdsFromResult\" sid=\"1\" type=\"hidden\" value=\"\"/><input name=\"EntrezSystem2.PEntrez.DbConnector.LastIdsFromResult\" sid=\"1\" type=\"hidden\" value=\"\"/><input name=\"EntrezSystem2.PEntrez.DbConnector.LinkName\" sid=\"1\" type=\"hidden\"/><input name=\"EntrezSystem2.PEntrez.DbConnector.LinkReadableName\" sid=\"1\" type=\"hidden\"/><input name=\"EntrezSystem2.PEntrez.DbConnector.LinkSrcDb\" sid=\"1\" type=\"hidden\"/><input name=\"EntrezSystem2.PEntrez.DbConnector.Cmd\" sid=\"1\" type=\"hidden\"/><input name=\"EntrezSystem2.PEntrez.DbConnector.TabCmd\" sid=\"1\" type=\"hidden\"/><input name=\"EntrezSystem2.PEntrez.DbConnector.QueryKey\" sid=\"1\" type=\"hidden\"/></div>\n",
       "<input id=\"p$a\" name=\"p$a\" type=\"hidden\"/><input id=\"p$l\" name=\"p$l\" type=\"hidden\" value=\"EntrezSystem2\"/><input id=\"p$st\" name=\"p$st\" type=\"hidden\" value=\"protein\"/><input disabled=\"disabled\" id=\"SessionId\" name=\"SessionId\" type=\"hidden\" value=\"CE884FD5E6AA2281_1606SID\"/><input disabled=\"disabled\" id=\"Snapshot\" name=\"Snapshot\" type=\"hidden\" value=\"/projects/Sequences/SeqDbRelease@1.123\"/></form>\n",
       "</div>\n",
       "</div>\n",
       "<!-- CE884FD5E6AA2281_1606SID /projects/Sequences/SeqDbRelease@1.123 portal101 v4.1.r585844 Mon, May 06 2019 02:53:16 -->\n",
       "<script src=\"/portal/js/portal.js\" type=\"text/javascript\"></script><script snapshot=\"protein\" src=\"//static.pubmed.gov/portal/portal3rc.fcgi/4189134/js/3812534/4087685/3812535/4080513/3815665/4186313/4184468/3758627/4184279/3908752/3423/4176330/4184195/3217400/4176568/4177091/4076482/4188354/4185583/31971/35962/2168002/33542/33996/31854/35641/33538/33997/33966/33536/3254321/3397055/4187370/4161206/4001808.js\" type=\"text/javascript\"></script></body>\n",
       "</html>>"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "html_page = requests.get(urls[0])\n",
    "soup = BeautifulSoup(html_page.content, 'html.parser')\n",
    "soup.prettify"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "proteins = soup.find('pre')\n",
    "proteins"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "References\n",
    "Stephen F. Altschul, Thomas L. Madden, Alejandro A. Schffer, Jinghui Zhang, Zheng Zhang, Webb Miller, and David J. Lipman (1997), \"Gapped BLAST and PSI-BLAST: a new generation of protein database search programs\", Nucleic Acids Res. 25:3389-3402.\n",
    "https://towardsdatascience.com/processing-xml-in-python-elementtree-c8992941efd2 # processing xml data\n",
    "https://stackoverflow.com/questions/10220412/convert-tab-delimited-txt-file-into-a-csv-file-using-python # import of a text file\n",
    "https://www.ncbi.nlm.nih.gov/assembly/model/\n",
    "ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/405/GCF_000001405.39_GRCh38.p13/GCF_000001405.39_GRCh38.p13_assembly_report.txt # dna data\n",
    "https://www.ncbi.nlm.nih.gov/genomes/Genome2BE/genome2srv.cgi?action=GetFeatures4Grid&amp;download=on&amp;type=Proteins&amp;genome_id=51&genome_assembly_id=582967&mode=2&is_locus=1&is_locus_tag=0&optcols=1,0,0,0,0 # protein data\n",
    "https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&PAGE_TYPE=BlastDocs&DOC_TYPE=References\n",
    "https://www.ncbi.nlm.nih.gov/pubmed/20003500?dopt=Citation # paper for the creation of blast+\n",
    "https://www.ncbi.nlm.nih.gov/books/NBK52637/ # stand alone blast+ set up for pc\n",
    "ftp://ftp.ncbi.nlm.nih.gov/blast/db/ # data base location\n",
    "http://seqanswers.com/forums/showthread.php?t=44370 # troubleshooting database set up\n",
    "https://widdowquinn.github.io/2018-03-06-ibioic/02-sequence_databases/03-programming_for_blast.html # python/blast+ integration\n",
    "http://biopython.org/DIST/docs/tutorial/Tutorial.pdf # use of entrez package for online searches using api\n",
    "https://www.ncbi.nlm.nih.gov/books/NBK25499/ # entrez documentation"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "learn-env",
   "language": "python",
   "name": "learn-env"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
